<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jovana Maksimovic, Alicia Oshlack and Belinda Phipson" />

<meta name="date" content="2021-03-29" />

<title>Gene set testing for Illumina HumanMethylation Arrays</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-sm-12 col-md-4 col-lg-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-sm-12 col-md-8 col-lg-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">methyl-geneset-testing</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/Oshlack/methyl-geneset-testing">
    <span class="fas fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Gene set testing for Illumina HumanMethylation Arrays</h1>
<h3 class="subtitle">Generating &quot;truth&quot; gene sets from RNA-seq data</h3>
<h4 class="author">Jovana Maksimovic, Alicia Oshlack and Belinda Phipson</h4>
<h4 class="date">March 29, 2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-03-29
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>methyl-geneset-testing/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it's best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200302code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200302)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200302code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200302)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsatlassianpetermacorgaubitbucketscmosmethylgenesettestingtreeb33d804ef53140a2f145ec4d6696b1352253e3b3targetblankb33d804a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/tree/b33d804ef53140a2f145ec4d6696b1352253e3b3" target="_blank">b33d804</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsatlassianpetermacorgaubitbucketscmosmethylgenesettestingtreeb33d804ef53140a2f145ec4d6696b1352253e3b3targetblankb33d804a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/tree/b33d804ef53140a2f145ec4d6696b1352253e3b3" target="_blank">b33d804</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/figures.nb.html
    Ignored:    code/.DS_Store
    Ignored:    code/.Rhistory
    Ignored:    code/.job/
    Ignored:    code/old/
    Ignored:    data/.DS_Store
    Ignored:    data/annotations/
    Ignored:    data/cache-intermediates/
    Ignored:    data/cache-region/
    Ignored:    data/cache-rnaseq/
    Ignored:    data/cache-runtime/
    Ignored:    data/datasets/.DS_Store
    Ignored:    data/datasets/GSE110554-data.RData
    Ignored:    data/datasets/GSE120854/
    Ignored:    data/datasets/GSE120854_RAW.tar
    Ignored:    data/datasets/GSE135446-data.RData
    Ignored:    data/datasets/GSE135446/
    Ignored:    data/datasets/GSE135446_RAW.tar
    Ignored:    data/datasets/GSE95460_RAW.tar
    Ignored:    data/datasets/GSE95460_RAW/
    Ignored:    data/datasets/GSE95462-data.RData
    Ignored:    data/datasets/GSE95462/
    Ignored:    data/datasets/GSE95462_RAW/
    Ignored:    data/datasets/SRP100803/
    Ignored:    data/datasets/SRP125125/.DS_Store
    Ignored:    data/datasets/SRP125125/SRR6298*/
    Ignored:    data/datasets/SRP125125/SRR_Acc_List.txt
    Ignored:    data/datasets/SRP125125/SRR_Acc_List_Full.txt
    Ignored:    data/datasets/SRP125125/SraRunTable.txt
    Ignored:    data/datasets/SRP125125/multiqc_data/
    Ignored:    data/datasets/SRP125125/multiqc_report.html
    Ignored:    data/datasets/SRP125125/quants/
    Ignored:    data/datasets/SRP166862/
    Ignored:    data/datasets/SRP217468/
    Ignored:    data/datasets/TCGA.BRCA.rds
    Ignored:    data/datasets/TCGA.KIRC.rds
    Ignored:    data/misc/
    Ignored:    output/.DS_Store
    Ignored:    output/FDR-analysis/
    Ignored:    output/compare-methods/
    Ignored:    output/figures/
    Ignored:    output/methylgsa-params/
    Ignored:    output/outputs.tar.gz
    Ignored:    output/random-cpg-sims/
    Ignored:    output/test/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/04_expressionGenesets.Rmd</code>) and HTML (<code>docs/04_expressionGenesets.html</code>) files. If you've configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/d44c029e0d104c81502cbc0b72fe1767bdf5e90d/analysis/04_expressionGenesets.Rmd" target="_blank">d44c029</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2021-03-29
</td>
<td>
Rename analysis files to reflect addition of new datasets
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tximport)
library(here)
library(tidyverse)
library(EnsDb.Hsapiens.v75)
library(readr)
library(limma)
library(edgeR)
library(glue)
library(patchwork)
library(biobroom)
library(ChAMP)
source(here(&quot;code/utility.R&quot;))</code></pre>
<div id="data-download-mapping-and-quantification" class="section level1">
<h1>Data download, mapping and quantification</h1>
<p>The RNAseq dataset used to generate the &quot;truth&quot; gene sets is avoilable from GEO at <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE107011">GSE107011</a> or SRA at <a href="https://www.ncbi.nlm.nih.gov/sra?term=SRP125125">SRP125125</a>.</p>
<p>The RNAseq data was quasi-mapped and quantified using Salmon (v1.2.1) with the <a href="http://refgenomes.databio.org/v2/asset/hg19_cdna/fasta/archive?tag=default">hg19_cdna</a> human transcriptome downloaded from <a href="http://refgenomes.databio.org/index">refgenie</a>. The code used to perform the Salmon quasi-mapping and quantification can be found in <code>code/salmon-quant.sh</code>.</p>
<p>The Salmon mapping statistics can be veiwed <a href="multiqc_report.html">here</a>.</p>
<p>For downstream analysis that quantification files (<code>quant.sf</code>) for each sample are expected to be present in the following directory structure:</p>
<ul>
<li><code>data</code>
<ul>
<li><code>datasets</code>
<ul>
<li><code>SRP125125</code>
<ul>
<li><code>quants</code>
<ul>
<li><code>SRR6298258_quant</code></li>
<li>...</li>
<li>...</li>
<li><code>SRR6298376_quant</code></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul>
<div id="data-import" class="section level2">
<h2>Data import</h2>
<p>Load sample information and file names.</p>
<pre class="r"><code>targets &lt;- read_csv(here(&quot;data/datasets/SRP125125/SraRunTableFull.txt&quot;))</code></pre>
<pre><code>
── Column specification ────────────────────────────────────────────────────────
cols(
  .default = col_character()
)
ℹ Use `spec()` for the full column specifications.</code></pre>
<pre class="r"><code>targets$Cell_type &lt;- gsub(&quot; &quot;, &quot;_&quot;, targets$Cell_type)
targets$Cell_type &lt;- gsub(&quot;-&quot;, &quot;_&quot;, targets$Cell_type)
targets</code></pre>
<pre><code># A tibble: 24 x 27
   Run    `Assay Type` AvgSpotLen Bases BioProject BioSample Bytes `Center Name`
   &lt;chr&gt;  &lt;chr&gt;        &lt;chr&gt;      &lt;chr&gt; &lt;chr&gt;      &lt;chr&gt;     &lt;chr&gt; &lt;chr&gt;        
 1 SRR62… RNA-Seq      100        1342… PRJNA4187… SAMN0803… 6876… GEO          
 2 SRR62… RNA-Seq      100        1427… PRJNA4187… SAMN0803… 7340… GEO          
 3 SRR62… RNA-Seq      100        1570… PRJNA4187… SAMN0803… 8071… GEO          
 4 SRR62… RNA-Seq      100        1904… PRJNA4187… SAMN0803… 9742… GEO          
 5 SRR62… RNA-Seq      100        1470… PRJNA4187… SAMN0803… 7564… GEO          
 6 SRR62… RNA-Seq      100        1248… PRJNA4187… SAMN0803… 6610… GEO          
 7 SRR62… RNA-Seq      100        2184… PRJNA4187… SAMN0803… 1114… GEO          
 8 SRR62… RNA-Seq      100        2065… PRJNA4187… SAMN0803… 1056… GEO          
 9 SRR62… RNA-Seq      100        1984… PRJNA4187… SAMN0803… 1014… GEO          
10 SRR62… RNA-Seq      100        2082… PRJNA4187… SAMN0803… 1064… GEO          
# … with 14 more rows, and 19 more variables: Consent &lt;chr&gt;,
#   DATASTORE filetype &lt;chr&gt;, DATASTORE provider &lt;chr&gt;, DATASTORE region &lt;chr&gt;,
#   Experiment &lt;chr&gt;, GEO_Accession (exp) &lt;chr&gt;, Instrument &lt;chr&gt;,
#   LibraryLayout &lt;chr&gt;, LibrarySelection &lt;chr&gt;, LibrarySource &lt;chr&gt;,
#   Organism &lt;chr&gt;, Platform &lt;chr&gt;, ReleaseDate &lt;chr&gt;, Sample Name &lt;chr&gt;,
#   SRA Study &lt;chr&gt;, Cell_type &lt;chr&gt;, disease_status &lt;chr&gt;, gender &lt;chr&gt;,
#   source_name &lt;chr&gt;</code></pre>
<p>Setup file paths and sample names.</p>
<pre class="r"><code>files &lt;- list.files(here(&quot;data/datasets/SRP125125/quants&quot;), 
                    recursive = TRUE, pattern = &quot;quant.sf&quot;, full.names = TRUE)
pos &lt;- regexpr(&quot;SRR6298[0-9][0-9][0-9]&quot;, files, perl = TRUE)
names(files) &lt;- unname(substr(files, pos, 
                              pos[1] + attr(pos, &quot;match.length&quot;) - 1))
head(files)</code></pre>
<pre><code>                                                                                                             SRR6298258 
&quot;/Users/maksimovicjovana/Work/Research/methyl-geneset-testing/data/datasets/SRP125125/quants/SRR6298258_quant/quant.sf&quot; 
                                                                                                             SRR6298271 
&quot;/Users/maksimovicjovana/Work/Research/methyl-geneset-testing/data/datasets/SRP125125/quants/SRR6298271_quant/quant.sf&quot; 
                                                                                                             SRR6298273 
&quot;/Users/maksimovicjovana/Work/Research/methyl-geneset-testing/data/datasets/SRP125125/quants/SRR6298273_quant/quant.sf&quot; 
                                                                                                             SRR6298278 
&quot;/Users/maksimovicjovana/Work/Research/methyl-geneset-testing/data/datasets/SRP125125/quants/SRR6298278_quant/quant.sf&quot; 
                                                                                                             SRR6298281 
&quot;/Users/maksimovicjovana/Work/Research/methyl-geneset-testing/data/datasets/SRP125125/quants/SRR6298281_quant/quant.sf&quot; 
                                                                                                             SRR6298284 
&quot;/Users/maksimovicjovana/Work/Research/methyl-geneset-testing/data/datasets/SRP125125/quants/SRR6298284_quant/quant.sf&quot; </code></pre>
<p>Associate transcripts with gene IDs for gene-level summarization.</p>
<pre class="r"><code>edb &lt;- EnsDb.Hsapiens.v75
tx2gene &lt;- transcripts(edb, columns = c(&quot;tx_id&quot;, &quot;gene_id&quot;), 
                       return.type = &quot;DataFrame&quot;)
tx2gene</code></pre>
<pre><code>DataFrame with 215647 rows and 2 columns
                 tx_id         gene_id
           &lt;character&gt;     &lt;character&gt;
1      ENST00000000233 ENSG00000004059
2      ENST00000000412 ENSG00000003056
3      ENST00000000442 ENSG00000173153
4      ENST00000001008 ENSG00000004478
5      ENST00000001146 ENSG00000003137
...                ...             ...
215643        LRG_94t1          LRG_94
215644        LRG_96t1          LRG_96
215645        LRG_97t1          LRG_97
215646        LRG_98t1          LRG_98
215647        LRG_99t1          LRG_99</code></pre>
<p>Import gene-level counts and abundances.</p>
<pre class="r"><code>txiG &lt;- tximport(files, type = &quot;salmon&quot;, tx2gene = tx2gene, 
                  countsFromAbundance = &quot;lengthScaledTPM&quot;, 
                  ignoreTxVersion = TRUE)</code></pre>
<pre><code>reading in files with read_tsv</code></pre>
<pre><code>1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 
summarizing abundance
summarizing counts
summarizing length</code></pre>
<pre class="r"><code>colnames(txiG$counts) &lt;- names(files)
head(txiG$counts)</code></pre>
<pre><code>                 SRR6298258 SRR6298271  SRR6298273   SRR6298278   SRR6298281
ENSG00000000003  61.8092088   64.34252    2.108605     7.108053     6.854211
ENSG00000000005   0.0000000    0.00000    0.000000     0.000000     0.000000
ENSG00000000419 308.0357168  453.78408  300.756056   184.887042   183.859446
ENSG00000000457 566.1655509  668.13279  536.765170   409.182211   554.068761
ENSG00000000460  88.1087011  149.58245  207.133830    96.734048   195.018761
ENSG00000000938   0.9993118  199.98401 6106.191865 17055.657470 15321.804086
                SRR6298284 SRR6298286 SRR6298299 SRR6298302   SRR6298307
ENSG00000000003   12.16932   22.37741   207.2632   17.27445     4.634087
ENSG00000000005    0.00000    0.00000     0.0000    0.00000     0.000000
ENSG00000000419  137.81081   94.11433   324.2569  153.66560   285.821284
ENSG00000000457  506.88286  724.10043   831.8078  819.55366   287.623292
ENSG00000000460   36.97186   88.45333   305.1499  380.72895   250.642210
ENSG00000000938 6591.27698   64.65507   231.8483 5019.86572 17168.696404
                  SRR6298310 SRR6298313 SRR6298315 SRR6298328 SRR6298331
ENSG00000000003     5.597539    0.00000   191.5922   193.9760   12.63314
ENSG00000000005     0.000000    0.00000     0.0000     0.0000    0.00000
ENSG00000000419   157.483545  169.61751   411.3714   354.7413  245.31879
ENSG00000000457   577.570811  427.52646   528.0102   886.7599  639.44243
ENSG00000000460   245.554096   10.19566   121.0524   166.5221  259.31743
ENSG00000000938 12360.377205 9068.06785    49.0607   128.5385 5460.09143
                  SRR6298336  SRR6298339  SRR6298342  SRR6298344  SRR6298365
ENSG00000000003     7.548887    13.24455    9.167896  150.442321    6.321174
ENSG00000000005     0.000000     0.00000    0.000000    0.000000    0.000000
ENSG00000000419   414.301512   295.38687  299.028565  303.925491  389.002470
ENSG00000000457   251.307082   387.21898  289.416840 1037.669788  423.635676
ENSG00000000460   213.774478   186.27445   77.771446  230.877293  230.062502
ENSG00000000938 19652.208550 11370.32565 9706.096203    7.556217 3718.765758
                  SRR6298370 SRR6298373  SRR6298376
ENSG00000000003     2.440773    34.5469    23.76297
ENSG00000000005     0.000000     0.0000     0.00000
ENSG00000000419   287.085979   326.0884   254.49838
ENSG00000000457   279.957458   597.5548   292.36927
ENSG00000000460   113.801983   131.8787    84.14768
ENSG00000000938 21057.678193 16233.1932 12282.18942</code></pre>
<p>Set up <code>DGElist</code> object for downstream analysis.</p>
<pre class="r"><code>z &lt;- DGEList(txiG$counts)
z$genes &lt;- ensembldb::genes(edb, filter = GeneIdFilter(rownames(z)), 
                 columns = c(&quot;gene_id&quot;, &quot;symbol&quot;, &quot;entrezid&quot;), 
                 return.type = &quot;DataFrame&quot;)
z$genes$entrezid &lt;- sapply(z$genes$entrezid, function(x) x[1])
z$genes$length &lt;- rowMedians(txiG$length)
targets &lt;- targets[match(colnames(z), targets$Run), ]
z$samples$group &lt;- targets$Cell_type
z</code></pre>
<pre><code>An object of class &quot;DGEList&quot;
$counts
                SRR6298258 SRR6298271 SRR6298273 SRR6298278 SRR6298281
ENSG00000000003   61.80921   64.34252   2.108605   7.108053   6.854211
ENSG00000000005    0.00000    0.00000   0.000000   0.000000   0.000000
ENSG00000000419  308.03572  453.78408 300.756056 184.887042 183.859446
ENSG00000000457  566.16555  668.13279 536.765170 409.182211 554.068761
ENSG00000000460   88.10870  149.58245 207.133830  96.734048 195.018761
                SRR6298284 SRR6298286 SRR6298299 SRR6298302 SRR6298307
ENSG00000000003   12.16932   22.37741   207.2632   17.27445   4.634087
ENSG00000000005    0.00000    0.00000     0.0000    0.00000   0.000000
ENSG00000000419  137.81081   94.11433   324.2569  153.66560 285.821284
ENSG00000000457  506.88286  724.10043   831.8078  819.55366 287.623292
ENSG00000000460   36.97186   88.45333   305.1499  380.72895 250.642210
                SRR6298310 SRR6298313 SRR6298315 SRR6298328 SRR6298331
ENSG00000000003   5.597539    0.00000   191.5922   193.9760   12.63314
ENSG00000000005   0.000000    0.00000     0.0000     0.0000    0.00000
ENSG00000000419 157.483545  169.61751   411.3714   354.7413  245.31879
ENSG00000000457 577.570811  427.52646   528.0102   886.7599  639.44243
ENSG00000000460 245.554096   10.19566   121.0524   166.5221  259.31743
                SRR6298336 SRR6298339 SRR6298342 SRR6298344 SRR6298365
ENSG00000000003   7.548887   13.24455   9.167896   150.4423   6.321174
ENSG00000000005   0.000000    0.00000   0.000000     0.0000   0.000000
ENSG00000000419 414.301512  295.38687 299.028565   303.9255 389.002470
ENSG00000000457 251.307082  387.21898 289.416840  1037.6698 423.635676
ENSG00000000460 213.774478  186.27445  77.771446   230.8773 230.062502
                SRR6298370 SRR6298373 SRR6298376
ENSG00000000003   2.440773    34.5469   23.76297
ENSG00000000005   0.000000     0.0000    0.00000
ENSG00000000419 287.085979   326.0884  254.49838
ENSG00000000457 279.957458   597.5548  292.36927
ENSG00000000460 113.801983   131.8787   84.14768
36675 more rows ...

$samples
                          group lib.size norm.factors
SRR6298258    Naive_CD8_T_cells  8546013            1
SRR6298271    Naive_CD4_T_cells  9992761            1
SRR6298273        Naive_B_cells 11476629            1
SRR6298278  Classical_monocytes 14825624            1
SRR6298281 Natural_killer_cells 10978058            1
18 more rows ...

$genes
DataFrame with 36680 rows and 4 columns
              gene_id       symbol  entrezid    length
          &lt;character&gt;  &lt;character&gt; &lt;integer&gt; &lt;numeric&gt;
1     ENSG00000000003       TSPAN6      7105  1966.238
2     ENSG00000000005         TNMD     64102   704.094
3     ENSG00000000419         DPM1      8813   838.868
4     ENSG00000000457        SCYL3     57147  3715.563
5     ENSG00000000460     C1orf112     55732  2687.244
...               ...          ...       ...       ...
36676 ENSG00000273439         ZNF8      7554 1972.2820
36677 ENSG00000273444 RP5-1147A1.2        NA   39.8542
36678 ENSG00000273452   AC005358.1        NA 1817.5722
36679 ENSG00000273463   AL138834.1        NA  426.5972
36680 ENSG00000273484       OR6R2P        NA   92.8252</code></pre>
</div>
</div>
<div id="quality-control" class="section level1">
<h1>Quality control</h1>
<p>Genes that do not have an adequate number of reads in any sample should be filtered out prior to downstream analyses. From a biological perspective, genes that are not expressed at a biologically meaningful level in any condition are not of interest. Statistically, we get a better estimate of the mean-variance relationship in the data and reduce the number of statistical tests that are performes during differential expression analyses.</p>
<p>Filter out lowly expressed genes and calculate TMM normalisation factors.</p>
<pre class="r"><code>keep &lt;- filterByExpr(z, group = z$samples$group)
x &lt;- z[keep, ]
y &lt;- calcNormFactors(x)
y</code></pre>
<pre><code>An object of class &quot;DGEList&quot;
$counts
                 SRR6298258 SRR6298271  SRR6298273   SRR6298278   SRR6298281
ENSG00000000003  61.8092088   64.34252    2.108605     7.108053     6.854211
ENSG00000000419 308.0357168  453.78408  300.756056   184.887042   183.859446
ENSG00000000457 566.1655509  668.13279  536.765170   409.182211   554.068761
ENSG00000000460  88.1087011  149.58245  207.133830    96.734048   195.018761
ENSG00000000938   0.9993118  199.98401 6106.191865 17055.657470 15321.804086
                SRR6298284 SRR6298286 SRR6298299 SRR6298302   SRR6298307
ENSG00000000003   12.16932   22.37741   207.2632   17.27445     4.634087
ENSG00000000419  137.81081   94.11433   324.2569  153.66560   285.821284
ENSG00000000457  506.88286  724.10043   831.8078  819.55366   287.623292
ENSG00000000460   36.97186   88.45333   305.1499  380.72895   250.642210
ENSG00000000938 6591.27698   64.65507   231.8483 5019.86572 17168.696404
                  SRR6298310 SRR6298313 SRR6298315 SRR6298328 SRR6298331
ENSG00000000003     5.597539    0.00000   191.5922   193.9760   12.63314
ENSG00000000419   157.483545  169.61751   411.3714   354.7413  245.31879
ENSG00000000457   577.570811  427.52646   528.0102   886.7599  639.44243
ENSG00000000460   245.554096   10.19566   121.0524   166.5221  259.31743
ENSG00000000938 12360.377205 9068.06785    49.0607   128.5385 5460.09143
                  SRR6298336  SRR6298339  SRR6298342  SRR6298344  SRR6298365
ENSG00000000003     7.548887    13.24455    9.167896  150.442321    6.321174
ENSG00000000419   414.301512   295.38687  299.028565  303.925491  389.002470
ENSG00000000457   251.307082   387.21898  289.416840 1037.669788  423.635676
ENSG00000000460   213.774478   186.27445   77.771446  230.877293  230.062502
ENSG00000000938 19652.208550 11370.32565 9706.096203    7.556217 3718.765758
                  SRR6298370 SRR6298373  SRR6298376
ENSG00000000003     2.440773    34.5469    23.76297
ENSG00000000419   287.085979   326.0884   254.49838
ENSG00000000457   279.957458   597.5548   292.36927
ENSG00000000460   113.801983   131.8787    84.14768
ENSG00000000938 21057.678193 16233.1932 12282.18942
18116 more rows ...

$samples
                          group lib.size norm.factors
SRR6298258    Naive_CD8_T_cells  8546013    1.2429013
SRR6298271    Naive_CD4_T_cells  9992761    1.2126078
SRR6298273        Naive_B_cells 11476629    0.9964633
SRR6298278  Classical_monocytes 14825624    0.7306961
SRR6298281 Natural_killer_cells 10978058    1.0729896
18 more rows ...

$genes
DataFrame with 18121 rows and 4 columns
              gene_id        symbol  entrezid    length
          &lt;character&gt;   &lt;character&gt; &lt;integer&gt; &lt;numeric&gt;
1     ENSG00000000003        TSPAN6      7105  1966.238
2     ENSG00000000419          DPM1      8813   838.868
3     ENSG00000000457         SCYL3     57147  3715.563
4     ENSG00000000460      C1orf112     55732  2687.244
5     ENSG00000000938           FGR      2268  1818.207
...               ...           ...       ...       ...
18117 ENSG00000273398 RP11-474G23.1        NA   2069.22
18118 ENSG00000273423       OR13I1P        NA   1026.28
18119 ENSG00000273425    AC073107.1        NA   1492.08
18120 ENSG00000273429    AC253572.2        NA   1540.08
18121 ENSG00000273439          ZNF8      7554   1972.28</code></pre>
<p>Plotting the distribution log-CPM values shows that a majority of genes within each sample are either not expressed or lowly-expressed with log-CPM values that are small or negative.</p>
<pre class="r"><code>L &lt;- mean(z$samples$lib.size) * 1e-6
M &lt;- median(z$samples$lib.size) * 1e-6

dat &lt;- tidy(z, addSamples = TRUE)</code></pre>
<pre><code>Warning: `tbl_df()` was deprecated in dplyr 1.0.0.
Please use `tibble::as_tibble()` instead.</code></pre>
<pre class="r"><code>dat$cpm &lt;- reshape2::melt(cpm(z, log = TRUE), value.name = &quot;cpm&quot;)$cpm
p1 &lt;- ggplot(dat, aes(x = cpm, colour = group)) +
    geom_density() +
    labs(colour = &quot;Cell Type&quot;, x = &quot;Log CPM&quot;, y = &quot;Density&quot;, 
         title = &quot;Unfiltered&quot;) +
    geom_vline(xintercept = log2(10/M + 2/L), linetype = &quot;dashed&quot;)

dat &lt;- tidy(y, addSamples = TRUE)
dat$cpm &lt;- reshape2::melt(cpm(y, log = TRUE), value.name = &quot;cpm&quot;)$cpm
p2 &lt;- ggplot(dat, aes(x = cpm, colour = group)) +
    geom_density() +
    labs(colour = &quot;Cell Type&quot;, x = &quot;Log CPM&quot;, y = &quot;Density&quot;,
         title = &quot;Filtered&quot;) +
    geom_vline(xintercept = log2(10/M + 2/L), linetype = &quot;dashed&quot;)

p1 + p2 + plot_layout(guides = &quot;collect&quot;) &amp; theme(legend.position = &quot;bottom&quot;)</code></pre>
<p><img src="figure/04_expressionGenesets.Rmd/unnamed-chunk-8-1.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cellPal &lt;- RColorBrewer::brewer.pal(6, &quot;Dark2&quot;)[c(4, 5, 1, 2, 3, 6)]
dat &lt;- tidy(y, addSamples = TRUE)
p1 &lt;- ggplot(dat, aes(x = sample, y = lib.size, fill = group)) +
    geom_bar(stat = &quot;identity&quot;) +
    labs(fill = &quot;Cell Type&quot;, x = &quot;Sample&quot;, y = &quot;Library Size&quot;) +
    scale_fill_manual(values = cellPal) +
    theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 7),
          legend.position = &quot;bottom&quot;)

dat$cpm &lt;- reshape2::melt(cpm(y, log = TRUE), value.name = &quot;cpm&quot;)$cpm
p2 &lt;- ggplot(dat, aes(x = sample, y = cpm, fill = group)) +
    geom_boxplot(show.legend = FALSE) +
    labs(x = &quot;Sample&quot;, y = &quot;CPM&quot;) +
    theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 7),
          legend.position = &quot;bottom&quot;) +
    scale_fill_manual(values = cellPal) 

p &lt;- p1 + p2 + plot_layout(guides = &quot;collect&quot;) &amp; theme(legend.position = &quot;bottom&quot;)
p</code></pre>
<p><img src="figure/04_expressionGenesets.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Save figure for use in manuscript.</p>
<pre class="r"><code>outDir &lt;- here::here(&quot;output/figures&quot;)
if (!dir.exists(outDir)) dir.create(outDir)

fig &lt;- here(&quot;output/figures/SFig-4A.rds&quot;)
saveRDS(p1, fig, compress = FALSE)


fig &lt;- here(&quot;output/figures/SFig-4B.rds&quot;)
saveRDS(p2, fig, compress = FALSE)</code></pre>
<p>Multi-dimensional scaling (MDS) plots show the largest sources of variation in the data. They are a good way of exploring the relationships between the samples and identifying structure in the data. The following series of MDS plots examines the first four principal components.</p>
<pre class="r"><code>lcpm &lt;- cpm(y, log = TRUE)
dims &lt;- list(c(1,2), c(1,3), c(2,3), c(3,4))
p &lt;- vector(&quot;list&quot;, length(dims))

for(i in 1:length(dims)){
    tmp &lt;- plotMDS(lcpm, top=1000, gene.selection=&quot;common&quot;, plot = FALSE,
                   dim.plot = dims[[i]])
    
    dat &lt;-data.frame(x = tmp$x, y = tmp$y, cellType = targets$Cell_type)
    p[[i]] &lt;- ggplot(dat, aes(x = x, y = y, colour = cellType)) +
        geom_point() +
        scale_colour_manual(values = cellPal)  +
        labs(colour = &quot;Cell Type&quot;, x = glue(&quot;PC {tmp$dim.plot[1]}&quot;),
             y = glue(&quot;PC {tmp$dim.plot[2]}&quot;))
    
}

(p[[1]] | p[[2]]) / (p[[3]] | p[[4]]) + plot_layout(guides = &quot;collect&quot;)</code></pre>
<p><img src="figure/04_expressionGenesets.Rmd/unnamed-chunk-11-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>Save figure for use in manuscript.</p>
<pre class="r"><code>fig &lt;- here(&quot;output/figures/SFig-4C.rds&quot;)
saveRDS(p[[1]], fig, compress = FALSE)</code></pre>
</div>
<div id="differential-expression-analysis" class="section level1">
<h1>Differential expression analysis</h1>
<p>The TMM normalised data was transformed using <code>voomWithQualityWeights</code>. This takes into account the differing library sizes and the mean variance relationship in the data as well as calculating sample-specific quality weights. Linear models were fit using <code>limma</code>, taking into account the <code>voom</code> weights.</p>
<pre class="r"><code>design &lt;- model.matrix(~0+y$samples$group, data = targets)
colnames(design) &lt;- c(levels(factor(y$samples$group)))
v &lt;- voomWithQualityWeights(y, design, plot = TRUE)</code></pre>
<p><img src="figure/04_expressionGenesets.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cont &lt;- makeContrasts(CD4vCD8=Naive_CD4_T_cells-Naive_CD8_T_cells,
                      MonovNeu=Classical_monocytes-Low_density_neutrophils,
                      BcellvNK=Naive_B_cells-Natural_killer_cells,
                      levels=design)
fit &lt;- lmFit(v, design)
cfit &lt;- contrasts.fit(fit, cont)
fit2 &lt;- eBayes(cfit, robust = TRUE)
summary(decideTests(fit2, p.value = 0.05))</code></pre>
<pre><code>       CD4vCD8 MonovNeu BcellvNK
Down        89     4295     2293
NotSig   17917     8703    13282
Up         115     5123     2546</code></pre>
<pre class="r"><code>fitSum &lt;- summary(decideTests(fit2, p.value = 0.05))
dat &lt;- reshape2::melt(fitSum[rownames(fitSum) != &quot;NotSig&quot;, ])
colnames(dat) &lt;- c(&quot;dir&quot;,&quot;comp&quot;,&quot;num&quot;)

p &lt;- ggplot(dat, aes(x = comp, y = num, fill = dir)) +
    geom_bar(stat = &quot;identity&quot;, position = &quot;dodge&quot;) +
    labs(x = &quot;Comparison&quot;, y = &quot;No. DE Genes (FDR &lt; 0.05)&quot;, fill = &quot;Direction&quot;) +
    scale_fill_brewer(palette = &quot;Set1&quot;, direction = -1)
p</code></pre>
<p><img src="figure/04_expressionGenesets.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Save figure for use in manuscript.</p>
<pre class="r"><code>fig &lt;- here(&quot;output/figures/SFig-4D.rds&quot;)
saveRDS(p, fig, compress = FALSE)</code></pre>
</div>
<div id="gene-set-testing" class="section level1">
<h1>Gene set testing</h1>
<p>Testing for enrichment of GO categories and KEGG pathways amongst statistically significant differentially expressed genes.</p>
<p>Save results as RDS objects for use as &quot;truth&quot; sets in methylation analyses.</p>
<pre class="r"><code>go &lt;- NULL
kegg &lt;- NULL

for(i in 1:ncol(cont)){
    top &lt;- topTable(fit2, coef = i, p.value = 0.05, number = Inf)
    
    tmp &lt;- goana(top$entrezid, universe = v$genes$entrezid, 
                 covariate = v$genes$length)
    tmp &lt;- topGO(tmp, number = Inf)
    tmp$FDR &lt;- p.adjust(tmp$P.DE, method = &quot;BH&quot;)
    tmp &lt;- rownames_to_column(tmp, var = &quot;ID&quot;)
    tmp$contrast &lt;- colnames(cont)[i] 
    go &lt;- bind_rows(go, tmp)
        
    tmp &lt;- kegga(top$entrezid, universe = v$genes$entrezid, species = &quot;Hs&quot;, 
                 covariate = v$genes$length, species.KEGG = &quot;hsa&quot;)
    tmp &lt;- topKEGG(tmp, number = Inf)
    tmp$FDR &lt;- p.adjust(tmp$P.DE, method = &quot;BH&quot;)
    tmp &lt;- rownames_to_column(tmp, var = &quot;PID&quot;)
    tmp$contrast &lt;- colnames(cont)[i] 
    kegg &lt;- bind_rows(kegg, tmp)
}</code></pre>
<pre><code>Warning in rowsum.default(covariate, group = universe, reorder = FALSE): missing
values for &#39;group&#39;</code></pre>
<pre><code>Warning in rowsum.default(rep_len(1L, length(universe)), group = universe, :
missing values for &#39;group&#39;</code></pre>
<pre><code>Warning in rowsum.default(covariate, group = universe, reorder = FALSE): missing
values for &#39;group&#39;</code></pre>
<pre><code>Warning in rowsum.default(rep_len(1L, length(universe)), group = universe, :
missing values for &#39;group&#39;</code></pre>
<pre><code>Warning in rowsum.default(covariate, group = universe, reorder = FALSE): missing
values for &#39;group&#39;</code></pre>
<pre><code>Warning in rowsum.default(rep_len(1L, length(universe)), group = universe, :
missing values for &#39;group&#39;</code></pre>
<pre class="r"><code>outDir &lt;- here::here(&quot;data/cache-rnaseq&quot;)
if (!dir.exists(outDir)) dir.create(outDir)

saveRDS(go, here(glue(&quot;data/cache-rnaseq/RNAseq-GO.rds&quot;)))
saveRDS(kegg, here(glue(&quot;data/cache-rnaseq/RNAseq-KEGG.rds&quot;)))</code></pre>
<p>Test whether the BROAD sets are enriched for the differentially expressed genes using the <code>gsaseq</code> function that can be found in <code>code/utility.R</code></p>
<pre class="r"><code>data(PathwayList)
keep &lt;- sapply(PathwayList, function(x) any(x %in% v$genes$symbol))
ensembl &lt;- suppressMessages(lapply(PathwayList[keep], function(x){
    tmp &lt;- v$genes$gene_id[v$genes$symbol %in% x]
    tmp[!is.na(tmp)]
}))

entrez &lt;- suppressMessages(lapply(ensembl, function(x){
    tmp &lt;- unname(v$genes$entrezid[v$genes$gene_id %in% x])
    tmp[!is.na(tmp)]
}))

gsa &lt;- NULL
for(i in 1:ncol(cont)){
    top &lt;- topTable(fit2, coef = i, p.value = 0.05, number = Inf)
    
    tmp &lt;- gsaseq(top$entrezid, universe = v$genes$entrezid, 
                  collection = entrez, gene.length = v$genes$length)
    tmp &lt;- rownames_to_column(data.frame(tmp), var = &quot;ID&quot;)
    tmp$contrast &lt;- colnames(cont)[i] 
    gsa &lt;- bind_rows(gsa, tmp)
    
}

saveRDS(gsa, here(&quot;data/cache-rnaseq/RNAseq-BROAD-GSA.rds&quot;))</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Mojave 10.14.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ChAMP_2.20.1                              
 [2] RPMM_1.25                                 
 [3] cluster_2.1.1                             
 [4] DT_0.17                                   
 [5] IlluminaHumanMethylationEPICmanifest_0.3.0
 [6] Illumina450ProbeVariants.db_1.26.0        
 [7] DMRcate_2.4.1                             
 [8] ChAMPdata_2.22.0                          
 [9] minfi_1.36.0                              
[10] bumphunter_1.32.0                         
[11] locfit_1.5-9.4                            
[12] iterators_1.0.13                          
[13] foreach_1.5.1                             
[14] Biostrings_2.58.0                         
[15] XVector_0.30.0                            
[16] SummarizedExperiment_1.20.0               
[17] MatrixGenerics_1.2.1                      
[18] matrixStats_0.58.0                        
[19] biobroom_1.22.0                           
[20] broom_0.7.5                               
[21] patchwork_1.1.1                           
[22] glue_1.4.2                                
[23] edgeR_3.32.1                              
[24] limma_3.46.0                              
[25] EnsDb.Hsapiens.v75_2.99.0                 
[26] ensembldb_2.14.0                          
[27] AnnotationFilter_1.14.0                   
[28] GenomicFeatures_1.42.2                    
[29] AnnotationDbi_1.52.0                      
[30] Biobase_2.50.0                            
[31] GenomicRanges_1.42.0                      
[32] GenomeInfoDb_1.26.4                       
[33] IRanges_2.24.1                            
[34] S4Vectors_0.28.1                          
[35] BiocGenerics_0.36.0                       
[36] forcats_0.5.1                             
[37] stringr_1.4.0                             
[38] dplyr_1.0.5                               
[39] purrr_0.3.4                               
[40] readr_1.4.0                               
[41] tidyr_1.1.3                               
[42] tibble_3.1.0                              
[43] ggplot2_3.3.3                             
[44] tidyverse_1.3.0                           
[45] here_1.0.1                                
[46] tximport_1.18.0                           
[47] workflowr_1.6.2                           

loaded via a namespace (and not attached):
  [1] rappdirs_0.3.3                                     
  [2] rtracklayer_1.50.0                                 
  [3] R.methodsS3_1.8.1                                  
  [4] wateRmelon_1.34.0                                  
  [5] bit64_4.0.5                                        
  [6] knitr_1.31                                         
  [7] DelayedArray_0.16.3                                
  [8] R.utils_2.10.1                                     
  [9] data.table_1.14.0                                  
 [10] rpart_4.1-15                                       
 [11] doParallel_1.0.16                                  
 [12] RCurl_1.98-1.3                                     
 [13] GEOquery_2.58.0                                    
 [14] generics_0.1.0                                     
 [15] preprocessCore_1.52.1                              
 [16] RSQLite_2.2.5                                      
 [17] combinat_0.0-8                                     
 [18] bit_4.0.4                                          
 [19] xml2_1.3.2                                         
 [20] lubridate_1.7.10                                   
 [21] httpuv_1.5.5                                       
 [22] assertthat_0.2.1                                   
 [23] IlluminaHumanMethylation450kmanifest_0.4.0         
 [24] viridis_0.5.1                                      
 [25] isva_1.9                                           
 [26] IlluminaHumanMethylationEPICanno.ilm10b4.hg19_0.6.0
 [27] xfun_0.22                                          
 [28] hms_1.0.0                                          
 [29] jquerylib_0.1.3                                    
 [30] DNAcopy_1.64.0                                     
 [31] evaluate_0.14                                      
 [32] missMethyl_1.24.0                                  
 [33] promises_1.2.0.1                                   
 [34] fansi_0.4.2                                        
 [35] scrime_1.3.5                                       
 [36] progress_1.2.2                                     
 [37] dendextend_1.14.0                                  
 [38] dbplyr_2.1.0                                       
 [39] readxl_1.3.1                                       
 [40] DBI_1.1.1                                          
 [41] htmlwidgets_1.5.3                                  
 [42] reshape_0.8.8                                      
 [43] ROC_1.66.0                                         
 [44] ellipsis_0.3.1                                     
 [45] backports_1.2.1                                    
 [46] permute_0.9-5                                      
 [47] annotate_1.68.0                                    
 [48] biomaRt_2.46.3                                     
 [49] sparseMatrixStats_1.2.1                            
 [50] vctrs_0.3.6                                        
 [51] cachem_1.0.4                                       
 [52] withr_2.4.1                                        
 [53] globaltest_5.44.0                                  
 [54] Gviz_1.34.1                                        
 [55] BSgenome_1.58.0                                    
 [56] checkmate_2.0.0                                    
 [57] GenomicAlignments_1.26.0                           
 [58] prettyunits_1.1.1                                  
 [59] mclust_5.4.7                                       
 [60] ExperimentHub_1.16.0                               
 [61] lazyeval_0.2.2                                     
 [62] crayon_1.4.1                                       
 [63] genefilter_1.72.1                                  
 [64] labeling_0.4.2                                     
 [65] pkgconfig_2.0.3                                    
 [66] nlme_3.1-152                                       
 [67] ProtGenerics_1.22.0                                
 [68] nnet_7.3-15                                        
 [69] rlang_0.4.10                                       
 [70] nleqslv_3.3.2                                      
 [71] lifecycle_1.0.0                                    
 [72] affyio_1.60.0                                      
 [73] BiocFileCache_1.14.0                               
 [74] modelr_0.1.8                                       
 [75] AnnotationHub_2.22.0                               
 [76] dichromat_2.0-0                                    
 [77] cellranger_1.1.0                                   
 [78] rprojroot_2.0.2                                    
 [79] rngtools_1.5                                       
 [80] IlluminaHumanMethylation450kanno.ilmn12.hg19_0.6.0 
 [81] base64_2.0                                         
 [82] Matrix_1.3-2                                       
 [83] Rhdf5lib_1.12.1                                    
 [84] reprex_1.0.0                                       
 [85] base64enc_0.1-3                                    
 [86] geneLenDataBase_1.26.0                             
 [87] whisker_0.4                                        
 [88] viridisLite_0.3.0                                  
 [89] png_0.1-7                                          
 [90] bitops_1.0-6                                       
 [91] R.oo_1.24.0                                        
 [92] KernSmooth_2.23-18                                 
 [93] rhdf5filters_1.2.0                                 
 [94] blob_1.2.1                                         
 [95] DelayedMatrixStats_1.12.3                          
 [96] doRNG_1.8.2                                        
 [97] qvalue_2.22.0                                      
 [98] nor1mix_1.3-0                                      
 [99] jpeg_0.1-8.1                                       
[100] scales_1.1.1                                       
[101] memoise_2.0.0                                      
[102] magrittr_2.0.1                                     
[103] plyr_1.8.6                                         
[104] zlibbioc_1.36.0                                    
[105] compiler_4.0.3                                     
[106] RColorBrewer_1.1-2                                 
[107] illuminaio_0.32.0                                  
[108] clue_0.3-58                                        
[109] JADE_2.0-3                                         
[110] affy_1.68.0                                        
[111] Rsamtools_2.6.0                                    
[112] cli_2.3.1                                          
[113] DSS_2.38.0                                         
[114] htmlTable_2.1.0                                    
[115] Formula_1.2-4                                      
[116] mgcv_1.8-34                                        
[117] MASS_7.3-53.1                                      
[118] tidyselect_1.1.0                                   
[119] stringi_1.5.3                                      
[120] highr_0.8                                          
[121] yaml_2.2.1                                         
[122] askpass_1.1                                        
[123] latticeExtra_0.6-29                                
[124] grid_4.0.3                                         
[125] sass_0.3.1                                         
[126] VariantAnnotation_1.36.0                           
[127] tools_4.0.3                                        
[128] rstudioapi_0.13                                    
[129] foreign_0.8-81                                     
[130] git2r_0.28.0                                       
[131] bsseq_1.26.0                                       
[132] gridExtra_2.3                                      
[133] farver_2.1.0                                       
[134] digest_0.6.27                                      
[135] BiocManager_1.30.12                                
[136] shiny_1.6.0                                        
[137] quadprog_1.5-8                                     
[138] Rcpp_1.0.6                                         
[139] siggenes_1.64.0                                    
[140] BiocVersion_3.12.0                                 
[141] later_1.1.0.1                                      
[142] org.Hs.eg.db_3.12.0                                
[143] httr_1.4.2                                         
[144] biovizBase_1.38.0                                  
[145] lumi_2.42.0                                        
[146] colorspace_2.0-0                                   
[147] rvest_1.0.0                                        
[148] XML_3.99-0.6                                       
[149] fs_1.5.0                                           
[150] splines_4.0.3                                      
[151] statmod_1.4.35                                     
[152] kpmt_0.1.0                                         
[153] multtest_2.46.0                                    
[154] shinythemes_1.2.0                                  
[155] plotly_4.9.3                                       
[156] xtable_1.8-4                                       
[157] jsonlite_1.7.2                                     
[158] marray_1.68.0                                      
[159] R6_2.5.0                                           
[160] Hmisc_4.5-0                                        
[161] pillar_1.5.1                                       
[162] htmltools_0.5.1.1                                  
[163] mime_0.10                                          
[164] fastmap_1.1.0                                      
[165] BiocParallel_1.24.1                                
[166] interactiveDisplayBase_1.28.0                      
[167] beanplot_1.2                                       
[168] codetools_0.2-18                                   
[169] utf8_1.2.1                                         
[170] sva_3.38.0                                         
[171] lattice_0.20-41                                    
[172] bslib_0.2.4                                        
[173] BiasedUrn_1.07                                     
[174] curl_4.3                                           
[175] gtools_3.8.2                                       
[176] GO.db_3.12.1                                       
[177] openssl_1.4.3                                      
[178] survival_3.2-10                                    
[179] methylumi_2.36.0                                   
[180] rmarkdown_2.7                                      
[181] fastICA_1.2-2                                      
[182] munsell_0.5.0                                      
[183] rhdf5_2.34.0                                       
[184] GenomeInfoDbData_1.2.4                             
[185] goseq_1.42.0                                       
[186] impute_1.64.0                                      
[187] HDF5Array_1.18.1                                   
[188] reshape2_1.4.4                                     
[189] haven_2.3.1                                        
[190] gtable_0.3.0                                       </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
