<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jovana Maksimovic, Alicia Oshlack and Belinda Phipson" />

<meta name="date" content="2020-08-14" />

<title>Gene set testing for Illumina HumanMethylation Arrays</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">methyl-geneset-testing</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/Oshlack/methyl-geneset-testing">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Gene set testing for Illumina HumanMethylation Arrays</h1>
<h3 class="subtitle">Evaluating the performance of different methods</h3>
<h4 class="author">Jovana Maksimovic, Alicia Oshlack and Belinda Phipson</h4>
<h4 class="date">August 14, 2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-08-14
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>methyl-geneset-testing/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it's best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200302code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200302)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200302code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200302)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomOshlackmethylgenesettestingtree91699a8d1b57e40650c7101a0b7db675e67658a1targetblank91699a8a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/Oshlack/methyl-geneset-testing/tree/91699a8d1b57e40650c7101a0b7db675e67658a1" target="_blank">91699a8</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomOshlackmethylgenesettestingtree91699a8d1b57e40650c7101a0b7db675e67658a1targetblank91699a8a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/Oshlack/methyl-geneset-testing/tree/91699a8d1b57e40650c7101a0b7db675e67658a1" target="_blank">91699a8</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/figures.nb.html
    Ignored:    code/.job/
    Ignored:    code/old/
    Ignored:    data/.DS_Store
    Ignored:    data/annotations/
    Ignored:    data/cache-intermediates/
    Ignored:    data/cache-region/
    Ignored:    data/cache-rnaseq/
    Ignored:    data/cache-runtime/
    Ignored:    data/datasets/GSE110554-data.RData
    Ignored:    data/datasets/SRP125125/SRR6298258/
    Ignored:    data/datasets/SRP125125/SRR6298271/
    Ignored:    data/datasets/SRP125125/SRR6298273/
    Ignored:    data/datasets/SRP125125/SRR6298278/
    Ignored:    data/datasets/SRP125125/SRR6298281/
    Ignored:    data/datasets/SRP125125/SRR6298284/
    Ignored:    data/datasets/SRP125125/SRR6298286/
    Ignored:    data/datasets/SRP125125/SRR6298299/
    Ignored:    data/datasets/SRP125125/SRR6298302/
    Ignored:    data/datasets/SRP125125/SRR6298307/
    Ignored:    data/datasets/SRP125125/SRR6298310/
    Ignored:    data/datasets/SRP125125/SRR6298313/
    Ignored:    data/datasets/SRP125125/SRR6298315/
    Ignored:    data/datasets/SRP125125/SRR6298328/
    Ignored:    data/datasets/SRP125125/SRR6298331/
    Ignored:    data/datasets/SRP125125/SRR6298336/
    Ignored:    data/datasets/SRP125125/SRR6298339/
    Ignored:    data/datasets/SRP125125/SRR6298342/
    Ignored:    data/datasets/SRP125125/SRR6298344/
    Ignored:    data/datasets/SRP125125/SRR6298365/
    Ignored:    data/datasets/SRP125125/SRR6298370/
    Ignored:    data/datasets/SRP125125/SRR6298373/
    Ignored:    data/datasets/SRP125125/SRR6298376/
    Ignored:    data/datasets/SRP125125/SRR_Acc_List.txt
    Ignored:    data/datasets/SRP125125/SRR_Acc_List_Full.txt
    Ignored:    data/datasets/SRP125125/SraRunTable.txt
    Ignored:    data/datasets/SRP125125/multiqc_data/
    Ignored:    data/datasets/SRP125125/multiqc_report.html
    Ignored:    data/datasets/SRP125125/quants/
    Ignored:    data/datasets/TCGA.KIRC.rds
    Ignored:    data/genesets/BROAD-sets.rds
    Ignored:    data/misc/
    Ignored:    output/.DS_Store
    Ignored:    output/FDR-analysis/
    Ignored:    output/cache-rnaseq/
    Ignored:    output/compare-methods/
    Ignored:    output/figures/
    Ignored:    output/methylgsa-params/
    Ignored:    output/random-cpg-sims/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/05_exploreData.Rmd</code>) and HTML (<code>docs/05_exploreData.html</code>) files. If you've configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Oshlack/methyl-geneset-testing/blob/91699a8d1b57e40650c7101a0b7db675e67658a1/analysis/05_exploreData.Rmd" target="_blank">91699a8</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2020-08-14
</td>
<td>
wflow_publish(&quot;analysis/_site.yml&quot;, republish = TRUE, all = TRUE)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Oshlack/methyl-geneset-testing/blob/39bdd220a669d7fe614f2f012975716365c979b4/analysis/05_exploreData.Rmd" target="_blank">39bdd22</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2020-08-14
</td>
<td>
Renamed analysis files with numbering for run order.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(here)
library(minfi)
library(paletteer)
library(limma)
library(BiocParallel)
library(reshape2)
library(gridExtra)
library(missMethyl)
library(ggplot2)
library(glue)
library(grid)
library(tidyverse)
library(rbin)
library(patchwork)
library(ChAMPdata)
library(lemon)
source(here(&quot;code/utility.R&quot;))</code></pre>
<div id="load-data" class="section level1">
<h1>Load data</h1>
<p>We are using publicly available EPIC data <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE110554">GSE110554</a> generated from flow sorted blood cells. The data is normalised and filtered (bad probes, multi-mapping probes, SNP probes, sex chromosomes).</p>
<pre class="r"><code># load data
dataFile &lt;- here(&quot;data/datasets/GSE110554-data.RData&quot;)
if(file.exists(dataFile)){
    # load processed data and sample information
    load(dataFile)
} else {
    # get data from experiment hub, normalise, filter and save objects
    readData(dataFile)
    # load processed data and sample information
    load(dataFile)
}</code></pre>
<div id="qc-plots" class="section level2">
<h2>QC plots</h2>
<pre class="r"><code># plot mean detection p-values across all samples
dat &lt;- tibble::tibble(mean = colMeans(detP), cellType = targets$CellType)
ggplot(dat, aes(y = mean, x = cellType, fill = cellType)) +
    geom_bar(stat = &quot;identity&quot;) +
    labs(fill = &quot;Cell Type&quot;) +
    scale_fill_brewer(palette = &quot;Dark2&quot;)</code></pre>
<p><img src="figure/05_exploreData.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># plot normalised beta value distribution
bVals &lt;- getBeta(normGr)
dat &lt;- data.frame(reshape2::melt(bVals))
colnames(dat) &lt;- c(&quot;cpg&quot;, &quot;sample&quot;, &quot;bVal&quot;)
dat &lt;- dplyr::bind_cols(dat, cellType = rep(targets$CellType, 
                                            each = nrow(bVals)))

ggplot(dat, aes(x = bVal, colour = cellType)) +
    geom_density() +
    labs(colour = &quot;Cell Type&quot;) +
    scale_color_brewer(palette = &quot;Dark2&quot;)</code></pre>
<p><img src="figure/05_exploreData.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># MDS plots to look at largest sources of variation
p &lt;- plotMDS(getM(fltGr), top=1000, gene.selection=&quot;common&quot;, plot = FALSE)
dat &lt;- tibble::tibble(x = p$x, y = p$y, cellType = targets$CellType)

p &lt;- ggplot(dat, aes(x = x, y = y, colour = cellType)) +
    geom_point(size = 2) +
    labs(colour = &quot;Cell type&quot;) +
    scale_colour_brewer(palette = &quot;Dark2&quot;) +
    labs(x = &quot;Principal component 1&quot;, y = &quot;Principal component 2&quot;)
p</code></pre>
<p><img src="figure/05_exploreData.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Save figure for use in manuscript.</p>
<pre class="r"><code>outDir &lt;- here::here(&quot;output/figures&quot;)
if (!dir.exists(outDir)) dir.create(outDir)

fig &lt;- here(&quot;output/figures/Fig-4A.rds&quot;)
saveRDS(p, fig, compress = FALSE)</code></pre>
</div>
<div id="statistical-analysis" class="section level2">
<h2>Statistical analysis</h2>
<p>Compare several sets of sorted immune cells. Consider results significant at FDR &lt; 0.05 and delta beta &gt; 10% (~ lfc = 0.5).</p>
<pre class="r"><code>mVals &lt;- getM(fltGr)
bVals &lt;- getBeta(fltGr)</code></pre>
<pre class="r"><code>design &lt;- model.matrix(~0+targets$CellType)
colnames(design) &lt;- levels(factor(targets$CellType))
fit &lt;- lmFit(mVals, design)
cont.matrix &lt;- makeContrasts(CD4vCD8=CD4T-CD8T,
                             MonovNeu=Mono-Neu,
                             BcellvNK=Bcell-NK,
                             levels=design)
fit2 &lt;- contrasts.fit(fit, cont.matrix)
tfit &lt;- eBayes(fit2, robust=TRUE, trend=TRUE)
tfit &lt;- treat(tfit, lfc = 0.5)
pval &lt;- 0.05
fitSum &lt;- summary(decideTests(tfit, p.value = pval))
fitSum</code></pre>
<pre><code>       CD4vCD8 MonovNeu BcellvNK
Down      5072     9324    34803
NotSig  725611   712480   667559
Up        3202    12081    31523</code></pre>
<pre class="r"><code>bDat &lt;- getBiasDat(rownames(topTreat(tfit, coef = &quot;BcellvNK&quot;, num = 5000)), 
                   array.type = &quot;EPIC&quot;)</code></pre>
<pre><code>Loading required package: IlluminaHumanMethylationEPICanno.ilm10b4.hg19</code></pre>
<pre class="r"><code>p &lt;- ggplot(bDat, aes(x = avgbias, y = propDM)) +
    geom_point(shape = 1, size = 2) +
    geom_smooth() +
    labs(x = &quot;No. CpGs per gene (binned)&quot;,
         y = &quot;Prop. differential methylation&quot;) +
    theme_minimal() +
    theme(panel.grid = element_blank(),
          axis.line = element_line(colour = &quot;black&quot;))
p</code></pre>
<pre><code>`geom_smooth()` using method = &#39;loess&#39; and formula &#39;y ~ x&#39;</code></pre>
<p><img src="figure/05_exploreData.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Save figure for use in manuscript.</p>
<pre class="r"><code>fig &lt;- here(&quot;output/figures/Fig-1B.rds&quot;)
saveRDS(p, fig, compress = FALSE)</code></pre>
<p>Examine only the independent contrasts.</p>
<pre class="r"><code>dat &lt;- melt(fitSum[rownames(fitSum) != &quot;NotSig&quot;, ])
colnames(dat) &lt;- c(&quot;dir&quot;,&quot;comp&quot;,&quot;num&quot;)

p &lt;- ggplot(dat, aes(x = comp, y = num, fill = dir)) +
    geom_bar(stat = &quot;identity&quot;, position = &quot;dodge&quot;) +
    labs(x = &quot;Comparison&quot;, y = &quot;No. DM CpGs (FDR &lt; 0.05)&quot;, fill = &quot;Direction&quot;) +
    scale_fill_brewer(palette = &quot;Set1&quot;, direction = -1)
p</code></pre>
<p><img src="figure/05_exploreData.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Save figure for use in manuscript.</p>
<pre class="r"><code>fig &lt;- here(&quot;output/figures/Fig-4B.rds&quot;)
saveRDS(p, fig, compress = FALSE)</code></pre>
</div>
</div>
<div id="compare-performance-of-different-gene-set-testing-methods" class="section level1">
<h1>Compare performance of different gene set testing methods</h1>
<p>We compare the performance of the following gene set testing methods available for methylation arrays: a standard hypergeometric test (HGT), GOmeth from the <em>missMethyl</em> package, methylglm (mGLM), methylRRA-ORA (mRRA (ORA)) and methylRRA-GSEA (mRRA (GSEA)) from the <em>methylGSA</em> package and ebayGSEA from the <em>ChAMP</em> package.</p>
<p>We perform gene set testing on the results of all three blood cell type comparisons: B-cells vs. NK cells, CD4 T-cells vs. CD8 T-cells and monocytes vs. neutrophils, using all of the different gene set testing methods.</p>
<p>Gene set testing is performed for each comparison using GO categories, KEGG pathways and the BROAD MSigDB gene sets for the following methods: HGT, GOmeth, mGLM, mRRA (ORA) and mRRA (GSEA)</p>
<p>As gene set testing can only be performed using the BROAD MSigDB gene sets provided with the <em>ChAMP</em> package, the ebayGSEA method is only tested using the BROAD MSigDB gene sets for the three blood cell comparisons.</p>
<p>Furthermore, as the methylGSA methods do not work well with sets that only contain very few genes or very many genes, we only test sets with at least 5 genes and at most 5000 genes.</p>
<p>Firstly, the results of the statistical analysis of the three blood cell comparisons are saved as an RDS object.</p>
<pre class="r"><code>minsize &lt;- 5
maxsize &lt;- 5000

outDir &lt;- here::here(&quot;output/cache-intermediates&quot;)
if (!dir.exists(outDir)) dir.create(outDir)

outFile &lt;- here(&quot;data/cache-intermediates/blood.contrasts.rds&quot;)

if(!file.exists(outFile)){
  obj &lt;- NULL
  obj$tfit &lt;- tfit
  obj$maxsize &lt;- maxsize
  obj$minsize &lt;- minsize
  obj$mVals &lt;- mVals
  obj$targets &lt;- targets

  saveRDS(obj, file = outFile)

} </code></pre>
<p>As some of the methods take a considerable amount of time to perform the gene set testing analysis, we have created several scripts in order to run the analyses in parallel on a HPC. The code used to run all the gene set testing analyses using the different methods can be found in the <code>code/compare-methods</code> directory. It consists of four scripts: <code>genRunMethodJob.R</code>, <code>runChAMP.R</code>, <code>runMethylGSA.R</code> and <code>runMissMethyl.R</code>. The <code>genRunMethodJob.R</code> script creates and submits <a href="https://slurm.schedmd.com/overview.html">Slurm</a> job scripts that run the <code>runChAMP.R</code>, <code>runMethylGSA.R</code> and <code>runMissMethyl.R</code> scripts, for <em>each</em> gene set type, in parallel, on a HPC. The results of each job are saved as an RDS file named <code>{package}.{set}.rds</code> in the <code>output/compare-methods</code> directory. Once all analysis jobs are complete, all of the suvbsequent analyses in this document can be executed.</p>
<div id="load-output-for-all-methods" class="section level2">
<h2>Load output for all methods</h2>
<p>The results of all the gene set testing analyses, using all the different methods, for the different types of gene sets, are loaded into a <code>list</code> of <code>data.frames</code>. All of the <code>data.frames</code> in the <code>list</code> are then concatenated into a <code>tibble</code> for downstream analysis and plotting.</p>
<pre class="r"><code>inFiles &lt;- list.files(here(&quot;output/compare-methods&quot;), pattern = &quot;rds&quot;, 
                      full.names = TRUE)

res &lt;- lapply(inFiles, function(file){
    readRDS(file)
})

dat &lt;- as_tibble(dplyr::bind_rows(res))</code></pre>
</div>
<div id="explore-analysis-results" class="section level2">
<h2>Explore analysis results</h2>
<div id="go-categories" class="section level3">
<h3>GO categories</h3>
<p>Examine the performance of the different methods when gene set testing was performed on GO categories.</p>
<pre class="r"><code>ann &lt;- loadAnnotation(&quot;EPIC&quot;)
flatAnn &lt;- loadFlatAnnotation(ann)
cpgEgGo &lt;- cpgsEgGoFreqs(flatAnn)

cpgEgGo %&gt;% 
  group_by(GO) %&gt;%
  summarise(med = median(Freq)) -&gt; medCpgEgGo</code></pre>
<pre><code>`summarise()` ungrouping output (override with `.groups` argument)</code></pre>
<p>In order to examine whether the probe-number bias influenced the significantly enriched GO categories for the five different methods, we split the GO categories into bins based on the median number of CpGs per gene per GO category. We then calculated the proportion of significantly enriched GO categories in each bin for each of the three comparisons. As GOmeth explicitly corrects for this bias it showed very little trend. However mRRA (ORA) tended to have more significant GO categories when the genes had larger numbers of CpGs. The hypergeometric test and mRRA (GSEA) tended to have more significant GO categories with average numbers of CpGs associated per gene ranging from 27 - 44. The mGLM method appears to control for the probe-number bias of the array fairly well.</p>
<pre class="r"><code>dat %&gt;% filter(set == &quot;GO&quot;) %&gt;%
    filter(sub %in% c(&quot;n&quot;,&quot;c1&quot;)) %&gt;% 
    mutate(method = unname(dict[method])) %&gt;% 
    inner_join(medCpgEgGo, by = c(&quot;ID&quot; = &quot;GO&quot;)) -&gt; sub

bins &lt;- rbin_quantiles(sub, ID, med, bins = 12)
sub$bin &lt;- as.factor(findInterval(sub$med, bins$upper_cut))

binLabs &lt;- paste0(&quot;&lt;&quot;, bins$upper_cut)
names(binLabs) &lt;- levels(sub$bin)
binLabs[length(binLabs)] &lt;- gsub(&quot;&lt;&quot;, &quot;\u2265&quot;, binLabs[length(binLabs) - 1])
    
sub %&gt;% group_by(contrast, method, bin) %&gt;%
  summarise(prop = sum(pvalue &lt; 0.05)/n()) -&gt; pdat</code></pre>
<pre><code>`summarise()` regrouping output by &#39;contrast&#39;, &#39;method&#39; (override with `.groups` argument)</code></pre>
<pre class="r"><code>p &lt;- ggplot(pdat, aes(x = as.numeric(bin), y = prop, color = method)) +
  geom_line() +
  facet_wrap(vars(contrast), ncol = 3) +
  scale_x_continuous(breaks = as.numeric(levels(pdat$bin)), labels = binLabs) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, vjust = 1,
                                   size = 7),
        legend.position = &quot;bottom&quot;) +
  labs(x = &quot;Med. No. CpGs per Gene per GO Cat. (binned)&quot;,
       y = &quot;Prop. GO Cat. with p-value &lt; 0.05&quot;,
       colour = &quot;Method&quot;) +
    scale_color_manual(values = methodCols)
p</code></pre>
<p><img src="figure/05_exploreData.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>As we are comparing immune cells, we expect GO categories related to the immune system and its processes to be highly ranked. To evalue this, we downloaded all of the child terms for the GO category &quot;immune system process&quot; (<a href="GO:002376" class="uri">GO:002376</a>) from AminGO 2; <a href="http://amigo.geneontology.org/amigo/term/GO:0002376" class="uri">http://amigo.geneontology.org/amigo/term/GO:0002376</a>. We also examine results when top ranked 100 GO terms from RNA-seq analysis of the same cell type comparisons is used as &quot;truth&quot;.</p>
<pre class="r"><code>immuneGO &lt;- unique(read.csv(here(&quot;data/genesets/GO-immune-system-process.txt&quot;),
                            stringsAsFactors = FALSE, header = FALSE,
                            col.names = &quot;GOID&quot;))

rnaseqGO &lt;- readRDS(here(&quot;data/cache-rnaseq/RNAseq-GO.rds&quot;))
rnaseqGO %&gt;% group_by(contrast) %&gt;%
    mutate(rank = 1:n()) %&gt;%
    filter(rank &lt;= 100) -&gt; topGOSets

p &lt;- vector(&quot;list&quot;, length(unique(dat$contrast)))

for(i in 1:length(unique(dat$contrast))){
    cont &lt;- sort(unique(dat$contrast))[i]
    
    dat %&gt;% filter(set == &quot;GO&quot;) %&gt;%
        filter(sub %in% c(&quot;n&quot;,&quot;c1&quot;)) %&gt;%
        filter(contrast == cont) %&gt;%
        mutate(method = unname(dict[method])) %&gt;% 
        arrange(method, pvalue) %&gt;%
        group_by(method) %&gt;%
        mutate(rank = 1:n()) %&gt;%
        filter(rank &lt;= 100) %&gt;%
        mutate(csum = cumsum(ID %in% immuneGO$GOID)) %&gt;%
        mutate(truth = &quot;ISP Terms&quot;) -&gt; immuneSum
    
    dat %&gt;% filter(set == &quot;GO&quot;) %&gt;%
        filter(sub %in% c(&quot;n&quot;,&quot;c1&quot;)) %&gt;%
        filter(contrast == cont) %&gt;%
        mutate(method = unname(dict[method])) %&gt;% 
        arrange(method, pvalue) %&gt;%
        group_by(method) %&gt;%
        mutate(rank = 1:n()) %&gt;%
        filter(rank &lt;= 100) %&gt;%
        mutate(csum = cumsum(ID %in% topGOSets$ID[topGOSets$contrast %in% 
                                                         contrast])) %&gt;%
        mutate(truth = &quot;RNAseq Terms&quot;) -&gt; rnaseqSum
    
    truthSum &lt;- bind_rows(immuneSum, rnaseqSum)
    
    p[[i]] &lt;- ggplot(truthSum, aes(x = rank, y = csum, colour = method)) +
    geom_line() +
    facet_wrap(vars(truth), ncol = 2) +
    geom_vline(xintercept = 10, linetype = &quot;dotted&quot;) +
    labs(colour = &quot;Method&quot;, x = &quot;Rank&quot;, 
         y = &quot;Cumulative no. sets in truth&quot;) +
    theme(legend.position = &quot;bottom&quot;) +
    scale_color_manual(values = methodCols)
}

p[[1]] + ggtitle(sort(unique(dat$contrast))[1])</code></pre>
<p><img src="figure/05_exploreData.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p[[2]] + ggtitle(sort(unique(dat$contrast))[2])</code></pre>
<p><img src="figure/05_exploreData.Rmd/unnamed-chunk-17-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p[[3]] + ggtitle(sort(unique(dat$contrast))[3])</code></pre>
<p><img src="figure/05_exploreData.Rmd/unnamed-chunk-17-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>Save figure for use in manuscript.</p>
<pre class="r"><code>fig &lt;- here(&quot;output/figures/Fig-4C.rds&quot;)
saveRDS(p[[1]], fig, compress = FALSE)

fig &lt;- here(&quot;output/figures/SFig-5A.rds&quot;)
saveRDS(p[[2]], fig, compress = FALSE)

fig &lt;- here(&quot;output/figures/SFig-6A.rds&quot;)
saveRDS(p[[3]], fig, compress = FALSE)</code></pre>
<p>Examine what the top 10 ranked gene sets are and how many genes they contain, for each method and comparison.</p>
<pre class="r"><code>terms &lt;- missMethyl:::.getGO()$idTable
nGenes &lt;- rownames_to_column(data.frame(n = sapply(missMethyl:::.getGO()$idList, 
                                                   length)), 
                             var = &quot;ID&quot;)

dat %&gt;% filter(set == &quot;GO&quot;) %&gt;%
    filter(sub %in% c(&quot;n&quot;,&quot;c1&quot;)) %&gt;% 
    mutate(method = unname(dict[method])) %&gt;% 
    arrange(contrast, method, pvalue) %&gt;%
    group_by(contrast, method) %&gt;%
    mutate(FDR = p.adjust(pvalue, method = &quot;BH&quot;)) %&gt;%
    mutate(rank = 1:n()) %&gt;%
    filter(rank &lt;= 10) %&gt;% 
    inner_join(terms, by = c(&quot;ID&quot; = &quot;GOID&quot;)) %&gt;%
    inner_join(nGenes) -&gt; sub</code></pre>
<pre><code>Joining, by = &quot;ID&quot;</code></pre>
<pre class="r"><code>p &lt;- vector(&quot;list&quot;, length(unique(sub$contrast)))
truthPal &lt;- scales::hue_pal()(4)
names(truthPal) &lt;- c(&quot;Both&quot;, &quot;ISP&quot;, &quot;Neither&quot;, &quot;RNAseq&quot;)

for(i in 1:length(p)){
    cont &lt;- sort(unique(sub$contrast))[i]
    sub %&gt;% filter(contrast == cont) %&gt;%
        arrange(method, -rank) %&gt;%
        ungroup() %&gt;%
        mutate(idx = as.factor(1:n())) -&gt; tmp
    
    setLabs &lt;- substr(tmp$TERM, 1, 40)
    names(setLabs) &lt;- tmp$idx
    
    tmp %&gt;% mutate(rna = ID %in% topGOSets$ID[topGOSets$contrast %in% cont],
                   isp = ID %in% immuneGO$GOID,
                   both = rna + isp,
                   col = ifelse(both == 2, &quot;Both&quot;, 
                                ifelse(both == 1 &amp; rna == 1, &quot;RNAseq&quot;,
                                       ifelse(both == 1 &amp; isp == 1, 
                                              &quot;ISP&quot;, &quot;Neither&quot;)))) %&gt;%
        mutate(col = factor(col, 
                            levels = c(&quot;Both&quot;, &quot;ISP&quot;, &quot;RNAseq&quot;, 
                                       &quot;Neither&quot;))) -&gt; tmp
    
    p[[i]] &lt;- ggplot(tmp, aes(x = -log10(FDR), y = idx, colour = col)) +
        geom_point(aes(size = n), alpha = 0.7) +
        scale_size(limits = c(min(sub$n), max(sub$n))) +
        facet_wrap(vars(method), ncol = 2, scales = &quot;free&quot;) +
        scale_y_discrete(labels = setLabs) +
        scale_colour_manual(values = truthPal) +
        labs(y = &quot;&quot;, size = &quot;No. genes&quot;, colour = &quot;In truth set&quot;) +
        theme(axis.text.y = element_text(size = 6),
              axis.text.x = element_text(size = 6),
              legend.box = &quot;horizontal&quot;,
              legend.margin = margin(0, 0, 0, 0, unit = &quot;lines&quot;),
              panel.spacing.x = unit(1, &quot;lines&quot;)) +
        coord_cartesian(xlim = c(-log10(0.99), -log10(10^-80))) +
        geom_vline(xintercept = -log10(0.05), linetype = &quot;dashed&quot;) +
        ggtitle(cont)
}

shift_legend(p[[1]], plot = TRUE, pos = &quot;left&quot;)</code></pre>
<p><img src="figure/05_exploreData.Rmd/unnamed-chunk-20-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>shift_legend(p[[2]], plot = TRUE, pos = &quot;left&quot;)</code></pre>
<p><img src="figure/05_exploreData.Rmd/unnamed-chunk-20-2.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>shift_legend(p[[3]], plot = TRUE, pos = &quot;left&quot;)</code></pre>
<p><img src="figure/05_exploreData.Rmd/unnamed-chunk-20-3.png" width="768" style="display: block; margin: auto;" /></p>
<p>Save figure for use in manuscript.</p>
<pre class="r"><code>fig &lt;- here(&quot;output/figures/Fig-4D.rds&quot;)
saveRDS(shift_legend(p[[1]] + theme(plot.title = element_blank()),
                     pos = &quot;left&quot;), 
        fig, compress = FALSE)

fig &lt;- here(&quot;output/figures/SFig-5B.rds&quot;)
saveRDS(shift_legend(p[[2]] + theme(plot.title = element_blank()),
                     pos = &quot;left&quot;), 
        fig, compress = FALSE)

fig &lt;- here(&quot;output/figures/SFig-6B.rds&quot;)
saveRDS(shift_legend(p[[3]] + theme(plot.title = element_blank()),
                     pos = &quot;left&quot;), 
        fig, compress = FALSE)</code></pre>
<p>P-value histograms for the different methods for all contrasts on GO categories.</p>
<pre class="r"><code>dat %&gt;% filter(set == &quot;GO&quot;) %&gt;%
    filter(sub %in% c(&quot;n&quot;,&quot;c1&quot;)) %&gt;%
    mutate(method = unname(dict[method])) -&gt; subDat 
    
ggplot(subDat, aes(pvalue, fill = method)) +
    geom_histogram(binwidth = 0.025) +
    facet_grid(cols = vars(contrast), rows = vars(method)) +
    theme(legend.position = &quot;bottom&quot;) +
    labs(x = &quot;P-value&quot;, y = &quot;Frequency&quot;, fill = &quot;Method&quot;) +
    scale_fill_manual(values = methodCols)</code></pre>
<p><img src="figure/05_exploreData.Rmd/unnamed-chunk-22-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="compare-gometh-results-using-different-dm-cpg-significance-cutoffs" class="section level4">
<h4>Compare GOmeth results using different DM CpG significance cutoffs</h4>
<p>As the results of GOmeth depend on the list of significant CpGs provided as input to the function, we explored the effect of selecting &quot;significant&quot; CpGs in different ways on the gene set testing performance of GOmeth.</p>
<pre class="r"><code>dat %&gt;% filter(set == &quot;GO&quot;) %&gt;%
    filter(grepl(&quot;mmethyl&quot;, method)) %&gt;%
    mutate(method = unname(dict[method])) %&gt;%
    arrange(contrast, method, pvalue) %&gt;%
    group_by(contrast, method, sub) %&gt;%
    mutate(csum = cumsum(ID %in% immuneGO$GOID)) %&gt;% 
    mutate(rank = 1:n()) %&gt;%
    mutate(cut = ifelse(sub == &quot;c1&quot;, &quot;Top 5000&quot;, 
                        ifelse(sub == &quot;c2&quot;, &quot;Top 10000&quot;,
                               ifelse(sub == &quot;p1&quot;, &quot;FDR &lt; 0.01&quot;, &quot;FDR &lt; 0.05&quot;)))) %&gt;%
    filter(rank &lt;= 100) -&gt; sub

p &lt;- ggplot(sub, aes(x = rank, y = csum, colour = cut)) +
    geom_line() +
    facet_wrap(vars(contrast, method), ncol=2, nrow = 3, scales = &quot;free&quot;) +
    geom_vline(xintercept = 10, linetype = &quot;dotted&quot;) +
    labs(colour = &quot;Sig. select&quot;, x = &quot;Rank&quot;, y = &quot;Cumulative no. immune sets&quot;) +
    theme(legend.position = &quot;bottom&quot;)
p</code></pre>
<p><img src="figure/05_exploreData.Rmd/unnamed-chunk-23-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dat %&gt;% filter(set == &quot;GO&quot;) %&gt;%
    filter(grepl(&quot;mmethyl&quot;, method)) %&gt;%
    mutate(method = unname(dict[method])) %&gt;%
    arrange(contrast, method, pvalue) %&gt;%
    group_by(contrast, method, sub) %&gt;%
    mutate(csum = cumsum(ID %in% topGOSets$ID[topGOSets$contrast %in% 
                                                  contrast])) %&gt;%
    mutate(rank = 1:n()) %&gt;%
    mutate(cut = ifelse(sub == &quot;c1&quot;, &quot;Top 5000&quot;, 
                        ifelse(sub == &quot;c2&quot;, &quot;Top 10000&quot;,
                               ifelse(sub == &quot;p1&quot;, &quot;FDR &lt; 0.01&quot;, &quot;FDR &lt; 0.05&quot;)))) %&gt;%
    filter(rank &lt;= 100) -&gt; sub

p &lt;- ggplot(sub, aes(x = rank, y = csum, colour = cut)) +
    geom_line() +
    facet_wrap(vars(contrast, method), ncol=2, nrow = 3, scales = &quot;free&quot;) +
    geom_vline(xintercept = 10, linetype = &quot;dotted&quot;) +
    labs(colour = &quot;Sig. select&quot;, x = &quot;Rank&quot;, 
         y = glue(&quot;Cumulative no. RNAseq sets&quot;)) +
    theme(legend.position = &quot;bottom&quot;)
p</code></pre>
<p><img src="figure/05_exploreData.Rmd/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="kegg-pathways" class="section level3">
<h3>KEGG pathways</h3>
<p>Now test KEGG pathways with at least 5 genes and 5000 at most.</p>
<p>Again, as we are comparing immune cells we expect pathways from the following categories to be highly ranked: Immune system, Immune disease, Signal transduction, Signaling molecules and interaction; <a href="https://www.genome.jp/kegg/pathway.html" class="uri">https://www.genome.jp/kegg/pathway.html</a>.</p>
<pre class="r"><code>immuneKEGG &lt;- read.csv(here(&quot;data/genesets/kegg-immune-related-pathways.csv&quot;), 
                            stringsAsFactors = FALSE, header = FALSE, 
                            col.names = c(&quot;ID&quot;,&quot;pathway&quot;))
immuneKEGG$PID &lt;- paste0(&quot;path:hsa0&quot;,immuneKEGG$ID)

dat %&gt;% filter(set == &quot;KEGG&quot;) %&gt;%
    filter(sub %in% c(&quot;n&quot;,&quot;c1&quot;)) %&gt;% 
    mutate(method = unname(dict[method])) %&gt;%
    arrange(contrast, method, pvalue) %&gt;%
    group_by(contrast, method) %&gt;%
    mutate(csum = cumsum(ID %in% immuneKEGG$PID)) %&gt;%
    mutate(rank = 1:n()) %&gt;%
    filter(rank &lt;= 100) -&gt; sub

p1 &lt;- ggplot(sub, aes(x = rank, y = csum, colour = method)) +
    geom_line() +
    facet_wrap(vars(contrast), ncol=3) +
    geom_vline(xintercept = 10, linetype = &quot;dotted&quot;) +
    labs(colour = &quot;Method&quot;, x = &quot;Rank&quot;, y = &quot;Cumulative no. immune sets&quot;) +
    theme(legend.position = &quot;bottom&quot;) +
    scale_color_manual(values = methodCols)</code></pre>
<p>Examine results when top ranked 100 KEGG pathways from RNA-seq analysis of the same cell type comparisons is used as &quot;truth&quot;.</p>
<pre class="r"><code>rnaseqKEGG &lt;- readRDS(here(&quot;data/cache-rnaseq/RNAseq-KEGG.rds&quot;))
rnaseqKEGG %&gt;% group_by(contrast) %&gt;%
    mutate(rank = 1:n()) %&gt;%
    filter(rank &lt;= 100) -&gt; topKEGG

dat %&gt;% filter(set == &quot;KEGG&quot;) %&gt;%
    filter(sub %in% c(&quot;n&quot;,&quot;c1&quot;)) %&gt;%
    mutate(method = unname(dict[method])) %&gt;%
    arrange(contrast, method, pvalue) %&gt;%
    group_by(contrast, method) %&gt;%
    mutate(csum = cumsum(ID %in% topKEGG$PID[topKEGG$contrast %in% contrast])) %&gt;%
    mutate(rank = 1:n()) %&gt;%
    filter(rank &lt;= 100) -&gt; sub

p2 &lt;- ggplot(sub, aes(x = rank, y = csum, colour = method)) +
    geom_line() +
    facet_wrap(vars(contrast), ncol=3, scales = &quot;free_y&quot;) +
    geom_vline(xintercept = 10, linetype = &quot;dotted&quot;) +
    labs(colour = &quot;Method&quot;, x = &quot;Rank&quot;, y = &quot;Cumulative no. RNAseq sets&quot;) +
    theme(legend.position = &quot;bottom&quot;) + 
    scale_color_manual(values = methodCols)

p3 &lt;- p1 / p2 + 
    plot_layout(guides = &quot;collect&quot;) &amp; 
    theme(legend.position = &quot;bottom&quot;)
p3</code></pre>
<p><img src="figure/05_exploreData.Rmd/unnamed-chunk-26-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>Save figure for use in manuscript.</p>
<pre class="r"><code>fig &lt;- here(&quot;output/figures/SFig-7A.rds&quot;)
saveRDS(p1, fig, compress = FALSE)

fig &lt;- here(&quot;output/figures/SFig-7B.rds&quot;)
saveRDS(p2, fig, compress = FALSE)</code></pre>
<p>Examine what the top 10 ranked gene sets are and how many genes they contain, for each method and comparison.</p>
<pre class="r"><code>terms &lt;- missMethyl:::.getKEGG()$idTable
nGenes &lt;- rownames_to_column(data.frame(n = sapply(missMethyl:::.getKEGG()$idList, 
                                                   length)), 
                             var = &quot;ID&quot;)

dat %&gt;% filter(set == &quot;KEGG&quot;) %&gt;%
    filter(sub %in% c(&quot;n&quot;,&quot;c1&quot;)) %&gt;% 
    mutate(method = unname(dict[method])) %&gt;%
    arrange(contrast, method, pvalue) %&gt;%
    group_by(contrast, method) %&gt;%
    mutate(FDR = p.adjust(pvalue, method = &quot;BH&quot;)) %&gt;%
    mutate(rank = 1:n()) %&gt;%
    filter(rank &lt;= 10) %&gt;% 
    inner_join(terms, by = c(&quot;ID&quot; = &quot;PathwayID&quot;)) %&gt;%
    inner_join(nGenes) -&gt; sub</code></pre>
<pre><code>Joining, by = &quot;ID&quot;</code></pre>
<pre class="r"><code>p &lt;- vector(&quot;list&quot;, length(unique(sub$contrast)))
for(i in 1:length(p)){
    cont &lt;- sort(unique(sub$contrast))[i]
    sub %&gt;% filter(contrast == cont) %&gt;%
        arrange(method, -rank) %&gt;%
        ungroup() %&gt;%
        mutate(idx = as.factor(1:n())) -&gt; tmp
    
    setLabs &lt;- substr(tmp$Description, 1, 40)
    names(setLabs) &lt;- tmp$idx
    
    tmp %&gt;% mutate(rna = ID %in% topKEGG$PID[topKEGG$contrast %in% cont],
                   isp = ID %in% immuneKEGG$PID,
                   both = rna + isp,
                   col = ifelse(both == 2, &quot;Both&quot;, 
                                ifelse(both == 1 &amp; rna == 1, &quot;RNAseq&quot;,
                                       ifelse(both == 1 &amp; isp == 1, 
                                              &quot;ISP&quot;, &quot;Neither&quot;)))) %&gt;%
        mutate(col = factor(col, 
                            levels = c(&quot;Both&quot;, &quot;ISP&quot;, &quot;RNAseq&quot;, 
                                       &quot;Neither&quot;))) -&gt; tmp
    
    p[[i]] &lt;- ggplot(tmp, aes(x = -log10(FDR), y = idx, colour = col)) +
        geom_point(aes(size = n), alpha = 0.7) +
        scale_size(limits = c(min(sub$n), max(sub$n))) +
        facet_wrap(vars(method), ncol = 2, scales = &quot;free&quot;) +
        scale_y_discrete(labels = setLabs) +
        scale_colour_manual(values = truthPal) +
        labs(y = &quot;&quot;, size = &quot;No. genes&quot;, colour = &quot;In truth set&quot;) +
        theme(axis.text.y = element_text(size = 6),
              axis.text.x = element_text(size = 6),
              legend.box = &quot;horizontal&quot;,
              legend.margin = margin(0, 0, 0, 0, unit = &quot;lines&quot;),
              panel.spacing.x = unit(1, &quot;lines&quot;)) +
        coord_cartesian(xlim = c(-log10(0.99), -log10(10^-80))) +
        geom_vline(xintercept = -log10(0.05), linetype = &quot;dashed&quot;) +
        ggtitle(cont)
}

shift_legend(p[[1]], plot = TRUE, pos = &quot;left&quot;)</code></pre>
<p><img src="figure/05_exploreData.Rmd/unnamed-chunk-29-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>shift_legend(p[[2]], plot = TRUE, pos = &quot;left&quot;)</code></pre>
<p><img src="figure/05_exploreData.Rmd/unnamed-chunk-29-2.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>shift_legend(p[[3]], plot = TRUE, pos = &quot;left&quot;)</code></pre>
<p><img src="figure/05_exploreData.Rmd/unnamed-chunk-29-3.png" width="768" style="display: block; margin: auto;" /></p>
<p>Save figure for use in manuscript.</p>
<pre class="r"><code>fig &lt;- here(&quot;output/figures/SFig-7C.rds&quot;)
saveRDS(shift_legend(p[[1]] + theme(plot.title = element_blank()),
                     pos = &quot;left&quot;), 
        fig, compress = FALSE)

fig &lt;- here(&quot;output/figures/SFig-7D.rds&quot;)
saveRDS(shift_legend(p[[2]] + theme(plot.title = element_blank()),
                     pos = &quot;left&quot;), 
        fig, compress = FALSE)

fig &lt;- here(&quot;output/figures/SFig-7E.rds&quot;)
saveRDS(shift_legend(p[[3]] + theme(plot.title = element_blank()),
                     pos = &quot;left&quot;), 
        fig, compress = FALSE)</code></pre>
<p>P-value histograms for the different methods for all contrasts on KEGG pathways.</p>
<pre class="r"><code>dat %&gt;% filter(set == &quot;KEGG&quot;) %&gt;%
    filter(sub %in% c(&quot;n&quot;,&quot;c1&quot;)) %&gt;%
    mutate(method = unname(dict[method])) -&gt; subDat 
    
ggplot(subDat, aes(pvalue, fill = method)) +
    geom_histogram(binwidth = 0.025) +
    facet_grid(cols = vars(contrast), rows = vars(method)) +
    theme(legend.position = &quot;bottom&quot;) +
    labs(x = &quot;P-value&quot;, y = &quot;Frequency&quot;, fill = &quot;Method&quot;) +
    scale_fill_manual(values = methodCols)</code></pre>
<p><img src="figure/05_exploreData.Rmd/unnamed-chunk-31-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="compare-gometh-results-using-different-dm-cpg-significance-cutoffs-1" class="section level4">
<h4>Compare GOmeth results using different DM CpG significance cutoffs</h4>
<p>As the results of GOmeth depend on the list of significant CpGs provided as input to the function, we explored the effect of selecting &quot;significant&quot; CpGs in different ways on the gene set testing performance of GOmeth.</p>
<pre class="r"><code>dat %&gt;% filter(set == &quot;KEGG&quot;) %&gt;%
    filter(grepl(&quot;mmethyl&quot;, method)) %&gt;% 
    mutate(method = unname(dict[method])) %&gt;%
    arrange(contrast, method, pvalue) %&gt;%
    group_by(contrast, method, sub) %&gt;%
    mutate(csum = cumsum(ID %in% immuneKEGG$PID)) %&gt;% 
    mutate(rank = 1:n()) %&gt;%
    mutate(cut = ifelse(sub == &quot;c1&quot;, &quot;Top 5000&quot;, 
                        ifelse(sub == &quot;c2&quot;, &quot;Top 10000&quot;,
                               ifelse(sub == &quot;p1&quot;, &quot;FDR &lt; 0.01&quot;, &quot;FDR &lt; 0.05&quot;)))) %&gt;%
    filter(rank &lt;= 100) -&gt; sub

p &lt;- ggplot(sub, aes(x = rank, y = csum, colour = cut)) +
    geom_line() +
    facet_wrap(vars(contrast, method), ncol=2, nrow = 3, scales = &quot;free&quot;) +
    geom_vline(xintercept = 10, linetype = &quot;dotted&quot;) +
    labs(colour = &quot;Sig. select&quot;, x = &quot;Rank&quot;, y = &quot;Cumulative no. immune sets&quot;) +
    theme(legend.position = &quot;bottom&quot;)
p</code></pre>
<p><img src="figure/05_exploreData.Rmd/unnamed-chunk-32-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dat %&gt;% filter(set == &quot;KEGG&quot;) %&gt;%
    filter(grepl(&quot;mmethyl&quot;, method)) %&gt;% 
    mutate(method = unname(dict[method])) %&gt;%
    arrange(contrast, method, pvalue) %&gt;%
    group_by(contrast, method, sub) %&gt;%
    mutate(csum = cumsum(ID %in% topKEGG$PID[topKEGG$contrast %in% contrast])) %&gt;%
    mutate(rank = 1:n()) %&gt;%
    mutate(cut = ifelse(sub == &quot;c1&quot;, &quot;Top 5000&quot;, 
                        ifelse(sub == &quot;c2&quot;, &quot;Top 10000&quot;,
                               ifelse(sub == &quot;p1&quot;, &quot;FDR &lt; 0.01&quot;, &quot;FDR &lt; 0.05&quot;)))) %&gt;%
    filter(rank &lt;= 100) -&gt; sub

p &lt;- ggplot(sub, aes(x = rank, y = csum, colour = cut)) +
    geom_line() +
    facet_wrap(vars(contrast, method), ncol=2, nrow = 3, scales = &quot;free&quot;) +
    geom_vline(xintercept = 10, linetype = &quot;dotted&quot;) +
    labs(colour = &quot;Sig. select&quot;, x = &quot;Rank&quot;, 
         y = glue(&quot;Cumulative no. RNAseq sets&quot;)) +
    theme(legend.position = &quot;bottom&quot;)
p</code></pre>
<p><img src="figure/05_exploreData.Rmd/unnamed-chunk-33-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="broad-gene-sets" class="section level3">
<h3>BROAD gene sets</h3>
<p>As the ebGSEA method from the ChAMP package only allows testing of their in-built database of Broad Institute gene sets, we will compare to other methods by testing these same gene sets. Using the top 100 ranked gene sets as identified by Belinda's <code>gsaseq</code> analysis of the corresponding RNAseq data as &quot;truth&quot;.</p>
<pre class="r"><code>rnaseqBROAD &lt;- readRDS(here(&quot;data/cache-rnaseq/RNAseq-BROAD-GSA.rds&quot;))
rnaseqBROAD %&gt;% group_by(contrast) %&gt;%
    mutate(rank = 1:n()) %&gt;%
    filter(rank &lt;= 100) -&gt; topBROAD

p &lt;- vector(&quot;list&quot;, length(unique(dat$contrast)))

for(i in 1:length(unique(dat$contrast))){
    cont &lt;- sort(unique(dat$contrast))[i]
    
    dat %&gt;% filter(set == &quot;BROAD&quot;) %&gt;%
        filter(sub %in% c(&quot;n&quot;,&quot;c1&quot;)) %&gt;%
        filter(contrast == cont) %&gt;%
        mutate(method = unname(dict[method])) %&gt;% 
    arrange(contrast, method, pvalue) %&gt;%
    group_by(contrast, method) %&gt;%
    mutate(csum = cumsum(ID %in% topBROAD$ID[topBROAD$contrast %in% contrast])) %&gt;%
    mutate(rank = 1:n()) %&gt;%
    filter(rank &lt;= 100) -&gt; sub
    
    p[[i]] &lt;- ggplot(sub, aes(x = rank, y = csum, colour = method)) +
    geom_line() +
    geom_vline(xintercept = 10, linetype = &quot;dotted&quot;) +
    labs(colour = &quot;Method&quot;, x = &quot;Rank&quot;, y = &quot;Cumulative no. RNAseq sets&quot;) +
    theme(legend.position = &quot;bottom&quot;) +
    scale_color_manual(values = methodCols) +
        ggtitle(cont)
}

p[[1]]</code></pre>
<p><img src="figure/05_exploreData.Rmd/unnamed-chunk-34-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p[[2]]</code></pre>
<p><img src="figure/05_exploreData.Rmd/unnamed-chunk-34-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p[[3]]</code></pre>
<p><img src="figure/05_exploreData.Rmd/unnamed-chunk-34-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>Save figure for use in manuscript.</p>
<pre class="r"><code>fig &lt;- here(&quot;output/figures/SFig-8A.rds&quot;)
saveRDS(p[[1]], fig, compress = FALSE)

fig &lt;- here(&quot;output/figures/SFig-9A.rds&quot;)
saveRDS(p[[2]], fig, compress = FALSE)

fig &lt;- here(&quot;output/figures/SFig-10A.rds&quot;)
saveRDS(p[[3]], fig, compress = FALSE)</code></pre>
<p>Examine what the top 10 ranked gene sets are and how many genes they contain, for each method and comparison.</p>
<pre class="r"><code>data(PathwayList)
nGenes &lt;- rownames_to_column(data.frame(n = sapply(PathwayList, 
                                                   length)), 
                             var = &quot;ID&quot;)

dat %&gt;% filter(set == &quot;BROAD&quot;) %&gt;%
    filter(sub %in% c(&quot;n&quot;,&quot;c1&quot;)) %&gt;% 
    mutate(method = unname(dict[method])) %&gt;% 
    arrange(contrast, method, pvalue) %&gt;%
    group_by(contrast, method) %&gt;%
    mutate(FDR = p.adjust(pvalue, method = &quot;BH&quot;)) %&gt;%
    mutate(rank = 1:n()) %&gt;%
    filter(rank &lt;= 10) %&gt;% 
    inner_join(nGenes) -&gt; sub</code></pre>
<pre><code>Joining, by = &quot;ID&quot;</code></pre>
<pre class="r"><code>p &lt;- vector(&quot;list&quot;, length(unique(sub$contrast)))
for(i in 1:length(p)){
    cont &lt;- sort(unique(sub$contrast))[i]
    sub %&gt;% filter(contrast == cont) %&gt;%
        arrange(method, -rank) %&gt;%
        ungroup() %&gt;%
        mutate(idx = as.factor(1:n())) -&gt; tmp
    
    setLabs &lt;- substr(tmp$ID, 1, 30)
    names(setLabs) &lt;- tmp$idx
    
    tmp %&gt;% mutate(rna = ID %in% topBROAD$ID[topGOSets$contrast %in% cont],
                   col = ifelse(rna == 1, &quot;RNAseq&quot;, &quot;None&quot;)) %&gt;%
        mutate(col = factor(col, levels = c(&quot;RNAseq&quot;, &quot;None&quot;))) -&gt; tmp
    
    p[[i]] &lt;- ggplot(tmp, aes(x = -log10(FDR), y = idx, colour = col)) +
        geom_point(aes(size = n), alpha = 0.7) +
        scale_size(limits = c(min(sub$n), max(sub$n))) +
        facet_wrap(vars(method), ncol = 2, scales = &quot;free&quot;) +
        scale_y_discrete(labels = setLabs) +
        scale_color_manual(values = scales::hue_pal()(4)[3:4]) +
        labs(y = &quot;&quot;, size = &quot;No. genes&quot;, colour = &quot;In truth set&quot;) +
        theme(axis.text.y = element_text(size = 6),
              axis.text.x = element_text(size = 6),
              legend.box = &quot;horizontal&quot;,
              legend.margin = margin(0, 0, 0, 0, unit = &quot;lines&quot;),
              panel.spacing.x = unit(1, &quot;lines&quot;)) +
        coord_cartesian(xlim = c(-log10(0.99), -log10(10^-100))) +
        geom_vline(xintercept = -log10(0.05), linetype = &quot;dashed&quot;) +
        ggtitle(cont)
}

shift_legend(p[[1]], plot = TRUE, pos = &quot;left&quot;)</code></pre>
<p><img src="figure/05_exploreData.Rmd/unnamed-chunk-37-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>shift_legend(p[[2]], plot = TRUE, pos = &quot;left&quot;)</code></pre>
<p><img src="figure/05_exploreData.Rmd/unnamed-chunk-37-2.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>shift_legend(p[[3]], plot = TRUE, pos = &quot;left&quot;)</code></pre>
<p><img src="figure/05_exploreData.Rmd/unnamed-chunk-37-3.png" width="768" style="display: block; margin: auto;" /></p>
<p>Save figure for use in manuscript.</p>
<pre class="r"><code>fig &lt;- here(&quot;output/figures/SFig-8B.rds&quot;)
saveRDS(shift_legend(p[[1]] + theme(plot.title = element_blank()),
                     pos = &quot;left&quot;), 
        fig, compress = FALSE)

fig &lt;- here(&quot;output/figures/SFig-9B.rds&quot;)
saveRDS(shift_legend(p[[2]] + theme(plot.title = element_blank()),
                     pos = &quot;left&quot;), 
        fig, compress = FALSE)

fig &lt;- here(&quot;output/figures/SFig-10B.rds&quot;)
saveRDS(shift_legend(p[[3]] + theme(plot.title = element_blank()),
                     pos = &quot;left&quot;), 
        fig, compress = FALSE)</code></pre>
<p>P-value histograms for the different methods for all contrasts on BROAD gene sets.</p>
<pre class="r"><code>dat %&gt;% filter(set == &quot;BROAD&quot;) %&gt;%
    filter(sub %in% c(&quot;n&quot;,&quot;c1&quot;)) %&gt;%
    mutate(method = unname(dict[method])) -&gt; subDat 
    
ggplot(subDat, aes(pvalue, fill = method)) +
    geom_histogram(binwidth = 0.025) +
    facet_grid(cols = vars(contrast), rows = vars(method)) +
    theme(legend.position = &quot;bottom&quot;) +
    labs(x = &quot;P-value&quot;, y = &quot;Frequency&quot;, fill = &quot;Method&quot;) +
    scale_fill_manual(values = methodCols)</code></pre>
<p><img src="figure/05_exploreData.Rmd/unnamed-chunk-39-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="compare-gometh-results-using-different-dm-cpg-significance-cutoffs-2" class="section level4">
<h4>Compare GOmeth results using different DM CpG significance cutoffs</h4>
<p>As the results of GOmeth depend on the list of significant CpGs provided as input to the function, we explored the effect of selecting &quot;significant&quot; CpGs in different ways on the gene set testing performance of GOmeth.</p>
<pre class="r"><code>dat %&gt;% filter(set == &quot;BROAD&quot;) %&gt;%
    filter(grepl(&quot;mmethyl&quot;, method)) %&gt;% 
    mutate(method = unname(dict[method])) %&gt;%
    arrange(contrast, method, pvalue) %&gt;%
    group_by(contrast, method, sub) %&gt;%
    mutate(csum = cumsum(ID %in% topBROAD$ID)) %&gt;% 
    mutate(rank = 1:n()) %&gt;%
    mutate(cut = ifelse(sub == &quot;c1&quot;, &quot;Top 5000&quot;, 
                        ifelse(sub == &quot;c2&quot;, &quot;Top 10000&quot;,
                               ifelse(sub == &quot;p1&quot;, &quot;FDR &lt; 0.01&quot;, &quot;FDR &lt; 0.05&quot;)))) %&gt;%
    filter(rank &lt;= 100) -&gt; sub

p &lt;- ggplot(sub, aes(x = rank, y = csum, colour = cut)) +
    geom_line() +
    facet_wrap(vars(contrast, method), ncol=2, nrow = 3, scales = &quot;free&quot;) +
    geom_vline(xintercept = 10, linetype = &quot;dotted&quot;) +
    labs(colour = &quot;Sig. select&quot;, x = &quot;Rank&quot;, y = &quot;Cumulative no. immune sets&quot;) +
    theme(legend.position = &quot;bottom&quot;)
p</code></pre>
<p><img src="figure/05_exploreData.Rmd/unnamed-chunk-40-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dat %&gt;% filter(set == &quot;BROAD&quot;) %&gt;%
    filter(grepl(&quot;mmethyl&quot;, method)) %&gt;% 
    mutate(method = unname(dict[method])) %&gt;%
    arrange(contrast, method, pvalue) %&gt;%
    group_by(contrast, method, sub) %&gt;%
    mutate(csum = cumsum(ID %in% topBROAD$ID)) %&gt;% 
    mutate(rank = 1:n()) %&gt;%
    mutate(cut = ifelse(sub == &quot;c1&quot;, &quot;Top 5000&quot;, 
                        ifelse(sub == &quot;c2&quot;, &quot;Top 10000&quot;,
                               ifelse(sub == &quot;p1&quot;, &quot;FDR &lt; 0.01&quot;, &quot;FDR &lt; 0.05&quot;)))) %&gt;%
    filter(rank &lt;= 100) -&gt; sub

p &lt;- ggplot(sub, aes(x = rank, y = csum, colour = cut)) +
    geom_line() +
    facet_wrap(vars(contrast, method), ncol=2, nrow = 3, scales = &quot;free&quot;) +
    geom_vline(xintercept = 10, linetype = &quot;dotted&quot;) +
    labs(colour = &quot;Sig. select&quot;, x = &quot;Rank&quot;, y = &quot;Cumulative no. immune sets&quot;) +
    theme(legend.position = &quot;bottom&quot;)
p</code></pre>
<p><img src="figure/05_exploreData.Rmd/unnamed-chunk-41-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.3 (2020-02-29)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Mojave 10.14.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8

attached base packages:
 [1] grid      stats4    parallel  stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] IlluminaHumanMethylationEPICanno.ilm10b4.hg19_0.6.0
 [2] lemon_0.4.5                                        
 [3] ChAMPdata_2.18.0                                   
 [4] patchwork_1.0.1                                    
 [5] rbin_0.2.0                                         
 [6] forcats_0.5.0                                      
 [7] stringr_1.4.0                                      
 [8] dplyr_1.0.0                                        
 [9] purrr_0.3.4                                        
[10] readr_1.3.1                                        
[11] tidyr_1.1.0                                        
[12] tibble_3.0.3                                       
[13] tidyverse_1.3.0                                    
[14] glue_1.4.1                                         
[15] ggplot2_3.3.2                                      
[16] missMethyl_1.20.4                                  
[17] gridExtra_2.3                                      
[18] reshape2_1.4.4                                     
[19] limma_3.42.2                                       
[20] paletteer_1.2.0                                    
[21] minfi_1.32.0                                       
[22] bumphunter_1.28.0                                  
[23] locfit_1.5-9.4                                     
[24] iterators_1.0.12                                   
[25] foreach_1.5.0                                      
[26] Biostrings_2.54.0                                  
[27] XVector_0.26.0                                     
[28] SummarizedExperiment_1.16.1                        
[29] DelayedArray_0.12.3                                
[30] BiocParallel_1.20.1                                
[31] matrixStats_0.56.0                                 
[32] Biobase_2.46.0                                     
[33] GenomicRanges_1.38.0                               
[34] GenomeInfoDb_1.22.1                                
[35] IRanges_2.20.2                                     
[36] S4Vectors_0.24.4                                   
[37] BiocGenerics_0.32.0                                
[38] here_0.1                                           
[39] workflowr_1.6.2                                    

loaded via a namespace (and not attached):
  [1] readxl_1.3.1                                      
  [2] backports_1.1.8                                   
  [3] BiocFileCache_1.10.2                              
  [4] plyr_1.8.6                                        
  [5] splines_3.6.3                                     
  [6] digest_0.6.25                                     
  [7] htmltools_0.5.0                                   
  [8] GO.db_3.10.0                                      
  [9] fansi_0.4.1                                       
 [10] magrittr_1.5                                      
 [11] memoise_1.1.0                                     
 [12] annotate_1.64.0                                   
 [13] modelr_0.1.8                                      
 [14] askpass_1.1                                       
 [15] siggenes_1.60.0                                   
 [16] prettyunits_1.1.1                                 
 [17] colorspace_1.4-1                                  
 [18] rvest_0.3.5                                       
 [19] blob_1.2.1                                        
 [20] rappdirs_0.3.1                                    
 [21] haven_2.3.1                                       
 [22] BiasedUrn_1.07                                    
 [23] xfun_0.15                                         
 [24] jsonlite_1.7.0                                    
 [25] crayon_1.3.4                                      
 [26] RCurl_1.98-1.2                                    
 [27] genefilter_1.68.0                                 
 [28] GEOquery_2.54.1                                   
 [29] IlluminaHumanMethylationEPICmanifest_0.3.0        
 [30] survival_3.2-3                                    
 [31] ruv_0.9.7.1                                       
 [32] gtable_0.3.0                                      
 [33] zlibbioc_1.32.0                                   
 [34] Rhdf5lib_1.8.0                                    
 [35] HDF5Array_1.14.4                                  
 [36] scales_1.1.1                                      
 [37] DBI_1.1.0                                         
 [38] rngtools_1.5                                      
 [39] Rcpp_1.0.5                                        
 [40] xtable_1.8-4                                      
 [41] progress_1.2.2                                    
 [42] bit_1.1-15.2                                      
 [43] mclust_5.4.6                                      
 [44] preprocessCore_1.48.0                             
 [45] httr_1.4.2                                        
 [46] RColorBrewer_1.1-2                                
 [47] ellipsis_0.3.1                                    
 [48] farver_2.0.3                                      
 [49] pkgconfig_2.0.3                                   
 [50] reshape_0.8.8                                     
 [51] XML_3.99-0.3                                      
 [52] dbplyr_1.4.4                                      
 [53] labeling_0.3                                      
 [54] tidyselect_1.1.0                                  
 [55] rlang_0.4.7                                       
 [56] later_1.1.0.1                                     
 [57] AnnotationDbi_1.48.0                              
 [58] cellranger_1.1.0                                  
 [59] munsell_0.5.0                                     
 [60] tools_3.6.3                                       
 [61] cli_2.0.2                                         
 [62] generics_0.0.2                                    
 [63] RSQLite_2.2.0                                     
 [64] broom_0.7.0                                       
 [65] evaluate_0.14                                     
 [66] yaml_2.2.1                                        
 [67] rematch2_2.1.2                                    
 [68] org.Hs.eg.db_3.10.0                               
 [69] knitr_1.29                                        
 [70] bit64_0.9-7.1                                     
 [71] fs_1.4.2                                          
 [72] beanplot_1.2                                      
 [73] scrime_1.3.5                                      
 [74] methylumi_2.32.0                                  
 [75] nlme_3.1-148                                      
 [76] doRNG_1.8.2                                       
 [77] whisker_0.4                                       
 [78] nor1mix_1.3-0                                     
 [79] xml2_1.3.2                                        
 [80] biomaRt_2.42.1                                    
 [81] compiler_3.6.3                                    
 [82] rstudioapi_0.11                                   
 [83] curl_4.3                                          
 [84] reprex_0.3.0                                      
 [85] statmod_1.4.34                                    
 [86] stringi_1.4.6                                     
 [87] IlluminaHumanMethylation450kanno.ilmn12.hg19_0.6.0
 [88] GenomicFeatures_1.38.2                            
 [89] lattice_0.20-41                                   
 [90] Matrix_1.2-18                                     
 [91] IlluminaHumanMethylation450kmanifest_0.4.0        
 [92] multtest_2.42.0                                   
 [93] vctrs_0.3.2                                       
 [94] pillar_1.4.6                                      
 [95] lifecycle_0.2.0                                   
 [96] cowplot_1.0.0                                     
 [97] data.table_1.12.8                                 
 [98] bitops_1.0-6                                      
 [99] httpuv_1.5.4                                      
[100] rtracklayer_1.46.0                                
[101] R6_2.4.1                                          
[102] promises_1.1.1                                    
[103] codetools_0.2-16                                  
[104] MASS_7.3-51.6                                     
[105] assertthat_0.2.1                                  
[106] rhdf5_2.30.1                                      
[107] openssl_1.4.2                                     
[108] rprojroot_1.3-2                                   
[109] withr_2.2.0                                       
[110] GenomicAlignments_1.22.1                          
[111] Rsamtools_2.2.3                                   
[112] GenomeInfoDbData_1.2.2                            
[113] mgcv_1.8-31                                       
[114] hms_0.5.3                                         
[115] quadprog_1.5-8                                    
[116] base64_2.0                                        
[117] rmarkdown_2.3                                     
[118] DelayedMatrixStats_1.8.0                          
[119] illuminaio_0.28.0                                 
[120] git2r_0.27.1                                      
[121] lubridate_1.7.9                                   </code></pre>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
