<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jovana Maksimovic, Alicia Oshlack and Belinda Phipson" />

<meta name="date" content="2021-04-06" />

<title>Gene set testing for Illumina HumanMethylation Arrays</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-sm-12 col-md-4 col-lg-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-sm-12 col-md-8 col-lg-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">methyl-geneset-testing</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/Oshlack/methyl-geneset-testing">
    <span class="fas fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Gene set testing for Illumina HumanMethylation Arrays</h1>
<h3 class="subtitle">Evaluating the functionality and performance of GOregion</h3>
<h4 class="author">Jovana Maksimovic, Alicia Oshlack and Belinda Phipson</h4>
<h4 class="date">April 06, 2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-04-06
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>methyl-geneset-testing/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it's best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200302code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200302)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200302code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200302)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsatlassianpetermacorgaubitbucketscmosmethylgenesettestingtreeea1abcfc172cffd7330ac366980d1979a0411e44targetblankea1abcfa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/tree/ea1abcfc172cffd7330ac366980d1979a0411e44" target="_blank">ea1abcf</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsatlassianpetermacorgaubitbucketscmosmethylgenesettestingtreeea1abcfc172cffd7330ac366980d1979a0411e44targetblankea1abcfa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/tree/ea1abcfc172cffd7330ac366980d1979a0411e44" target="_blank">ea1abcf</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/figures.nb.html
    Ignored:    code/.DS_Store
    Ignored:    code/.Rhistory
    Ignored:    code/.job/
    Ignored:    code/old/
    Ignored:    data/.DS_Store
    Ignored:    data/annotations/
    Ignored:    data/cache-intermediates/
    Ignored:    data/cache-region/
    Ignored:    data/cache-rnaseq/
    Ignored:    data/cache-runtime/
    Ignored:    data/datasets/.DS_Store
    Ignored:    data/datasets/GSE110554-data.RData
    Ignored:    data/datasets/GSE120854/
    Ignored:    data/datasets/GSE120854_RAW.tar
    Ignored:    data/datasets/GSE135446-data.RData
    Ignored:    data/datasets/GSE135446/
    Ignored:    data/datasets/GSE135446_RAW.tar
    Ignored:    data/datasets/GSE45459-data.RData
    Ignored:    data/datasets/GSE45459_Matrix_signal_intensities.txt
    Ignored:    data/datasets/GSE45460/
    Ignored:    data/datasets/GSE45460_RAW.tar
    Ignored:    data/datasets/GSE95460_RAW.tar
    Ignored:    data/datasets/GSE95460_RAW/
    Ignored:    data/datasets/GSE95462-data.RData
    Ignored:    data/datasets/GSE95462/
    Ignored:    data/datasets/GSE95462_RAW/
    Ignored:    data/datasets/SRP100803/
    Ignored:    data/datasets/SRP125125/.DS_Store
    Ignored:    data/datasets/SRP125125/SRR6298*/
    Ignored:    data/datasets/SRP125125/SRR_Acc_List.txt
    Ignored:    data/datasets/SRP125125/SRR_Acc_List_Full.txt
    Ignored:    data/datasets/SRP125125/SraRunTable.txt
    Ignored:    data/datasets/SRP125125/multiqc_data/
    Ignored:    data/datasets/SRP125125/multiqc_report.html
    Ignored:    data/datasets/SRP125125/quants/
    Ignored:    data/datasets/SRP166862/
    Ignored:    data/datasets/SRP217468/
    Ignored:    data/datasets/TCGA.BRCA.rds
    Ignored:    data/datasets/TCGA.KIRC.rds
    Ignored:    data/misc/
    Ignored:    output/.DS_Store
    Ignored:    output/FDR-analysis/
    Ignored:    output/compare-methods/
    Ignored:    output/figures/
    Ignored:    output/methylgsa-params/
    Ignored:    output/outputs.tar.gz
    Ignored:    output/random-cpg-sims/
    Ignored:    output/test/

Unstaged changes:
    Modified:   analysis/04_expressionGenesetsBcells.Rmd
    Modified:   analysis/05_compareMethods.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/07_regionAnalysisBcells.Rmd</code>) and HTML (<code>docs/07_regionAnalysisBcells.html</code>) files. If you've configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/ea1abcfc172cffd7330ac366980d1979a0411e44/analysis/07_regionAnalysisBcells.Rmd" target="_blank">ea1abcf</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2021-04-06
</td>
<td>
wflow_publish(c(&quot;analysis/07_regionAnalysisBcells.Rmd&quot;))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/a9a3a86439630816564e1ec217381de691d4d4f0/docs/07_regionAnalysisBcells.html" target="_blank">a9a3a86</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2021-04-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/98431578adb30dcaa808cddbfe752948d7eb5385/analysis/07_regionAnalysisBcells.Rmd" target="_blank">9843157</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2021-04-06
</td>
<td>
wflow_publish(c(&quot;analysis/index.Rmd&quot;, &quot;analysis/07_regionAnalysisBcells.Rmd&quot;,
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(here)
library(ChAMP)
library(minfi)
library(paletteer)
library(limma)
library(BiocParallel)
library(reshape2)
library(DMRcate)
library(missMethyl)
library(ggplot2)
library(glue)
library(UpSetR)
library(dplyr)
library(patchwork)
library(tibble)
library(grid)
library(ggupset)
library(ggpubr)
library(TxDb.Hsapiens.UCSC.hg19.knownGene)
source(here(&quot;code/utility.R&quot;))</code></pre>
<div id="load-data" class="section level1">
<h1>Load data</h1>
<p>We are using publicly available EPIC data <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE45459">GSE45459</a> generated from developing human B-cells. The data is normalised and filtered (bad probes, multi-mapping probes, SNP probes, sex chromosomes).</p>
<pre class="r"><code># load data
dataFile &lt;- here(&quot;data/datasets/GSE45459-data.RData&quot;)

if(file.exists(dataFile)){
    # load processed data and sample information
    load(dataFile)
    
} else {
    
    raw &lt;- read.delim(here(&quot;data/datasets/GSE45459_Matrix_signal_intensities.txt&quot;),
                      row.names = &quot;ID_REF&quot;)
    M &lt;- raw[,grepl(&quot;Methylated&quot;, colnames(raw))]
    U &lt;- raw[,grepl(&quot;Unmethylated&quot;, colnames(raw))]
    detP &lt;- raw[,grepl(&quot;Pval&quot;, colnames(raw))]
    targets &lt;- as.data.frame(strsplit2(colnames(detP),&quot;_&quot;)[,1:2])
    colnames(targets) &lt;- c(&quot;ID&quot;,&quot;Stage&quot;)
    mSet &lt;- MethylSet(Meth = as.matrix(M), Unmeth = as.matrix(U),
                      annotation = c(array = &quot;IlluminaHumanMethylation450k&quot;, 
                                     annotation = &quot;ilmn12.hg19&quot;))
    colnames(mSet) &lt;- targets$ID
    grRaw &lt;- minfi::mapToGenome(mSet)

    #normalise
    normGr &lt;- preprocessQuantile(grRaw)
    #filter
    fltGr &lt;- filterQual(normGr = normGr, detP = detP)
    fltGr &lt;- filterProbes(fltGr)
    
    save(detP, normGr, fltGr, targets, file = dataFile)
    
}</code></pre>
</div>
<div id="statistical-analysis" class="section level1">
<h1>Statistical analysis</h1>
<p>Compare several sets of sorted immune cells. Consider results significant at FDR &lt; 0.05 and delta beta ~ 10% (~ lfc = 0.5).</p>
<pre class="r"><code>mVals &lt;- getM(fltGr)
bVals &lt;- getBeta(fltGr)</code></pre>
<pre class="r"><code>Stage &lt;- factor(targets$Stage)
design &lt;- model.matrix(~0+Stage)

fit &lt;- lmFit(mVals, design)
cont.matrix &lt;- makeContrasts(S1vS2=StageS1-StageS2,
                             S2vS3=StageS2-StageS3,
                             levels=design)
fit2 &lt;- contrasts.fit(fit, cont.matrix)
fit2 &lt;- eBayes(fit2, robust=TRUE, trend=TRUE)
tfit &lt;- treat(fit2, lfc = 0.5)
pval &lt;- 0.05
fitSum &lt;- summary(decideTests(tfit, p.value = pval))
fitSum</code></pre>
<pre><code>        S1vS2  S2vS3
Down     1083      0
NotSig 366973 369336
Up       2065    785</code></pre>
</div>
<div id="find-differentially-methylated-regions" class="section level1">
<h1>Find differentially methylated regions</h1>
<p>Identify differentially methylated regions using the <em>DMRcate</em> package.</p>
<pre class="r"><code>outDir &lt;- here::here(&quot;data/cache-region&quot;)
if (!dir.exists(outDir)) dir.create(outDir)

outFile &lt;- here(&quot;data/cache-region/dmrcate-GSE45459-results.rds&quot;)

if(!file.exists(outFile)){
  dmrList &lt;- vector(&quot;list&quot;, ncol(cont.matrix))

  for(i in 1:ncol(cont.matrix)){
    cpgAnn &lt;- cpg.annotate(&quot;array&quot;, mVals, what = &quot;M&quot;, arraytype = &quot;450K&quot;,
                           analysis.type = &quot;differential&quot;, design = design, 
                           contrasts = TRUE, cont.matrix = cont.matrix, 
                           coef = colnames(fitSum)[i])
    dmrList[[i]] &lt;- extractRanges(dmrcate(cpgAnn))

  }
  
  names(dmrList) &lt;- colnames(cont.matrix)
  saveRDS(dmrList, file = outFile)
  
} else {
  dmrList &lt;- readRDS(outFile)
  
}</code></pre>
<pre class="r"><code>dat &lt;- rownames_to_column(melt(sapply(dmrList, length)), var = &quot;contrast&quot;)
dat$filtering = &quot;before&quot;

meanDiff &lt;- 0
noCpgs &lt;- 2

for(i in 1:length(dmrList)){
    keep &lt;- (abs(dmrList[[i]]$meandiff) &gt; meanDiff &amp; dmrList[[i]]$no.cpgs &gt;=noCpgs)
    dat &lt;- bind_rows(dat, data.frame(contrast = names(dmrList)[i],
                                value = length(dmrList[[i]][keep, ]),
                                filtering = &quot;after&quot;, 
                                stringsAsFactors = FALSE))
}

p &lt;- ggplot(dat, aes(x = contrast, y = value, fill = filtering)) +
    geom_bar(stat = &quot;identity&quot;, position = &quot;dodge&quot;) +
    labs(x = &quot;Contrast&quot;, y = &quot;No. DMRs&quot;, fill = &quot;Filtering&quot;)
p</code></pre>
<p><img src="figure/07_regionAnalysisBcells.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/a9a3a86439630816564e1ec217381de691d4d4f0/docs/figure/07_regionAnalysisBcells.Rmd/unnamed-chunk-6-1.png" target="_blank">a9a3a86</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2021-04-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Save figure for use in manuscript.</p>
<pre class="r"><code>outDir &lt;- here::here(&quot;output/figures&quot;)
if (!dir.exists(outDir)) dir.create(outDir)

fig &lt;- here(&quot;output/figures/Fig-5B.rds&quot;)
saveRDS(p, fig, compress = FALSE)</code></pre>
<div id="go-analysis-of-dmrs" class="section level2">
<h2>GO analysis of DMRs</h2>
<p>Run GO analysis on the differentially methylated regions (DMRs) identified by DMRcate for each of the contrasts.</p>
<pre class="r"><code>outFile &lt;- here(&quot;data/cache-region/dmrcate-GSE45459-go.rds&quot;)
anno &lt;- loadAnnotation(arrayType = &quot;450k&quot;)
txdb &lt;- TxDb.Hsapiens.UCSC.hg19.knownGene
hg19Genes &lt;- GenomicFeatures::genes(txdb)</code></pre>
<pre><code>  403 genes were dropped because they have exons located on both strands
  of the same reference sequence or on more than one reference sequence,
  so cannot be represented by a single genomic range.
  Use &#39;single.strand.genes.only=FALSE&#39; to get all the genes in a
  GRangesList object, or use suppressMessages() to suppress this message.</code></pre>
<pre class="r"><code>dmrGo &lt;- NULL

if(!file.exists(outFile)){
    for(i in 1:length(dmrList)){
        
        keep &lt;- (abs(dmrList[[i]]$meandiff) &gt; meanDiff &amp; dmrList[[i]]$no.cpgs &gt;= noCpgs)
        
        overlaps &lt;- findOverlaps(hg19Genes, dmrList[[i]][keep, ], 
                                 minoverlap = 1)
        sigGenes &lt;- hg19Genes$gene_id[from(overlaps)]
        tmp &lt;- topGO(goana(sigGenes, universe = hg19Genes$gene_id), 
                     number = Inf)
        tmp &lt;- rownames_to_column(tmp, var = &quot;GO&quot;)[, c(&quot;GO&quot;, &quot;P.DE&quot;)]
        tmp$method &lt;- &quot;goana&quot;
        tmp$contrast &lt;- colnames(cont.matrix)[i]
        dmrGo &lt;- bind_rows(dmrGo, tmp)
  
        tmp &lt;- topGSA(goregion(dmrList[[i]][keep, ], anno = anno, 
                               prior.prob = FALSE, array.type = &quot;450k&quot;), 
                      number = Inf)
        tmp &lt;- rownames_to_column(tmp, var = &quot;GO&quot;)[, c(&quot;GO&quot;, &quot;P.DE&quot;)]
        tmp$method &lt;- &quot;goregion-hgt&quot;
        tmp$contrast &lt;- colnames(cont.matrix)[i]
        dmrGo &lt;- bind_rows(dmrGo, tmp)
        
        tmp &lt;- topGSA(goregion(dmrList[[i]][keep, ], anno = anno, 
                               array.type = &quot;450k&quot;, plot.bias = FALSE), 
                      number = Inf)
        tmp &lt;- rownames_to_column(tmp, var = &quot;GO&quot;)[, c(&quot;GO&quot;, &quot;P.DE&quot;)]
        tmp$method &lt;- &quot;goregion-gometh&quot;
        tmp$contrast &lt;- colnames(cont.matrix)[i]
        dmrGo &lt;- bind_rows(dmrGo, tmp)
        
        tmp &lt;- topGSA(gometh(rownames(topTreat(tfit, coef = i, num = 5000)), 
                             anno = anno, array.type = &quot;450k&quot;), number = Inf)
        tmp &lt;- rownames_to_column(tmp, var = &quot;GO&quot;)[, c(&quot;GO&quot;, &quot;P.DE&quot;)]
        tmp$method &lt;- &quot;gometh-probe-top&quot;
        tmp$contrast &lt;- colnames(cont.matrix)[i]
        dmrGo &lt;- bind_rows(dmrGo, tmp)
        
        tmp &lt;- topGSA(gometh(rownames(topTreat(tfit, coef = i, num = Inf, 
                                               p.value = pval)), anno = anno, 
                             array.type = &quot;450k&quot;), number = Inf)
        tmp &lt;- rownames_to_column(tmp, var = &quot;GO&quot;)[, c(&quot;GO&quot;, &quot;P.DE&quot;)]
        tmp$method &lt;- &quot;gometh-probe-fdr&quot;
        tmp$contrast &lt;- colnames(cont.matrix)[i]
        dmrGo &lt;- bind_rows(dmrGo, tmp)
    }
    
    saveRDS(dmrGo, file = outFile)
    
} else {
    dmrGo &lt;- readRDS(outFile)
    
}</code></pre>
<div id="probe-bias-in-dmrs" class="section level3">
<h3>Probe bias in DMRs</h3>
<pre class="r"><code>bDat &lt;- vector(&quot;list&quot;, length(dmrList))
cpgs &lt;- GRanges(seqnames = anno$chr, 
                ranges = IRanges(start = anno$pos, end = anno$pos), 
                strand = anno$strand, 
                name = anno$Name)

for(i in 1:length(dmrList)){
    keep &lt;- (abs(dmrList[[i]]$meandiff) &gt; meanDiff &amp; dmrList[[i]]$no.cpgs &gt;=noCpgs)
    overlaps &lt;- findOverlaps(cpgs, dmrList[[i]][keep, ])
    dmrCpgs &lt;- cpgs$name[from(overlaps)]
    bDat[[i]] &lt;- getBiasDat(dmrCpgs, array.type = &quot;450k&quot;,
                            anno = anno)
    
}</code></pre>
<pre><code>All input CpGs are used for testing.
All input CpGs are used for testing.</code></pre>
<pre class="r"><code>p &lt;- vector(&quot;list&quot;, length(bDat))

for(i in 1:length(p)){
p[[i]] &lt;- ggplot(bDat[[i]], aes(x = avgbias, y = propDM)) +
    geom_point(shape = 1, size = 2) +
    geom_smooth() +
    labs(x = &quot;No. CpGs per gene (binned)&quot;,
         y = &quot;Prop. differential methylation&quot;) +
    theme_minimal() +
    theme(panel.grid = element_blank(),
          axis.line = element_line(colour = &quot;black&quot;))
}

(p[[1]] + labs(title = names(dmrList)[1]) | 
        p[[2]] + labs(title = names(dmrList)[2]))</code></pre>
<pre><code>`geom_smooth()` using method = &#39;loess&#39; and formula &#39;y ~ x&#39;
`geom_smooth()` using method = &#39;loess&#39; and formula &#39;y ~ x&#39;</code></pre>
<p><img src="figure/07_regionAnalysisBcells.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/a9a3a86439630816564e1ec217381de691d4d4f0/docs/figure/07_regionAnalysisBcells.Rmd/unnamed-chunk-10-1.png" target="_blank">a9a3a86</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2021-04-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Save figure for use in manuscript.</p>
<pre class="r"><code>fig &lt;- here(&quot;output/figures/SFig-11A.rds&quot;)
saveRDS(p[[1]], fig, compress = FALSE)

fig &lt;- here(&quot;output/figures/SFig-12A.rds&quot;)
saveRDS(p[[2]], fig, compress = FALSE)</code></pre>
</div>
<div id="compare-goregion-with-other-approaches" class="section level3">
<h3>Compare GOregion with other approaches</h3>
<pre class="r"><code>immuneGO &lt;- unique(read.csv(here(&quot;data/genesets/GO-immune-system-process.txt&quot;), 
                            stringsAsFactors = FALSE, header = FALSE, 
                            col.names = &quot;GOID&quot;))

rnaseqGO &lt;- readRDS(here(&quot;data/cache-rnaseq/RNAseq-GSE45460-GO.rds&quot;))
rnaseqGO %&gt;% group_by(contrast) %&gt;%
    mutate(rank = 1:n()) %&gt;%
    dplyr::filter(rank &lt;= 100) -&gt; topGOSets

dmrGo %&gt;% arrange(contrast, method, P.DE) %&gt;%
    dplyr::filter(method %in% c(&quot;goana&quot;, &quot;goregion-gometh&quot;)) %&gt;%
    mutate(method = unname((dict[method]))) %&gt;%
    group_by(contrast, method) %&gt;%
    mutate(rank = 1:n()) %&gt;%
    dplyr::filter(rank &lt;= 100) -&gt; dat

p &lt;- vector(&quot;list&quot;, length(unique(dat$contrast)))

for(i in 1:length(p)){
    cont &lt;- sort(unique(dat$contrast))[i]
    
    dat %&gt;% dplyr::filter(contrast == cont) %&gt;%
        arrange(method, P.DE) %&gt;%
        group_by(method) %&gt;%
        mutate(csum = cumsum(GO %in% immuneGO$GOID)) %&gt;%
        mutate(truth = &quot;ISP Terms&quot;) -&gt; immuneSum
    
    dat %&gt;% dplyr::filter(contrast == cont) %&gt;%
        arrange(method, P.DE) %&gt;%
        group_by(method) %&gt;%
        mutate(csum = cumsum(GO %in% topGOSets$ID[topGOSets$contrast %in% 
                                                      contrast])) %&gt;%
        mutate(truth = &quot;RNAseq Terms&quot;) -&gt; rnaseqSum
    
    truthSum &lt;- bind_rows(immuneSum, rnaseqSum)
    
    p[[i]] &lt;- ggplot(truthSum, aes(x = rank, y = csum, colour = method)) +
    geom_line() +
        facet_wrap(vars(truth)) +
    geom_vline(xintercept = 10, linetype = &quot;dotted&quot;) +
    labs(colour = &quot;Method&quot;, x = &quot;Rank&quot;, 
         y = glue(&quot;Cumulative no. RNAseq sets&quot;)) +
    theme(legend.position = &quot;bottom&quot;) +
    scale_color_manual(values = methodCols) +
        ggtitle(cont)
}

p[[1]]</code></pre>
<p><img src="figure/07_regionAnalysisBcells.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/a9a3a86439630816564e1ec217381de691d4d4f0/docs/figure/07_regionAnalysisBcells.Rmd/unnamed-chunk-12-1.png" target="_blank">a9a3a86</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2021-04-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>p[[2]]</code></pre>
<p><img src="figure/07_regionAnalysisBcells.Rmd/unnamed-chunk-12-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-2">
Past versions of unnamed-chunk-12-2.png
</button>
</p>
<div id="fig-unnamed-chunk-12-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/a9a3a86439630816564e1ec217381de691d4d4f0/docs/figure/07_regionAnalysisBcells.Rmd/unnamed-chunk-12-2.png" target="_blank">a9a3a86</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2021-04-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Save figure for use in manuscript.</p>
<pre class="r"><code>fig &lt;- here(&quot;output/figures/Fig-5C.rds&quot;)
saveRDS(p[[1]], fig, compress = FALSE)

fig &lt;- here(&quot;output/figures/SFig-11B.rds&quot;)
saveRDS(p[[2]], fig, compress = FALSE)</code></pre>
<p>Terms NOT in top 100 &quot;truth&quot; GO categories for Stage 2 versus Stage 3.</p>
<pre class="r"><code>terms &lt;- missMethyl:::.getGO()$idTable
dat %&gt;% dplyr::filter(contrast == &quot;S2vS3&quot;) %&gt;%
    arrange(method, P.DE) %&gt;%
    group_by(method) %&gt;%
    dplyr::filter(!(GO %in% topGOSets$ID[topGOSets$contrast %in% &quot;S2vS3&quot;])) %&gt;%
    inner_join(terms, by = c(&quot;GO&quot; = &quot;GOID&quot;)) %&gt;% data.frame</code></pre>
<pre><code>            GO         P.DE   method contrast rank ONTOLOGY
1   GO:0042100 1.871886e-07 GOregion    S2vS3    8       BP
2   GO:0051251 2.361494e-06 GOregion    S2vS3   14       BP
3   GO:0030888 2.919664e-06 GOregion    S2vS3   15       BP
4   GO:0046651 3.954230e-06 GOregion    S2vS3   17       BP
5   GO:0032943 4.354933e-06 GOregion    S2vS3   18       BP
6   GO:1903037 6.441511e-06 GOregion    S2vS3   19       BP
7   GO:0002696 1.172146e-05 GOregion    S2vS3   22       BP
8   GO:0070661 1.353316e-05 GOregion    S2vS3   23       BP
9   GO:0050867 1.810620e-05 GOregion    S2vS3   24       BP
10  GO:0007159 1.986455e-05 GOregion    S2vS3   25       BP
11  GO:0030890 3.240636e-05 GOregion    S2vS3   26       BP
12  GO:0050870 3.576705e-05 GOregion    S2vS3   27       BP
13  GO:0001772 4.064320e-05 GOregion    S2vS3   28       CC
14  GO:0050864 4.433351e-05 GOregion    S2vS3   29       BP
15  GO:0002695 5.137454e-05 GOregion    S2vS3   30       BP
16  GO:0050670 5.356112e-05 GOregion    S2vS3   31       BP
17  GO:0032944 5.631251e-05 GOregion    S2vS3   32       BP
18  GO:0050862 7.195065e-05 GOregion    S2vS3   33       BP
19  GO:1903039 9.189073e-05 GOregion    S2vS3   36       BP
20  GO:0070663 1.034611e-04 GOregion    S2vS3   37       BP
21  GO:0022407 1.072928e-04 GOregion    S2vS3   38       BP
22  GO:0071356 1.256294e-04 GOregion    S2vS3   40       BP
23  GO:0045619 1.304243e-04 GOregion    S2vS3   42       BP
24  GO:0002790 1.633360e-04 GOregion    S2vS3   44       BP
25  GO:0045580 1.667860e-04 GOregion    S2vS3   45       BP
26  GO:0022409 2.457559e-04 GOregion    S2vS3   50       BP
27  GO:0034612 2.549419e-04 GOregion    S2vS3   51       BP
28  GO:0031295 2.606491e-04 GOregion    S2vS3   52       BP
29  GO:0050866 2.726386e-04 GOregion    S2vS3   53       BP
30  GO:0050856 2.796375e-04 GOregion    S2vS3   54       BP
31  GO:0031294 2.826779e-04 GOregion    S2vS3   55       BP
32  GO:0023061 3.216904e-04 GOregion    S2vS3   59       BP
33  GO:0050851 4.230004e-04 GOregion    S2vS3   61       BP
34  GO:0008597 4.785944e-04 GOregion    S2vS3   62       MF
35  GO:0002429 5.059610e-04 GOregion    S2vS3   63       BP
36  GO:0002757 5.059610e-04 GOregion    S2vS3   64       BP
37  GO:0050671 5.798341e-04 GOregion    S2vS3   66       BP
38  GO:0032946 5.952831e-04 GOregion    S2vS3   67       BP
39  GO:0050857 7.496028e-04 GOregion    S2vS3   68       BP
40  GO:0032735 8.410294e-04 GOregion    S2vS3   69       BP
41  GO:0050871 9.054775e-04 GOregion    S2vS3   71       BP
42  GO:0002024 9.402317e-04 GOregion    S2vS3   72       BP
43  GO:2000106 9.626609e-04 GOregion    S2vS3   73       BP
44  GO:0070665 1.012407e-03 GOregion    S2vS3   74       BP
45  GO:0051209 1.046106e-03 GOregion    S2vS3   75       BP
46  GO:0051283 1.072826e-03 GOregion    S2vS3   76       BP
47  GO:0051282 1.122256e-03 GOregion    S2vS3   77       BP
48  GO:0043605 1.177601e-03 GOregion    S2vS3   78       BP
49  GO:0019722 1.214673e-03 GOregion    S2vS3   79       BP
50  GO:0051208 1.229678e-03 GOregion    S2vS3   80       BP
51  GO:0051250 1.325522e-03 GOregion    S2vS3   81       BP
52  GO:2000275 1.541309e-03 GOregion    S2vS3   83       BP
53  GO:0060453 1.629881e-03 GOregion    S2vS3   84       BP
54  GO:0090277 1.808017e-03 GOregion    S2vS3   86       BP
55  GO:0038063 1.820508e-03 GOregion    S2vS3   87       BP
56  GO:0051281 1.823752e-03 GOregion    S2vS3   88       BP
57  GO:0002791 1.839700e-03 GOregion    S2vS3   89       BP
58  GO:2000510 2.047714e-03 GOregion    S2vS3   91       BP
59  GO:0031587 2.159552e-03 GOregion    S2vS3   92       BP
60  GO:0051651 2.188118e-03 GOregion    S2vS3   93       BP
61  GO:0002443 2.205549e-03 GOregion    S2vS3   94       BP
62  GO:0002253 2.297889e-03 GOregion    S2vS3   95       BP
63  GO:2001190 2.471676e-03 GOregion    S2vS3   96       BP
64  GO:0015870 2.518229e-03 GOregion    S2vS3   97       BP
65  GO:0061526 2.518229e-03 GOregion    S2vS3   98       BP
66  GO:0002366 2.547678e-03 GOregion    S2vS3   99       BP
67  GO:0042100 1.165126e-07      HGT    S2vS3   13       BP
68  GO:0065008 2.489250e-07      HGT    S2vS3   16       BP
69  GO:0030888 3.845382e-07      HGT    S2vS3   18       BP
70  GO:0051251 4.312737e-07      HGT    S2vS3   19       BP
71  GO:0032879 6.277272e-07      HGT    S2vS3   22       BP
72  GO:0065007 6.299087e-07      HGT    S2vS3   23       BP
73  GO:0050789 1.415538e-06      HGT    S2vS3   27       BP
74  GO:1903037 2.546568e-06      HGT    S2vS3   30       BP
75  GO:0002696 2.968925e-06      HGT    S2vS3   32       BP
76  GO:0048519 3.509679e-06      HGT    S2vS3   33       BP
77  GO:0001772 3.986710e-06      HGT    S2vS3   34       CC
78  GO:0050794 4.488290e-06      HGT    S2vS3   37       BP
79  GO:0022407 4.945817e-06      HGT    S2vS3   40       BP
80  GO:0050867 4.985423e-06      HGT    S2vS3   41       BP
81  GO:0030890 5.394265e-06      HGT    S2vS3   42       BP
82  GO:0060341 5.505027e-06      HGT    S2vS3   44       BP
83  GO:0050864 8.371982e-06      HGT    S2vS3   47       BP
84  GO:0007159 9.019335e-06      HGT    S2vS3   49       BP
85  GO:0046651 1.119393e-05      HGT    S2vS3   50       BP
86  GO:0032943 1.257385e-05      HGT    S2vS3   51       BP
87  GO:0050862 1.764620e-05      HGT    S2vS3   56       BP
88  GO:0050870 1.862370e-05      HGT    S2vS3   57       BP
89  GO:0007275 1.871765e-05      HGT    S2vS3   58       BP
90  GO:0048522 2.000668e-05      HGT    S2vS3   60       BP
91  GO:0031256 2.270777e-05      HGT    S2vS3   63       CC
92  GO:0048523 2.364600e-05      HGT    S2vS3   64       BP
93  GO:0002695 2.900358e-05      HGT    S2vS3   65       BP
94  GO:0045619 3.042762e-05      HGT    S2vS3   67       BP
95  GO:0002790 3.116909e-05      HGT    S2vS3   68       BP
96  GO:0048856 3.301259e-05      HGT    S2vS3   69       BP
97  GO:0070661 3.343787e-05      HGT    S2vS3   70       BP
98  GO:0031295 3.394475e-05      HGT    S2vS3   71       BP
99  GO:0019722 3.415117e-05      HGT    S2vS3   72       BP
100 GO:0045580 3.928101e-05      HGT    S2vS3   73       BP
101 GO:0031294 4.154762e-05      HGT    S2vS3   74       BP
102 GO:1903039 4.191872e-05      HGT    S2vS3   75       BP
103 GO:0022409 4.448138e-05      HGT    S2vS3   76       BP
104 GO:0023061 4.525688e-05      HGT    S2vS3   78       BP
105 GO:0051209 4.905469e-05      HGT    S2vS3   79       BP
106 GO:0032589 5.045763e-05      HGT    S2vS3   80       CC
107 GO:0051283 5.209781e-05      HGT    S2vS3   81       BP
108 GO:0050851 5.320690e-05      HGT    S2vS3   82       BP
109 GO:0048518 5.702087e-05      HGT    S2vS3   83       BP
110 GO:0051282 5.865712e-05      HGT    S2vS3   84       BP
111 GO:0032502 5.973700e-05      HGT    S2vS3   85       BP
112 GO:0051208 6.977448e-05      HGT    S2vS3   87       BP
113 GO:0005515 8.055737e-05      HGT    S2vS3   88       MF
114 GO:0050866 8.087841e-05      HGT    S2vS3   89       BP
115 GO:0048731 1.013109e-04      HGT    S2vS3   90       BP
116 GO:0050856 1.052406e-04      HGT    S2vS3   91       BP
117 GO:0050857 1.068240e-04      HGT    S2vS3   92       BP
118 GO:0006109 1.073701e-04      HGT    S2vS3   93       BP
119 GO:0050670 1.116954e-04      HGT    S2vS3   95       BP
120 GO:0002429 1.174476e-04      HGT    S2vS3   96       BP
121 GO:0002757 1.174476e-04      HGT    S2vS3   97       BP
122 GO:0005829 1.183893e-04      HGT    S2vS3   99       CC
123 GO:0032944 1.207893e-04      HGT    S2vS3  100       BP
                                                                                                                                  TERM
1                                                                                                                 B cell proliferation
2                                                                                         positive regulation of lymphocyte activation
3                                                                                                   regulation of B cell proliferation
4                                                                                                             lymphocyte proliferation
5                                                                                                       mononuclear cell proliferation
6                                                                                           regulation of leukocyte cell-cell adhesion
7                                                                                          positive regulation of leukocyte activation
8                                                                                                              leukocyte proliferation
9                                                                                               positive regulation of cell activation
10                                                                                                        leukocyte cell-cell adhesion
11                                                                                         positive regulation of B cell proliferation
12                                                                                            positive regulation of T cell activation
13                                                                                                               immunological synapse
14                                                                                                     regulation of B cell activation
15                                                                                         negative regulation of leukocyte activation
16                                                                                              regulation of lymphocyte proliferation
17                                                                                        regulation of mononuclear cell proliferation
18                                                                            positive regulation of T cell receptor signaling pathway
19                                                                                 positive regulation of leukocyte cell-cell adhesion
20                                                                                               regulation of leukocyte proliferation
21                                                                                                    regulation of cell-cell adhesion
22                                                                                          cellular response to tumor necrosis factor
23                                                                                            regulation of lymphocyte differentiation
24                                                                                                                   peptide secretion
25                                                                                                regulation of T cell differentiation
26                                                                                           positive regulation of cell-cell adhesion
27                                                                                                   response to tumor necrosis factor
28                                                                                                                T cell costimulation
29                                                                                              negative regulation of cell activation
30                                                                                     regulation of T cell receptor signaling pathway
31                                                                                                            lymphocyte costimulation
32                                                                                                                      signal release
33                                                                                         antigen receptor-mediated signaling pathway
34                                                           calcium-dependent protein serine/threonine phosphatase regulator activity
35                                                                  immune response-activating cell surface receptor signaling pathway
36                                                                                      immune response-activating signal transduction
37                                                                                     positive regulation of lymphocyte proliferation
38                                                                               positive regulation of mononuclear cell proliferation
39                                                                  positive regulation of antigen receptor-mediated signaling pathway
40                                                                                    positive regulation of interleukin-12 production
41                                                                                            positive regulation of B cell activation
42                                                                                                          diet induced thermogenesis
43                                                                                           regulation of leukocyte apoptotic process
44                                                                                      positive regulation of leukocyte proliferation
45                                                                                     release of sequestered calcium ion into cytosol
46                                                                                  negative regulation of sequestering of calcium ion
47                                                                                           regulation of sequestering of calcium ion
48                                                                                                    cellular amide catabolic process
49                                                                                                          calcium-mediated signaling
50                                                                                                         sequestering of calcium ion
51                                                                                        negative regulation of lymphocyte activation
52                                                                          regulation of oxidative phosphorylation uncoupler activity
53                                                                                                regulation of gastric acid secretion
54                                                                                    positive regulation of peptide hormone secretion
55                                                                       collagen-activated tyrosine kinase receptor signaling pathway
56                                                              positive regulation of release of sequestered calcium ion into cytosol
57                                                                                                     regulation of peptide secretion
58                                                                                    positive regulation of dendritic cell chemotaxis
59                                      positive regulation of inositol 1,4,5-trisphosphate-sensitive calcium-release channel activity
60                                                                                                     maintenance of location in cell
61                                                                                                         leukocyte mediated immunity
62                                                                                                       activation of immune response
63  positive regulation of T cell activation via T cell receptor contact with antigen bound to MHC molecule on antigen presenting cell
64                                                                                                             acetylcholine transport
65                                                                                                             acetylcholine secretion
66                                                                                    leukocyte activation involved in immune response
67                                                                                                                B cell proliferation
68                                                                                                    regulation of biological quality
69                                                                                                  regulation of B cell proliferation
70                                                                                        positive regulation of lymphocyte activation
71                                                                                                          regulation of localization
72                                                                                                               biological regulation
73                                                                                                    regulation of biological process
74                                                                                          regulation of leukocyte cell-cell adhesion
75                                                                                         positive regulation of leukocyte activation
76                                                                                           negative regulation of biological process
77                                                                                                               immunological synapse
78                                                                                                      regulation of cellular process
79                                                                                                    regulation of cell-cell adhesion
80                                                                                              positive regulation of cell activation
81                                                                                         positive regulation of B cell proliferation
82                                                                                                 regulation of cellular localization
83                                                                                                     regulation of B cell activation
84                                                                                                        leukocyte cell-cell adhesion
85                                                                                                            lymphocyte proliferation
86                                                                                                      mononuclear cell proliferation
87                                                                            positive regulation of T cell receptor signaling pathway
88                                                                                            positive regulation of T cell activation
89                                                                                                  multicellular organism development
90                                                                                             positive regulation of cellular process
91                                                                                                               leading edge membrane
92                                                                                             negative regulation of cellular process
93                                                                                         negative regulation of leukocyte activation
94                                                                                            regulation of lymphocyte differentiation
95                                                                                                                   peptide secretion
96                                                                                                    anatomical structure development
97                                                                                                             leukocyte proliferation
98                                                                                                                T cell costimulation
99                                                                                                          calcium-mediated signaling
100                                                                                               regulation of T cell differentiation
101                                                                                                           lymphocyte costimulation
102                                                                                positive regulation of leukocyte cell-cell adhesion
103                                                                                          positive regulation of cell-cell adhesion
104                                                                                                                     signal release
105                                                                                    release of sequestered calcium ion into cytosol
106                                                                                                         neuron projection membrane
107                                                                                 negative regulation of sequestering of calcium ion
108                                                                                        antigen receptor-mediated signaling pathway
109                                                                                          positive regulation of biological process
110                                                                                          regulation of sequestering of calcium ion
111                                                                                                              developmental process
112                                                                                                        sequestering of calcium ion
113                                                                                                                    protein binding
114                                                                                             negative regulation of cell activation
115                                                                                                                 system development
116                                                                                    regulation of T cell receptor signaling pathway
117                                                                 positive regulation of antigen receptor-mediated signaling pathway
118                                                                                       regulation of carbohydrate metabolic process
119                                                                                             regulation of lymphocyte proliferation
120                                                                 immune response-activating cell surface receptor signaling pathway
121                                                                                     immune response-activating signal transduction
122                                                                                                                            cytosol
123                                                                                       regulation of mononuclear cell proliferation</code></pre>
<p>Top 100 &quot;truth&quot; terms for Stage 2 versus Stage 3 - these look very broad.</p>
<pre class="r"><code>topGOSets[topGOSets$contrast == &quot;S2vS3&quot;,] %&gt;% data.frame</code></pre>
<pre><code>            ID
1   GO:0005886
2   GO:0071944
3   GO:0002376
4   GO:0006955
5   GO:0051716
6   GO:0001775
7   GO:0048583
8   GO:0050896
9   GO:0005737
10  GO:0016020
11  GO:0045321
12  GO:0070887
13  GO:0002252
14  GO:0023052
15  GO:0071310
16  GO:0007154
17  GO:0007165
18  GO:0050853
19  GO:0012505
20  GO:0007166
21  GO:0050865
22  GO:0046649
23  GO:0010033
24  GO:0009607
25  GO:0002682
26  GO:0002694
27  GO:0002521
28  GO:0019899
29  GO:0051707
30  GO:0043207
31  GO:0002250
32  GO:0023051
33  GO:0009605
34  GO:0098542
35  GO:0050790
36  GO:0006952
37  GO:0051249
38  GO:0022610
39  GO:0006950
40  GO:0065009
41  GO:0007155
42  GO:0010646
43  GO:0035556
44  GO:0048584
45  GO:0030098
46  GO:0033993
47  GO:0009986
48  GO:0016192
49  GO:0046903
50  GO:0050776
51  GO:0006915
52  GO:0016477
53  GO:0042221
54  GO:0008219
55  GO:0140352
56  GO:0043168
57  GO:0051179
58  GO:0030155
59  GO:0042110
60  GO:0009966
61  GO:0042113
62  GO:0006810
63  GO:0098552
64  GO:0098805
65  GO:0032940
66  GO:0051234
67  GO:0002768
68  GO:0048534
69  GO:0012501
70  GO:0051336
71  GO:0002520
72  GO:0034097
73  GO:0002764
74  GO:0006928
75  GO:0030217
76  GO:0050863
77  GO:0045087
78  GO:1903034
79  GO:0005543
80  GO:0009611
81  GO:0043067
82  GO:0031982
83  GO:0002684
84  GO:0030097
85  GO:0002274
86  GO:0050778
87  GO:0040011
88  GO:0048870
89  GO:0051674
90  GO:0008289
91  GO:0005794
92  GO:0050854
93  GO:0010941
94  GO:0110165
95  GO:0042060
96  GO:0045785
97  GO:0051239
98  GO:0005575
99  GO:0035091
100 GO:0001932
                                                                  Term Ont
1                                                      plasma membrane  CC
2                                                       cell periphery  CC
3                                                immune system process  BP
4                                                      immune response  BP
5                                        cellular response to stimulus  BP
6                                                      cell activation  BP
7                                   regulation of response to stimulus  BP
8                                                 response to stimulus  BP
9                                                            cytoplasm  CC
10                                                            membrane  CC
11                                                leukocyte activation  BP
12                              cellular response to chemical stimulus  BP
13                                             immune effector process  BP
14                                                           signaling  BP
15                              cellular response to organic substance  BP
16                                                  cell communication  BP
17                                                 signal transduction  BP
18                                   B cell receptor signaling pathway  BP
19                                                 endomembrane system  CC
20                             cell surface receptor signaling pathway  BP
21                                       regulation of cell activation  BP
22                                               lymphocyte activation  BP
23                                       response to organic substance  BP
24                                         response to biotic stimulus  BP
25                                 regulation of immune system process  BP
26                                  regulation of leukocyte activation  BP
27                                           leukocyte differentiation  BP
28                                                      enzyme binding  MF
29                                          response to other organism  BP
30                                response to external biotic stimulus  BP
31                                            adaptive immune response  BP
32                                             regulation of signaling  BP
33                                       response to external stimulus  BP
34                                  defense response to other organism  BP
35                                    regulation of catalytic activity  BP
36                                                    defense response  BP
37                                 regulation of lymphocyte activation  BP
38                                                 biological adhesion  BP
39                                                  response to stress  BP
40                                    regulation of molecular function  BP
41                                                       cell adhesion  BP
42                                    regulation of cell communication  BP
43                                   intracellular signal transduction  BP
44                         positive regulation of response to stimulus  BP
45                                          lymphocyte differentiation  BP
46                                                   response to lipid  BP
47                                                        cell surface  CC
48                                          vesicle-mediated transport  BP
49                                                           secretion  BP
50                                       regulation of immune response  BP
51                                                   apoptotic process  BP
52                                                      cell migration  BP
53                                                response to chemical  BP
54                                                          cell death  BP
55                                                    export from cell  BP
56                                                       anion binding  MF
57                                                        localization  BP
58                                         regulation of cell adhesion  BP
59                                                   T cell activation  BP
60                                   regulation of signal transduction  BP
61                                                   B cell activation  BP
62                                                           transport  BP
63                                                    side of membrane  CC
64                                                      whole membrane  CC
65                                                   secretion by cell  BP
66                                       establishment of localization  BP
67  immune response-regulating cell surface receptor signaling pathway  BP
68                         hematopoietic or lymphoid organ development  BP
69                                               programmed cell death  BP
70                                    regulation of hydrolase activity  BP
71                                           immune system development  BP
72                                                response to cytokine  BP
73                        immune response-regulating signaling pathway  BP
74                           movement of cell or subcellular component  BP
75                                              T cell differentiation  BP
76                                     regulation of T cell activation  BP
77                                              innate immune response  BP
78                                  regulation of response to wounding  BP
79                                                phospholipid binding  MF
80                                                response to wounding  BP
81                                 regulation of programmed cell death  BP
82                                                             vesicle  CC
83                        positive regulation of immune system process  BP
84                                                         hemopoiesis  BP
85                                        myeloid leukocyte activation  BP
86                              positive regulation of immune response  BP
87                                                          locomotion  BP
88                                                       cell motility  BP
89                                                localization of cell  BP
90                                                       lipid binding  MF
91                                                     Golgi apparatus  CC
92           regulation of antigen receptor-mediated signaling pathway  BP
93                                            regulation of cell death  BP
94                                          cellular anatomical entity  CC
95                                                       wound healing  BP
96                                positive regulation of cell adhesion  BP
97                      regulation of multicellular organismal process  BP
98                                                  cellular_component  CC
99                                        phosphatidylinositol binding  MF
100                              regulation of protein phosphorylation  BP
        N  DE         P.DE          FDR contrast rank
1    4953 332 1.795893e-20 4.047404e-16    S2vS3    1
2    5055 334 1.433078e-19 1.614864e-15    S2vS3    2
3    2750 213 7.033036e-19 5.283451e-15    S2vS3    3
4    1837 159 3.420205e-18 1.927029e-14    S2vS3    4
5    6912 414 1.528796e-17 6.890893e-14    S2vS3    5
6    1320 125 2.969794e-17 1.115504e-13    S2vS3    6
7    3935 270 3.897365e-17 1.254785e-13    S2vS3    7
8    8272 471 1.169129e-16 3.293584e-13    S2vS3    8
9   10568 565 4.478514e-16 1.121470e-12    S2vS3    9
10   8379 473 5.672035e-16 1.278307e-12    S2vS3   10
11   1168 112 7.166047e-16 1.468193e-12    S2vS3   11
12   3095 222 8.297393e-16 1.558320e-12    S2vS3   12
13   1112 108 1.059416e-15 1.836619e-12    S2vS3   13
14   6052 367 1.428695e-15 2.299893e-12    S2vS3   14
15   2550 190 5.868897e-15 8.817823e-12    S2vS3   15
16   6072 365 7.900451e-15 1.112828e-11    S2vS3   16
17   5571 339 3.403649e-14 4.512238e-11    S2vS3   17
18     61  21 4.063621e-14 5.087879e-11    S2vS3   18
19   4186 271 6.372785e-14 7.559130e-11    S2vS3   19
20   2845 202 8.252831e-14 9.299703e-11    S2vS3   20
21    545  65 9.776876e-14 1.049245e-10    S2vS3   21
22    647  72 1.436143e-13 1.471198e-10    S2vS3   22
23   3130 216 1.537215e-13 1.506271e-10    S2vS3   23
24   1351 117 2.110087e-13 1.981459e-10    S2vS3   24
25   1403 120 2.339648e-13 2.109145e-10    S2vS3   25
26    505  61 3.336694e-13 2.892272e-10    S2vS3   26
27    510  61 5.155813e-13 4.303576e-10    S2vS3   27
28   2123 160 6.269131e-13 5.045979e-10    S2vS3   28
29   1317 113 1.072229e-12 8.332700e-10    S2vS3   29
30   1319 113 1.183864e-12 8.893579e-10    S2vS3   30
31    388  51 1.269515e-12 9.229374e-10    S2vS3   31
32   3407 226 2.121755e-12 1.494312e-09    S2vS3   32
33   2588 183 2.765554e-12 1.888705e-09    S2vS3   33
34    996  92 3.208860e-12 2.127002e-09    S2vS3   34
35   2199 161 5.582140e-12 3.594419e-09    S2vS3   35
36   1525 123 6.461705e-12 4.045207e-09    S2vS3   36
37    421  52 8.481605e-12 5.166215e-09    S2vS3   37
38   1381 114 1.020484e-11 6.052274e-09    S2vS3   38
39   3701 238 1.097718e-11 6.343401e-09    S2vS3   39
40   2822 193 1.191926e-11 6.715611e-09    S2vS3   40
41   1374 113 1.613942e-11 8.871566e-09    S2vS3   41
42   3370 220 2.357471e-11 1.265008e-08    S2vS3   42
43   2641 182 3.028834e-11 1.587461e-08    S2vS3   43
44   2165 156 4.492621e-11 2.301141e-08    S2vS3   44
45    348  45 4.801728e-11 2.404812e-08    S2vS3   45
46    865  80 8.960797e-11 4.390206e-08    S2vS3   46
47    782  74 1.700350e-10 8.066675e-08    S2vS3   47
48   1909 140 1.718065e-10 8.066675e-08    S2vS3   48
49   1466 115 1.964490e-10 9.035449e-08    S2vS3   49
50    836  77 2.519324e-10 1.135560e-07    S2vS3   50
51   1799 133 3.213711e-10 1.420145e-07    S2vS3   51
52   1448 113 3.949766e-10 1.711844e-07    S2vS3   52
53   4308 261 4.221859e-10 1.795246e-07    S2vS3   53
54   2065 147 4.485295e-10 1.871946e-07    S2vS3   54
55   1385 109 5.232796e-10 2.144209e-07    S2vS3   55
56   2638 177 5.419505e-10 2.147408e-07    S2vS3   56
57   6212 350 5.431170e-10 2.147408e-07    S2vS3   57
58    695  67 6.055320e-10 2.352909e-07    S2vS3   58
59    446  50 6.842476e-10 2.613710e-07    S2vS3   59
60   2978 194 7.176083e-10 2.695456e-07    S2vS3   60
61    238  34 8.407218e-10 3.106122e-07    S2vS3   61
62   4715 279 9.067048e-10 3.286608e-07    S2vS3   62
63    507  54 9.187395e-10 3.286608e-07    S2vS3   63
64   1597 120 1.059567e-09 3.731166e-07    S2vS3   64
65   1338 105 1.335210e-09 4.629481e-07    S2vS3   65
66   4839 284 1.394085e-09 4.719711e-07    S2vS3   66
67    372  44 1.403118e-09 4.719711e-07    S2vS3   67
68    918  80 1.504443e-09 4.986121e-07    S2vS3   68
69   1935 138 1.549339e-09 5.060500e-07    S2vS3   69
70   1187  96 1.601824e-09 5.157185e-07    S2vS3   70
71    969  83 1.639218e-09 5.203247e-07    S2vS3   71
72   1120  92 1.677739e-09 5.251555e-07    S2vS3   72
73    375  44 1.805274e-09 5.573350e-07    S2vS3   73
74   2055 144 2.062569e-09 6.281638e-07    S2vS3   74
75    247  34 2.254293e-09 6.774001e-07    S2vS3   75
76    312  39 2.548908e-09 7.558517e-07    S2vS3   76
77    800  72 2.717326e-09 7.953295e-07    S2vS3   77
78    167  27 3.056853e-09 8.832345e-07    S2vS3   78
79    427  47 4.070664e-09 1.161273e-06    S2vS3   79
80    605  59 4.391898e-09 1.237253e-06    S2vS3   80
81   1441 109 4.865845e-09 1.353846e-06    S2vS3   81
82   3672 225 5.006167e-09 1.361665e-06    S2vS3   82
83    926  79 5.014786e-09 1.361665e-06    S2vS3   83
84    877  76 5.099015e-09 1.368053e-06    S2vS3   84
85    624  60 5.493161e-09 1.456463e-06    S2vS3   85
86    609  59 5.607710e-09 1.459438e-06    S2vS3   86
87   1789 128 5.679343e-09 1.459438e-06    S2vS3   87
88   1607 118 5.763410e-09 1.459438e-06    S2vS3   88
89   1607 118 5.763410e-09 1.459438e-06    S2vS3   89
90    720  66 6.295686e-09 1.576510e-06    S2vS3   90
91   1468 110 6.881440e-09 1.697261e-06    S2vS3   91
92     63  16 6.928520e-09 1.697261e-06    S2vS3   92
93   1559 115 7.203634e-09 1.736009e-06    S2vS3   93
94  16443 756 7.300624e-09 1.736009e-06    S2vS3   94
95    493  51 7.317784e-09 1.736009e-06    S2vS3   95
96    407  45 7.627266e-09 1.790580e-06    S2vS3   96
97   3126 197 7.806555e-09 1.813777e-06    S2vS3   97
98  16606 761 8.517103e-09 1.958673e-06    S2vS3   98
99    248  33 9.162603e-09 2.085834e-06    S2vS3   99
100  1387 105 9.315912e-09 2.097525e-06    S2vS3  100</code></pre>
<p>Examine what the top 10 ranked gene sets are and how many genes they contain, for each method and comparison.</p>
<pre class="r"><code>nGenes &lt;- rownames_to_column(data.frame(n = sapply(missMethyl:::.getGO()$idList, 
                                                   length)), 
                             var = &quot;ID&quot;)

dat %&gt;% arrange(contrast, method, P.DE) %&gt;%
    group_by(contrast, method) %&gt;%
    mutate(FDR = p.adjust(P.DE, method = &quot;BH&quot;)) %&gt;%
    dplyr::filter(rank &lt;= 10) %&gt;% 
    inner_join(terms, by = c(&quot;GO&quot; = &quot;GOID&quot;)) %&gt;%
    inner_join(nGenes, by = c(&quot;GO&quot; = &quot;ID&quot;)) -&gt; sub</code></pre>
<pre class="r"><code>p &lt;- vector(&quot;list&quot;, length(unique(sub$contrast)))
truthPal &lt;- scales::hue_pal()(4)
names(truthPal) &lt;- c(&quot;Both&quot;, &quot;ISP&quot;, &quot;Neither&quot;, &quot;RNAseq&quot;)

for(i in 1:length(p)){
    cont &lt;- sort(unique(sub$contrast))[i]
    sub %&gt;% dplyr::filter(contrast == cont) %&gt;%
        arrange(method, -rank) %&gt;%
        ungroup() %&gt;%
        mutate(idx = as.factor(1:n())) -&gt; tmp
    
    setLabs &lt;- substr(tmp$TERM, 1, 40)
    names(setLabs) &lt;- tmp$idx
    
    tmp %&gt;% mutate(rna = GO %in% topGOSets$ID[topGOSets$contrast %in% cont],
                   isp = GO %in% immuneGO$GOID,
                   both = rna + isp,
                   col = ifelse(both == 2, &quot;Both&quot;, 
                                ifelse(both == 1 &amp; rna == 1, &quot;RNAseq&quot;,
                                       ifelse(both == 1 &amp; isp == 1, 
                                              &quot;ISP&quot;, &quot;Neither&quot;)))) %&gt;%
        mutate(col = factor(col, 
                            levels = c(&quot;Both&quot;, &quot;ISP&quot;, &quot;RNAseq&quot;, 
                                       &quot;Neither&quot;))) -&gt; tmp
    
    p[[i]] &lt;- ggplot(tmp, aes(x = -log10(FDR), y = idx, colour = col)) +
        geom_point(aes(size = n), alpha = 0.7) +
        scale_size(limits = c(min(sub$n), max(sub$n))) +
        facet_wrap(vars(method), ncol = 2, scales = &quot;free&quot;) +
        scale_y_discrete(labels = setLabs) +
        scale_colour_manual(values = truthPal) +
        labs(y = &quot;&quot;, size = &quot;No. genes&quot;, colour = &quot;In truth set&quot;) +
        theme(axis.text = element_text(size = 8),
              legend.box = &quot;vertical&quot;,
              legend.position = &quot;bottom&quot;,
              legend.margin = margin(0, 0, 0, 0, unit = &quot;lines&quot;),
              panel.spacing.x = unit(1, &quot;lines&quot;)) +
        coord_cartesian(xlim = c(-log10(0.99), -log10(10^-80))) +
        geom_vline(xintercept = -log10(0.05), linetype = &quot;dashed&quot;) +
        ggtitle(cont)
}

shift_legend(p[[1]], plot = TRUE, pos = &quot;left&quot;)</code></pre>
<p><img src="figure/07_regionAnalysisBcells.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/a9a3a86439630816564e1ec217381de691d4d4f0/docs/figure/07_regionAnalysisBcells.Rmd/unnamed-chunk-17-1.png" target="_blank">a9a3a86</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2021-04-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>shift_legend(p[[2]], plot = TRUE, pos = &quot;left&quot;)</code></pre>
<p><img src="figure/07_regionAnalysisBcells.Rmd/unnamed-chunk-17-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-2">
Past versions of unnamed-chunk-17-2.png
</button>
</p>
<div id="fig-unnamed-chunk-17-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/a9a3a86439630816564e1ec217381de691d4d4f0/docs/figure/07_regionAnalysisBcells.Rmd/unnamed-chunk-17-2.png" target="_blank">a9a3a86</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2021-04-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Save figure for use in manuscript.</p>
<pre class="r"><code>fig &lt;- here(&quot;output/figures/Fig-5D.rds&quot;)
saveRDS(p[[1]], fig, compress = FALSE)

fig &lt;- here(&quot;output/figures/SFig-11C.rds&quot;)
saveRDS(p[[2]], fig, compress = FALSE)</code></pre>
</div>
<div id="compare-goregion-with-probe-wise-analysis" class="section level3">
<h3>Compare GOregion with probe-wise analysis</h3>
<pre class="r"><code>dmrGo %&gt;% dplyr::filter(method %in% c(&quot;goregion-gometh&quot;, &quot;gometh-probe-top&quot;, 
                         &quot;gometh-probe-fdr&quot;)) %&gt;%
    mutate(method = unname((dict[method]))) %&gt;% 
    arrange(contrast, method, P.DE) %&gt;%
    group_by(contrast, method) %&gt;%
    mutate(csum = cumsum(GO %in% immuneGO$GOID)) %&gt;%
    mutate(rank = 1:n()) %&gt;%
    dplyr::filter(rank &lt;= 100) -&gt; sub

p1 &lt;- ggplot(sub, aes(x = rank, y = csum, colour = method)) +
    geom_line() +
    facet_wrap(vars(contrast), ncol=3) +
    geom_vline(xintercept = 10, linetype = &quot;dotted&quot;) +
    labs(colour = &quot;Method&quot;, x = &quot;Rank&quot;, y = &quot;Cumulative no. immune sets&quot;) +
    theme(legend.position = &quot;bottom&quot;) +
    scale_color_manual(values = methodCols)</code></pre>
<pre class="r"><code>dmrGo %&gt;% dplyr::filter(method %in% c(&quot;goregion-gometh&quot;, &quot;gometh-probe-top&quot;, 
                         &quot;gometh-probe-fdr&quot;)) %&gt;%
    mutate(method = unname((dict[method]))) %&gt;% 
    arrange(contrast, method, P.DE) %&gt;%
    group_by(contrast, method) %&gt;%
    mutate(csum = cumsum(GO %in% topGOSets$ID[topGOSets$contrast %in% 
                                                  contrast])) %&gt;%
    mutate(rank = 1:n()) %&gt;%
    dplyr::filter(rank &lt;= 100) -&gt; sub

p2 &lt;- ggplot(sub, aes(x = rank, y = csum, colour = method)) +
    geom_line() +
    facet_wrap(vars(contrast), ncol=3, scales = &quot;free_y&quot;) +
    geom_vline(xintercept = 10, linetype = &quot;dotted&quot;) +
    labs(colour = &quot;Method&quot;, x = &quot;Rank&quot;, y = &quot;Cumulative no. RNAseq sets&quot;) +
    theme(legend.position = &quot;bottom&quot;) + 
    scale_color_manual(values = methodCols)

p3 &lt;- p1 / p2 + 
    plot_layout(guides = &quot;collect&quot;) &amp; 
    theme(legend.position = &quot;bottom&quot;)
p3</code></pre>
<p><img src="figure/07_regionAnalysisBcells.Rmd/unnamed-chunk-20-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-20-1">
Past versions of unnamed-chunk-20-1.png
</button>
</p>
<div id="fig-unnamed-chunk-20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/a9a3a86439630816564e1ec217381de691d4d4f0/docs/figure/07_regionAnalysisBcells.Rmd/unnamed-chunk-20-1.png" target="_blank">a9a3a86</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2021-04-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Save figure for use in manuscript.</p>
<pre class="r"><code>fig &lt;- here(&quot;output/figures/SFig-13A.rds&quot;)
saveRDS(p1, fig, compress = FALSE)

fig &lt;- here(&quot;output/figures/SFig-13B.rds&quot;)
saveRDS(p2, fig, compress = FALSE)</code></pre>
<p>Examine what the top 10 ranked gene sets are and how many genes they contain, for each method and comparison.</p>
<pre class="r"><code>terms &lt;- missMethyl:::.getGO()$idTable
nGenes &lt;- rownames_to_column(data.frame(n = sapply(missMethyl:::.getGO()$idList, 
                                                   length)), 
                             var = &quot;ID&quot;)

dmrGo %&gt;% dplyr::filter(method %in% c(&quot;goregion-gometh&quot;, &quot;gometh-probe-top&quot;, 
                         &quot;gometh-probe-fdr&quot;)) %&gt;%
    mutate(method = unname((dict[method]))) %&gt;% 
    arrange(contrast, method, P.DE) %&gt;%
    group_by(contrast, method) %&gt;%
    mutate(FDR = p.adjust(P.DE, method = &quot;BH&quot;)) %&gt;%
    mutate(rank = 1:n()) %&gt;%
    dplyr::filter(rank &lt;= 10) %&gt;% 
    inner_join(terms, by = c(&quot;GO&quot; = &quot;GOID&quot;)) %&gt;%
    inner_join(nGenes, by = c(&quot;GO&quot; = &quot;ID&quot;)) -&gt; sub</code></pre>
<pre class="r"><code>p &lt;- vector(&quot;list&quot;, length(unique(sub$contrast)))
for(i in 1:length(p)){
    cont &lt;- sort(unique(sub$contrast))[i]
    sub %&gt;% dplyr::filter(contrast == cont) %&gt;%
        arrange(method, -rank) %&gt;%
        ungroup() %&gt;%
        mutate(idx = as.factor(1:n())) -&gt; tmp
    
    setLabs &lt;- substr(tmp$TERM, 1, 40)
    names(setLabs) &lt;- tmp$idx
    
    tmp %&gt;% mutate(rna = GO %in% topGOSets$ID[topGOSets$contrast %in% cont],
                   isp = GO %in% immuneGO$GOID,
                   both = rna + isp,
                   col = ifelse(both == 2, &quot;Both&quot;, 
                                ifelse(both == 1 &amp; rna == 1, &quot;RNAseq&quot;,
                                       ifelse(both == 1 &amp; isp == 1, 
                                              &quot;ISP&quot;, &quot;Neither&quot;)))) %&gt;%
        mutate(col = factor(col, 
                            levels = c(&quot;Both&quot;, &quot;ISP&quot;, &quot;RNAseq&quot;, 
                                       &quot;Neither&quot;))) -&gt; tmp
    
    p[[i]] &lt;- ggplot(tmp, aes(x = -log10(FDR), y = idx, colour = col)) +
        geom_point(aes(size = n), alpha = 0.7) +
        scale_size(limits = c(min(sub$n), max(sub$n))) +
        facet_wrap(vars(method), ncol = 2, scales = &quot;free&quot;) +
        scale_y_discrete(labels = setLabs) +
        scale_color_manual(values = truthPal) +
        labs(y = &quot;&quot;, size = &quot;No. genes&quot;, colour = &quot;In truth set&quot;) +
        theme(axis.text = element_text(size = 7),
              legend.margin = margin(0, 0, 0, 0, unit = &quot;lines&quot;),
              legend.box = &quot;horizontal&quot;,
              panel.spacing.x = unit(1, &quot;lines&quot;)) +
        coord_cartesian(xlim = c(-log10(0.99), -log10(10^-80))) +
        geom_vline(xintercept = -log10(0.05), linetype = &quot;dashed&quot;) +
        ggtitle(cont)
}

shift_legend(p[[1]], plot = TRUE, pos = &quot;left&quot;)</code></pre>
<p><img src="figure/07_regionAnalysisBcells.Rmd/unnamed-chunk-23-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-1">
Past versions of unnamed-chunk-23-1.png
</button>
</p>
<div id="fig-unnamed-chunk-23-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/a9a3a86439630816564e1ec217381de691d4d4f0/docs/figure/07_regionAnalysisBcells.Rmd/unnamed-chunk-23-1.png" target="_blank">a9a3a86</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2021-04-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>shift_legend(p[[2]], plot = TRUE, pos = &quot;left&quot;)</code></pre>
<p><img src="figure/07_regionAnalysisBcells.Rmd/unnamed-chunk-23-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-2">
Past versions of unnamed-chunk-23-2.png
</button>
</p>
<div id="fig-unnamed-chunk-23-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/a9a3a86439630816564e1ec217381de691d4d4f0/docs/figure/07_regionAnalysisBcells.Rmd/unnamed-chunk-23-2.png" target="_blank">a9a3a86</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2021-04-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Save figure for use in manuscript.</p>
<pre class="r"><code>fig &lt;- here(&quot;output/figures/SFig-13C.rds&quot;)
saveRDS(shift_legend(p[[1]] + theme(plot.title = element_blank()),
                     pos = &quot;left&quot;), 
        fig, compress = FALSE)

fig &lt;- here(&quot;output/figures/SFig-13D.rds&quot;)
saveRDS(shift_legend(p[[2]] + theme(plot.title = element_blank()),
                     pos = &quot;left&quot;), 
        fig, compress = FALSE)</code></pre>
</div>
</div>
<div id="compare-characteristics-of-region-wise-and-probe-wise-results" class="section level2">
<h2>Compare characteristics of region-wise and probe-wise results</h2>
<pre class="r"><code>cpgs &lt;- GRanges(seqnames = anno$chr, 
                ranges = IRanges(start = anno$pos, 
                                 end = anno$pos),
                strand = anno$strand,
                name = anno$Name)
dat &lt;- NULL

for(i in 1:ncol(cont.matrix)){
    keep &lt;- (abs(dmrList[[i]]$meandiff) &gt; meanDiff &amp; dmrList[[i]]$no.cpgs &gt;= noCpgs)
    
    overlaps &lt;- findOverlaps(cpgs, dmrList[[i]][keep,])
    tmp &lt;- data.frame(cpgs = cpgs$name[from(overlaps)],
                      method = &quot;DMRcate&quot;, 
                      contrast = colnames(cont.matrix)[i],
                      stringsAsFactors = FALSE)
    dat &lt;- bind_rows(dat, tmp)
    
    tmp &lt;- data.frame(cpgs = rownames(topTreat(tfit, coef = i, num = 5000)),
                      method = &quot;Top 5000&quot;,
                      contrast = colnames(cont.matrix)[i],
                      stringsAsFactors = FALSE)
    dat &lt;- bind_rows(dat, tmp)
    
    tmp &lt;- data.frame(cpgs = rownames(topTreat(tfit, coef = i, num = Inf, 
                                               p.value = pval)),
                      method = &quot;FDR &lt; 0.05&quot;,
                      contrast = colnames(cont.matrix)[i],
                      stringsAsFactors = FALSE)
    dat &lt;- bind_rows(dat, tmp)
    
}

dat %&gt;% group_by(contrast, method) %&gt;% 
    tally() -&gt; sub

selectCols &lt;- c(&quot;#ff6b97&quot;, &quot;#48bf8e&quot;, &quot;#a41415&quot;)
names(selectCols) &lt;- unique(dat$method)
    
ggplot(sub, aes(x = method, y = n, fill = method)) +
    geom_bar(stat = &quot;identity&quot;, show.legend = FALSE) +
    facet_wrap(vars(contrast)) + 
    labs(y = &quot;No. significant CpGs&quot;, x = &quot;Sig. CpGs selected using&quot;) + 
    theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
    scale_fill_manual(values = selectCols)</code></pre>
<p><img src="figure/07_regionAnalysisBcells.Rmd/unnamed-chunk-25-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-25-1">
Past versions of unnamed-chunk-25-1.png
</button>
</p>
<div id="fig-unnamed-chunk-25-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/a9a3a86439630816564e1ec217381de691d4d4f0/docs/figure/07_regionAnalysisBcells.Rmd/unnamed-chunk-25-1.png" target="_blank">a9a3a86</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2021-04-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>flatAnn &lt;- loadFlatAnnotation(anno)

dat %&gt;% group_by(contrast, method) %&gt;%
    inner_join(flatAnn, by = c(&quot;cpgs&quot; = &quot;cpg&quot;)) %&gt;% 
    group_by(contrast, method) %&gt;%
    dplyr::select(group_cols(), entrezid) %&gt;%
    distinct() %&gt;%
    tally() -&gt; sub

ggplot(sub, aes(x = method, y = n, fill = method)) +
    geom_bar(stat = &quot;identity&quot;, show.legend = FALSE) +
    facet_wrap(vars(contrast)) + 
    labs(y = &quot;No. genes with sig. CpGs&quot;, x = &quot;Sig. CpGs selected using&quot;) + 
    theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
    scale_fill_manual(values = selectCols)</code></pre>
<p><img src="figure/07_regionAnalysisBcells.Rmd/unnamed-chunk-26-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-26-1">
Past versions of unnamed-chunk-26-1.png
</button>
</p>
<div id="fig-unnamed-chunk-26-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/a9a3a86439630816564e1ec217381de691d4d4f0/docs/figure/07_regionAnalysisBcells.Rmd/unnamed-chunk-26-1.png" target="_blank">a9a3a86</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2021-04-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>dat %&gt;% group_by(contrast, method) %&gt;%
    left_join(flatAnn, by = c(&quot;cpgs&quot; = &quot;cpg&quot;)) %&gt;% 
    group_by(contrast, method) %&gt;%
    dplyr::select(group_cols(), entrezid, cpgs) %&gt;%
    summarise(prop = sum(!is.na(entrezid[!duplicated(cpgs)]))/
                  length(unique(cpgs))) -&gt; sub</code></pre>
<pre><code>`summarise()` has grouped output by &#39;contrast&#39;. You can override using the `.groups` argument.</code></pre>
<pre class="r"><code>p &lt;- vector(&quot;list&quot;, length(unique(sub$contrast)))

for(i in 1:length(p)){
    cont &lt;- sort(unique(sub$contrast))[i]
    p[[i]] &lt;- ggplot(sub[sub$contrast == cont,], 
                     aes(x = method, y = prop)) +
        geom_bar(stat = &quot;identity&quot;, 
                 show.legend = FALSE, 
                 fill=&quot;black&quot;) +
        labs(y = &quot;Prop. sig. CpGs mapped to genes&quot;, 
             x = &quot;Sig. CpGs selected using&quot;) + 
        theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
        scale_x_discrete(limits = c(&quot;FDR &lt; 0.05&quot;, &quot;DMRcate&quot;, &quot;Top 5000&quot;))
}

p[[1]] + ggtitle(sort(unique(sub$contrast))[1]) | 
    p[[2]] + ggtitle(sort(unique(sub$contrast))[2])</code></pre>
<p><img src="figure/07_regionAnalysisBcells.Rmd/unnamed-chunk-27-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-27-1">
Past versions of unnamed-chunk-27-1.png
</button>
</p>
<div id="fig-unnamed-chunk-27-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/a9a3a86439630816564e1ec217381de691d4d4f0/docs/figure/07_regionAnalysisBcells.Rmd/unnamed-chunk-27-1.png" target="_blank">a9a3a86</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2021-04-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Save figure for use in manuscript.</p>
<pre class="r"><code>fig &lt;- here(&quot;output/figures/Fig-5F.rds&quot;)
saveRDS(p[[1]], fig, compress = FALSE)

fig &lt;- here(&quot;output/figures/SFig-11E.rds&quot;)
saveRDS(p[[2]], fig, compress = FALSE)</code></pre>
<pre class="r"><code>dat %&gt;% group_by(contrast, method) %&gt;%
    left_join(flatAnn, by = c(&quot;cpgs&quot; = &quot;cpg&quot;)) %&gt;% 
    group_by(contrast, method) %&gt;%
    dplyr::select(group_cols(), group, cpgs) %&gt;%
    group_by(contrast, method, group) %&gt;%
    tally() -&gt; sub

ggplot(sub, aes(x = group, y = n, fill = method)) +
    geom_bar(stat = &quot;identity&quot;, position = &quot;dodge&quot;) +
    facet_wrap(vars(contrast), nrow = 3, ncol = 1, scales = &quot;free_y&quot;) + 
    labs(fill = &quot;Sig. CpGs selected using&quot;, y = &quot;No. sig. CpGs mapped to genomic features&quot;, 
         x = &quot;Feature&quot;) + 
    theme(axis.text.x = element_text(angle = 45, hjust = 1))</code></pre>
<p><img src="figure/07_regionAnalysisBcells.Rmd/unnamed-chunk-29-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-29-1">
Past versions of unnamed-chunk-29-1.png
</button>
</p>
<div id="fig-unnamed-chunk-29-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/a9a3a86439630816564e1ec217381de691d4d4f0/docs/figure/07_regionAnalysisBcells.Rmd/unnamed-chunk-29-1.png" target="_blank">a9a3a86</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2021-04-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="cpg-overlap-between-region-wise-and-probe-wise-approaches" class="section level3">
<h3>CpG overlap between region-wise and probe-wise approaches</h3>
<p>Compare the CpGs covered by the different approaches, for the three contrasts.</p>
<pre class="r"><code>p &lt;- vector(&quot;list&quot;, ncol(cont.matrix))
d &lt;- vector(&quot;list&quot;, length(p))

for(i in 1:length(p)){
    cont &lt;- sort(colnames(cont.matrix))[i]
    dat %&gt;% dplyr::filter(contrast == cont) %&gt;%
        dplyr::select(-contrast) %&gt;%
        group_by(cpgs) %&gt;%
        summarize(meth = list(method)) %&gt;%
        ggplot(aes(x = meth)) +
        geom_bar() +
        labs(y = &quot;Intersection size&quot;, x = &quot;&quot;) +
        scale_x_upset(sets = c(&quot;FDR &lt; 0.05&quot;, &quot;DMRcate&quot;, &quot;Top 5000&quot;)) +
        theme(axis.title.y = element_text(size = 10)) -&gt; int
    
    dat %&gt;% dplyr::filter(contrast == cont) %&gt;%
        group_by(contrast, method) %&gt;% 
        tally() %&gt;%
        ggplot(aes(x = method, y = n)) +
        geom_col(fill=&quot;black&quot;, position = &quot;dodge&quot;) +
        geom_text(aes(label = n),
                  position = position_dodge(0.9),
                  size = 1.5, hjust = 1.1, vjust = 0.5) +
        labs(y = &quot;Set size&quot;) +
        scale_x_discrete(position = &quot;top&quot;,
                         limits = c(&quot;Top 5000&quot;, &quot;DMRcate&quot;, &quot;FDR &lt; 0.05&quot;)) +
        scale_y_reverse(labels = scales::format_format(big.mark = &quot; &quot;, 
                                                       decimal.mark = &quot;.&quot;, 
                                                       scientific = FALSE, 
                                                       digits = 0),
                        expand = expansion(mult = c(0.6, 0))) +
        coord_flip() +
        theme_minimal() +
        theme(legend.position = &quot;none&quot;,
              axis.title.y = element_blank(),
              axis.title.x = element_text(size = 8),
              axis.text.x = element_blank(),
              axis.text.y = element_blank(),
              panel.grid = element_blank(),
              plot.margin = margin(0, 0, 0, 0,&quot;cm&quot;)) -&gt; sets
    
    p[[i]] &lt;- ggarrange(ggarrange(plotlist = list(NULL, sets, NULL), 
                                  nrow = 3, heights = c(2.5, 1, 0.1)), int, 
                        ncol = 2,
                        widths = c(1, 3.5))
    d[[i]] &lt;- ggarrange(ggarrange(plotlist = list(NULL, sets, NULL), 
                                  nrow = 3, heights = c(4.75, 1, 0.1)), int, 
                        ncol = 2,
                        widths = c(1, 3.5))
        
}

d[[1]] </code></pre>
<p><img src="figure/07_regionAnalysisBcells.Rmd/unnamed-chunk-30-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-30-1">
Past versions of unnamed-chunk-30-1.png
</button>
</p>
<div id="fig-unnamed-chunk-30-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/a9a3a86439630816564e1ec217381de691d4d4f0/docs/figure/07_regionAnalysisBcells.Rmd/unnamed-chunk-30-1.png" target="_blank">a9a3a86</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2021-04-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>d[[2]]</code></pre>
<p><img src="figure/07_regionAnalysisBcells.Rmd/unnamed-chunk-30-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-30-2">
Past versions of unnamed-chunk-30-2.png
</button>
</p>
<div id="fig-unnamed-chunk-30-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/a9a3a86439630816564e1ec217381de691d4d4f0/docs/figure/07_regionAnalysisBcells.Rmd/unnamed-chunk-30-2.png" target="_blank">a9a3a86</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2021-04-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Save figure for use in manuscript.</p>
<pre class="r"><code>fig &lt;- here(&quot;output/figures/Fig-5E.rds&quot;)
saveRDS(p[[1]], fig, compress = FALSE)

fig &lt;- here(&quot;output/figures/SFig-11D.rds&quot;)
saveRDS(p[[2]], fig, compress = FALSE)</code></pre>
</div>
<div id="gene-overlap-between-region-wise-and-probe-wise-approaches" class="section level3">
<h3>Gene overlap between region-wise and probe-wise approaches</h3>
<p>Compare the genes covered by the different approaches, for the three contrasts.</p>
<pre class="r"><code>p &lt;- vector(&quot;list&quot;, ncol(cont.matrix))
d &lt;- vector(&quot;list&quot;, length(p))

for(i in 1:length(p)){
    cont &lt;- sort(colnames(cont.matrix))[i]
    dat %&gt;% dplyr::filter(contrast == cont) %&gt;%
        left_join(flatAnn, by = c(&quot;cpgs&quot; = &quot;cpg&quot;)) %&gt;%
        dplyr::select(method, entrezid) %&gt;%
        distinct() %&gt;%
        group_by(entrezid) %&gt;%
        summarize(meth = list(method)) %&gt;%
        ggplot(aes(x = meth)) +
        geom_bar() +
        labs(y = &quot;Intersection size&quot;, x = &quot;&quot;) +
        scale_x_upset(sets = c(&quot;FDR &lt; 0.05&quot;, &quot;DMRcate&quot;, &quot;Top 5000&quot;)) +
        theme(axis.title.y = element_text(size = 10)) -&gt; int
    
    dat %&gt;% group_by(contrast, method) %&gt;%
        inner_join(flatAnn, by = c(&quot;cpgs&quot; = &quot;cpg&quot;)) %&gt;%
        group_by(contrast, method) %&gt;%
        dplyr::select(group_cols(), entrezid) %&gt;%
        distinct() %&gt;%
        dplyr::filter(contrast == cont) %&gt;%
        tally() %&gt;%
        ggplot(aes(x = method, y = n)) +
        geom_col(fill=&quot;black&quot;, position = &quot;dodge&quot;) +
        geom_text(aes(label = n),
                  position = position_dodge(0.9),
                  size = 1.5, hjust = 1.1, vjust = 0.5) +
        labs(y = &quot;Set size&quot;) +
        scale_x_discrete(position = &quot;top&quot;,
                         limits = c(&quot;Top 5000&quot;, &quot;DMRcate&quot;, &quot;FDR &lt; 0.05&quot;)) +
        scale_y_reverse(labels = scales::format_format(big.mark = &quot; &quot;,
                                                       decimal.mark = &quot;.&quot;,
                                                       scientific = FALSE,
                                                       digits = 0),
                        expand = expansion(mult = c(0.6, 0))) +
        coord_flip() +
        theme_minimal() +
        theme(legend.position = &quot;none&quot;,
              axis.title.y = element_blank(),
              axis.title.x = element_text(size = 9),
              axis.text.x = element_blank(),
              axis.text.y = element_blank(),
              panel.grid = element_blank(),
              plot.margin = margin(0, 0, 0, 0,&quot;cm&quot;)) -&gt; sets
    
    p[[i]] &lt;- ggarrange(ggarrange(plotlist = list(NULL, sets, NULL),
                                  nrow = 3, heights = c(2.5, 1, 0.1)), int,
                        ncol = 2,
                        widths = c(1, 3.5))
    d[[i]] &lt;- ggarrange(ggarrange(plotlist = list(NULL, sets, NULL),
                                  nrow = 3, heights = c(4.75, 1, 0.1)), int,
                        ncol = 2,
                        widths = c(1, 3.5))
    
}

d[[1]]</code></pre>
<p><img src="figure/07_regionAnalysisBcells.Rmd/unnamed-chunk-32-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-32-1">
Past versions of unnamed-chunk-32-1.png
</button>
</p>
<div id="fig-unnamed-chunk-32-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/a9a3a86439630816564e1ec217381de691d4d4f0/docs/figure/07_regionAnalysisBcells.Rmd/unnamed-chunk-32-1.png" target="_blank">a9a3a86</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2021-04-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>d[[2]]</code></pre>
<p><img src="figure/07_regionAnalysisBcells.Rmd/unnamed-chunk-32-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-32-2">
Past versions of unnamed-chunk-32-2.png
</button>
</p>
<div id="fig-unnamed-chunk-32-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/a9a3a86439630816564e1ec217381de691d4d4f0/docs/figure/07_regionAnalysisBcells.Rmd/unnamed-chunk-32-2.png" target="_blank">a9a3a86</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2021-04-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Save figure for use in manuscript.</p>
<pre class="r"><code>fig &lt;- here(&quot;output/figures/Fig-5G.rds&quot;)
saveRDS(p[[1]], fig, compress = FALSE)

fig &lt;- here(&quot;output/figures/SFig-11F.rds&quot;)
saveRDS(p[[2]], fig, compress = FALSE)</code></pre>
</div>
</div>
<div id="effect-of-dmr-cut-offs-i.e.-num-probes-in-region-and-absolute-delta-beta" class="section level2">
<h2>Effect of DMR cut offs i.e. num probes in region and absolute delta beta</h2>
<pre class="r"><code>outFile &lt;- here(&quot;data/cache-region/dmrcate-GSE45459-params.rds&quot;)
dmrParams &lt;- NULL

meanDiffs &lt;- seq(0, 0.2, by = 0.1)
noCpgs &lt;- 2:4

if(!file.exists(outFile)){
    for(i in 1:length(dmrList)){
        for(j in meanDiffs){
            for(k in noCpgs){
                keep &lt;- (abs(dmrList[[i]]$meandiff) &gt; j &amp; 
                             dmrList[[i]]$no.cpgs &gt;= k)
                
                tmp &lt;- topGSA(goregion(dmrList[[i]][keep, ], anno = anno, 
                                       array.type = &quot;450k&quot;), 
                              number = Inf)
                tmp &lt;- rownames_to_column(tmp, var = &quot;GO&quot;)[, c(&quot;GO&quot;, &quot;P.DE&quot;)]
                tmp$params &lt;- glue(&quot;|\u0394\u03B2| = {j}; No. CpGs = {k}&quot;)
                tmp$contrast &lt;- colnames(cont.matrix)[i]
                dmrParams &lt;- bind_rows(dmrParams, tmp)
            }
        }
    }
    
    saveRDS(dmrParams, file = outFile)
    
} else {
    dmrParams &lt;- readRDS(outFile)
    
}</code></pre>
<p>Examine effect of changing DMR parameter cut offs on gene set rankings of GO categories in &quot;immune system process&quot;.</p>
<pre class="r"><code>dmrParams %&gt;% arrange(contrast, params, P.DE) %&gt;%
    group_by(contrast, params) %&gt;%
    mutate(csum = cumsum(GO %in% immuneGO$GOID)) %&gt;%
    mutate(rank = 1:n()) %&gt;%
    dplyr::filter(rank &lt;= 100) -&gt; dat

p1 &lt;- ggplot(dat, aes(x = rank, y = csum, colour = params)) +
    geom_line() +
    facet_wrap(vars(contrast), ncol=3) +
    geom_vline(xintercept = 10, linetype = &quot;dotted&quot;) +
    labs(colour = &quot;Parameters&quot;, x = &quot;Rank&quot;, y = &quot;Cumulative no. immune sets&quot;)
p1</code></pre>
<p><img src="figure/07_regionAnalysisBcells.Rmd/unnamed-chunk-35-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-35-1">
Past versions of unnamed-chunk-35-1.png
</button>
</p>
<div id="fig-unnamed-chunk-35-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/a9a3a86439630816564e1ec217381de691d4d4f0/docs/figure/07_regionAnalysisBcells.Rmd/unnamed-chunk-35-1.png" target="_blank">a9a3a86</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2021-04-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Examine effect of changing DMR parameter cut offs on gene set rankings on GO categories derived from RNAseq analysis.</p>
<pre class="r"><code>dmrParams %&gt;% arrange(contrast, params, P.DE) %&gt;%
    group_by(contrast, params) %&gt;%
    mutate(csum = cumsum(GO %in% topGOSets$ID[topGOSets$contrast %in% contrast])) %&gt;%
    mutate(rank = 1:n()) %&gt;%
    dplyr::filter(rank &lt;= 100) -&gt; sub

p2 &lt;- ggplot(sub, aes(x = rank, y = csum, colour = params)) +
    geom_line() +
    facet_wrap(vars(contrast), ncol=3) +
    geom_vline(xintercept = 10, linetype = &quot;dotted&quot;) +
    labs(colour = &quot;Parameters&quot;, x = &quot;Rank&quot;, 
         y = glue(&quot;Cumulative no. RNAseq sets&quot;))
p2</code></pre>
<p><img src="figure/07_regionAnalysisBcells.Rmd/unnamed-chunk-36-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-36-1">
Past versions of unnamed-chunk-36-1.png
</button>
</p>
<div id="fig-unnamed-chunk-36-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/a9a3a86439630816564e1ec217381de691d4d4f0/docs/figure/07_regionAnalysisBcells.Rmd/unnamed-chunk-36-1.png" target="_blank">a9a3a86</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2021-04-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Save figure for use in manuscript.</p>
<pre class="r"><code>fig &lt;- here(&quot;output/figures/SFig-14A.rds&quot;)
saveRDS(p1, fig, compress = FALSE)

fig &lt;- here(&quot;output/figures/SFig-14B.rds&quot;)
saveRDS(p2, fig, compress = FALSE)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Mojave 10.14.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8

attached base packages:
 [1] grid      stats4    parallel  stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2            
 [2] GenomicFeatures_1.42.3                             
 [3] AnnotationDbi_1.52.0                               
 [4] ggpubr_0.4.0                                       
 [5] ggupset_0.3.0                                      
 [6] tibble_3.1.0                                       
 [7] patchwork_1.1.1                                    
 [8] dplyr_1.0.5                                        
 [9] UpSetR_1.4.0                                       
[10] glue_1.4.2                                         
[11] ggplot2_3.3.3                                      
[12] missMethyl_1.24.0                                  
[13] IlluminaHumanMethylationEPICanno.ilm10b4.hg19_0.6.0
[14] IlluminaHumanMethylation450kanno.ilmn12.hg19_0.6.0 
[15] reshape2_1.4.4                                     
[16] BiocParallel_1.24.1                                
[17] limma_3.46.0                                       
[18] paletteer_1.3.0                                    
[19] ChAMP_2.20.1                                       
[20] RPMM_1.25                                          
[21] cluster_2.1.1                                      
[22] DT_0.17                                            
[23] IlluminaHumanMethylationEPICmanifest_0.3.0         
[24] Illumina450ProbeVariants.db_1.26.0                 
[25] DMRcate_2.4.1                                      
[26] ChAMPdata_2.22.0                                   
[27] minfi_1.36.0                                       
[28] bumphunter_1.32.0                                  
[29] locfit_1.5-9.4                                     
[30] iterators_1.0.13                                   
[31] foreach_1.5.1                                      
[32] Biostrings_2.58.0                                  
[33] XVector_0.30.0                                     
[34] SummarizedExperiment_1.20.0                        
[35] Biobase_2.50.0                                     
[36] MatrixGenerics_1.2.1                               
[37] matrixStats_0.58.0                                 
[38] GenomicRanges_1.42.0                               
[39] GenomeInfoDb_1.26.4                                
[40] IRanges_2.24.1                                     
[41] S4Vectors_0.28.1                                   
[42] BiocGenerics_0.36.0                                
[43] here_1.0.1                                         
[44] workflowr_1.6.2                                    

loaded via a namespace (and not attached):
  [1] Hmisc_4.5-0                               
  [2] Rsamtools_2.6.0                           
  [3] rprojroot_2.0.2                           
  [4] crayon_1.4.1                              
  [5] MASS_7.3-53.1                             
  [6] rhdf5filters_1.2.0                        
  [7] nlme_3.1-152                              
  [8] backports_1.2.1                           
  [9] sva_3.38.0                                
 [10] impute_1.64.0                             
 [11] rlang_0.4.10                              
 [12] readxl_1.3.1                              
 [13] DSS_2.38.0                                
 [14] globaltest_5.44.0                         
 [15] bit64_4.0.5                               
 [16] isva_1.9                                  
 [17] rngtools_1.5                              
 [18] methylumi_2.36.0                          
 [19] haven_2.3.1                               
 [20] tidyselect_1.1.0                          
 [21] rio_0.5.26                                
 [22] XML_3.99-0.6                              
 [23] nleqslv_3.3.2                             
 [24] tidyr_1.1.3                               
 [25] GenomicAlignments_1.26.0                  
 [26] xtable_1.8-4                              
 [27] magrittr_2.0.1                            
 [28] evaluate_0.14                             
 [29] zlibbioc_1.36.0                           
 [30] rstudioapi_0.13                           
 [31] doRNG_1.8.2                               
 [32] whisker_0.4                               
 [33] bslib_0.2.4                               
 [34] rpart_4.1-15                              
 [35] ensembldb_2.14.0                          
 [36] shiny_1.6.0                               
 [37] xfun_0.22                                 
 [38] askpass_1.1                               
 [39] clue_0.3-58                               
 [40] multtest_2.46.0                           
 [41] interactiveDisplayBase_1.28.0             
 [42] base64_2.0                                
 [43] biovizBase_1.38.0                         
 [44] scrime_1.3.5                              
 [45] dendextend_1.14.0                         
 [46] png_0.1-7                                 
 [47] permute_0.9-5                             
 [48] reshape_0.8.8                             
 [49] withr_2.4.1                               
 [50] lumi_2.42.0                               
 [51] bitops_1.0-6                              
 [52] plyr_1.8.6                                
 [53] cellranger_1.1.0                          
 [54] AnnotationFilter_1.14.0                   
 [55] JADE_2.0-3                                
 [56] pillar_1.5.1                              
 [57] cachem_1.0.4                              
 [58] fs_1.5.0                                  
 [59] DelayedMatrixStats_1.12.3                 
 [60] vctrs_0.3.7                               
 [61] ellipsis_0.3.1                            
 [62] generics_0.1.0                            
 [63] tools_4.0.3                               
 [64] foreign_0.8-81                            
 [65] munsell_0.5.0                             
 [66] DelayedArray_0.16.3                       
 [67] fastmap_1.1.0                             
 [68] compiler_4.0.3                            
 [69] abind_1.4-5                               
 [70] httpuv_1.5.5                              
 [71] rtracklayer_1.50.0                        
 [72] geneLenDataBase_1.26.0                    
 [73] ExperimentHub_1.16.0                      
 [74] lemon_0.4.5                               
 [75] beanplot_1.2                              
 [76] Gviz_1.34.1                               
 [77] plotly_4.9.3                              
 [78] GenomeInfoDbData_1.2.4                    
 [79] gridExtra_2.3                             
 [80] DNAcopy_1.64.0                            
 [81] edgeR_3.32.1                              
 [82] lattice_0.20-41                           
 [83] utf8_1.2.1                                
 [84] later_1.1.0.1                             
 [85] BiocFileCache_1.14.0                      
 [86] jsonlite_1.7.2                            
 [87] affy_1.68.0                               
 [88] scales_1.1.1                              
 [89] carData_3.0-4                             
 [90] sparseMatrixStats_1.2.1                   
 [91] genefilter_1.72.1                         
 [92] lazyeval_0.2.2                            
 [93] promises_1.2.0.1                          
 [94] car_3.0-10                                
 [95] doParallel_1.0.16                         
 [96] latticeExtra_0.6-29                       
 [97] R.utils_2.10.1                            
 [98] goseq_1.42.0                              
 [99] checkmate_2.0.0                           
[100] rmarkdown_2.7                             
[101] openxlsx_4.2.3                            
[102] nor1mix_1.3-0                             
[103] cowplot_1.1.1                             
[104] statmod_1.4.35                            
[105] siggenes_1.64.0                           
[106] forcats_0.5.1                             
[107] dichromat_2.0-0                           
[108] BSgenome_1.58.0                           
[109] HDF5Array_1.18.1                          
[110] bsseq_1.26.0                              
[111] survival_3.2-10                           
[112] yaml_2.2.1                                
[113] htmltools_0.5.1.1                         
[114] memoise_2.0.0                             
[115] VariantAnnotation_1.36.0                  
[116] quadprog_1.5-8                            
[117] viridisLite_0.3.0                         
[118] digest_0.6.27                             
[119] assertthat_0.2.1                          
[120] mime_0.10                                 
[121] rappdirs_0.3.3                            
[122] BiasedUrn_1.07                            
[123] RSQLite_2.2.5                             
[124] data.table_1.14.0                         
[125] blob_1.2.1                                
[126] R.oo_1.24.0                               
[127] preprocessCore_1.52.1                     
[128] fastICA_1.2-2                             
[129] shinythemes_1.2.0                         
[130] splines_4.0.3                             
[131] Formula_1.2-4                             
[132] labeling_0.4.2                            
[133] rematch2_2.1.2                            
[134] Rhdf5lib_1.12.1                           
[135] illuminaio_0.32.0                         
[136] AnnotationHub_2.22.0                      
[137] ProtGenerics_1.22.0                       
[138] RCurl_1.98-1.3                            
[139] broom_0.7.6                               
[140] hms_1.0.0                                 
[141] rhdf5_2.34.0                              
[142] colorspace_2.0-0                          
[143] base64enc_0.1-3                           
[144] BiocManager_1.30.12                       
[145] nnet_7.3-15                               
[146] sass_0.3.1                                
[147] GEOquery_2.58.0                           
[148] Rcpp_1.0.6                                
[149] mclust_5.4.7                              
[150] fansi_0.4.2                               
[151] R6_2.5.0                                  
[152] lifecycle_1.0.0                           
[153] zip_2.1.1                                 
[154] curl_4.3                                  
[155] kpmt_0.1.0                                
[156] ggsignif_0.6.1                            
[157] affyio_1.60.0                             
[158] jquerylib_0.1.3                           
[159] Matrix_1.3-2                              
[160] qvalue_2.22.0                             
[161] ROC_1.66.0                                
[162] org.Hs.eg.db_3.12.0                       
[163] RColorBrewer_1.1-2                        
[164] stringr_1.4.0                             
[165] IlluminaHumanMethylation450kmanifest_0.4.0
[166] htmlwidgets_1.5.3                         
[167] biomaRt_2.46.3                            
[168] purrr_0.3.4                               
[169] marray_1.68.0                             
[170] mgcv_1.8-34                               
[171] openssl_1.4.3                             
[172] htmlTable_2.1.0                           
[173] codetools_0.2-18                          
[174] GO.db_3.12.1                              
[175] gtools_3.8.2                              
[176] prettyunits_1.1.1                         
[177] dbplyr_2.1.0                              
[178] R.methodsS3_1.8.1                         
[179] gtable_0.3.0                              
[180] DBI_1.1.1                                 
[181] git2r_0.28.0                              
[182] wateRmelon_1.34.0                         
[183] httr_1.4.2                                
[184] highr_0.8                                 
[185] KernSmooth_2.23-18                        
[186] stringi_1.5.3                             
[187] progress_1.2.2                            
[188] farver_2.1.0                              
[189] annotate_1.68.0                           
[190] viridis_0.5.1                             
[191] xml2_1.3.2                                
[192] combinat_0.0-8                            
[193] readr_1.4.0                               
[194] BiocVersion_3.12.0                        
[195] bit_4.0.4                                 
[196] jpeg_0.1-8.1                              
[197] pkgconfig_2.0.3                           
[198] rstatix_0.7.0                             
[199] knitr_1.31                                </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
