<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jovana Maksimovic, Alicia Oshlack and Belinda Phipson" />

<meta name="date" content="2020-03-31" />

<title>Gene set testing for Illumina HumanMethylation Arrays</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">methyl-geneset-testing</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Gene set testing for Illumina HumanMethylation Arrays</h1>
<h4 class="author">Jovana Maksimovic, Alicia Oshlack and Belinda Phipson</h4>
<h4 class="date">March 31, 2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-03-31
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>methyl-geneset-testing/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.1). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200302code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200302)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200302code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200302)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsatlassianpetermacorgaubitbucketscmosmethylgenesettestingtree43ca6b605415c516f644cf92bb2c17e03e6663a0targetblank43ca6b6a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/tree/43ca6b605415c516f644cf92bb2c17e03e6663a0" target="_blank">43ca6b6</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsatlassianpetermacorgaubitbucketscmosmethylgenesettestingtree43ca6b605415c516f644cf92bb2c17e03e6663a0targetblank43ca6b6a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/tree/43ca6b605415c516f644cf92bb2c17e03e6663a0" target="_blank">43ca6b6</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/figure/
    Ignored:    code/.job/
    Ignored:    code/old/
    Ignored:    data/
    Ignored:    output/compare-methods/
    Ignored:    output/random-cpg-sims/

Unstaged changes:
    Modified:   methyl-geneset-testing.Rproj

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/exploreData.Rmd</code>) and HTML (<code>docs/exploreData.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/43ca6b605415c516f644cf92bb2c17e03e6663a0/analysis/exploreData.Rmd" target="_blank">43ca6b6</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-03-31
</td>
<td>
wflow_publish(“analysis/exploreData.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/2bcb4c962044ae6c3919813331bee66aa5df9882/analysis/exploreData.Rmd" target="_blank">2bcb4c9</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2020-03-23
</td>
<td>
New script and modifications for methylGSA analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/250d8aea7c2d438e4487aaed37e6c0764ef00a29/analysis/exploreData.Rmd" target="_blank">250d8ae</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2020-03-16
</td>
<td>
Comitting local changes.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/24fb1661352656ee8c129a6c7bc3a23eb0c94ba4/analysis/exploreData.Rmd" target="_blank">24fb166</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-03-16
</td>
<td>
Adding Rmd file changes.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/d7cd66e771da5cd568aab4e5999ae8b7be7a88ec/analysis/exploreData.Rmd" target="_blank">d7cd66e</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-03-02
</td>
<td>
Initial Commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(here)
library(minfi)
library(paletteer)
library(limma)
library(BiocParallel)
library(reshape2)
library(gridExtra)
library(missMethyl)
library(ggplot2)
library(glue)
library(tidyverse)
library(rbin)
library(patchwork)
source(here(&quot;code/utility.R&quot;))</code></pre>
<div id="load-data" class="section level1">
<h1>Load data</h1>
<p>We are using publicly available EPIC data <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE110554">GSE110554</a> generated from flow sorted blood cells. The data is normalised and filtered (bad probes, multi-mapping probes, SNP probes, sex chromosomes).</p>
<pre class="r"><code># load data
dataFile &lt;- here(&quot;data/GSE110554-data.RData&quot;)
if(file.exists(dataFile)){
  # load processed data and sample information
  load(dataFile)
} else {
  # get data from experiment hub, normalise, filter and save objects
  readData(dataFile)
  # load processed data and sample information
  load(dataFile)
}</code></pre>
<div id="qc-plots" class="section level2">
<h2>QC plots</h2>
<pre class="r"><code># plot mean detection p-values across all samples
dat &lt;- tibble::tibble(mean = colMeans(detP), cellType = targets$CellType)
ggplot(dat, aes(y = mean, x = cellType, fill = cellType)) +
  geom_bar(stat = &quot;identity&quot;) +
  labs(fill = &quot;Cell Type&quot;)</code></pre>
<p><img src="figure/exploreData.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># plot normalised beta value distribution
bVals &lt;- getBeta(normGr)
dat &lt;- data.frame(reshape2::melt(bVals))
colnames(dat) &lt;- c(&quot;cpg&quot;, &quot;sample&quot;, &quot;bVal&quot;)
dat &lt;- dplyr::bind_cols(dat, cellType = rep(targets$CellType, 
                                            each = nrow(bVals)))

ggplot(dat, aes(x = bVal, colour = cellType)) +
  geom_density() +
  labs(colour = &quot;Cell Type&quot;)</code></pre>
<p><img src="figure/exploreData.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># MDS plots to look at largest sources of variation
p &lt;- plotMDS(getM(fltGr), top=1000, gene.selection=&quot;common&quot;, plot = FALSE)
dat &lt;- tibble::tibble(x = p$x, y = p$y, cellType = targets$CellType)

ggplot(dat, aes(x = x, y = y, colour = cellType)) +
  geom_point() +
  labs(colour = &quot;Cell Type&quot;)</code></pre>
<p><img src="figure/exploreData.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="statistical-analysis" class="section level1">
<h1>Statistical analysis</h1>
<p>Compare several sets of sorted immune cells. Consider results significant at FDR &lt; 0.05 and delta beta ~ 10% (~ lfc = 0.5).</p>
<pre class="r"><code>mVals &lt;- getM(fltGr)
bVals &lt;- getBeta(fltGr)</code></pre>
<pre class="r"><code>design &lt;- model.matrix(~0+targets$CellType)
colnames(design) &lt;- levels(factor(targets$CellType))
fit &lt;- lmFit(mVals, design)
cont.matrix &lt;- makeContrasts(CD4vCD8=CD4T-CD8T,
                             #BcellvMono=Bcell-Mono,
                             MonovNeu=Mono-Neu,
                             BcellvNK=Bcell-NK,
                             #NeuvNK=Neu-NK,
                             #MonovNK=Mono-NK,
                             levels=design)
fit2 &lt;- contrasts.fit(fit, cont.matrix)
tfit &lt;- eBayes(fit2, robust=TRUE, trend=TRUE)
tfit &lt;- treat(tfit, lfc = 0.5)
pval &lt;- 0.05
fitSum &lt;- summary(decideTests(tfit, p.value = pval))
fitSum</code></pre>
<pre><code>       CD4vCD8 MonovNeu BcellvNK
Down      5072     9324    34803
NotSig  725611   712480   667559
Up        3202    12081    31523</code></pre>
<p>Examine only the independent contrasts.</p>
<pre class="r"><code>dat &lt;- melt(fitSum[rownames(fitSum) != &quot;NotSig&quot;, ])
colnames(dat) &lt;- c(&quot;dir&quot;,&quot;comp&quot;,&quot;num&quot;)

ggplot(dat, aes(x = comp, y = num, fill = dir)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;dodge&quot;) +
  labs(x = &quot;Comparison&quot;, y = &quot;No. DM CpGs (FDR &lt; 0.05)&quot;, fill = &quot;Direction&quot;)</code></pre>
<p><img src="figure/exploreData.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="test-go-categories" class="section level2">
<h2>Test GO categories</h2>
<p>Compare the gene set testing methods available for methylation arrays; hypergeometric test (HGT), gometh (best), methylRRA (GLM), methylRRA (ORA) and methylRRA (GSEA). As methylRRA does not work well with sets that only contain very few genes or very large sets, we will only test sets with at least 5 genes and maximum of 5000 genes. First test GO categories.</p>
<div id="gometh" class="section level3">
<h3>gometh</h3>
<pre class="r"><code>minsize &lt;- 5
maxsize &lt;- 5000

outFile &lt;- here(&quot;data/blood.contrasts.rds&quot;)

if(!file.exists(outFile)){
  obj &lt;- NULL
  obj$tfit &lt;- tfit
  obj$maxsize &lt;- maxsize
  obj$minsize &lt;- minsize
  obj$mVals &lt;- mVals
  obj$targets &lt;- targets

  saveRDS(obj, file = outFile)

} </code></pre>
<pre class="r"><code>inFiles &lt;- list.files(here(&quot;output/compare-methods&quot;), pattern = &quot;rds&quot;, 
                      full.names = TRUE)

res &lt;- lapply(inFiles, function(file){
    readRDS(file)
})

dat &lt;- as_tibble(dplyr::bind_rows(res))</code></pre>
<pre class="r"><code>sub &lt;- dat %&gt;% filter(method %in% c(&quot;mmethyl.hgt&quot;, &quot;mmethyl.gometh&quot;))

ggplot(sub, aes(x = pvalue, colour = sub, linetype = method)) +
    geom_density() +
    facet_wrap(vars(contrast), ncol = 3) +
    labs(colour = &quot;Sig. CpGs&quot;, linetype = &quot;Method&quot;, x = &quot;P-value&quot;, 
         y = &quot;Density&quot;) +
    scale_color_hue(labels = c(&quot;Top 5000&quot;, &quot;Top 10000&quot;, &quot;FDR &lt; 0.01&quot;, 
                                  &quot;FDR &lt; 0.05&quot;)) + 
    theme(legend.position = &quot;bottom&quot;, legend.text = element_text(size=6))</code></pre>
<p><img src="figure/exploreData.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ann &lt;- loadAnnotation(&quot;EPIC&quot;)
flatAnn &lt;- missMethyl:::.getFlatAnnotation(&quot;EPIC&quot;, anno = ann)
cpgEgGo &lt;- cpgsEgGoFreqs(flatAnn)</code></pre>
<pre class="r"><code>cpgEgGo %&gt;% 
  group_by(GO) %&gt;%
  summarise(med = median(Freq)) -&gt; medCpgEgGo

dat %&gt;% filter(set == &quot;GO&quot;) %&gt;%
    filter(sub %in% c(&quot;n&quot;,&quot;c1&quot;)) %&gt;% 
    inner_join(medCpgEgGo, by = c(&quot;ID&quot; = &quot;GO&quot;)) -&gt; sub

bins &lt;- rbin_quantiles(sub, ID, med, bins = 12)
sub$bin &lt;- as.factor(findInterval(sub$med, bins$upper_cut))

binLabs &lt;- paste0(&quot;&lt;&quot;, bins$upper_cut)
names(binLabs) &lt;- levels(sub$bin)
binLabs[length(binLabs)] &lt;- gsub(&quot;&lt;&quot;, &quot;\u2265&quot;, binLabs[length(binLabs) - 1])
    
sub %&gt;% group_by(contrast, method, bin) %&gt;%
  summarise(prop = sum(pvalue &lt; 0.05)/n()) -&gt; pdat

p &lt;- ggplot(pdat, aes(x = as.numeric(bin), y = prop, color = method)) +
  geom_line() +
  facet_wrap(vars(contrast), ncol = 3) +
  scale_x_continuous(breaks = as.numeric(levels(pdat$bin)), labels = binLabs) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, vjust = 1,
                                   size = 7),
        legend.position = &quot;bottom&quot;) +
  labs(x = &quot;Med. No. CpGs per Gene per GO Cat. (binned)&quot;,
       y = &quot;Prop. GO Cat. with p-value &lt; 0.05&quot;)
p</code></pre>
<p><img src="figure/exploreData.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="methylgsa" class="section level3">
<h3>methylGSA</h3>
<p>As we are comparing immune cells, we expect GO categories related to the immune system and its processes to be highly ranked. To evalue this, we downloaded all of the child terms for the GO category “immune system process” (<a href="GO:002376" class="uri">GO:002376</a>) from AminGO 2; <a href="http://amigo.geneontology.org/amigo/term/GO:0002376" class="uri">http://amigo.geneontology.org/amigo/term/GO:0002376</a>.</p>
<pre class="r"><code>immuneGO &lt;- unique(read.csv(here(&quot;data/GO-immune-system-process.txt&quot;), 
                            stringsAsFactors = FALSE, header = FALSE, 
                            col.names = &quot;GOID&quot;))

dat %&gt;% filter(set == &quot;GO&quot;) %&gt;%
    filter(sub %in% c(&quot;n&quot;,&quot;c1&quot;)) %&gt;% 
    arrange(contrast, method, pvalue) %&gt;%
    group_by(contrast, method) %&gt;%
    mutate(csum = cumsum(ID %in% immuneGO$GOID)) %&gt;%
    mutate(rank = 1:n()) %&gt;%
    filter(rank &lt;= 100) -&gt; sub

p &lt;- ggplot(sub, aes(x = rank, y = csum, colour = method)) +
    geom_line() +
    facet_wrap(vars(contrast), ncol=3) +
    geom_vline(xintercept = 10, linetype = &quot;dotted&quot;) +
    labs(colour = &quot;Method&quot;, x = &quot;Rank&quot;, y = &quot;Cumulative no. immune sets&quot;) +
    theme(legend.position = &quot;bottom&quot;)
p</code></pre>
<p><img src="figure/exploreData.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Examine what the top 10 ranked gene sets are and how many genes they contain, for each method and comparison.</p>
<pre class="r"><code>terms &lt;- missMethyl:::.getGO()$idTable
nGenes &lt;- rownames_to_column(data.frame(n = sapply(missMethyl:::.getGO()$idList, 
                                                   length)), 
                             var = &quot;ID&quot;)

dat %&gt;% filter(set == &quot;GO&quot;) %&gt;%
    filter(sub %in% c(&quot;n&quot;,&quot;c1&quot;)) %&gt;% 
    arrange(contrast, method, pvalue) %&gt;%
    group_by(contrast, method) %&gt;%
    mutate(rank = 1:n()) %&gt;%
    mutate(FDR = p.adjust(pvalue)) %&gt;%
    filter(rank &lt;= 10) %&gt;% 
    inner_join(terms, by = c(&quot;ID&quot; = &quot;GOID&quot;)) %&gt;%
    inner_join(nGenes) -&gt; sub</code></pre>
<pre><code>Joining, by = &quot;ID&quot;</code></pre>
<pre class="r"><code>p &lt;- vector(&quot;list&quot;, length(unique(sub$contrast)) * length(unique(sub$method)))
i = 1
for(cont in unique(sub$contrast)){
    c = 1
    for(meth in unique(sub$method)){
        tmp &lt;- sub %&gt;% filter(contrast == cont &amp; method == meth)
        p[[i]] &lt;- ggplot(tmp, aes(x = -log10(FDR), y = rank)) + 
            geom_point(aes(size = n), alpha = 0.5, 
                colour = scales::hue_pal()(length(unique(sub$method)))[c]) +
            scale_y_continuous(labels = tmp$TERM, breaks = tmp$rank) +
            labs(y = &quot;&quot;, size = &quot;No. genes&quot;, title = meth) +
            theme(axis.text.y = element_text(size = 6),
                  plot.title = element_text(size = 8),
                  legend.position = &quot;right&quot;, 
                  legend.key.size = unit(0.25, &quot;cm&quot;),
                  legend.text = element_text(size = 6),
                  legend.title = element_text(size = 8),
                  axis.text.x = element_text(size = 6),
                  axis.title.x = element_text(size = 8)) + 
            coord_cartesian(xlim = c(-log10(0.99), -log10(10^-100))) +
            geom_vline(xintercept = -log10(0.05), linetype = &quot;dashed&quot;)
        i = i + 1
        c = c + 1
    }
}

(p[[1]] / p[[2]] / p[[3]] / p[[4]] / p[[5]]) + 
    plot_annotation(title = unique(sub$contrast)[1],
                    theme = theme(plot.title = element_text(size = 10))) </code></pre>
<p><img src="figure/exploreData.Rmd/unnamed-chunk-14-1.png" width="576" style="display: block; margin: auto;" /></p>
<pre class="r"><code>(p[[6]] / p[[7]] / p[[8]] / p[[9]] / p[[10]]) + 
    plot_annotation(title = unique(sub$contrast)[2],
                    theme = theme(plot.title = element_text(size = 10))) </code></pre>
<p><img src="figure/exploreData.Rmd/unnamed-chunk-14-2.png" width="576" style="display: block; margin: auto;" /></p>
<pre class="r"><code>(p[[11]] / p[[12]] / p[[13]] / p[[14]] / p[[15]]) + 
    plot_annotation(title = unique(sub$contrast)[3],
                    theme = theme(plot.title = element_text(size = 10))) </code></pre>
<p><img src="figure/exploreData.Rmd/unnamed-chunk-14-3.png" width="576" style="display: block; margin: auto;" /></p>
<p>P-value histograms for the different methods for all contrasts on GO categories.</p>
<pre class="r"><code>dat %&gt;% filter(set == &quot;GO&quot;) %&gt;%
    filter(sub %in% c(&quot;n&quot;,&quot;c1&quot;)) -&gt; sub 
    
ggplot(sub, aes(pvalue, fill = method)) +
    geom_histogram(binwidth = 0.025) +
    facet_grid(cols = vars(contrast), rows = vars(method)) +
    theme(legend.position = &quot;bottom&quot;) +
    labs(x = &quot;P-value&quot;, y = &quot;Frequency&quot;, fill = &quot;Method&quot;)</code></pre>
<p><img src="figure/exploreData.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="test-kegg-pathways" class="section level2">
<h2>Test KEGG pathways</h2>
<p>Now test KEGG pathways with at least 5 genes and 5000 at most.</p>
<p>Again, as we are comparing immune cells we expecte pathways from the following categories to be highly ranked: Immune system, Immune disease, Signal transduction, Signaling molecules and interaction; <a href="https://www.genome.jp/kegg/pathway.html" class="uri">https://www.genome.jp/kegg/pathway.html</a>.</p>
<pre class="r"><code>immuneKEGG &lt;- read.csv(here(&quot;data/kegg-immune-related-pathways.csv&quot;), 
                            stringsAsFactors = FALSE, header = FALSE, 
                            col.names = c(&quot;ID&quot;,&quot;pathway&quot;))
immuneKEGG$PID &lt;- paste0(&quot;path:hsa0&quot;,immuneKEGG$ID)

dat %&gt;% filter(set == &quot;KEGG&quot;) %&gt;%
    filter(sub %in% c(&quot;n&quot;,&quot;c1&quot;)) %&gt;% 
    arrange(contrast, method, pvalue) %&gt;%
    group_by(contrast, method) %&gt;%
    mutate(csum = cumsum(ID %in% immuneKEGG$PID)) %&gt;%
    mutate(rank = 1:n()) %&gt;%
    filter(rank &lt;= 100) -&gt; sub

p &lt;- ggplot(sub, aes(x = rank, y = csum, colour = method)) +
    geom_line() +
    facet_wrap(vars(contrast), ncol=3) +
    geom_vline(xintercept = 10, linetype = &quot;dotted&quot;) +
    labs(colour = &quot;Method&quot;, x = &quot;Rank&quot;, y = &quot;Cumulative no. immune sets&quot;) +
    theme(legend.position = &quot;bottom&quot;)
p</code></pre>
<p><img src="figure/exploreData.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Examine what the top 10 ranked gene sets are and how many genes they contain, for each method and comparison.</p>
<pre class="r"><code>terms &lt;- missMethyl:::.getKEGG()$idTable
nGenes &lt;- rownames_to_column(data.frame(n = sapply(missMethyl:::.getKEGG()$idList, 
                                                   length)), 
                             var = &quot;ID&quot;)

dat %&gt;% filter(set == &quot;KEGG&quot;) %&gt;%
    filter(sub %in% c(&quot;n&quot;,&quot;c1&quot;)) %&gt;% 
    arrange(contrast, method, pvalue) %&gt;%
    group_by(contrast, method) %&gt;%
    mutate(rank = 1:n()) %&gt;%
    mutate(FDR = p.adjust(pvalue)) %&gt;%
    filter(rank &lt;= 10) %&gt;% 
    inner_join(terms, by = c(&quot;ID&quot; = &quot;PathwayID&quot;)) %&gt;%
    inner_join(nGenes) -&gt; sub</code></pre>
<pre><code>Joining, by = &quot;ID&quot;</code></pre>
<pre class="r"><code>p &lt;- vector(&quot;list&quot;, length(unique(sub$contrast)) * length(unique(sub$method)))
i = 1
for(cont in unique(sub$contrast)){
    c = 1
    for(meth in unique(sub$method)){
        tmp &lt;- sub %&gt;% filter(contrast == cont &amp; method == meth)
        p[[i]] &lt;- ggplot(tmp, aes(x = -log10(FDR), y = rank)) + 
            geom_point(aes(size = n), alpha = 0.5, 
                colour = scales::hue_pal()(length(unique(sub$method)))[c]) +
            scale_y_continuous(labels = tmp$Description, breaks = tmp$rank) +
            labs(y = &quot;&quot;, size = &quot;No. genes&quot;, title = meth) +
            theme(axis.text.y = element_text(size = 6),
                  plot.title = element_text(size = 8),
                  legend.position = &quot;right&quot;, 
                  legend.key.size = unit(0.25, &quot;cm&quot;),
                  legend.text = element_text(size = 6),
                  legend.title = element_text(size = 8),
                  axis.text.x = element_text(size = 6),
                  axis.title.x = element_text(size = 8)) + 
            coord_cartesian(xlim = c(-log10(0.99), -log10(10^-15))) +
            geom_vline(xintercept = -log10(0.05), linetype = &quot;dashed&quot;)
        i = i + 1
        c = c + 1
    }
}

(p[[1]] / p[[2]] / p[[3]] / p[[4]] / p[[5]]) + 
    plot_annotation(title = unique(sub$contrast)[1],
                    theme = theme(plot.title = element_text(size = 10))) </code></pre>
<p><img src="figure/exploreData.Rmd/unnamed-chunk-17-1.png" width="576" style="display: block; margin: auto;" /></p>
<pre class="r"><code>(p[[6]] / p[[7]] / p[[8]] / p[[9]] / p[[10]]) + 
    plot_annotation(title = unique(sub$contrast)[2],
                    theme = theme(plot.title = element_text(size = 10))) </code></pre>
<p><img src="figure/exploreData.Rmd/unnamed-chunk-17-2.png" width="576" style="display: block; margin: auto;" /></p>
<pre class="r"><code>(p[[11]] / p[[12]] / p[[13]] / p[[14]] / p[[15]]) + 
    plot_annotation(title = unique(sub$contrast)[3],
                    theme = theme(plot.title = element_text(size = 10))) </code></pre>
<p><img src="figure/exploreData.Rmd/unnamed-chunk-17-3.png" width="576" style="display: block; margin: auto;" /></p>
<p>P-value histograms for the different methods for all contrasts on KEGG pathways.</p>
<pre class="r"><code>dat %&gt;% filter(set == &quot;KEGG&quot;) %&gt;%
    filter(sub %in% c(&quot;n&quot;,&quot;c1&quot;)) -&gt; sub 
    
ggplot(sub, aes(pvalue, fill = method)) +
    geom_histogram(binwidth = 0.025) +
    facet_grid(cols = vars(contrast), rows = vars(method)) +
    theme(legend.position = &quot;bottom&quot;) +
    labs(x = &quot;P-value&quot;, y = &quot;Frequency&quot;, fill = &quot;Method&quot;)</code></pre>
<p><img src="figure/exploreData.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="test-broad-gene-sets" class="section level2">
<h2>Test Broad gene sets</h2>
<p>As the ebGSEA method from the ChAMP package only allows testing of their in-built database of Broad Institute gene sets, we will compare to other methods by testing these same gene sets. Note, As methylRRA (GSEA) consistenly performed very poorly in previous comparisons, it is being omitted from the following results.</p>
<pre class="r"><code>library(ChAMP)</code></pre>
<pre><code>Loading required package: ChAMPdata</code></pre>
<pre><code>Loading required package: FEM</code></pre>
<pre><code>Loading required package: AnnotationDbi</code></pre>
<pre><code>
Attaching package: &#39;AnnotationDbi&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:dplyr&#39;:

    select</code></pre>
<pre><code>Loading required package: Matrix</code></pre>
<pre><code>
Attaching package: &#39;Matrix&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:tidyr&#39;:

    expand, pack, unpack</code></pre>
<pre><code>The following object is masked from &#39;package:S4Vectors&#39;:

    expand</code></pre>
<pre><code>Loading required package: marray</code></pre>
<pre><code>Loading required package: corrplot</code></pre>
<pre><code>corrplot 0.84 loaded</code></pre>
<pre><code>Loading required package: igraph</code></pre>
<pre><code>
Attaching package: &#39;igraph&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    as_data_frame, groups, union</code></pre>
<pre><code>The following objects are masked from &#39;package:purrr&#39;:

    compose, simplify</code></pre>
<pre><code>The following object is masked from &#39;package:tidyr&#39;:

    crossing</code></pre>
<pre><code>The following object is masked from &#39;package:tibble&#39;:

    as_data_frame</code></pre>
<pre><code>The following object is masked from &#39;package:Biostrings&#39;:

    union</code></pre>
<pre><code>The following objects are masked from &#39;package:DelayedArray&#39;:

    path, simplify</code></pre>
<pre><code>The following object is masked from &#39;package:GenomicRanges&#39;:

    union</code></pre>
<pre><code>The following object is masked from &#39;package:IRanges&#39;:

    union</code></pre>
<pre><code>The following object is masked from &#39;package:S4Vectors&#39;:

    union</code></pre>
<pre><code>The following objects are masked from &#39;package:BiocGenerics&#39;:

    normalize, path, union</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    decompose, spectrum</code></pre>
<pre><code>The following object is masked from &#39;package:base&#39;:

    union</code></pre>
<pre><code>Loading required package: impute</code></pre>
<pre><code>Loading required package: org.Hs.eg.db</code></pre>
<pre><code>Loading required package: graph</code></pre>
<pre><code>
Attaching package: &#39;graph&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:igraph&#39;:

    degree, edges, intersection</code></pre>
<pre><code>The following object is masked from &#39;package:stringr&#39;:

    boundary</code></pre>
<pre><code>The following object is masked from &#39;package:Biostrings&#39;:

    complement</code></pre>
<pre><code>Warning: replacing previous import &#39;igraph::edges&#39; by &#39;graph::edges&#39; when
loading &#39;FEM&#39;</code></pre>
<pre><code>Warning: replacing previous import &#39;igraph::intersection&#39; by
&#39;graph::intersection&#39; when loading &#39;FEM&#39;</code></pre>
<pre><code>Warning: replacing previous import &#39;igraph::degree&#39; by &#39;graph::degree&#39; when
loading &#39;FEM&#39;</code></pre>
<pre><code>Warning: replacing previous import &#39;igraph::union&#39; by &#39;graph::union&#39; when
loading &#39;FEM&#39;</code></pre>
<pre><code>Warning: replacing previous import &#39;limma::plotMA&#39; by &#39;BiocGenerics::plotMA&#39;
when loading &#39;FEM&#39;</code></pre>
<pre><code>Warning: replacing previous import &#39;igraph::path&#39; by &#39;BiocGenerics::path&#39; when
loading &#39;FEM&#39;</code></pre>
<pre><code>Warning: replacing previous import &#39;Matrix::which&#39; by &#39;BiocGenerics::which&#39; when
loading &#39;FEM&#39;</code></pre>
<pre><code>Warning: replacing previous import &#39;igraph::normalize&#39; by
&#39;BiocGenerics::normalize&#39; when loading &#39;FEM&#39;</code></pre>
<pre><code>Loading required package: DMRcate</code></pre>
<pre><code>No methods found in package &#39;IRanges&#39; for request: &#39;splitAsList&#39; when loading &#39;ensembldb&#39;</code></pre>
<pre><code>Registered S3 method overwritten by &#39;R.oo&#39;:
  method        from       
  throw.default R.methodsS3</code></pre>
<pre><code>Warning: replacing previous import &#39;minfi::getMeth&#39; by &#39;bsseq::getMeth&#39; when
loading &#39;DMRcate&#39;</code></pre>
<pre><code>Loading required package: Illumina450ProbeVariants.db</code></pre>
<pre><code>Loading required package: IlluminaHumanMethylationEPICmanifest</code></pre>
<pre><code>Loading required package: DT</code></pre>
<pre><code>Warning: replacing previous import &#39;plyr::summarise&#39; by &#39;plotly::summarise&#39; when
loading &#39;ChAMP&#39;</code></pre>
<pre><code>Warning: replacing previous import &#39;plyr::rename&#39; by &#39;plotly::rename&#39; when
loading &#39;ChAMP&#39;</code></pre>
<pre><code>Warning: replacing previous import &#39;plyr::arrange&#39; by &#39;plotly::arrange&#39; when
loading &#39;ChAMP&#39;</code></pre>
<pre><code>Warning: replacing previous import &#39;plyr::mutate&#39; by &#39;plotly::mutate&#39; when
loading &#39;ChAMP&#39;</code></pre>
<pre><code>Warning: replacing previous import &#39;plyr::is.discrete&#39; by &#39;Hmisc::is.discrete&#39;
when loading &#39;ChAMP&#39;</code></pre>
<pre><code>Warning: replacing previous import &#39;plyr::summarize&#39; by &#39;Hmisc::summarize&#39; when
loading &#39;ChAMP&#39;</code></pre>
<pre><code>Warning: replacing previous import &#39;plotly::subplot&#39; by &#39;Hmisc::subplot&#39; when
loading &#39;ChAMP&#39;</code></pre>
<pre><code>Warning: replacing previous import &#39;GenomicRanges::sort&#39; by &#39;globaltest::sort&#39;
when loading &#39;ChAMP&#39;</code></pre>
<pre><code>Warning: replacing previous import &#39;plotly::last_plot&#39; by &#39;ggplot2::last_plot&#39;
when loading &#39;ChAMP&#39;</code></pre>
<pre><code>Warning: replacing previous import &#39;globaltest::model.matrix&#39; by
&#39;stats::model.matrix&#39; when loading &#39;ChAMP&#39;</code></pre>
<pre><code>Warning: replacing previous import &#39;globaltest::p.adjust&#39; by &#39;stats::p.adjust&#39;
when loading &#39;ChAMP&#39;</code></pre>
<pre><code>&gt;&gt; Package version 2.16.1 loaded &lt;&lt;
       ___ _      _   __  __ ___ 
      / __| |_   /_\ |  \/  | _ \
     | (__| &#39; \ / _ \| |\/| |  _/
      \___|_||_/_/ \_\_|  |_|_|  
      ------------------------------
    If you have any question or suggestion about ChAMP, please email to champ450k@gmail.com or tianyuan1991hit@gmail.com.
    ChAMP paper just get published on Bioconductor! ^_^

    Yuan Tian, Tiffany J Morris, Amy P Webster, Zhen Yang, Stephan Beck, Andrew Feber, Andrew E Teschendorff; ChAMP: updated methylation analysis pipeline for Illumina BeadChips, Bioinformatics, btx513, https://doi.org/10.1093/bioinformatics/btx513
     --------------------------</code></pre>
<pre class="r"><code>data(PathwayList)
nGenes &lt;- rownames_to_column(data.frame(n = sapply(PathwayList, 
                                                   length)), 
                             var = &quot;ID&quot;)

dat %&gt;% filter(set == &quot;BROAD&quot;) %&gt;%
    filter(sub %in% c(&quot;n&quot;,&quot;c1&quot;)) %&gt;% 
    arrange(contrast, method, pvalue) %&gt;%
    group_by(contrast, method) %&gt;%
    mutate(rank = 1:n()) %&gt;%
    mutate(FDR = p.adjust(pvalue)) %&gt;%
    filter(rank &lt;= 10) %&gt;% 
    inner_join(nGenes) -&gt; sub</code></pre>
<pre><code>Joining, by = &quot;ID&quot;</code></pre>
<pre class="r"><code>p &lt;- vector(&quot;list&quot;, length(unique(sub$contrast)) * length(unique(sub$method)))
i = 1
for(cont in unique(sub$contrast)){
    c = 1
    for(meth in unique(sub$method)){
        tmp &lt;- sub %&gt;% filter(contrast == cont &amp; method == meth)
        p[[i]] &lt;- ggplot(tmp, aes(x = -log10(FDR), y = rank)) + 
            geom_point(aes(size = n), alpha = 0.5, 
                colour = scales::hue_pal()(length(unique(sub$method)))[c]) +
            scale_y_continuous(labels = tmp$ID, breaks = tmp$rank) +
            labs(y = &quot;&quot;, size = &quot;No. genes&quot;, title = meth) +
            theme(axis.text.y = element_text(size = 6),
                  plot.title = element_text(size = 8),
                  legend.position = &quot;right&quot;, 
                  legend.key.size = unit(0.25, &quot;cm&quot;),
                  legend.text = element_text(size = 6),
                  legend.title = element_text(size = 8),
                  axis.text.x = element_text(size = 6),
                  axis.title.x = element_text(size = 8)) + 
            coord_cartesian(xlim = c(-log10(0.99), -log10(10^-100))) +
            geom_vline(xintercept = -log10(0.05), linetype = &quot;dashed&quot;)
        i = i + 1
        c = c + 1
    }
}

(p[[1]] / p[[2]] / p[[3]] / p[[4]] / p[[5]] / p[[6]] / p[[7]]) + 
    plot_annotation(title = unique(sub$contrast)[1],
                    theme = theme(plot.title = element_text(size = 10))) </code></pre>
<p><img src="figure/exploreData.Rmd/unnamed-chunk-19-1.png" width="576" style="display: block; margin: auto;" /></p>
<pre class="r"><code>(p[[8]] / p[[9]] / p[[10]] / p[[11]] / p[[12]] / p[[13]] / p[[14]]) + 
    plot_annotation(title = unique(sub$contrast)[2],
                    theme = theme(plot.title = element_text(size = 10))) </code></pre>
<p><img src="figure/exploreData.Rmd/unnamed-chunk-19-2.png" width="576" style="display: block; margin: auto;" /></p>
<pre class="r"><code>(p[[15]] / p[[16]] / p[[17]] / p[[18]] / p[[19]] / p[[20]] / p[[21]]) + 
    plot_annotation(title = unique(sub$contrast)[3],
                    theme = theme(plot.title = element_text(size = 10))) </code></pre>
<p><img src="figure/exploreData.Rmd/unnamed-chunk-19-3.png" width="576" style="display: block; margin: auto;" /> P-value histograms for the different methods for all contrasts on BROAD gene sets.</p>
<pre class="r"><code>dat %&gt;% filter(set == &quot;BROAD&quot;) %&gt;%
    filter(sub %in% c(&quot;n&quot;,&quot;c1&quot;)) -&gt; sub 
    
ggplot(sub, aes(pvalue, fill = method)) +
    geom_histogram(binwidth = 0.025) +
    facet_grid(cols = vars(contrast), rows = vars(method)) +
    theme(legend.position = &quot;bottom&quot;) +
    labs(x = &quot;P-value&quot;, y = &quot;Frequency&quot;, fill = &quot;Method&quot;)</code></pre>
<p><img src="figure/exploreData.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS:   /config/RStudio/R/3.6.1/lib64/R/lib/libRblas.so
LAPACK: /config/RStudio/R/3.6.1/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ChAMP_2.16.1                              
 [2] DT_0.9                                    
 [3] IlluminaHumanMethylationEPICmanifest_0.3.0
 [4] Illumina450ProbeVariants.db_1.22.0        
 [5] DMRcate_2.0.7                             
 [6] FEM_3.14.0                                
 [7] graph_1.62.0                              
 [8] org.Hs.eg.db_3.8.2                        
 [9] impute_1.58.0                             
[10] igraph_1.2.5                              
[11] corrplot_0.84                             
[12] marray_1.62.0                             
[13] Matrix_1.2-18                             
[14] AnnotationDbi_1.46.1                      
[15] ChAMPdata_2.18.0                          
[16] patchwork_1.0.0                           
[17] rbin_0.1.2                                
[18] forcats_0.4.0                             
[19] stringr_1.4.0                             
[20] dplyr_0.8.3                               
[21] purrr_0.3.3                               
[22] readr_1.3.1                               
[23] tidyr_1.0.2                               
[24] tibble_2.1.3                              
[25] tidyverse_1.3.0                           
[26] glue_1.3.2                                
[27] ggplot2_3.3.0                             
[28] missMethyl_1.20.4                         
[29] gridExtra_2.3                             
[30] reshape2_1.4.3                            
[31] limma_3.42.2                              
[32] paletteer_1.1.0                           
[33] minfi_1.32.0                              
[34] bumphunter_1.26.0                         
[35] locfit_1.5-9.1                            
[36] iterators_1.0.12                          
[37] foreach_1.4.8                             
[38] Biostrings_2.54.0                         
[39] XVector_0.24.0                            
[40] SummarizedExperiment_1.16.1               
[41] DelayedArray_0.12.2                       
[42] BiocParallel_1.20.1                       
[43] matrixStats_0.56.0                        
[44] Biobase_2.46.0                            
[45] GenomicRanges_1.36.1                      
[46] GenomeInfoDb_1.22.1                       
[47] IRanges_2.20.2                            
[48] S4Vectors_0.24.3                          
[49] BiocGenerics_0.32.0                       
[50] here_0.1                                  
[51] workflowr_1.6.1                           

loaded via a namespace (and not attached):
  [1] Hmisc_4.2-0                                        
  [2] Rsamtools_2.0.1                                    
  [3] rprojroot_1.3-2                                    
  [4] crayon_1.3.4                                       
  [5] jcolors_0.0.4                                      
  [6] MASS_7.3-51.5                                      
  [7] nlme_3.1-145                                       
  [8] backports_1.1.5                                    
  [9] reprex_0.3.0                                       
 [10] sva_3.34.0                                         
 [11] rlang_0.4.5                                        
 [12] readxl_1.3.1                                       
 [13] DSS_2.34.0                                         
 [14] globaltest_5.40.0                                  
 [15] bit64_0.9-7                                        
 [16] isva_1.9                                           
 [17] rngtools_1.4                                       
 [18] methylumi_2.30.0                                   
 [19] haven_2.2.0                                        
 [20] tidyselect_0.2.5                                   
 [21] XML_3.98-1.20                                      
 [22] nleqslv_3.3.2                                      
 [23] GenomicAlignments_1.20.1                           
 [24] xtable_1.8-4                                       
 [25] magrittr_1.5                                       
 [26] evaluate_0.14                                      
 [27] bibtex_0.4.2                                       
 [28] cli_2.0.2                                          
 [29] zlibbioc_1.30.0                                    
 [30] rstudioapi_0.11                                    
 [31] doRNG_1.7.1                                        
 [32] whisker_0.4                                        
 [33] rpart_4.1-15                                       
 [34] ensembldb_2.8.0                                    
 [35] IlluminaHumanMethylationEPICanno.ilm10b4.hg19_0.6.0
 [36] maps_3.3.0                                         
 [37] shiny_1.3.2                                        
 [38] xfun_0.12                                          
 [39] askpass_1.1                                        
 [40] clue_0.3-57                                        
 [41] multtest_2.40.0                                    
 [42] cluster_2.1.0                                      
 [43] interactiveDisplayBase_1.22.0                      
 [44] base64_2.0                                         
 [45] biovizBase_1.32.0                                  
 [46] scrime_1.3.5                                       
 [47] dendextend_1.13.4                                  
 [48] permute_0.9-5                                      
 [49] reshape_0.8.8                                      
 [50] withr_2.1.2                                        
 [51] lumi_2.38.0                                        
 [52] bitops_1.0-6                                       
 [53] plyr_1.8.6                                         
 [54] cellranger_1.1.0                                   
 [55] AnnotationFilter_1.8.0                             
 [56] JADE_2.0-3                                         
 [57] pillar_1.4.3                                       
 [58] GenomicFeatures_1.36.4                             
 [59] fs_1.3.2                                           
 [60] DelayedMatrixStats_1.8.0                           
 [61] vctrs_0.2.4                                        
 [62] generics_0.0.2                                     
 [63] tools_3.6.1                                        
 [64] foreign_0.8-72                                     
 [65] munsell_0.5.0                                      
 [66] compiler_3.6.1                                     
 [67] httpuv_1.5.2                                       
 [68] rtracklayer_1.44.4                                 
 [69] geneLenDataBase_1.20.0                             
 [70] ExperimentHub_1.12.0                               
 [71] beanplot_1.2                                       
 [72] Gviz_1.28.3                                        
 [73] pkgmaker_0.27                                      
 [74] scico_1.1.0                                        
 [75] plotly_4.9.0                                       
 [76] GenomeInfoDbData_1.2.1                             
 [77] DNAcopy_1.58.0                                     
 [78] edgeR_3.26.8                                       
 [79] lattice_0.20-40                                    
 [80] later_1.0.0                                        
 [81] BiocFileCache_1.10.2                               
 [82] jsonlite_1.6.1                                     
 [83] affy_1.62.0                                        
 [84] scales_1.1.0                                       
 [85] genefilter_1.68.0                                  
 [86] lazyeval_0.2.2                                     
 [87] promises_1.1.0                                     
 [88] doParallel_1.0.15                                  
 [89] latticeExtra_0.6-28                                
 [90] R.utils_2.9.0                                      
 [91] goseq_1.36.0                                       
 [92] checkmate_1.9.4                                    
 [93] rmarkdown_2.1                                      
 [94] nor1mix_1.3-0                                      
 [95] statmod_1.4.32                                     
 [96] siggenes_1.60.0                                    
 [97] dichromat_2.0-0                                    
 [98] BSgenome_1.52.0                                    
 [99] HDF5Array_1.14.3                                   
[100] bsseq_1.22.0                                       
[101] survival_2.44-1.1                                  
[102] yaml_2.2.1                                         
[103] htmltools_0.4.0                                    
[104] memoise_1.1.0                                      
[105] VariantAnnotation_1.30.1                           
[106] quadprog_1.5-8                                     
[107] viridisLite_0.3.0                                  
[108] digest_0.6.25                                      
[109] assertthat_0.2.1                                   
[110] mime_0.9                                           
[111] rappdirs_0.3.1                                     
[112] registry_0.5-1                                     
[113] BiasedUrn_1.07                                     
[114] palr_0.2.0                                         
[115] pals_1.6                                           
[116] RSQLite_2.1.2                                      
[117] mapproj_1.2.6                                      
[118] data.table_1.12.8                                  
[119] blob_1.2.0                                         
[120] R.oo_1.22.0                                        
[121] preprocessCore_1.48.0                              
[122] fastICA_1.2-2                                      
[123] shinythemes_1.1.2                                  
[124] splines_3.6.1                                      
[125] Formula_1.2-3                                      
[126] labeling_0.3                                       
[127] rematch2_2.1.0                                     
[128] Rhdf5lib_1.6.1                                     
[129] illuminaio_0.28.0                                  
[130] AnnotationHub_2.18.0                               
[131] ProtGenerics_1.16.0                                
[132] RCurl_1.95-4.12                                    
[133] broom_0.5.2                                        
[134] hms_0.5.3                                          
[135] modelr_0.1.6                                       
[136] rhdf5_2.28.0                                       
[137] colorspace_1.4-1                                   
[138] base64enc_0.1-3                                    
[139] BiocManager_1.30.10                                
[140] nnet_7.3-12                                        
[141] GEOquery_2.54.1                                    
[142] Rcpp_1.0.4                                         
[143] mclust_5.4.5                                       
[144] fansi_0.4.1                                        
[145] R6_2.4.1                                           
[146] grid_3.6.1                                         
[147] lifecycle_0.2.0                                    
[148] acepack_1.4.1                                      
[149] oompaBase_3.2.9                                    
[150] kpmt_0.1.0                                         
[151] curl_4.3                                           
[152] affyio_1.54.0                                      
[153] RPMM_1.25                                          
[154] qvalue_2.16.0                                      
[155] ROC_1.62.0                                         
[156] RColorBrewer_1.1-2                                 
[157] IlluminaHumanMethylationEPICanno.ilm10b2.hg19_0.6.0
[158] IlluminaHumanMethylation450kmanifest_0.4.0         
[159] htmlwidgets_1.3                                    
[160] biomaRt_2.42.1                                     
[161] rvest_0.3.5                                        
[162] mgcv_1.8-29                                        
[163] openssl_1.4.1                                      
[164] htmlTable_1.13.2                                   
[165] codetools_0.2-16                                   
[166] IlluminaHumanMethylation450kanno.ilmn12.hg19_0.6.0 
[167] lubridate_1.7.4                                    
[168] GO.db_3.8.2                                        
[169] gtools_3.8.1                                       
[170] prettyunits_1.0.2                                  
[171] dbplyr_1.4.2                                       
[172] R.methodsS3_1.7.1                                  
[173] gtable_0.3.0                                       
[174] DBI_1.0.0                                          
[175] git2r_0.26.1                                       
[176] wateRmelon_1.30.0                                  
[177] httr_1.4.1                                         
[178] KernSmooth_2.23-15                                 
[179] stringi_1.4.6                                      
[180] progress_1.2.2                                     
[181] farver_2.0.3                                       
[182] annotate_1.62.0                                    
[183] viridis_0.5.1                                      
[184] xml2_1.2.5                                         
[185] combinat_0.0-8                                     
[186] BiocVersion_3.10.1                                 
[187] bit_1.1-14                                         
[188] pkgconfig_2.0.3                                    
[189] ruv_0.9.7.1                                        
[190] knitr_1.28                                         </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
