<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Getting started</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">methyl-geneset-testing</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/Oshlack/methyl-geneset-testing">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Getting started</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-08-28
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>methyl-geneset-testing/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it's best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200302code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200302)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200302code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200302)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomOshlackmethylgenesettestingtree562f140aa61ac43c285f2da11a48bb1bffe03d11targetblank562f140a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/Oshlack/methyl-geneset-testing/tree/562f140aa61ac43c285f2da11a48bb1bffe03d11" target="_blank">562f140</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomOshlackmethylgenesettestingtree562f140aa61ac43c285f2da11a48bb1bffe03d11targetblank562f140a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/Oshlack/methyl-geneset-testing/tree/562f140aa61ac43c285f2da11a48bb1bffe03d11" target="_blank">562f140</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/figures.nb.html
    Ignored:    code/.job/
    Ignored:    code/old/
    Ignored:    data/.DS_Store
    Ignored:    data/annotations/
    Ignored:    data/cache-intermediates/
    Ignored:    data/cache-region/
    Ignored:    data/cache-rnaseq/
    Ignored:    data/cache-runtime/
    Ignored:    data/datasets/GSE110554-data.RData
    Ignored:    data/datasets/SRP125125/SRR6298258/
    Ignored:    data/datasets/SRP125125/SRR6298271/
    Ignored:    data/datasets/SRP125125/SRR6298273/
    Ignored:    data/datasets/SRP125125/SRR6298278/
    Ignored:    data/datasets/SRP125125/SRR6298281/
    Ignored:    data/datasets/SRP125125/SRR6298284/
    Ignored:    data/datasets/SRP125125/SRR6298286/
    Ignored:    data/datasets/SRP125125/SRR6298299/
    Ignored:    data/datasets/SRP125125/SRR6298302/
    Ignored:    data/datasets/SRP125125/SRR6298307/
    Ignored:    data/datasets/SRP125125/SRR6298310/
    Ignored:    data/datasets/SRP125125/SRR6298313/
    Ignored:    data/datasets/SRP125125/SRR6298315/
    Ignored:    data/datasets/SRP125125/SRR6298328/
    Ignored:    data/datasets/SRP125125/SRR6298331/
    Ignored:    data/datasets/SRP125125/SRR6298336/
    Ignored:    data/datasets/SRP125125/SRR6298339/
    Ignored:    data/datasets/SRP125125/SRR6298342/
    Ignored:    data/datasets/SRP125125/SRR6298344/
    Ignored:    data/datasets/SRP125125/SRR6298365/
    Ignored:    data/datasets/SRP125125/SRR6298370/
    Ignored:    data/datasets/SRP125125/SRR6298373/
    Ignored:    data/datasets/SRP125125/SRR6298376/
    Ignored:    data/datasets/SRP125125/SRR_Acc_List.txt
    Ignored:    data/datasets/SRP125125/SRR_Acc_List_Full.txt
    Ignored:    data/datasets/SRP125125/SraRunTable.txt
    Ignored:    data/datasets/SRP125125/multiqc_data/
    Ignored:    data/datasets/SRP125125/multiqc_report.html
    Ignored:    data/datasets/SRP125125/quants/
    Ignored:    data/datasets/TCGA.KIRC.rds
    Ignored:    data/misc/
    Ignored:    output/.DS_Store
    Ignored:    output/FDR-analysis/
    Ignored:    output/compare-methods/
    Ignored:    output/figures/
    Ignored:    output/methylgsa-params/
    Ignored:    output/outputs.tar.gz
    Ignored:    output/random-cpg-sims/
    Ignored:    output/test/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/gettingStarted.Rmd</code>) and HTML (<code>docs/gettingStarted.html</code>) files. If you've configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Oshlack/methyl-geneset-testing/blob/562f140aa61ac43c285f2da11a48bb1bffe03d11/analysis/gettingStarted.Rmd" target="_blank">562f140</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2020-08-28
</td>
<td>
wflow_publish(c(&quot;analysis/03_expressionGenesets.Rmd&quot;, &quot;analysis/gettingStarted.Rmd&quot;,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Oshlack/methyl-geneset-testing/e66c39ab282fc1b404f5dd82c8c9ec1ff71eee81/docs/gettingStarted.html" target="_blank">e66c39a</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2020-08-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Oshlack/methyl-geneset-testing/blob/ad9e7be3060369217fecb4b7fd98df9570a7b22a/analysis/gettingStarted.Rmd" target="_blank">ad9e7be</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2020-08-14
</td>
<td>
wflow_publish(c(&quot;analysis/figures.Rmd&quot;, &quot;analysis/gettingStarted.Rmd&quot;,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Oshlack/methyl-geneset-testing/555069b9dae4b12c11c1f78b902882688bc37a7b/docs/gettingStarted.html" target="_blank">555069b</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2020-08-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Oshlack/methyl-geneset-testing/blob/91699a8d1b57e40650c7101a0b7db675e67658a1/analysis/gettingStarted.Rmd" target="_blank">91699a8</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2020-08-14
</td>
<td>
wflow_publish(&quot;analysis/_site.yml&quot;, republish = TRUE, all = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Oshlack/methyl-geneset-testing/e16272544567fe917fbc3e2de666e31899209b81/docs/gettingStarted.html" target="_blank">e162725</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2020-07-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Oshlack/methyl-geneset-testing/blob/ea6f88dda55a42c7a1514b69150d73c28bcc2ee0/analysis/gettingStarted.Rmd" target="_blank">ea6f88d</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2020-07-27
</td>
<td>
wflow_publish(c(&quot;analysis/index.Rmd&quot;, &quot;analysis/gettingStarted.Rmd&quot;))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Oshlack/methyl-geneset-testing/d439b3224d9c6dc4ab48db1f14a361d28fb0bbbc/docs/gettingStarted.html" target="_blank">d439b32</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2020-07-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Oshlack/methyl-geneset-testing/blob/6278674aaa36496fbfdb518beff75a6d45e70323/analysis/gettingStarted.Rmd" target="_blank">6278674</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2020-07-27
</td>
<td>
wflow_publish(c(&quot;analysis/index.Rmd&quot;, &quot;analysis/gettingStarted.Rmd&quot;))
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>This page describes how to download the data and code used in this analysis, set up the project directory and rerun the analysis. We have use the <a href="https://jdblischak.github.io/workflowr/" title="Workflowr"><code>workflowr</code></a> package to organise the analysis and insert reproducibilty information into the output documents.</p>
<div id="getting-the-code" class="section level1">
<h1>Getting the code</h1>
<p>All the code and outputs of analysis are available from GitHub at <a href="https://github.com/Oshlack/methyl-geneset-testing" class="uri">https://github.com/Oshlack/methyl-geneset-testing</a>. If you want to replicate the analysis you can either fork the repository and clone it or download the repository as a zipped directory.</p>
<p>Once you have a local copy of the repository you should see the following directory structure:</p>
<ul>
<li><code>analysis/</code> - Contains the RMarkdown documents with the various stages of analysis. These are numbered according to the order they should be run.</li>
<li><code>data/</code> - This directory contains the data files used in the analysis with sub-directories for different data types (see <a href="#data">Getting the data</a> for details). Processed intermediate data files will also be placed here.</li>
<li><code>output/</code> - Directory for output files produced by the analysis, each analysis step has it's own sub-directory.</li>
<li><code>docs/</code> - This directory contains the analysis website hosted at <a href="http://oshlacklab.com/methyl-geneset-testing" class="uri">http://oshlacklab.com/methyl-geneset-testing</a>, including image files.</li>
<li><code>code/</code> - R scripts with custom functions used in some analysis stages. There are sub-directories for scripts associated with different steps in the anlaysis.</li>
<li><code>README.md</code> - README describing the project.</li>
<li><code>.Rprofile</code> - Custom R profile for the project including set up for <code>workflowr</code>.</li>
<li><code>.gitignore</code> - Details of files and directories that are excluded from the repository.</li>
<li><code>_workflowr.yml</code> - Workflowr configuration file.</li>
<li><code>methyl-geneset-testing.Rproj</code> - RStudio project file.</li>
</ul>
</div>
<div id="data" class="section level1">
<h1>Getting the data</h1>
<p>In this project we have used data from several publicly avilable datasets. Flow-sorted, blood cell methylation data generated using Illumina HumanMethylationEPIC arrays. Normal kidney methylation data from The Cancer Genome Atlas (TCGA) kidney clear-cell carcinoma (KIRC) cohort, which was generated using Illumina HumanMethylation450 arrays. These are both automatically downloaded as part of the analysis directly from the Bioconductor <a href="expt-hub">ExperimentHub</a>.</p>
<p>We also use a flow-sorted, blood cell RNAseq dataset, which can be downloaded from GEO at <a href="geo-rnaseq">GSE107011</a> or SRA at <a href="sra-rnaseq">SRP125125</a>.</p>
<p>Once the RNAseq data has been downloaded it needs to be extracted, placed in the correct directories and quasi-mapped and quantified using <a href="https://combine-lab.github.io/salmon/">Salmon</a>. The approach we took is described <a href="expressionGenesets.html">here</a>. The downstream analysis code assumes the following directory structure inside the <code>data/</code> directory:</p>
<ul>
<li><code>datasets</code>
<ul>
<li><code>SRP125125</code>
<ul>
<li><code>quants</code>
<ul>
<li><code>SRR6298258_quant</code></li>
<li>...</li>
<li>...</li>
<li><code>SRR6298376_quant</code></li>
</ul></li>
</ul></li>
</ul></li>
</ul>
<p>Some additional data files used during the analysis are provided as part of the repository.</p>
<ul>
<li><code>genesets</code>
<ul>
<li><code>GO-immune-system-process.txt</code></li>
<li><code>kegg-immune-related-pathways.csv</code></li>
</ul></li>
<li><code>datasets</code>
<ul>
<li><code>SRP125125</code>
<ul>
<li><code>SraRunTableFull.txt</code></li>
</ul></li>
</ul></li>
</ul>
<p>Intermediate data files created during the analysis will be placed in:</p>
<ul>
<li><code>annotations</code></li>
<li><code>cache-intermediates</code></li>
<li><code>cache-region</code></li>
<li><code>cache-rnaseq</code></li>
<li><code>cache-runtime</code></li>
</ul>
<p>These are used by later stages of the analysis so should not be moved, altered or deleted.</p>
</div>
<div id="running-the-analysis" class="section level1">
<h1>Running the analysis</h1>
<p>The analysis directory contains the following analysis files:</p>
<pre><code>[1] &quot;01_exploreArrayBiasEPIC.Rmd&quot; &quot;02_exploreArrayBias450.Rmd&quot; 
[3] &quot;03_expressionGenesets.Rmd&quot;   &quot;04_fdrAnalysis.Rmd&quot;         
[5] &quot;05_exploreData.Rmd&quot;          &quot;06_runTimeComparison.Rmd&quot;   
[7] &quot;07_regionAnalysis.Rmd&quot;       &quot;08_methylGSAParamSweep.Rmd&quot; </code></pre>
<p>As indicated by the numbering they should be run in this order. If you want to rerun the entire analysis this can be easily done using <code>workflowr</code>.</p>
<pre class="r"><code>workflowr::wflow_build(republish = TRUE)</code></pre>
<p>It is important to consider the computer and environment you are using before doing this. Running this analysis from scratch requires a considerable amount of time, disk space and memory. <strong>Some stages of the analysis need to be executed on a HPC to generate results required by downstream steps.</strong> If you do no have access to a HPC to perform these analyses using the code provided, you can download pre-computed RDS files containing the results from <a href="https://doi.org/10.5281/zenodo.4005288"><img src="https://zenodo.org/badge/DOI/10.5281/zenodo.4005288.svg" alt="DOI" /></a>.</p>
<p>To use the pre-computed RDS objects, after cloning or downloading the GitHub repository to your computer, please extract the <code>outputs.tar.gz</code> archive under the <code>output</code> directory, using <code>tar -xvf outputs.tar.gz</code>.</p>
<p>It is also possible to run individual stages of the analysis, either by providing the names of the file you want to run to <code>workflowr::wflow_build()</code> or by manually knitting the document (for example using the 'Kint' button in RStudio).</p>
<p>Once <strong>all</strong> the analyses have been rerun, the manuscript figures can be generated using the code provided <a href="figures.html">here</a>.</p>
<pre class="r"><code>devtools::session_info()</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.3 (2020-02-29)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Mojave 10.14.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.5      rstudioapi_0.11 whisker_0.4     knitr_1.29     
 [5] magrittr_1.5    here_0.1        R6_2.4.1        rlang_0.4.7    
 [9] stringr_1.4.0   tools_3.6.3     xfun_0.15       git2r_0.27.1   
[13] htmltools_0.5.0 ellipsis_0.3.1  rprojroot_1.3-2 yaml_2.2.1     
[17] digest_0.6.25   tibble_3.0.3    lifecycle_0.2.0 crayon_1.3.4   
[21] later_1.1.0.1   vctrs_0.3.2     promises_1.1.1  fs_1.4.2       
[25] glue_1.4.1      evaluate_0.14   rmarkdown_2.3   stringi_1.4.6  
[29] compiler_3.6.3  pillar_1.4.6    backports_1.1.8 httpuv_1.5.4   
[33] pkgconfig_2.0.3</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
