<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Home</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">methyl-geneset-testing</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/Oshlack/methyl-geneset-testing">
    <span class="fas fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Home</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-05-21
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>methyl-geneset-testing/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsatlassianpetermacorgaubitbucketscmosmethylgenesettestingtreeae9d8f7bcf6675246b3afb0c57e906b1b9e27ec1targetblankae9d8f7a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/tree/ae9d8f7bcf6675246b3afb0c57e906b1b9e27ec1" target="_blank">ae9d8f7</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsatlassianpetermacorgaubitbucketscmosmethylgenesettestingtreeae9d8f7bcf6675246b3afb0c57e906b1b9e27ec1targetblankae9d8f7a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/tree/ae9d8f7bcf6675246b3afb0c57e906b1b9e27ec1" target="_blank">ae9d8f7</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/figures.nb.html
    Ignored:    code/.DS_Store
    Ignored:    code/.Rhistory
    Ignored:    code/.job/
    Ignored:    code/old/
    Ignored:    data/.DS_Store
    Ignored:    data/annotations/
    Ignored:    data/cache-intermediates/
    Ignored:    data/cache-region/
    Ignored:    data/cache-rnaseq/
    Ignored:    data/cache-runtime/
    Ignored:    data/datasets/.DS_Store
    Ignored:    data/datasets/GSE110554-data.RData
    Ignored:    data/datasets/GSE120854/
    Ignored:    data/datasets/GSE120854_RAW.tar
    Ignored:    data/datasets/GSE135446-data.RData
    Ignored:    data/datasets/GSE135446/
    Ignored:    data/datasets/GSE135446_RAW.tar
    Ignored:    data/datasets/GSE45459-data.RData
    Ignored:    data/datasets/GSE45459_Matrix_signal_intensities.txt
    Ignored:    data/datasets/GSE45460/
    Ignored:    data/datasets/GSE45460_RAW.tar
    Ignored:    data/datasets/GSE95460_RAW.tar
    Ignored:    data/datasets/GSE95460_RAW/
    Ignored:    data/datasets/GSE95462-data.RData
    Ignored:    data/datasets/GSE95462/
    Ignored:    data/datasets/GSE95462_RAW/
    Ignored:    data/datasets/SRP100803/
    Ignored:    data/datasets/SRP125125/.DS_Store
    Ignored:    data/datasets/SRP125125/SRR6298*/
    Ignored:    data/datasets/SRP125125/SRR_Acc_List.txt
    Ignored:    data/datasets/SRP125125/SRR_Acc_List_Full.txt
    Ignored:    data/datasets/SRP125125/SraRunTable.txt
    Ignored:    data/datasets/SRP125125/multiqc_data/
    Ignored:    data/datasets/SRP125125/multiqc_report.html
    Ignored:    data/datasets/SRP125125/quants/
    Ignored:    data/datasets/SRP166862/
    Ignored:    data/datasets/SRP217468/
    Ignored:    data/datasets/TCGA.BRCA.rds
    Ignored:    data/datasets/TCGA.KIRC.rds
    Ignored:    data/misc/
    Ignored:    output/--exclude
    Ignored:    output/.DS_Store
    Ignored:    output/FDR-analysis/
    Ignored:    output/compare-methods/
    Ignored:    output/figures/
    Ignored:    output/methylgsa-params/
    Ignored:    output/outputs-1.tar.gz
    Ignored:    output/outputs.tar.gz
    Ignored:    output/random-cpg-sims/

Untracked files:
    Untracked:  analysis/old/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/index.Rmd</code>) and HTML (<code>docs/index.html</code>) files. If you've configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/ae9d8f7bcf6675246b3afb0c57e906b1b9e27ec1/analysis/index.Rmd" target="_blank">ae9d8f7</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2021-05-21
</td>
<td>
wflow_publish(&quot;analysis/index.Rmd&quot;)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/4c57176ec3664b1804f67dfc771a928a291d4f46/docs/index.html" target="_blank">4c57176</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2021-04-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/62fdddc3d48739e5e8f80a02b8ecef78a9b0c456/analysis/index.Rmd" target="_blank">62fdddc</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2021-04-13
</td>
<td>
Removed unused links
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/a9a3a86439630816564e1ec217381de691d4d4f0/docs/index.html" target="_blank">a9a3a86</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2021-04-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/98431578adb30dcaa808cddbfe752948d7eb5385/analysis/index.Rmd" target="_blank">9843157</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2021-04-06
</td>
<td>
wflow_publish(c(&quot;analysis/index.Rmd&quot;, &quot;analysis/07_regionAnalysisBcells.Rmd&quot;,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/17cc1c91ddc1dcb9963ff89eb6b69915c4624d60/docs/index.html" target="_blank">17cc1c9</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2021-04-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/58c95d6825d9f7381119d3ed6edcaa38097e71de/analysis/index.Rmd" target="_blank">58c95d6</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2021-04-06
</td>
<td>
wflow_publish(c(&quot;analysis/index.Rmd&quot;, &quot;analysis/04_expressionGenesetsBcells.Rmd&quot;,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/6ccbf3402d72b3c25d91d2db7a00f729a5b73082/docs/index.html" target="_blank">6ccbf34</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-04-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/44ef1e473eb9f4af9031d959f96b37f28ce58b2a/analysis/index.Rmd" target="_blank">44ef1e4</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-04-01
</td>
<td>
wflow_publish(&quot;analysis/index.Rmd&quot;)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/41ab80497b556b873a370074f05612968f034df7/docs/index.html" target="_blank">41ab804</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-04-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/eeaf08090de92c819b1a352f466978fafac385cc/analysis/index.Rmd" target="_blank">eeaf080</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-04-01
</td>
<td>
wflow_publish(c(&quot;analysis/index.Rmd&quot;, &quot;analysis/04_expressionGenesetsCellLines.Rmd&quot;,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/c7c54db725866058c6a178f7205e10e8d818d65e/docs/index.html" target="_blank">c7c54db</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-03-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/f8067882394d49c33a2e9026964ccd9018d940d8/analysis/index.Rmd" target="_blank">f806788</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-03-31
</td>
<td>
wflow_publish(c(&quot;analysis/index.Rmd&quot;, &quot;analysis/04_expressionGenesetsFibroid.Rmd&quot;,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/ee6a73dfa3c339fe0be8412b22689d84debdbe67/docs/index.html" target="_blank">ee6a73d</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2021-03-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/586dde98f653e76657e1db55ed1e1a58ad1a93b1/analysis/index.Rmd" target="_blank">586dde9</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2021-03-29
</td>
<td>
wflow_publish(&quot;analysis/index.Rmd&quot;)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/7ce65655613253187ad026539bacb5f648a4ba27/docs/index.html" target="_blank">7ce6565</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2021-03-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/f9bf9cd5042ec30c20992007fa1cc908eccd5f6a/analysis/index.Rmd" target="_blank">f9bf9cd</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2021-03-29
</td>
<td>
wflow_publish(&quot;analysis/index.Rmd&quot;)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/f657fd060959f28c9175ea1e3765317dd35c8a1d/docs/index.html" target="_blank">f657fd0</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2021-03-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/319cb749a3e786531e79e1d2e9470c66bb041934/analysis/index.Rmd" target="_blank">319cb74</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2021-03-29
</td>
<td>
wflow_publish(&quot;analysis/index.Rmd&quot;)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/e42657140af0ea6bbb0a8d47fdeb197cb9c0d75e/docs/index.html" target="_blank">e426571</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2021-03-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/a27f986ea208aca7433bc0752fc4a8454ad1381f/analysis/index.Rmd" target="_blank">a27f986</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2021-03-29
</td>
<td>
wflow_publish(&quot;analysis/index.Rmd&quot;)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/c65d2654a544153b6933fd32a2bc956da38e2dbf/docs/index.html" target="_blank">c65d265</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-03-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/bda4d93c1d423cda32ca14878ab9215bf7d3991f/analysis/index.Rmd" target="_blank">bda4d93</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-03-22
</td>
<td>
wflow_publish(c(&quot;analysis/index.Rmd&quot;, &quot;analysis/04_fdrAnalysis.Rmd&quot;))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/59d648cac03da63b7bf28b7705ddf7114868d1a7/docs/index.html" target="_blank">59d648c</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-03-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/82d6037a16a3ac870482ee6dda1534690bd9bc4b/analysis/index.Rmd" target="_blank">82d6037</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-03-22
</td>
<td>
wflow_publish(c(&quot;analysis/index.Rmd&quot;))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/190014a05845d1e138bf03eddbcdfc544673dafb/docs/index.html" target="_blank">190014a</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-03-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/0f8a97968b88ba8826d391134657d9723bb208ef/analysis/index.Rmd" target="_blank">0f8a979</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-03-22
</td>
<td>
wflow_publish(c(&quot;analysis/03_expressionGenesetsNew.Rmd&quot;, &quot;analysis/index.Rmd&quot;))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/3af3b82135ee41031301b39fb8059195e0c893c1/docs/index.html" target="_blank">3af3b82</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2020-08-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/864b388a17115cf5252e11efc6896a08c3165715/analysis/index.Rmd" target="_blank">864b388</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2020-08-28
</td>
<td>
wflow_publish(&quot;analysis/index.Rmd&quot;)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/df00043c6144ec1b2e33ce19eb94be6996db4c2b/docs/index.html" target="_blank">df00043</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2020-08-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/59480d3f934fd9b9ba8f72d90d8c08728bb28f6f/analysis/index.Rmd" target="_blank">59480d3</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2020-08-28
</td>
<td>
wflow_publish(&quot;analysis/index.Rmd&quot;)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/d3675c57e18b80e964f6c175e76e1e9fdde07120/docs/index.html" target="_blank">d3675c5</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2020-08-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/562f140aa61ac43c285f2da11a48bb1bffe03d11/analysis/index.Rmd" target="_blank">562f140</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2020-08-28
</td>
<td>
wflow_publish(c(&quot;analysis/03_expressionGenesets.Rmd&quot;, &quot;analysis/gettingStarted.Rmd&quot;,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/555069b9dae4b12c11c1f78b902882688bc37a7b/docs/index.html" target="_blank">555069b</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2020-08-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/91699a8d1b57e40650c7101a0b7db675e67658a1/analysis/index.Rmd" target="_blank">91699a8</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2020-08-14
</td>
<td>
wflow_publish(&quot;analysis/_site.yml&quot;, republish = TRUE, all = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/696481cde2d1b9ab3e031fda924e15f6d340ba9b/docs/index.html" target="_blank">696481c</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2020-08-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/5e79b2e0bd7880aa01f837ca3784a8de595653e4/analysis/index.Rmd" target="_blank">5e79b2e</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2020-08-10
</td>
<td>
wflow_publish(c(&quot;analysis/regionAnalysis.Rmd&quot;, &quot;analysis/index.Rmd&quot;,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/e16272544567fe917fbc3e2de666e31899209b81/docs/index.html" target="_blank">e162725</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2020-07-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/ea6f88dda55a42c7a1514b69150d73c28bcc2ee0/analysis/index.Rmd" target="_blank">ea6f88d</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2020-07-27
</td>
<td>
wflow_publish(c(&quot;analysis/index.Rmd&quot;, &quot;analysis/gettingStarted.Rmd&quot;))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/d439b3224d9c6dc4ab48db1f14a361d28fb0bbbc/docs/index.html" target="_blank">d439b32</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2020-07-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/6278674aaa36496fbfdb518beff75a6d45e70323/analysis/index.Rmd" target="_blank">6278674</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2020-07-27
</td>
<td>
wflow_publish(c(&quot;analysis/index.Rmd&quot;, &quot;analysis/gettingStarted.Rmd&quot;))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/e631347d2fe362ed5b7fb0ec0a237be681daa314/docs/index.html" target="_blank">e631347</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2020-07-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/fd3abe58ef898e25fc2aa4ef58b8a25a672c4f77/docs/index.html" target="_blank">fd3abe5</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-06-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/1be6051747a985e644fc44036fca1c5f3d478b13/analysis/index.Rmd" target="_blank">1be6051</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-06-16
</td>
<td>
wflow_publish(&quot;analysis/index.Rmd&quot;)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/2460d188cad32fa6996662a68e73e4591e6a964c/docs/index.html" target="_blank">2460d18</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-06-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/5221ff045f63160f03b8b32015bee9b583395c1c/analysis/index.Rmd" target="_blank">5221ff0</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-06-16
</td>
<td>
wflow_publish(&quot;analysis/index.Rmd&quot;)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/61f5fff9f6d243878a74ee7ee4114f63c99d090a/docs/index.html" target="_blank">61f5fff</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2020-06-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/4e771035387622cad43bb9ebd7c6a8327c36efdc/analysis/index.Rmd" target="_blank">4e77103</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2020-06-01
</td>
<td>
wflow_publish(c(&quot;analysis/index.Rmd&quot;, &quot;analysis/gomethByFeature.Rmd&quot;))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/922496602738ae496a3b3caff5183b0c0fd4140e/docs/index.html" target="_blank">9224966</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-05-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/49cf1670562da834f16a26dd224cc38fb074216a/analysis/index.Rmd" target="_blank">49cf167</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-05-29
</td>
<td>
wflow_publish(c(&quot;analysis/methylGSAParamSweep.Rmd&quot;, &quot;analysis/index.Rmd&quot;))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/c381d87e61054c0eba8a0695885875eda65fa260/docs/index.html" target="_blank">c381d87</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-05-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/ff86a788617ffec04756bd48e5d5ff574aac6ca8/analysis/index.Rmd" target="_blank">ff86a78</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-05-19
</td>
<td>
wflow_publish(&quot;analysis/index.Rmd&quot;)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/2c18577124cc295d7e159311cad8754577ec37f9/docs/index.html" target="_blank">2c18577</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-05-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/b7daadd4026c6dab6bc73614ae0152d84d807df2/analysis/index.Rmd" target="_blank">b7daadd</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-05-19
</td>
<td>
wflow_publish(&quot;analysis/index.Rmd&quot;)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/f2da7f99969ab14f267f78072bbee83baec2c0d6/docs/index.html" target="_blank">f2da7f9</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-05-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/68a0f24f7acb1a1be1d1d154739e0022d1733735/analysis/index.Rmd" target="_blank">68a0f24</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-05-15
</td>
<td>
wflow_publish(c(&quot;analysis/index.Rmd&quot;, &quot;analysis/runTimeComparison.Rmd&quot;))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/06648a4b8d1af9a4113520a3844b4400395f7a21/docs/index.html" target="_blank">06648a4</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2020-04-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/89af32354ddfff9490bb55a4340a7d9a4d0b9605/analysis/index.Rmd" target="_blank">89af323</a>
</td>
<td>
JovMaksimovic
</td>
<td>
2020-04-27
</td>
<td>
wflow_publish(c(&quot;analysis/index.Rmd&quot;, &quot;analysis/exploreData.Rmd&quot;,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/64432deef9b7ce9126b6f4b7edd30520829879fb/docs/index.html" target="_blank">64432de</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-04-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/90b90efadca81fd1af36d08cbf7427ab4349b479/analysis/index.Rmd" target="_blank">90b90ef</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-04-17
</td>
<td>
Updated home link to FDR analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/244474d1ededdea7d83b837736931b8a5e75713b/docs/index.html" target="_blank">244474d</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-03-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/d7cd66e771da5cd568aab4e5999ae8b7be7a88ec/analysis/index.Rmd" target="_blank">d7cd66e</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-03-02
</td>
<td>
Initial Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/methyl-geneset-testing/blob/18404093182d57d32414b057613f96813408d5f5/analysis/index.Rmd" target="_blank">1840409</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-03-02
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="gene-set-enrichment-analysis-for-genome-wide-dna-methylation-data" class="section level2">
<h2>Gene set enrichment analysis for genome-wide DNA methylation data</h2>
<p>This site contains the results of the analyses presented in <a href="https://www.biorxiv.org/content/10.1101/2020.08.24.265702v1"><em>“Gene set enrichment analysis for genome-wide DNA methylation data”</em></a>. Follow the links below to view the different parts of the analysis. For details on how to reproduce the complete analysis, please see the <a href="gettingStarted.html">Getting started</a> page.</p>
</div>
<div id="abstract" class="section level2">
<h2>Abstract</h2>
<p>DNA methylation is one of the most commonly studied epigenetic marks, due to its role in disease and development. Illumina methylation arrays have been extensively used to measure methylation across the human genome. Methylation array analysis has primarily focused on preprocessing, normalisation and identification of differentially methylated CpGs and regions. GOmeth and GOregion are new methods for performing unbiased gene set testing following differential methylation analysis. Benchmarking analyses demonstrate GOmeth outperforms other approaches and GOregion is the first method for gene set testing of differentially methylated regions. Both methods are publicly available in the <a href="https://www.bioconductor.org/packages/release/bioc/html/missMethyl.html">missMethyl</a> Bioconductor R package.</p>
</div>
<div id="authors" class="section level2">
<h2>Authors</h2>
<p>Jovana Maksimovic<sup>1,2,3</sup>, Alicia Oshlack<sup>1,4</sup>, Belinda Phipson<sup>1,2+</sup></p>
<p><sup>1</sup> Peter MacCallum Cancer Centre, Melbourne, Victoria, 3000, Australia <sup>2</sup> Department of Pediatrics, University of Melbourne, Parkville, Victoria, 3010, Australia <sup>3</sup> Murdoch Children’s Research Institute, Parkville, Victoria, 3052, Australia <sup>4</sup> School of Biosciences, University of Melbourne, Parkville, Victoria, 3010, Australia</p>
<p><sup>+</sup> corresponding author</p>
</div>
<div id="analysis" class="section level2">
<h2>Analysis</h2>
<ol style="list-style-type: decimal">
<li><p><a href="01_exploreArrayBiasEPIC.html">Explore EPIC array bias</a> Explore the various array biases on the EPIC array that affect gene set testing.</p></li>
<li><p><a href="02_exploreArrayBias450.html">Explore 450k array bias</a> Explore the various array biases on the 450k array that affect gene set testing.</p></li>
<li><ol style="list-style-type: lower-alpha">
<li><p><a href="03_fdrAnalysisKIRC.html">Compare FDR of different methods (KIRC data)</a> Analyse the normal samples from a 450k array KIRC TCGA dataset using various gene set testing methods to estimate their false discovery rate control.</p></li>
<li><p><a href="03_fdrAnalysisBRCA.html">Compare FDR of different methods (BRCA data)</a> Analyse the normal samples from a 450k array BRCA TCGA dataset using various gene set testing methods to estimate their false discovery rate control.</p></li>
</ol></li>
<li><ol style="list-style-type: lower-alpha">
<li><p><a href="04_expressionGenesets.html">Generate a blood cell RNAseq &quot;truth&quot; set</a> Analyse an RNAseq sorted blood cell dataset and identify the top ranked gene sets for each cell type comparison.</p></li>
<li><p><a href="04_expressionGenesetsBcells.html">Generate a B-cell development gene exression &quot;truth&quot; set</a> Analyse Affymetrix gene expression microarray data of B-cell development and identify the top ranked gene sets for each stage comparison.</p></li>
</ol></li>
<li><ol style="list-style-type: lower-alpha">
<li><p><a href="05_compareMethods.html">Compare performance of different methods (EPIC data)</a> Analyse an EPIC array sorted blood cell dataset using various gene set testing methods. Compare how well the different methods perform using several metrics.</p></li>
<li><p><a href="05_compareMethodsBcells.html">Compare performance of different methods (450K data)</a> Analyse a 450k array dataset of B-cell development data using various gene set testing methods. Compare how well the different methods perform using several metrics.</p></li>
</ol></li>
<li><p><a href="06_runTimeComparison.html">Compare run-time of different methods</a> Analyse an EPIC array sorted blood cell dataset using various gene set testing methods. Compare the run-time of the different methods.</p></li>
<li><ol style="list-style-type: lower-alpha">
<li><p><a href="07_regionAnalysis.html">Evaluate GOregion (EPIC data)</a> Evalulate GOregion, our extension of gometh for geneset testing of differentially methylated regions (DMRs) identified by DMR finding software.</p></li>
<li><p><a href="07_regionAnalysisBcells.html">Evaluate GOregion (450K data)</a> Evalulate GOregion, our extension of gometh for geneset testing of differentially methylated regions (DMRs) identified by DMR finding software.</p></li>
</ol></li>
<li><p><a href="methylGSAParamSweep.html">Effect of gene set size parameters on methylGSA</a> Analyse an EPIC array sorted blood cell dataset using various gene set testing methods. Compare the run-time of the different methods.</p></li>
</ol>
</div>
<div id="licenses" class="section level2">
<h2>Licenses</h2>
<p>The code in this analysis is covered by the <a href="https://choosealicense.com/licenses/mit/" title="MIT License">MIT license</a> and the written content on this website is covered by a <a href="https://choosealicense.com/licenses/cc-by-4.0/" title="CC-BY License">Creative Commons CC-BY</a> license.</p>
</div>
<div id="citations" class="section level2">
<h2>Citations</h2>
<blockquote>
<p>Maksimovic J, Oshlack O, Phipson B. Gene set enrichment analysis for genome-wide DNA methylation data. bioRxiv. 2020. DOI: <a href="https://www.biorxiv.org/content/10.1101/2020.08.24.265702v1" title="Pre-print">https://doi.org/10.1101/2020.08.24.265702</a>.</p>
</blockquote>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
