---
title: "Null sims with normal data"
author: "Belinda Phipson"
date: "17 May 2019"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Introduction

Here we use the normal samples from the TCGA kidney cancer dataset to assess the performance of gometh, methylGSA and ebGSEA.

```{r}
# First load missMethyl
library(limma)
library(minfi)
library(BiasedUrn)
library(org.Hs.eg.db)
library(IlluminaHumanMethylationEPICanno.ilm10b4.hg19)
library(IlluminaHumanMethylation450kanno.ilmn12.hg19)
#library(missMethyl)
library(ChAMP)
library(methylGSA)
```

```{r}
ggplotColors <- function(g){

  d <- 360/g

  h <- cumsum(c(15, rep(d,g - 1)))

  hcl(h = h, c = 100, l = 65)

}
```

```{r}
makeqq <- function(pvals, main=NULL, ...) {
    o <- -log10(sort(pvals,decreasing=F))
    e <- -log10(1:length(o)/length(o))
    plot(e, o, pch=19, cex=1, main=main, ...,
        xlab=expression(Expected~~-log[10](italic(p))),
        ylab=expression(Observed~~-log[10](italic(p))),
        xlim=c(0,max(e)), ylim=c(0,max(o)))
    lines(e,e,col="red")
}
```

# Reading in the data
```{r}
eh = ExperimentHub()
```

```{r}
setwd("//data.mcri.edu.au/bioi1/shared/public_data/kidneyTCGA450k")
kNorm450 = read.csv("./kidney_normal/file_manifest_normal_kidney450.txt",sep="\t",header=TRUE,stringsAsFactors=FALSE)
kNorm450$File.Name = substr(kNorm450$File.Name,1,17)
kNorm450 = unique(kNorm450)
kNorm450$Sentrix_ID = substr(kNorm450$File.Name,1,10)
kNorm450$Sentrix_Position = substr(kNorm450$File.Name,12,17)
kNorm450$Sample_Group = "normal"
kNorm450$Basename = paste("./kidney_normal/",kNorm450$File.Name,sep="")
kNorm450 <- kNorm450[-1,]

rgSet450 = read.metharray.exp(targets = kNorm450)

ann450k = getAnnotation(IlluminaHumanMethylation450kanno.ilmn12.hg19)
```

# Sample QC

```{r}
detP<-detectionP(rgSet450)

graphics:::barplot(colMeans(detP))
```

# Normalisation

```{r}
mSetSq <- preprocessQuantile(rgSet450)
par(mfrow=c(1,2))
densityPlot(rgSet450)
densityPlot(getBeta(mSetSq))

par(mfrow=c(1,1))
sex <- getSex(mSetSq)

plotMDS(getM(mSetSq), top=1000, gene.selection="common",col=ggplotColors(2)[factor(sex$predictedSex)])
legend("top",legend=levels(factor(sex$predictedSex)),fill=ggplotColors(2))
```

# Filtering probes
```{r}
detP <- detP[match(featureNames(mSetSq),rownames(detP)),]

# remove any probes that have failed in one or more samples
keep <- rowSums(detP < 0.01) == ncol(mSetSq)
table(keep)
mSetSqFlt <- mSetSq[keep,]

# Filter out probes on X and Y chromosome
keep <- !(featureNames(mSetSqFlt) %in% ann450k$Name[ann450k$chr %in%
                                                               c("chrX","chrY")])
table(keep)
mSetSqFlt <- mSetSqFlt[keep,]

mSetSqFlt <- dropLociWithSnps(mSetSqFlt)

plotMDS(getM(mSetSqFlt), top=1000, gene.selection="common",col=ggplotColors(2)[factor(sex$predictedSex)])
```

# Save object

```{r}
#save(mSetSqFlt, 
#     file="C:/Users/belinda.phipson/_Work/_MCRI/_Research\ projects/MethylationGST/data/GRSetKidNorms.Rdata")
```

# Extract Mvalues and Beta values

```{r}
#load(file="C:/Users/belinda.phipson/_Work/_MCRI/_Research\ projects/MethylationGST/data/GRSetKidNorms.Rdata")
mVals <- getM(mSetSqFlt)
bVals <- getBeta(mSetSqFlt)
```

# Differential methylation analysis

```{r}
set.seed(1)
group <- rbinom(ncol(mVals),1,0.5)
```

```{r}
design <- model.matrix(~group)
fit <- lmFit(mVals,design)
fit <- eBayes(fit)
summary(decideTests(fit))
topTable(fit,coef=2)
top <- topTable(fit,coef=2,n=Inf)
```

# Gene set testing

## gometh

```{r}
gom <- gometh(sig.cpg = rownames(top)[1:500],array.type = "450K",plot.bias = TRUE)
topGSA(gom)
n <- sum(gom$P.DE<0.05)
```

## ebayGSEA

```{r}
myebayGSEA <- champ.ebGSEA(beta=bVals,pheno=group,arraytype="450K")
```

## methylGSA

```{r}
pval <- top$P.Value
names(pval) <- rownames(top)
res1 = methylglm(cpg.pval = pval, minsize = 1, 
                 maxsize = 9664, GS.type = "GO")
```

```{r}
res2 = methylRRA(cpg.pval = pval, method = "ORA", 
                    minsize = 1, maxsize = 9644)
```

```{r}
res3 = methylRRA(cpg.pval = pval, method = "GSEA", 
                    minsize = 1, maxsize = 9644)
```

