---
title: 'Gene set testing for Illumina HumanMethylation Arrays'
subtitle: 'Generating "truth" gene sets from RNA-seq data'
author: "Jovana Maksimovic, Alicia Oshlack and Belinda Phipson"
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output: html_document
---

```{r, message=FALSE, warning=FALSE, results='hide'}
library(tximport)
library(here)
library(tidyverse)
library(EnsDb.Hsapiens.v75)
library(readr)
library(limma)
library(edgeR)
library(glue)
library(patchwork)
library(biobroom)
library(ChAMP)
source(here("code/utility.R"))
```

# Data download, mapping and quantification

The RNAseq dataset used to generate the "truth" gene sets is avoilable from GEO at [GSE107011](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE107011) or SRA at [SRP125125](https://www.ncbi.nlm.nih.gov/sra?term=SRP125125).

The RNAseq data was quasi-mapped and quantified using Salmon (v1.2.1) with the [hg19_cdna](http://refgenomes.databio.org/v2/asset/hg19_cdna/fasta/archive?tag=default) human transcriptome downloaded from [refgenie](http://refgenomes.databio.org/index). The code used to perform the Salmon quasi-mapping and quantification can be found in `code/salmon-quant.sh`. 

The Salmon mapping statistics can be veiwed [here](multiqc_report.html).

For downstream analysis that quantification files (`quant.sf`) for each sample are expected to be present in the following directory structure:

* `data`
    * `datasets`
        * `SRP125125`
            * `quants`
                * `SRR6298258_quant`
                * ...
                * ...
                * `SRR6298376_quant`
                
## Data import

Load sample information and file names.

```{r}
targets <- read_csv(here("data/datasets/SRP125125/SraRunTableFull.txt"))
targets$Cell_type <- gsub(" ", "_", targets$Cell_type)
targets$Cell_type <- gsub("-", "_", targets$Cell_type)
targets
```

Setup file paths and sample names.

```{r}
files <- list.files(here("data/datasets/SRP125125/quants"), 
                    recursive = TRUE, pattern = "quant.sf", full.names = TRUE)
pos <- regexpr("SRR6298[0-9][0-9][0-9]", files, perl = TRUE)
names(files) <- unname(substr(files, pos, 
                              pos[1] + attr(pos, "match.length") - 1))
head(files)
```

Associate transcripts with gene IDs for gene-level summarization.

```{r}
edb <- EnsDb.Hsapiens.v75
tx2gene <- transcripts(edb, columns = c("tx_id", "gene_id"), 
                       return.type = "DataFrame")
tx2gene
```

Import gene-level counts and abundances.

```{r}
txiG <- tximport(files, type = "salmon", tx2gene = tx2gene, 
                  countsFromAbundance = "lengthScaledTPM", 
                  ignoreTxVersion = TRUE)
colnames(txiG$counts) <- names(files)
head(txiG$counts)
```

Set up `DGElist` object for downstream analysis.

```{r}
z <- DGEList(txiG$counts)
z$genes <- ensembldb::genes(edb, filter = GeneIdFilter(rownames(z)), 
                 columns = c("gene_id", "symbol", "entrezid"), 
                 return.type = "DataFrame")
z$genes$entrezid <- sapply(z$genes$entrezid, function(x) x[1])
z$genes$length <- rowMedians(txiG$length)
targets <- targets[match(colnames(z), targets$Run), ]
z$samples$group <- targets$Cell_type
z
```

# Quality control

Genes that do not have an adequate number of reads in any sample should be filtered out prior to downstream analyses. From a biological perspective, genes that are not expressed at a biologically meaningful level in any condition are not of interest. Statistically, we get a better estimate of the mean-variance relationship in the data and reduce the number of statistical tests that are performes during differential expression analyses.

Filter out lowly expressed genes and calculate TMM normalisation factors.

```{r}
keep <- filterByExpr(z, group = z$samples$group)
x <- z[keep, ]
y <- calcNormFactors(x)
y
```

Plotting the distribution log-CPM values shows that a majority of genes within each sample are either not expressed or lowly-expressed with log-CPM values that are small or negative.

```{r, fig.width=9}
L <- mean(z$samples$lib.size) * 1e-6
M <- median(z$samples$lib.size) * 1e-6

dat <- tidy(z, addSamples = TRUE)
dat$cpm <- reshape2::melt(cpm(z, log = TRUE), value.name = "cpm")$cpm
p1 <- ggplot(dat, aes(x = cpm, colour = group)) +
    geom_density() +
    labs(colour = "Cell Type", x = "Log CPM", y = "Density", 
         title = "Unfiltered") +
    geom_vline(xintercept = log2(10/M + 2/L), linetype = "dashed")

dat <- tidy(y, addSamples = TRUE)
dat$cpm <- reshape2::melt(cpm(y, log = TRUE), value.name = "cpm")$cpm
p2 <- ggplot(dat, aes(x = cpm, colour = group)) +
    geom_density() +
    labs(colour = "Cell Type", x = "Log CPM", y = "Density",
         title = "Filtered") +
    geom_vline(xintercept = log2(10/M + 2/L), linetype = "dashed")

p1 + p2 + plot_layout(guides = "collect") & theme(legend.position = "bottom")
```


```{r}
cellPal <- RColorBrewer::brewer.pal(6, "Dark2")[c(4, 5, 1, 2, 3, 6)]
dat <- tidy(y, addSamples = TRUE)
p1 <- ggplot(dat, aes(x = sample, y = lib.size, fill = group)) +
    geom_bar(stat = "identity") +
    labs(fill = "Cell Type", x = "Sample", y = "Library Size") +
    scale_fill_manual(values = cellPal) +
    theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 7),
          legend.position = "bottom")

dat$cpm <- reshape2::melt(cpm(y, log = TRUE), value.name = "cpm")$cpm
p2 <- ggplot(dat, aes(x = sample, y = cpm, fill = group)) +
    geom_boxplot(show.legend = FALSE) +
    labs(x = "Sample", y = "CPM") +
    theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 7),
          legend.position = "bottom") +
    scale_fill_manual(values = cellPal) 

p <- p1 + p2 + plot_layout(guides = "collect") & theme(legend.position = "bottom")
p
```

Save figure for use in manuscript.

```{r}
outDir <- here::here("output/figures")
if (!dir.exists(outDir)) dir.create(outDir)

fig <- here("output/figures/SFig-4A.rds")
saveRDS(p1, fig, compress = FALSE)


fig <- here("output/figures/SFig-4B.rds")
saveRDS(p2, fig, compress = FALSE)
```

Multi-dimensional scaling (MDS) plots show the largest sources of variation in the data. They are a good way of exploring the relationships between the samples and identifying structure in the data. The following series of MDS plots examines the first four principal components. 

```{r, fig.height=8, fig.width=9}
lcpm <- cpm(y, log = TRUE)
dims <- list(c(1,2), c(1,3), c(2,3), c(3,4))
p <- vector("list", length(dims))

for(i in 1:length(dims)){
    tmp <- plotMDS(lcpm, top=1000, gene.selection="common", plot = FALSE,
                   dim.plot = dims[[i]])
    
    dat <-data.frame(x = tmp$x, y = tmp$y, cellType = targets$Cell_type)
    p[[i]] <- ggplot(dat, aes(x = x, y = y, colour = cellType)) +
        geom_point() +
        scale_colour_manual(values = cellPal)  +
        labs(colour = "Cell Type", x = glue("PC {tmp$dim.plot[1]}"),
             y = glue("PC {tmp$dim.plot[2]}"))
    
}

(p[[1]] | p[[2]]) / (p[[3]] | p[[4]]) + plot_layout(guides = "collect")
```

Save figure for use in manuscript.

```{r}
fig <- here("output/figures/SFig-4C.rds")
saveRDS(p[[1]], fig, compress = FALSE)
```

# Differential expression analysis

The TMM normalised data was transformed using `voomWithQualityWeights`. This takes into account the differing library sizes and the mean variance relationship in the data as well as calculating sample-specific quality weights. Linear models were fit using `limma`, taking into account the `voom` weights. 

```{r}
design <- model.matrix(~0+y$samples$group, data = targets)
colnames(design) <- c(levels(factor(y$samples$group)))
v <- voomWithQualityWeights(y, design, plot = TRUE)
cont <- makeContrasts(CD4vCD8=Naive_CD4_T_cells-Naive_CD8_T_cells,
                      MonovNeu=Classical_monocytes-Low_density_neutrophils,
                      BcellvNK=Naive_B_cells-Natural_killer_cells,
                      levels=design)
fit <- lmFit(v, design)
cfit <- contrasts.fit(fit, cont)
fit2 <- eBayes(cfit, robust = TRUE)
summary(decideTests(fit2, p.value = 0.05))
```

```{r}
fitSum <- summary(decideTests(fit2, p.value = 0.05))
dat <- reshape2::melt(fitSum[rownames(fitSum) != "NotSig", ])
colnames(dat) <- c("dir","comp","num")

p <- ggplot(dat, aes(x = comp, y = num, fill = dir)) +
    geom_bar(stat = "identity", position = "dodge") +
    labs(x = "Comparison", y = "No. DE Genes (FDR < 0.05)", fill = "Direction") +
    scale_fill_brewer(palette = "Set1", direction = -1)
p
```

Save figure for use in manuscript.

```{r}
fig <- here("output/figures/SFig-4D.rds")
saveRDS(p, fig, compress = FALSE)
```

# Gene set testing

Testing for enrichment of GO categories and KEGG pathways amongst statistically significant differentially expressed genes.

Save results as RDS objects for use as "truth" sets in methylation analyses.

```{r, message=FALSE}
go <- NULL
kegg <- NULL

for(i in 1:ncol(cont)){
    top <- topTable(fit2, coef = i, p.value = 0.05, number = Inf)
    
    tmp <- goana(top$entrezid, universe = v$genes$entrezid, 
                 covariate = v$genes$length)
    tmp <- topGO(tmp, number = Inf)
    tmp$FDR <- p.adjust(tmp$P.DE, method = "BH")
    tmp <- rownames_to_column(tmp, var = "ID")
    tmp$contrast <- colnames(cont)[i] 
    go <- bind_rows(go, tmp)
        
    tmp <- kegga(top$entrezid, universe = v$genes$entrezid, species = "Hs", 
                 covariate = v$genes$length, species.KEGG = "hsa")
    tmp <- topKEGG(tmp, number = Inf)
    tmp$FDR <- p.adjust(tmp$P.DE, method = "BH")
    tmp <- rownames_to_column(tmp, var = "PID")
    tmp$contrast <- colnames(cont)[i] 
    kegg <- bind_rows(kegg, tmp)
}

outDir <- here::here("data/cache-rnaseq")
if (!dir.exists(outDir)) dir.create(outDir)

saveRDS(go, here(glue("data/cache-rnaseq/RNAseq-GO.rds")))
saveRDS(kegg, here(glue("data/cache-rnaseq/RNAseq-KEGG.rds")))
```

Test whether the BROAD sets are enriched for the differentially expressed genes using the `gsaseq` function that can be found in `code/utility.R`

```{r}
data(PathwayList)
keep <- sapply(PathwayList, function(x) any(x %in% v$genes$symbol))
ensembl <- suppressMessages(lapply(PathwayList[keep], function(x){
    tmp <- v$genes$gene_id[v$genes$symbol %in% x]
    tmp[!is.na(tmp)]
}))

entrez <- suppressMessages(lapply(ensembl, function(x){
    tmp <- unname(v$genes$entrezid[v$genes$gene_id %in% x])
    tmp[!is.na(tmp)]
}))

gsa <- NULL
for(i in 1:ncol(cont)){
    top <- topTable(fit2, coef = i, p.value = 0.05, number = Inf)
    
    tmp <- gsaseq(top$entrezid, universe = v$genes$entrezid, 
                  collection = entrez, gene.length = v$genes$length)
    tmp <- rownames_to_column(data.frame(tmp), var = "ID")
    tmp$contrast <- colnames(cont)[i] 
    gsa <- bind_rows(gsa, tmp)
    
}

saveRDS(gsa, here("data/cache-rnaseq/RNAseq-BROAD-GSA.rds"))
```

