---
title: 'Gene set testing for Illumina HumanMethylation Arrays'
subtitle: "Evaluating the performance of different methods"
author: "Jovana Maksimovic, Alicia Oshlack and Belinda Phipson"
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r, message=FALSE, warning=FALSE, results='hide'}
library(here)
library(minfi)
library(paletteer)
library(limma)
library(BiocParallel)
library(reshape2)
library(gridExtra)
library(missMethyl)
library(ggplot2)
library(glue)
library(grid)
library(tidyverse)
library(rbin)
library(patchwork)
library(ChAMPdata)
library(lemon)
source(here("code/utility.R"))
```

# Load data

We are using publicly available EPIC data [GSE95462](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE95462) generated from SBR3 and AU565 cell lines none, treated or treated with CAF-conditionned medium or with Decitabine in duplicate. The data is normalised and filtered (bad probes, multi-mapping probes, SNP probes, sex chromosomes).

```{r, message=FALSE, warning=FALSE}
# load data
dataFile <- here("data/datasets/GSE95462-data.RData")

if(file.exists(dataFile)){
  # load processed data and sample information
  load(dataFile)
  
} else {
  
  dir <- here("data/datasets/GSE95462_RAW")
  targets <- read_csv(file = glue("{dir}/Sample-info.csv"))
  targets$Group <- paste(targets$Cell_line, 
                         gsub("(11|15)","", targets$Treatment), sep = "_")
  targets$Group <- gsub("-", "_", targets$Group)
  
  idats <- list.files(dir, pattern = "idat")
  basenames <- glue('{dir}/{unique(gsub("_(Red|Grn).idat","", idats))}')
  rawRg <- minfi::read.metharray(basenames = as.character(basenames))
  
  detP <- minfi::detectionP(rawRg)
  
  # normalise
  normGr <- minfi::preprocessQuantile(rawRg)
  
  # filtering
  fltGr <- filterQual(normGr, detP)
  fltGr <- filterProbes(fltGr)
  
  save(detP, normGr, fltGr, targets, file = dataFile)
  
}
```

## QC plots

```{r}
# plot mean detection p-values across all samples
dat <- tibble::tibble(mean = colMeans(detP), group = targets$Group, 
                      sample = targets$Sample)
ggplot(dat, aes(y = mean, x = sample, fill = group)) +
    geom_bar(stat = "identity") +
    labs(fill = "Treatment", 
         x = "Sample", 
         y = "Mean detection p-value") +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5))
```

```{r}
# plot normalised beta value distribution
bVals <- getBeta(fltGr)
dat <- as_tibble(reshape2::melt(bVals))
colnames(dat) <- c("cpg", "ID", "beta")
dat$group <- rep(targets$Group, each = nrow(bVals))

p <- ggplot(dat, aes(x = beta, colour = group, group = ID)) + 
    geom_density(size=0.5, alpha=0.5) +
    labs(color = "Treatment", x = "Beta value", y = "Density")

p 
```

```{r}
dims <- list(c(1,2), c(1,3), c(2,3), c(3,4))
p <- vector("list", length(dims))

for(i in 1:length(dims)){
    tmp <- plotMDS(getM(fltGr), top=1000, 
                   gene.selection="common", plot = FALSE,
                   dim.plot = dims[[i]])
    
    dat <- data.frame(x = tmp$x, y = tmp$y, group = targets$Group)
    p[[i]] <- ggplot(dat, aes(x = x, y = y, colour = group)) +
        geom_point() +
        labs(colour = "Treatment", x = glue("PC {tmp$dim.plot[1]}"),
             y = glue("PC {tmp$dim.plot[2]}"))
    
}

(p[[1]] | p[[2]]) / (p[[3]] | p[[4]]) + plot_layout(guides = "collect")
```

Save figure for use in manuscript.

```{r}
outDir <- here::here("output/figures")
if (!dir.exists(outDir)) dir.create(outDir)

fig <- here("output/figures/Fig-4A.rds")
saveRDS(p, fig, compress = FALSE)
```

## Statistical analysis

Compare several sets of sorted immune cells. Consider results significant at FDR < 0.05 and delta beta > 10% (~ lfc = 0.5).

```{r}
mVals <- getM(fltGr)
bVals <- getBeta(fltGr)
```

```{r, fig.width=9,fig.height=6}
design <- model.matrix(~0+targets$Group)
colnames(design) <- levels(factor(targets$Group))
colnames(design) <- gsub("-","_", colnames(design))

fit <- lmFit(mVals, design)
cont.matrix <- makeContrasts(AU565_nonevSKBR3_none=AU565_none-SKBR3_none,
                             #AU565_nonevAU565_DAC=AU565_none-AU565_DAC,
                             #SKBR3_nonevSKBR3_DAC=SKBR3_none-SKBR3_DAC,
                             NonevDAC=0.5*(AU565_none+SKBR3_none)-0.5*(AU565_DAC+SKBR3_DAC),
                             levels=design)
fit2 <- contrasts.fit(fit, cont.matrix)
fit2 <- eBayes(fit2, robust=TRUE, trend=TRUE)
tfit <- treat(fit2, lfc = 0.5)
pval <- 0.01
fitSum <- summary(decideTests(tfit, p.value = pval))
fitSum
```

```{r}
bDat <- getBiasDat(rownames(topTreat(tfit, coef = 1, num = 5000)), 
                   array.type = "EPIC")

p <- ggplot(bDat, aes(x = avgbias, y = propDM)) +
    geom_point(shape = 1, size = 2) +
    geom_smooth() +
    labs(x = "No. CpGs per gene (binned)",
         y = "Prop. differential methylation") +
    theme_minimal() +
    theme(panel.grid = element_blank(),
          axis.line = element_line(colour = "black"))
p
```

Save figure for use in manuscript.

```{r}
fig <- here("output/figures/Fig-1B.rds")
saveRDS(p, fig, compress = FALSE)
```

Examine only the independent contrasts.

```{r}
dat <- reshape2::melt(fitSum[rownames(fitSum) != "NotSig", ])
colnames(dat) <- c("dir","comp","num")

p <- ggplot(dat, aes(x = comp, y = num, fill = dir)) +
    geom_bar(stat = "identity", position = "dodge") +
    labs(x = "Comparison", y = "No. DM CpGs (FDR < 0.05)", fill = "Direction") +
    scale_fill_brewer(palette = "Set1", direction = -1)
p
```

Save figure for use in manuscript.

```{r}
fig <- here("output/figures/Fig-4B.rds")
saveRDS(p, fig, compress = FALSE)
```

# Compare performance of different gene set testing methods

We compare the performance of the following gene set testing methods available for methylation arrays: a standard hypergeometric test (HGT), GOmeth from the *missMethyl* package, methylglm (mGLM), methylRRA-ORA (mRRA (ORA)) and methylRRA-GSEA (mRRA (GSEA)) from the *methylGSA* package and ebayGSEA from the *ChAMP* package. 

We perform gene set testing on the results of all three blood cell type comparisons: B-cells vs. NK cells, CD4 T-cells vs. CD8 T-cells and monocytes vs. neutrophils, using all of the different gene set testing methods.

Gene set testing is performed for each comparison using GO categories, KEGG pathways and the BROAD MSigDB gene sets for the following methods: HGT, GOmeth, mGLM, mRRA (ORA) and mRRA (GSEA)

As gene set testing can only be performed using the BROAD MSigDB gene sets provided with the *ChAMP* package, the ebayGSEA method is only tested using the BROAD MSigDB gene sets for the three blood cell comparisons.

Furthermore, as the methylGSA methods do not work well with sets that only contain very few genes or very many genes, we only test sets with at least 5 genes and at most 5000 genes. 

Firstly, the results of the statistical analysis of the three blood cell comparisons are saved as an RDS object.

```{r}
minsize <- 5
maxsize <- 5000

outDir <- here::here("data/cache-intermediates/CANCER")
if (!dir.exists(outDir)) dir.create(outDir)

outFile <- here("data/cache-intermediates/cancer.cellline.contrasts.rds")

if(!file.exists(outFile)){
  obj <- NULL
  obj$tfit <- tfit
  obj$maxsize <- maxsize
  obj$minsize <- minsize
  obj$mVals <- mVals
  obj$targets <- targets

  saveRDS(obj, file = outFile)

} 
```

As some of the methods take a considerable amount of time to perform the gene set testing analysis, we have created several scripts in order to run the analyses in parallel on a HPC.
The code used to run all the gene set testing analyses using the different methods can be found in the `code/compare-methods` directory. It consists of four scripts: `genRunMethodJob.R`, `runChAMP.R`, `runMethylGSA.R` and `runMissMethyl.R`. The `genRunMethodJob.R` script creates and submits [Slurm](https://slurm.schedmd.com/overview.html) job scripts that run the `runChAMP.R`, `runMethylGSA.R` and `runMissMethyl.R` scripts, for *each* gene set type, in parallel, on a HPC. The results of each job are saved as an RDS file named `{package}.{set}.rds` in the `output/compare-methods` directory. Once all analysis jobs are complete, all of the suvbsequent analyses in this document can be executed.

## Load output for all methods

The results of all the gene set testing analyses, using all the different methods, for the different types of gene sets, are loaded into a `list` of `data.frames`. All of the `data.frames` in the `list` are then concatenated into a `tibble` for downstream analysis and plotting.

```{r}
inFiles <- list.files(here("output/compare-methods/CANCER"), pattern = "rds", 
                      full.names = TRUE)

res <- lapply(inFiles, function(file){
    readRDS(file)
})

dat <- as_tibble(dplyr::bind_rows(res))
```

## Explore analysis results
### GO categories

Examine the performance of the different methods when gene set testing was performed on GO categories.

```{r}
ann <- loadAnnotation("EPIC")
flatAnn <- loadFlatAnnotation(ann)
cpgEgGo <- cpgsEgGoFreqs(flatAnn)

cpgEgGo %>% 
  group_by(GO) %>%
  summarise(med = median(Freq)) -> medCpgEgGo
```
In order to examine whether the probe-number bias influenced the significantly enriched GO categories for the five different methods, we split the GO categories into bins based on the median number of CpGs per gene per GO category. We then calculated the proportion of significantly enriched GO categories in each bin for each of the three comparisons. As GOmeth explicitly corrects for this bias it showed very little trend. However mRRA (ORA) tended to have more significant GO categories when the genes had larger numbers of CpGs. The hypergeometric test and mRRA (GSEA) tended to have more significant GO categories with average numbers of CpGs associated per gene ranging from 27 - 44. The mGLM method appears to control for the probe-number bias of the array fairly well.

```{r, fig.width=7}
dat %>% filter(set == "GO") %>%
    filter(sub %in% c("n","c1")) %>% 
    mutate(method = unname(dict[method])) %>% 
    inner_join(medCpgEgGo, by = c("ID" = "GO")) -> sub

bins <- rbin_quantiles(sub, ID, med, bins = 12)
sub$bin <- as.factor(findInterval(sub$med, bins$upper_cut))

binLabs <- paste0("<", bins$upper_cut)
names(binLabs) <- levels(sub$bin)
binLabs[length(binLabs)] <- gsub("<", "\u2265", binLabs[length(binLabs) - 1])
    
sub %>% group_by(contrast, method, bin) %>%
  summarise(prop = sum(pvalue < 0.05)/n()) -> pdat

p <- ggplot(pdat, aes(x = as.numeric(bin), y = prop, color = method)) +
  geom_line() +
  facet_wrap(vars(contrast), ncol = 3) +
  scale_x_continuous(breaks = as.numeric(levels(pdat$bin)), labels = binLabs) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, vjust = 1,
                                   size = 7),
        legend.position = "bottom") +
  labs(x = "Med. No. CpGs per Gene per GO Cat. (binned)",
       y = "Prop. GO Cat. with p-value < 0.05",
       colour = "Method") +
    scale_color_manual(values = methodCols)
p
```

We also examine results when top ranked 100 GO terms from RNA-seq analysis of the same cell type comparisons is used as "truth".

```{r}
rnaseqGO <- readRDS(here("data/cache-rnaseq/RNAseq-SRP100803-GO.rds"))
rnaseqGO %>% group_by(contrast) %>%
    mutate(rank = 1:n()) %>%
    filter(rank <= 100) -> topGOSets

p <- vector("list", length(unique(dat$contrast)))

for(i in 1:length(unique(dat$contrast))){
    cont <- sort(unique(dat$contrast))[i]
    
    dat %>% filter(set == "GO") %>%
        filter(sub %in% c("n","c1")) %>%
        filter(contrast == cont) %>%
        mutate(method = unname(dict[method])) %>% 
        arrange(method, pvalue) %>%
        group_by(method) %>%
        mutate(rank = 1:n()) %>%
        filter(rank <= 100) %>%
        mutate(csum = cumsum(ID %in% topGOSets$ID[topGOSets$contrast %in% 
                                                         contrast])) %>%
        mutate(truth = "RNAseq Terms") -> rnaseqSum
    
    p[[i]] <- ggplot(rnaseqSum, aes(x = rank, y = csum, colour = method)) +
    geom_line() +
    facet_wrap(vars(truth), ncol = 2) +
    geom_vline(xintercept = 10, linetype = "dotted") +
    labs(colour = "Method", x = "Rank", 
         y = "Cumulative no. sets in truth") +
    theme(legend.position = "bottom") +
    scale_color_manual(values = methodCols)
}

(p[[1]] + ggtitle(sort(unique(dat$contrast))[1]) | 
  p[[2]] + ggtitle(sort(unique(dat$contrast))[2])) + 
    plot_layout(guides = "collect") & 
    theme(legend.position = "bottom")
  
```

Save figure for use in manuscript.

```{r}
fig <- here("output/figures/Fig-4C.rds")
saveRDS(p[[1]], fig, compress = FALSE)

fig <- here("output/figures/SFig-5A.rds")
saveRDS(p[[2]], fig, compress = FALSE)

fig <- here("output/figures/SFig-6A.rds")
saveRDS(p[[3]], fig, compress = FALSE)
```

Examine what the top 10 ranked gene sets are and how many genes they contain, for each method and comparison.

```{r}
terms <- missMethyl:::.getGO()$idTable
nGenes <- rownames_to_column(data.frame(n = sapply(missMethyl:::.getGO()$idList, 
                                                   length)), 
                             var = "ID")

dat %>% filter(set == "GO") %>%
    filter(sub %in% c("n","c1")) %>% 
    mutate(method = unname(dict[method])) %>% 
    arrange(contrast, method, pvalue) %>%
    group_by(contrast, method) %>%
    mutate(FDR = p.adjust(pvalue, method = "BH")) %>%
    mutate(rank = 1:n()) %>%
    filter(rank <= 10) %>% 
    inner_join(terms, by = c("ID" = "GOID")) %>%
    inner_join(nGenes) -> sub
```

```{r, fig.height=7, fig.width=8, message=FALSE, warning=FALSE}
p <- vector("list", length(unique(sub$contrast)))
truthPal <- scales::hue_pal()(4)
names(truthPal) <- c("Both", "ISP", "Neither", "RNAseq")

for(i in 1:length(p)){
    cont <- sort(unique(sub$contrast))[i]
    sub %>% filter(contrast == cont) %>%
        arrange(method, -rank) %>%
        ungroup() %>%
        mutate(idx = as.factor(1:n())) -> tmp
    
    setLabs <- substr(tmp$TERM, 1, 40)
    names(setLabs) <- tmp$idx
    
    tmp %>% mutate(rna = ID %in% topGOSets$ID[topGOSets$contrast %in% cont],
                   isp = ID %in% immuneGO$GOID,
                   both = rna + isp,
                   col = ifelse(both == 2, "Both", 
                                ifelse(both == 1 & rna == 1, "RNAseq",
                                       ifelse(both == 1 & isp == 1, 
                                              "ISP", "Neither")))) %>%
        mutate(col = factor(col, 
                            levels = c("Both", "ISP", "RNAseq", 
                                       "Neither"))) -> tmp
    
    p[[i]] <- ggplot(tmp, aes(x = -log10(FDR), y = idx, colour = col)) +
        geom_point(aes(size = n), alpha = 0.7) +
        scale_size(limits = c(min(sub$n), max(sub$n))) +
        facet_wrap(vars(method), ncol = 2, scales = "free") +
        scale_y_discrete(labels = setLabs) +
        scale_colour_manual(values = truthPal) +
        labs(y = "", size = "No. genes", colour = "In truth set") +
        theme(axis.text.y = element_text(size = 6),
              axis.text.x = element_text(size = 6),
              legend.box = "horizontal",
              legend.margin = margin(0, 0, 0, 0, unit = "lines"),
              panel.spacing.x = unit(1, "lines")) +
        coord_cartesian(xlim = c(-log10(0.99), -log10(10^-80))) +
        geom_vline(xintercept = -log10(0.05), linetype = "dashed") +
        ggtitle(cont)
}

shift_legend(p[[1]], plot = TRUE, pos = "left")
shift_legend(p[[2]], plot = TRUE, pos = "left")

```

Save figure for use in manuscript.

```{r}
fig <- here("output/figures/Fig-4D.rds")
saveRDS(shift_legend(p[[1]] + theme(plot.title = element_blank()),
                     pos = "left"), 
        fig, compress = FALSE)

fig <- here("output/figures/SFig-5B.rds")
saveRDS(shift_legend(p[[2]] + theme(plot.title = element_blank()),
                     pos = "left"), 
        fig, compress = FALSE)

fig <- here("output/figures/SFig-6B.rds")
saveRDS(shift_legend(p[[3]] + theme(plot.title = element_blank()),
                     pos = "left"), 
        fig, compress = FALSE)
```

P-value histograms for the different methods for all contrasts on GO categories.

```{r, fig.height=8, fig.width=7}
dat %>% filter(set == "GO") %>%
    filter(sub %in% c("n","c1")) %>%
    mutate(method = unname(dict[method])) -> subDat 
    
ggplot(subDat, aes(pvalue, fill = method)) +
    geom_histogram(binwidth = 0.025) +
    facet_grid(cols = vars(contrast), rows = vars(method)) +
    theme(legend.position = "bottom") +
    labs(x = "P-value", y = "Frequency", fill = "Method") +
    scale_fill_manual(values = methodCols)
```

#### Compare GOmeth results using different DM CpG significance cutoffs

As the results of GOmeth depend on the list of significant CpGs provided as input to the function, we explored the effect of selecting "significant" CpGs in different ways on the gene set testing performance of GOmeth.

```{r, fig.width=7, fig.height=8}
dat %>% filter(set == "GO") %>%
    filter(grepl("mmethyl", method)) %>%
    mutate(method = unname(dict[method])) %>%
    arrange(contrast, method, pvalue) %>%
    group_by(contrast, method, sub) %>%
    mutate(csum = cumsum(ID %in% topGOSets$ID[topGOSets$contrast %in% 
                                                  contrast])) %>%
    mutate(rank = 1:n()) %>%
    mutate(cut = ifelse(sub == "c1", "Top 5000", 
                        ifelse(sub == "c2", "Top 10000",
                               ifelse(sub == "p1", "FDR < 0.01", "FDR < 0.05")))) %>%
    filter(rank <= 100) -> sub

p <- ggplot(sub, aes(x = rank, y = csum, colour = cut)) +
    geom_line() +
    facet_wrap(vars(contrast, method), ncol=2, nrow = 3, scales = "free") +
    geom_vline(xintercept = 10, linetype = "dotted") +
    labs(colour = "Sig. select", x = "Rank", 
         y = glue("Cumulative no. RNAseq sets")) +
    theme(legend.position = "bottom")
p
```

### KEGG pathways

Now test KEGG pathways with at least 5 genes and 5000 at most.

Examine results when top ranked 100 KEGG pathways from RNA-seq analysis of the same cell type comparisons is used as "truth".

```{r, fig.height=7, fig.width=9}
rnaseqKEGG <- readRDS(here("data/cache-rnaseq/RNAseq-SRP100803-KEGG.rds"))
rnaseqKEGG %>% group_by(contrast) %>%
    mutate(rank = 1:n()) %>%
    filter(rank <= 100) -> topKEGG

dat %>% filter(set == "KEGG") %>%
    filter(sub %in% c("n","c1")) %>%
    mutate(method = unname(dict[method])) %>%
    arrange(contrast, method, pvalue) %>%
    group_by(contrast, method) %>%
    mutate(csum = cumsum(ID %in% topKEGG$PID[topKEGG$contrast %in% contrast])) %>%
    mutate(rank = 1:n()) %>%
    filter(rank <= 100) -> sub

p <- ggplot(sub, aes(x = rank, y = csum, colour = method)) +
    geom_line() +
    facet_wrap(vars(contrast), ncol=3, scales = "free_y") +
    geom_vline(xintercept = 10, linetype = "dotted") +
    labs(colour = "Method", x = "Rank", y = "Cumulative no. RNAseq sets") +
    theme(legend.position = "bottom") + 
    scale_color_manual(values = methodCols)

p
```

Save figure for use in manuscript.

```{r}
fig <- here("output/figures/SFig-7A.rds")
saveRDS(p1, fig, compress = FALSE)

fig <- here("output/figures/SFig-7B.rds")
saveRDS(p2, fig, compress = FALSE)
```

Examine what the top 10 ranked gene sets are and how many genes they contain, for each method and comparison.

```{r, fig.height=7, fig.width=9}
terms <- missMethyl:::.getKEGG()$idTable
nGenes <- rownames_to_column(data.frame(n = sapply(missMethyl:::.getKEGG()$idList, 
                                                   length)), 
                             var = "ID")

dat %>% filter(set == "KEGG") %>%
    filter(sub %in% c("n","c1")) %>% 
    mutate(method = unname(dict[method])) %>%
    arrange(contrast, method, pvalue) %>%
    group_by(contrast, method) %>%
    mutate(FDR = p.adjust(pvalue, method = "BH")) %>%
    mutate(rank = 1:n()) %>%
    filter(rank <= 10) %>% 
    inner_join(terms, by = c("ID" = "PathwayID")) %>%
    inner_join(nGenes) -> sub
```

```{r, fig.height=7, fig.width=8, message=FALSE, warning=FALSE}
p <- vector("list", length(unique(sub$contrast)))
for(i in 1:length(p)){
    cont <- sort(unique(sub$contrast))[i]
    sub %>% filter(contrast == cont) %>%
        arrange(method, -rank) %>%
        ungroup() %>%
        mutate(idx = as.factor(1:n())) -> tmp
    
    setLabs <- substr(tmp$Description, 1, 40)
    names(setLabs) <- tmp$idx
    
    tmp %>% mutate(rna = ID %in% topKEGG$PID[topKEGG$contrast %in% cont],
                   isp = ID %in% immuneKEGG$PID,
                   both = rna + isp,
                   col = ifelse(both == 2, "Both", 
                                ifelse(both == 1 & rna == 1, "RNAseq",
                                       ifelse(both == 1 & isp == 1, 
                                              "ISP", "Neither")))) %>%
        mutate(col = factor(col, 
                            levels = c("Both", "ISP", "RNAseq", 
                                       "Neither"))) -> tmp
    
    p[[i]] <- ggplot(tmp, aes(x = -log10(FDR), y = idx, colour = col)) +
        geom_point(aes(size = n), alpha = 0.7) +
        scale_size(limits = c(min(sub$n), max(sub$n))) +
        facet_wrap(vars(method), ncol = 2, scales = "free") +
        scale_y_discrete(labels = setLabs) +
        scale_colour_manual(values = truthPal) +
        labs(y = "", size = "No. genes", colour = "In truth set") +
        theme(axis.text.y = element_text(size = 6),
              axis.text.x = element_text(size = 6),
              legend.box = "horizontal",
              legend.margin = margin(0, 0, 0, 0, unit = "lines"),
              panel.spacing.x = unit(1, "lines")) +
        coord_cartesian(xlim = c(-log10(0.99), -log10(10^-80))) +
        geom_vline(xintercept = -log10(0.05), linetype = "dashed") +
        ggtitle(cont)
}

shift_legend(p[[1]], plot = TRUE, pos = "left")
shift_legend(p[[2]], plot = TRUE, pos = "left")
```

Save figure for use in manuscript.

```{r}
fig <- here("output/figures/SFig-7C.rds")
saveRDS(shift_legend(p[[1]] + theme(plot.title = element_blank()),
                     pos = "left"), 
        fig, compress = FALSE)

fig <- here("output/figures/SFig-7D.rds")
saveRDS(shift_legend(p[[2]] + theme(plot.title = element_blank()),
                     pos = "left"), 
        fig, compress = FALSE)

fig <- here("output/figures/SFig-7E.rds")
saveRDS(shift_legend(p[[3]] + theme(plot.title = element_blank()),
                     pos = "left"), 
        fig, compress = FALSE)
```

P-value histograms for the different methods for all contrasts on KEGG pathways.

```{r, fig.height=8, fig.width=7}
dat %>% filter(set == "KEGG") %>%
    filter(sub %in% c("n","c1")) %>%
    mutate(method = unname(dict[method])) -> subDat 
    
ggplot(subDat, aes(pvalue, fill = method)) +
    geom_histogram(binwidth = 0.025) +
    facet_grid(cols = vars(contrast), rows = vars(method)) +
    theme(legend.position = "bottom") +
    labs(x = "P-value", y = "Frequency", fill = "Method") +
    scale_fill_manual(values = methodCols)
```

#### Compare GOmeth results using different DM CpG significance cutoffs

```{r, fig.width=7, fig.height=8}
dat %>% filter(set == "KEGG") %>%
    filter(grepl("mmethyl", method)) %>% 
    mutate(method = unname(dict[method])) %>%
    arrange(contrast, method, pvalue) %>%
    group_by(contrast, method, sub) %>%
    mutate(csum = cumsum(ID %in% topKEGG$PID[topKEGG$contrast %in% contrast])) %>%
    mutate(rank = 1:n()) %>%
    mutate(cut = ifelse(sub == "c1", "Top 5000", 
                        ifelse(sub == "c2", "Top 10000",
                               ifelse(sub == "p1", "FDR < 0.01", "FDR < 0.05")))) %>%
    filter(rank <= 100) -> sub

p <- ggplot(sub, aes(x = rank, y = csum, colour = cut)) +
    geom_line() +
    facet_wrap(vars(contrast, method), ncol=2, nrow = 3, scales = "free") +
    geom_vline(xintercept = 10, linetype = "dotted") +
    labs(colour = "Sig. select", x = "Rank", 
         y = glue("Cumulative no. RNAseq sets")) +
    theme(legend.position = "bottom")
p
```


### BROAD gene sets

As the ebGSEA method from the ChAMP package only allows testing of their in-built database of Broad Institute gene sets, we will compare to other methods by testing these same gene sets. Using the top 100 ranked gene sets as identified by Belinda's `gsaseq` analysis of the corresponding RNAseq data as "truth".

```{r}
rnaseqBROAD <- readRDS(here("data/cache-rnaseq/RNAseq-SRP100803-BROAD-GSA.rds"))
rnaseqBROAD %>% group_by(contrast) %>%
    mutate(rank = 1:n()) %>%
    filter(rank <= 100) -> topBROAD

p <- vector("list", length(unique(dat$contrast)))

for(i in 1:length(unique(dat$contrast))){
    cont <- sort(unique(dat$contrast))[i]
    
    dat %>% filter(set == "BROAD") %>%
        filter(sub %in% c("n","c1")) %>%
        filter(contrast == cont) %>%
        mutate(method = unname(dict[method])) %>% 
    arrange(contrast, method, pvalue) %>%
    group_by(contrast, method) %>%
    mutate(csum = cumsum(ID %in% topBROAD$ID[topBROAD$contrast %in% contrast])) %>%
    mutate(rank = 1:n()) %>%
    filter(rank <= 100) -> sub
    
    p[[i]] <- ggplot(sub, aes(x = rank, y = csum, colour = method)) +
    geom_line() +
    geom_vline(xintercept = 10, linetype = "dotted") +
    labs(colour = "Method", x = "Rank", y = "Cumulative no. RNAseq sets") +
    theme(legend.position = "bottom") +
    scale_color_manual(values = methodCols) +
        ggtitle(cont)
}

p[[1]]
p[[2]]
```

Save figure for use in manuscript.

```{r}
fig <- here("output/figures/SFig-8A.rds")
saveRDS(p[[1]], fig, compress = FALSE)

fig <- here("output/figures/SFig-9A.rds")
saveRDS(p[[2]], fig, compress = FALSE)

fig <- here("output/figures/SFig-10A.rds")
saveRDS(p[[3]], fig, compress = FALSE)
```

Examine what the top 10 ranked gene sets are and how many genes they contain, for each method and comparison.

```{r, fig.height=8, fig.width=9}
data(PathwayList)
nGenes <- rownames_to_column(data.frame(n = sapply(PathwayList, 
                                                   length)), 
                             var = "ID")

dat %>% filter(set == "BROAD") %>%
    filter(sub %in% c("n","c1")) %>% 
    mutate(method = unname(dict[method])) %>% 
    arrange(contrast, method, pvalue) %>%
    group_by(contrast, method) %>%
    mutate(FDR = p.adjust(pvalue, method = "BH")) %>%
    mutate(rank = 1:n()) %>%
    filter(rank <= 10) %>% 
    inner_join(nGenes) -> sub
```

```{r, fig.height=7, fig.width=8, message=FALSE, warning=FALSE}
p <- vector("list", length(unique(sub$contrast)))
for(i in 1:length(p)){
    cont <- sort(unique(sub$contrast))[i]
    sub %>% filter(contrast == cont) %>%
        arrange(method, -rank) %>%
        ungroup() %>%
        mutate(idx = as.factor(1:n())) -> tmp
    
    setLabs <- substr(tmp$ID, 1, 30)
    names(setLabs) <- tmp$idx
    
    tmp %>% mutate(rna = ID %in% topBROAD$ID[topGOSets$contrast %in% cont],
                   col = ifelse(rna == 1, "RNAseq", "None")) %>%
        mutate(col = factor(col, levels = c("RNAseq", "None"))) -> tmp
    
    p[[i]] <- ggplot(tmp, aes(x = -log10(FDR), y = idx, colour = col)) +
        geom_point(aes(size = n), alpha = 0.7) +
        scale_size(limits = c(min(sub$n), max(sub$n))) +
        facet_wrap(vars(method), ncol = 2, scales = "free") +
        scale_y_discrete(labels = setLabs) +
        scale_color_manual(values = scales::hue_pal()(4)[3:4]) +
        labs(y = "", size = "No. genes", colour = "In truth set") +
        theme(axis.text.y = element_text(size = 6),
              axis.text.x = element_text(size = 6),
              legend.box = "horizontal",
              legend.margin = margin(0, 0, 0, 0, unit = "lines"),
              panel.spacing.x = unit(1, "lines")) +
        coord_cartesian(xlim = c(-log10(0.99), -log10(10^-100))) +
        geom_vline(xintercept = -log10(0.05), linetype = "dashed") +
        ggtitle(cont)
}

shift_legend(p[[1]], plot = TRUE, pos = "left")
shift_legend(p[[2]], plot = TRUE, pos = "left")
```

Save figure for use in manuscript.

```{r}
fig <- here("output/figures/SFig-8B.rds")
saveRDS(shift_legend(p[[1]] + theme(plot.title = element_blank()),
                     pos = "left"), 
        fig, compress = FALSE)

fig <- here("output/figures/SFig-9B.rds")
saveRDS(shift_legend(p[[2]] + theme(plot.title = element_blank()),
                     pos = "left"), 
        fig, compress = FALSE)

fig <- here("output/figures/SFig-10B.rds")
saveRDS(shift_legend(p[[3]] + theme(plot.title = element_blank()),
                     pos = "left"), 
        fig, compress = FALSE)
```

P-value histograms for the different methods for all contrasts on BROAD gene sets.

```{r, fig.height=10, fig.width=7}
dat %>% filter(set == "BROAD") %>%
    filter(sub %in% c("n","c1")) %>%
    mutate(method = unname(dict[method])) -> subDat 
    
ggplot(subDat, aes(pvalue, fill = method)) +
    geom_histogram(binwidth = 0.025) +
    facet_grid(cols = vars(contrast), rows = vars(method)) +
    theme(legend.position = "bottom") +
    labs(x = "P-value", y = "Frequency", fill = "Method") +
    scale_fill_manual(values = methodCols)
```

#### Compare GOmeth results using different DM CpG significance cutoffs

As the results of GOmeth depend on the list of significant CpGs provided as input to the function, we explored the effect of selecting "significant" CpGs in different ways on the gene set testing performance of GOmeth.

```{r, fig.width=7, fig.height=8}
dat %>% filter(set == "BROAD") %>%
    filter(grepl("mmethyl", method)) %>% 
    mutate(method = unname(dict[method])) %>%
    arrange(contrast, method, pvalue) %>%
    group_by(contrast, method, sub) %>%
    mutate(csum = cumsum(ID %in% topBROAD$ID)) %>% 
    mutate(rank = 1:n()) %>%
    mutate(cut = ifelse(sub == "c1", "Top 5000", 
                        ifelse(sub == "c2", "Top 10000",
                               ifelse(sub == "p1", "FDR < 0.01", "FDR < 0.05")))) %>%
    filter(rank <= 100) -> sub

p <- ggplot(sub, aes(x = rank, y = csum, colour = cut)) +
    geom_line() +
    facet_wrap(vars(contrast, method), ncol=2, nrow = 3, scales = "free") +
    geom_vline(xintercept = 10, linetype = "dotted") +
    labs(colour = "Sig. select", x = "Rank", y = "Cumulative no. immune sets") +
    theme(legend.position = "bottom")
p
```

