---
title: 'Gene set testing for Illumina HumanMethylation Arrays'
subtitle: "Evaluating the performance of different methods"
author: "Jovana Maksimovic, Alicia Oshlack and Belinda Phipson"
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r, message=FALSE, warning=FALSE, results='hide'}
library(here)
library(minfi)
library(paletteer)
library(limma)
library(BiocParallel)
library(reshape2)
library(gridExtra)
library(missMethyl)
library(ggplot2)
library(glue)
library(tidyverse)
library(rbin)
library(patchwork)
library(ChAMPdata)
source(here("code/utility.R"))
```

# Load data

We are using publicly available EPIC data [GSE110554](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE110554) generated from flow sorted blood cells. The data is normalised and filtered (bad probes, multi-mapping probes, SNP probes, sex chromosomes).

```{r}
# load data
dataFile <- here("data/GSE110554-data.RData")
if(file.exists(dataFile)){
    # load processed data and sample information
    load(dataFile)
} else {
    # get data from experiment hub, normalise, filter and save objects
    readData(dataFile)
    # load processed data and sample information
    load(dataFile)
}
```

## QC plots

```{r}
# plot mean detection p-values across all samples
dat <- tibble::tibble(mean = colMeans(detP), cellType = targets$CellType)
ggplot(dat, aes(y = mean, x = cellType, fill = cellType)) +
    geom_bar(stat = "identity") +
    labs(fill = "Cell Type") +
    scale_fill_brewer(palette = "Dark2")
```

```{r}
# plot normalised beta value distribution
bVals <- getBeta(normGr)
dat <- data.frame(reshape2::melt(bVals))
colnames(dat) <- c("cpg", "sample", "bVal")
dat <- dplyr::bind_cols(dat, cellType = rep(targets$CellType, 
                                            each = nrow(bVals)))

ggplot(dat, aes(x = bVal, colour = cellType)) +
    geom_density() +
    labs(colour = "Cell Type") +
    scale_color_brewer(palette = "Dark2")
```

### Figure 4A {.tabset}
#### View
```{r}
# MDS plots to look at largest sources of variation
p <- plotMDS(getM(fltGr), top=1000, gene.selection="common", plot = FALSE)
dat <- tibble::tibble(x = p$x, y = p$y, cellType = targets$CellType)

p <- ggplot(dat, aes(x = x, y = y, colour = cellType)) +
    geom_point(size = 3) +
    labs(colour = "Cell type") +
    scale_colour_brewer(palette = "Dark2") +
    labs(x = "Principal component 1", y = "Principal component 2")
p
```

#### Create PDF
```{r, eval=FALSE}
fig <- here("output/Fig-4A.pdf")

pdf(file = fig, width = 9)
p
dev.off()
```

### {-}

MDS plots using *only* the CpGs that map to genes.

```{r}
ann <- loadAnnotation("EPIC")
flatAnn <- missMethyl:::.getFlatAnnotation("EPIC", anno = ann)
geneM <- getM(fltGr[rownames(fltGr) %in% flatAnn$cpg,])

p <- plotMDS(geneM, top=1000, gene.selection="common", plot = FALSE)
dat <- tibble::tibble(x = p$x, y = p$y, cellType = targets$CellType)

ggplot(dat, aes(x = x, y = y, colour = cellType)) +
    geom_point(size = 3) +
    labs(colour = "Cell Type") +
    scale_colour_brewer(palette = "Dark2")
```

# Statistical analysis

Compare several sets of sorted immune cells. Consider results significant at FDR < 0.05 and delta beta ~ 10% (~ lfc = 0.5).

```{r}
mVals <- getM(fltGr)
bVals <- getBeta(fltGr)
```

```{r, fig.width=9,fig.height=6}
design <- model.matrix(~0+targets$CellType)
colnames(design) <- levels(factor(targets$CellType))
fit <- lmFit(mVals, design)
cont.matrix <- makeContrasts(CD4vCD8=CD4T-CD8T,
                             #BcellvMono=Bcell-Mono,
                             MonovNeu=Mono-Neu,
                             BcellvNK=Bcell-NK,
                             #NeuvNK=Neu-NK,
                             #MonovNK=Mono-NK,
                             levels=design)
fit2 <- contrasts.fit(fit, cont.matrix)
tfit <- eBayes(fit2, robust=TRUE, trend=TRUE)
tfit <- treat(tfit, lfc = 0.5)
pval <- 0.05
fitSum <- summary(decideTests(tfit, p.value = pval))
fitSum
```

## Figure 1B {.tabset}
### View

```{r}
tmp <- gometh(rownames(topTreat(tfit, coef = "CD4vCD8", num = 5000)), 
              anno = ann, array.type = "EPIC", plot.bias = TRUE)
```

### Create PDF
```{r, eval=FALSE}
fig <- here("output/Fig-1B.pdf")

pdf(file = fig, width = 9)
tmp <- gometh(rownames(topTreat(tfit, coef = "CD4vCD8", num = 5000)), 
              anno = ann, array.type = "EPIC", plot.bias = TRUE)
dev.off()
```

## {-}

How correlated is differential methylation with differential expression for the same contrasts (same cell types but not identical samples)?

```{r, fig.height=6, fig.width=7, warning=FALSE, message=FALSE}
rfit <- readRDS(here("data/rnaseq-fit.rds"))
rdat <- rownames_to_column(data.frame(rfit$coefficients), var = "gene_id")

rdat %>% inner_join(data.frame(rfit$genes)) %>%
    mutate(entrezid = as.character(entrezid)) %>%
    dplyr::select(-length) -> rdat
rdat <- reshape2::melt(rdat, value.name = "coefficient")

reshape2::melt(rfit$p.value, value.name = "p.value") %>%
    group_by(Contrasts) %>%
    mutate(FDR = p.adjust(p.value, method = "BH")) -> rfdr

rdat %>% inner_join(rfdr, 
                    by = c("gene_id" = "Var1", "variable" = "Contrasts")) -> rdat

mdat <- rownames_to_column(data.frame(tfit$coefficients), var = "cpg")
mdat %>% inner_join(flatAnn) %>%
    dplyr::select(-symbol, -alias) -> mdat
mdat <- reshape2::melt(mdat, value.name = "coefficient") -> mdat

mdat %>% dplyr::select(-cpg) %>%
    mutate(promoter = ifelse(group %in% c("TSS1500", "TSS200", "1stExon"), 
                             "prom", "other")) %>%
    group_by(variable, entrezid, promoter) %>%
    #summarise(coef = coefficient[which.max(abs(coefficient))]) -> mdat
    summarise(coef = median(coefficient)) %>%
    dplyr::filter(promoter == "prom") -> mdat

mdat %>% inner_join(rdat, by = c("entrezid", "variable")) %>%
    mutate(colour = ifelse(FDR < 0.05, "Sig.", "Not sig.")) -> dat

ggplot(dat, aes(x = coef, y = coefficient, colour = colour)) +
    geom_point(data = dat[dat$colour == 'Not sig.',], alpha = 0.25, 
               size = 1) +
    geom_point(data = dat[dat$colour == 'Sig.',], size = 0.75, alpha = 0.5) +
    facet_wrap(vars(variable), ncol = 2, scales = "free") +
    labs(x = "Median logFC for CpGs in gene promoter", 
         y = "Gene expression Log FC", color = "Diff. Gene Exp.")
```

## Supp. Figure X1
### View
Examine only the independent contrasts.

```{r}
dat <- melt(fitSum[rownames(fitSum) != "NotSig", ])
colnames(dat) <- c("dir","comp","num")

p <- ggplot(dat, aes(x = comp, y = num, fill = dir)) +
    geom_bar(stat = "identity", position = "dodge") +
    labs(x = "Comparison", y = "No. DM CpGs (FDR < 0.05)", fill = "Direction") +
    scale_fill_brewer(palette = "Set1", direction = -1)
p
```
### Create PDF
```{r, eval=FALSE}
fig <- here("output/SFig-X1.pdf")

pdf(file = fig, width = 9)
p
dev.off()
```

## {-}

## Compare performance of different gene set testing methods

Compare the gene set testing methods available for methylation arrays; hypergeometric test (HGT), gometh (best), methylRRA (GLM), methylRRA (ORA) and methylRRA (GSEA). As methylRRA does not work well with sets that only contain very few genes or very large sets, we will only test sets with at least 5 genes and maximum of 5000 genes.

### Save analysis results

```{r}
minsize <- 5
maxsize <- 5000

outFile <- here("data/blood.contrasts.rds")

if(!file.exists(outFile)){
  obj <- NULL
  obj$tfit <- tfit
  obj$maxsize <- maxsize
  obj$minsize <- minsize
  obj$mVals <- mVals
  obj$targets <- targets

  saveRDS(obj, file = outFile)

} 
```

### Load output for all methods

```{r}
inFiles <- list.files(here("output/compare-methods"), pattern = "rds", 
                      full.names = TRUE)

res <- lapply(inFiles, function(file){
    readRDS(file)
})

dat <- as_tibble(dplyr::bind_rows(res))
```

```{r}
dat %>% filter(grepl("mmethyl", method)) %>%
    mutate(method = unname(dict[method])) -> subDat

ggplot(subDat, aes(x = pvalue, colour = sub)) +
    geom_density() +
    facet_grid(cols = vars(contrast), rows = vars(method)) +
    labs(colour = "Sig. CpG Selection", x = "P-value", y = "Density") +
    scale_color_hue(labels = c("Top 5000", "Top 10000", "FDR < 0.01", 
                                  "FDR < 0.05")) + 
    theme(legend.position = "bottom", legend.text = element_text(size=6))
```

## Test GO categories

```{r}
cpgEgGo <- cpgsEgGoFreqs(flatAnn)
```

### Supp. Figure X2
#### View

```{r, fig.width=7}
cpgEgGo %>% 
  group_by(GO) %>%
  summarise(med = median(Freq)) -> medCpgEgGo

dat %>% filter(set == "GO") %>%
    filter(sub %in% c("n","c1")) %>% 
    mutate(method = unname(dict[method])) %>% 
    inner_join(medCpgEgGo, by = c("ID" = "GO")) -> sub

bins <- rbin_quantiles(sub, ID, med, bins = 12)
sub$bin <- as.factor(findInterval(sub$med, bins$upper_cut))

binLabs <- paste0("<", bins$upper_cut)
names(binLabs) <- levels(sub$bin)
binLabs[length(binLabs)] <- gsub("<", "\u2265", binLabs[length(binLabs) - 1])
    
sub %>% group_by(contrast, method, bin) %>%
  summarise(prop = sum(pvalue < 0.05)/n()) -> pdat

p <- ggplot(pdat, aes(x = as.numeric(bin), y = prop, color = method)) +
  geom_line(size = 1.25) +
  facet_wrap(vars(contrast), ncol = 3) +
  scale_x_continuous(breaks = as.numeric(levels(pdat$bin)), labels = binLabs) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, vjust = 1,
                                   size = 7),
        legend.position = "bottom") +
  labs(x = "Med. No. CpGs per Gene per GO Cat. (binned)",
       y = "Prop. GO Cat. with p-value < 0.05",
       colour = "Method") +
    scale_color_manual(values = methodCols)
p
```

#### Create PDF
```{r, eval=FALSE}
fig <- here("output/SFig-X2.pdf")

pdf(file = fig, width = 9)
p
dev.off()
```

### {-}

As we are comparing immune cells, we expect GO categories related to the immune system and its processes to be highly ranked. To evalue this, we downloaded all of the child terms for the GO category "immune system process" (GO:002376) from AminGO 2;  [http://amigo.geneontology.org/amigo/term/GO:0002376](http://amigo.geneontology.org/amigo/term/GO:0002376). 

```{r, fig.width=7}
immuneGO <- unique(read.csv(here("data/GO-immune-system-process.txt"),
                            stringsAsFactors = FALSE, header = FALSE,
                            col.names = "GOID"))

dat %>% filter(set == "GO") %>%
    filter(sub %in% c("n","c1")) %>% 
    mutate(method = unname(dict[method])) %>% 
    arrange(contrast, method, pvalue) %>%
    group_by(contrast, method) %>%
    mutate(csum = cumsum(ID %in% immuneGO$GOID)) %>% 
    mutate(rank = 1:n()) %>%
    filter(rank <= 100) -> sub

p <- ggplot(sub, aes(x = rank, y = csum, colour = method)) +
    geom_line() +
    facet_wrap(vars(contrast), ncol=3) +
    geom_vline(xintercept = 10, linetype = "dotted") +
    labs(colour = "Method", x = "Rank", y = "Cumulative no. immune sets") +
    theme(legend.position = "bottom") +
    scale_color_manual(values = methodCols)
p
```

Examine results when top ranked 100 GO terms from RNA-seq analysis of the same cell type comparisons is used as "truth".

```{r, fig.width=7}
rnaseqGO <- readRDS(here("data/RNAseq-GO.rds"))
rnaseqGO %>% group_by(contrast) %>%
    mutate(rank = 1:n()) %>%
    filter(rank <= 100) -> topGOSets
    
dat %>% filter(set == "GO") %>%
    filter(sub %in% c("n","c1")) %>% 
    mutate(method = unname(dict[method])) %>% 
    arrange(contrast, method, pvalue) %>%
    group_by(contrast, method) %>%
    mutate(csum = cumsum(ID %in% topGOSets$ID[topGOSets$contrast %in% contrast])) %>%
    mutate(rank = 1:n()) %>%
    filter(rank <= 100) -> sub

p <- ggplot(sub, aes(x = rank, y = csum, colour = method)) +
    geom_line() +
    facet_wrap(vars(contrast), ncol=3) +
    geom_vline(xintercept = 10, linetype = "dotted") +
    labs(colou = "Method", x = "Rank", 
         y = glue("Cumulative no. RNAseq sets")) +
    theme(legend.position = "bottom") +
    scale_color_manual(values = methodCols)
p
```

### Figure 4B, Supp. Figure 4B, Supp. Figure 5B {.tabset}
#### View

```{r}
p <- vector("list", length(unique(dat$contrast)))

for(i in 1:length(unique(dat$contrast))){
    cont <- sort(unique(dat$contrast))[i]
    
    dat %>% filter(set == "GO") %>%
        filter(sub %in% c("n","c1")) %>%
        filter(contrast == cont) %>%
        mutate(method = unname(dict[method])) %>% 
        arrange(method, pvalue) %>%
        group_by(method) %>%
        mutate(rank = 1:n()) %>%
        filter(rank <= 100) %>%
        summarise(prop = sum(ID %in% immuneGO$GOID)/n()) %>%
        mutate(truth = "ISP Terms") -> immuneSum
    
    dat %>% filter(set == "GO") %>%
        filter(sub %in% c("n","c1")) %>%
        filter(contrast == cont) %>%
        mutate(method = unname(dict[method])) %>% 
        arrange(method, pvalue) %>%
        group_by(method) %>%
        mutate(rank = 1:n()) %>%
        filter(rank <= 100) %>%
        summarise(prop = sum(ID %in% topGOSets$ID[topGOSets$contrast %in% 
                                                      cont])/n()) %>%
        mutate(truth = "RNAseq Terms") -> rnaseqSum
    
    truthSum <- bind_rows(immuneSum, rnaseqSum)
    
    p[[i]] <- ggplot(truthSum, aes(x = truth, y = prop, fill = method)) + 
        geom_bar(stat = "identity", position = "dodge") +
        labs(fill = "Method", 
             x = "Truth set", 
             y = "Prop. 'truth' terms in top 100") +
        scale_fill_manual(values = methodCols) +
        ggtitle(cont)
}

p[[1]]
p[[2]]
p[[3]]
```

#### Create PDF
```{r, eval=FALSE}
fig <- here("output/Fig-4B.pdf")

pdf(file = fig, width = 9)
p[[1]]
dev.off()

fig <- here("output/SFig-4B.pdf")

pdf(file = fig, width = 9)
p[[2]]
dev.off()

fig <- here("output/SFig-5B.pdf")

pdf(file = fig, width = 9)
p[[3]]
dev.off()
```

### Figure 4C-G, Supp. Figure 4C-G, Supp Figure 5C-G {.tabset}
#### View

Examine what the top 10 ranked gene sets are and how many genes they contain, for each method and comparison.

```{r, fig.height=7, fig.width=9}
terms <- missMethyl:::.getGO()$idTable
nGenes <- rownames_to_column(data.frame(n = sapply(missMethyl:::.getGO()$idList, 
                                                   length)), 
                             var = "ID")

dat %>% filter(set == "GO") %>%
    filter(sub %in% c("n","c1")) %>% 
    mutate(method = unname(dict[method])) %>% 
    arrange(contrast, method, pvalue) %>%
    group_by(contrast, method) %>%
    mutate(FDR = p.adjust(pvalue, method = "BH")) %>%
    mutate(rank = 1:n()) %>%
    filter(rank <= 10) %>% 
    inner_join(terms, by = c("ID" = "GOID")) %>%
    inner_join(nGenes) -> sub

p <- vector("list", length(unique(sub$contrast)))
for(i in 1:length(unique(sub$contrast))){
    cont <- sort(unique(sub$contrast))[i]
    sub %>% filter(contrast == cont) %>%
        arrange(method, -rank) %>%
        ungroup() %>%
        mutate(idx = as.factor(1:n())) -> tmp
    
    ord <- unlist(lapply(1:length(unique(tmp$method)), function(i){
        c((i*10):(i*10-9))
    }))
    termLabs <- substr(tmp$TERM[ord], 1, 50)
    names(termLabs) <- tmp$idx[ord]
    
    rnaTruth <- tmp$ID[ord] %in% topGOSets$ID[topGOSets$contrast %in% cont]
    ispTruth <- tmp$ID[ord] %in% immuneGO$GOID
    sumTruth <- rnaTruth + ispTruth
    
    truthCol <- ifelse(sumTruth == 2, "purple", 
                       ifelse(sumTruth == 1 & rnaTruth == 1, "red",
                              ifelse(sumTruth == 1 & ispTruth == 1, "blue", 
                                     "black")))
        
    p[[i]] <- ggplot(tmp, aes(x = -log10(FDR), y = idx)) +
        geom_point(aes(size = n), alpha = 0.8, 
                   colour = methodCols[tmp$method]) +
        scale_size(limits = c(min(sub$n), max(sub$n))) +
        facet_wrap(vars(method), ncol = 2, scales = "free") +
        scale_y_discrete(labels = termLabs) +
        labs(y = "", size = "No. genes") +
        theme(axis.text.y = element_text(size = 8),
              plot.title = element_text(size = 10),
              legend.position = "bottom",
              legend.text = element_text(size = 8),
              legend.title = element_text(size = 10),
              axis.text.x = element_text(size = 8),
              axis.title.x = element_text(size = 10)) +
        geom_vline(xintercept = -log10(0.05), linetype = "dashed") +
        ggtitle(cont)  
}

p[[1]]
p[[2]]
p[[3]]
```

#### Create PDF
```{r, eval=FALSE}
fig <- here("output/Fig-4C-G.pdf")

pdf(file = fig, width = 9)
p[[1]]
dev.off()

fig <- here("output/SFig-4C-G.pdf")

pdf(file = fig, width = 9)
p[[2]]
dev.off()

fig <- here("output/SFig-5C-G.pdf")

pdf(file = fig, width = 9)
p[[3]]
dev.off()
```

### {-}

P-value histograms for the different methods for all contrasts on GO categories.

```{r, fig.height=8, fig.width=7}
dat %>% filter(set == "GO") %>%
    filter(sub %in% c("n","c1")) %>%
    mutate(method = unname(dict[method])) -> subDat 
    
ggplot(subDat, aes(pvalue, fill = method)) +
    geom_histogram(binwidth = 0.025) +
    facet_grid(cols = vars(contrast), rows = vars(method)) +
    theme(legend.position = "bottom") +
    labs(x = "P-value", y = "Frequency", fill = "Method") +
    scale_fill_manual(values = methodCols)
```

### Compare gometh results using different DM CpG significance cutoffs

```{r, fig.width=7, fig.height=8}
dat %>% filter(set == "GO") %>%
    filter(grepl("mmethyl", method)) %>%
    mutate(method = unname(dict[method])) %>%
    arrange(contrast, method, pvalue) %>%
    group_by(contrast, method, sub) %>%
    mutate(csum = cumsum(ID %in% immuneGO$GOID)) %>% 
    mutate(rank = 1:n()) %>%
    mutate(cut = ifelse(sub == "c1", "Top 5000", 
                        ifelse(sub == "c2", "Top 10000",
                               ifelse(sub == "p1", "FDR < 0.01", "FDR < 0.05")))) %>%
    filter(rank <= 100) -> sub

p <- ggplot(sub, aes(x = rank, y = csum, colour = cut)) +
    geom_line() +
    facet_wrap(vars(contrast, method), ncol=2, nrow = 3, scales = "free") +
    geom_vline(xintercept = 10, linetype = "dotted") +
    labs(colour = "Sig. select", x = "Rank", y = "Cumulative no. immune sets") +
    theme(legend.position = "bottom")
p
```

```{r, fig.width=7, fig.height=8}
dat %>% filter(set == "GO") %>%
    filter(grepl("mmethyl", method)) %>%
    mutate(method = unname(dict[method])) %>%
    arrange(contrast, method, pvalue) %>%
    group_by(contrast, method, sub) %>%
    mutate(csum = cumsum(ID %in% topGOSets$ID[topGOSets$contrast %in% 
                                                  contrast])) %>%
    mutate(rank = 1:n()) %>%
    mutate(cut = ifelse(sub == "c1", "Top 5000", 
                        ifelse(sub == "c2", "Top 10000",
                               ifelse(sub == "p1", "FDR < 0.01", "FDR < 0.05")))) %>%
    filter(rank <= 100) -> sub

p <- ggplot(sub, aes(x = rank, y = csum, colour = cut)) +
    geom_line() +
    facet_wrap(vars(contrast, method), ncol=2, nrow = 3, scales = "free") +
    geom_vline(xintercept = 10, linetype = "dotted") +
    labs(colour = "Sig. select", x = "Rank", 
         y = glue("Cumulative no. RNAseq sets")) +
    theme(legend.position = "bottom")
p
```

## Test KEGG pathways

Now test KEGG pathways with at least 5 genes and 5000 at most.

Again, as we are comparing immune cells we expecte pathways from the following categories to be highly ranked: Immune system, Immune disease, Signal transduction, Signaling molecules and interaction; [https://www.genome.jp/kegg/pathway.html](https://www.genome.jp/kegg/pathway.html). 

```{r, fig.width=7}
immuneKEGG <- read.csv(here("data/kegg-immune-related-pathways.csv"), 
                            stringsAsFactors = FALSE, header = FALSE, 
                            col.names = c("ID","pathway"))
immuneKEGG$PID <- paste0("path:hsa0",immuneKEGG$ID)

dat %>% filter(set == "KEGG") %>%
    filter(sub %in% c("n","c1")) %>% 
    mutate(method = unname(dict[method])) %>%
    arrange(contrast, method, pvalue) %>%
    group_by(contrast, method) %>%
    mutate(csum = cumsum(ID %in% immuneKEGG$PID)) %>%
    mutate(rank = 1:n()) %>%
    filter(rank <= 100) -> sub

p <- ggplot(sub, aes(x = rank, y = csum, colour = method)) +
    geom_line() +
    facet_wrap(vars(contrast), ncol=3) +
    geom_vline(xintercept = 10, linetype = "dotted") +
    labs(colour = "Method", x = "Rank", y = "Cumulative no. immune sets") +
    theme(legend.position = "bottom") +
    scale_color_manual(values = methodCols)
p
```

Examine results when top ranked 100 KEGG pathways from RNA-seq analysis of the same cell type comparisons is used as "truth".

```{r, fig.width=7}
rnaseqKEGG <- readRDS(here("data/RNAseq-KEGG.rds"))
rnaseqKEGG %>% group_by(contrast) %>%
    mutate(rank = 1:n()) %>%
    filter(rank <= 100) -> topKEGG

dat %>% filter(set == "KEGG") %>%
    filter(sub %in% c("n","c1")) %>%
    mutate(method = unname(dict[method])) %>%
    arrange(contrast, method, pvalue) %>%
    group_by(contrast, method) %>%
    mutate(csum = cumsum(ID %in% topKEGG$PID[topKEGG$contrast %in% contrast])) %>%
    mutate(rank = 1:n()) %>%
    filter(rank <= 100) -> sub

p <- ggplot(sub, aes(x = rank, y = csum, colour = method)) +
    geom_line() +
    facet_wrap(vars(contrast), ncol=3, scales = "free_y") +
    geom_vline(xintercept = 10, linetype = "dotted") +
    labs(colour = "Method", x = "Rank", y = "Cumulative no. RNAseq sets") +
    theme(legend.position = "bottom") + 
    scale_color_manual(values = methodCols)
p
```

Examine what the top 10 ranked gene sets are and how many genes they contain, for each method and comparison.

```{r, fig.height=7, fig.width=9}
terms <- missMethyl:::.getKEGG()$idTable
nGenes <- rownames_to_column(data.frame(n = sapply(missMethyl:::.getKEGG()$idList, 
                                                   length)), 
                             var = "ID")

dat %>% filter(set == "KEGG") %>%
    filter(sub %in% c("n","c1")) %>% 
    mutate(method = unname(dict[method])) %>%
    arrange(contrast, method, pvalue) %>%
    group_by(contrast, method) %>%
    mutate(FDR = p.adjust(pvalue, method = "BH")) %>%
    mutate(rank = 1:n()) %>%
    filter(rank <= 10) %>% 
    inner_join(terms, by = c("ID" = "PathwayID")) %>%
    inner_join(nGenes) -> sub

p <- vector("list", length(unique(sub$contrast)))
for(i in 1:length(unique(sub$contrast))){
    cont <- unique(sub$contrast)[i]
    sub %>% filter(contrast == cont) %>%
        arrange(method, -rank) %>%
        ungroup() %>%
        mutate(idx = as.factor(1:n())) -> tmp
    
    ord <- unlist(lapply(1:length(unique(tmp$method)), function(i){
        c((i*10):(i*10-9))
    }))
    termLabs <- substr(tmp$Description[ord], 1, 50)
    names(termLabs) <- tmp$idx[ord]
    
    p[[i]] <- ggplot(tmp, aes(x = -log10(FDR), y = idx)) +
        geom_point(aes(size = n), alpha = 0.8, colour = methodCols[tmp$method]) +
        facet_wrap(vars(method), ncol = 2, scales = "free") +
        scale_y_discrete(labels = termLabs) +
        labs(y = "", size = "No. genes") +
        theme(axis.text.y = element_text(size = 8),
              plot.title = element_text(size = 10),
              legend.position = "bottom",
              legend.text = element_text(size = 8),
              legend.title = element_text(size = 10),
              axis.text.x = element_text(size = 8),
              axis.title.x = element_text(size = 10)) +
        geom_vline(xintercept = -log10(0.05), linetype = "dashed") +
        ggtitle(cont)  
}

p[[1]]
p[[2]]
p[[3]]
```


P-value histograms for the different methods for all contrasts on KEGG pathways.

```{r, fig.height=8, fig.width=7}
dat %>% filter(set == "KEGG") %>%
    filter(sub %in% c("n","c1")) %>%
    mutate(method = unname(dict[method])) -> subDat 
    
ggplot(subDat, aes(pvalue, fill = method)) +
    geom_histogram(binwidth = 0.025) +
    facet_grid(cols = vars(contrast), rows = vars(method)) +
    theme(legend.position = "bottom") +
    labs(x = "P-value", y = "Frequency", fill = "Method") +
    scale_fill_manual(values = methodCols)
```

### Compare gometh results using different DM CpG significance cutoffs

```{r, fig.width=7, fig.height=8}
dat %>% filter(set == "KEGG") %>%
    filter(grepl("mmethyl", method)) %>% 
    mutate(method = unname(dict[method])) %>%
    arrange(contrast, method, pvalue) %>%
    group_by(contrast, method, sub) %>%
    mutate(csum = cumsum(ID %in% immuneKEGG$PID)) %>% 
    mutate(rank = 1:n()) %>%
    mutate(cut = ifelse(sub == "c1", "Top 5000", 
                        ifelse(sub == "c2", "Top 10000",
                               ifelse(sub == "p1", "FDR < 0.01", "FDR < 0.05")))) %>%
    filter(rank <= 100) -> sub

p <- ggplot(sub, aes(x = rank, y = csum, colour = cut)) +
    geom_line() +
    facet_wrap(vars(contrast, method), ncol=2, nrow = 3, scales = "free") +
    geom_vline(xintercept = 10, linetype = "dotted") +
    labs(colour = "Sig. select", x = "Rank", y = "Cumulative no. immune sets") +
    theme(legend.position = "bottom")
p
```

```{r, fig.width=7, fig.height=8}
dat %>% filter(set == "KEGG") %>%
    filter(grepl("mmethyl", method)) %>% 
    mutate(method = unname(dict[method])) %>%
    arrange(contrast, method, pvalue) %>%
    group_by(contrast, method, sub) %>%
    mutate(csum = cumsum(ID %in% topKEGG$PID[topKEGG$contrast %in% contrast])) %>%
    mutate(rank = 1:n()) %>%
    mutate(cut = ifelse(sub == "c1", "Top 5000", 
                        ifelse(sub == "c2", "Top 10000",
                               ifelse(sub == "p1", "FDR < 0.01", "FDR < 0.05")))) %>%
    filter(rank <= 100) -> sub

p <- ggplot(sub, aes(x = rank, y = csum, colour = cut)) +
    geom_line() +
    facet_wrap(vars(contrast, method), ncol=2, nrow = 3, scales = "free") +
    geom_vline(xintercept = 10, linetype = "dotted") +
    labs(colour = "Sig. select", x = "Rank", 
         y = glue("Cumulative no. RNAseq sets")) +
    theme(legend.position = "bottom")
p
```


## Test BROAD gene sets

As the ebGSEA method from the ChAMP package only allows testing of their in-built database of Broad Institute gene sets, we will compare to other methods by testing these same gene sets. Using the top 100 ranked gene sets as identified by Belinda's `gsaseq` analysis of the corresponding RNAseq data as "truth".

```{r, fig.width=7}
rnaseqBROAD <- readRDS(here("data/RNAseq-BROAD-GSA.rds"))
rnaseqBROAD %>% group_by(contrast) %>%
    mutate(rank = 1:n()) %>%
    filter(rank <= 100) -> topBROAD

dat %>% filter(set == "BROAD") %>%
    filter(sub %in% c("n","c1")) %>% 
    mutate(method = unname(dict[method])) %>% 
    arrange(contrast, method, pvalue) %>%
    group_by(contrast, method) %>%
    mutate(csum = cumsum(ID %in% topBROAD$ID[topBROAD$contrast %in% contrast])) %>%
    mutate(rank = 1:n()) %>%
    filter(rank <= 100) -> sub

p <- ggplot(sub, aes(x = rank, y = csum, colour = method)) +
    geom_line() +
    facet_wrap(vars(contrast), ncol=3, scales = "free_y") +
    geom_vline(xintercept = 10, linetype = "dotted") +
    labs(colour = "Method", x = "Rank", y = "Cumulative no. RNAseq sets") +
    theme(legend.position = "bottom") +
    scale_color_manual(values = methodCols)
p
```

### Supp. Figure 6A, Supp. Figure 7A, Supp. Figure 8A {.tabset}
#### View

```{r}
p <- vector("list", length(unique(dat$contrast)))

for(i in 1:length(unique(dat$contrast))){
    cont <- sort(unique(dat$contrast))[i]
    
    dat %>% filter(set == "BROAD") %>%
        filter(sub %in% c("n","c1")) %>%
        filter(contrast == cont) %>%
        mutate(method = unname(dict[method])) %>% 
        arrange(method, pvalue) %>%
        group_by(method) %>%
        mutate(rank = 1:n()) %>%
        filter(rank <= 100) %>%
        summarise(prop = sum(ID %in% topBROAD$ID[topBROAD$contrast %in% cont])/n()) %>%
        mutate(truth = "RNAseq Terms") -> rnaseqSum
    
    p[[i]] <- ggplot(rnaseqSum, aes(x = truth, y = prop, fill = method)) + 
        geom_bar(stat = "identity", position = "dodge") +
        labs(fill = "Method", 
             x = "Truth set", 
             y = "Prop. 'truth' terms in top 100") +
        scale_fill_manual(values = methodCols) +
        ggtitle(cont)
}

p[[1]]
p[[2]]
p[[3]]
```

#### Create PDF
```{r, eval=FALSE}
fig <- here("output/SFig-6A.pdf")

pdf(file = fig, width = 9)
p[[1]]
dev.off()

fig <- here("output/SFig-7A.pdf")

pdf(file = fig, width = 9)
p[[2]]
dev.off()

fig <- here("output/SFig-8A.pdf")

pdf(file = fig, width = 9)
p[[3]]
dev.off()
```

### Supp Figure 6B-F, 7B-F, 8B-F{.tabset}
#### View

```{r, fig.height=8, fig.width=9}
data(PathwayList)
nGenes <- rownames_to_column(data.frame(n = sapply(PathwayList, 
                                                   length)), 
                             var = "ID")

dat %>% filter(set == "BROAD") %>%
    filter(sub %in% c("n","c1")) %>% 
    mutate(method = unname(dict[method])) %>% 
    arrange(contrast, method, pvalue) %>%
    group_by(contrast, method) %>%
    mutate(FDR = p.adjust(pvalue, method = "BH")) %>%
    mutate(rank = 1:n()) %>%
    filter(rank <= 10) %>% 
    inner_join(nGenes) -> sub

p <- vector("list", length(unique(sub$contrast)))
for(i in 1:length(unique(sub$contrast))){
    cont <- sort(unique(sub$contrast))[i]
    sub %>% filter(contrast == cont) %>%
        arrange(method, -rank) %>%
        ungroup() %>%
        mutate(idx = as.factor(1:n())) -> tmp
    
    ord <- unlist(lapply(1:length(unique(tmp$method)), function(i){
        c((i*10):(i*10-9))
    }))
    termLabs <- substr(tmp$ID[ord], 1, 50)
    names(termLabs) <- tmp$idx[ord]
    
    truthCol <- ifelse(termLabs %in% topBROAD$ID[topBROAD$contrast %in% cont]
                       [1:100], "red", "black")
    
    p[[i]] <- ggplot(tmp, aes(x = -log10(FDR), y = idx)) +
        geom_point(aes(size = n), alpha = 0.8, colour = methodCols[tmp$method]) +
        facet_wrap(vars(method), ncol = 2, scales = "free") +
        scale_y_discrete(labels = termLabs) +
        labs(y = "", size = "No. genes") +
        theme(axis.text.y = element_text(size = 6, colour = truthCol),
              plot.title = element_text(size = 10),
              legend.position = "bottom",
              legend.text = element_text(size = 8),
              legend.title = element_text(size = 10),
              axis.text.x = element_text(size = 8),
              axis.title.x = element_text(size = 10)) +
        geom_vline(xintercept = -log10(0.05), linetype = "dashed") +
        ggtitle(cont)  
}

p[[1]]
p[[2]]
p[[3]]
```

#### Create PDF
```{r, eval=FALSE}
fig <- here("output/SFig-6B-F.pdf")

pdf(file = fig, width = 9)
p[[1]]
dev.off()

fig <- here("output/SFig-7B-F.pdf")

pdf(file = fig, width = 9)
p[[2]]
dev.off()

fig <- here("output/SFig-8B-F.pdf")

pdf(file = fig, width = 9)
p[[3]]
dev.off()
```

### {-}

P-value histograms for the different methods for all contrasts on BROAD gene sets.

```{r, fig.height=10, fig.width=7}
dat %>% filter(set == "BROAD") %>%
    filter(sub %in% c("n","c1")) %>%
    mutate(method = unname(dict[method])) -> subDat 
    
ggplot(subDat, aes(pvalue, fill = method)) +
    geom_histogram(binwidth = 0.025) +
    facet_grid(cols = vars(contrast), rows = vars(method)) +
    theme(legend.position = "bottom") +
    labs(x = "P-value", y = "Frequency", fill = "Method") +
    scale_fill_manual(values = methodCols)
```

### Compare gometh results using different DM CpG significance cutoffs

```{r, fig.width=7, fig.height=8}
dat %>% filter(set == "BROAD") %>%
    filter(grepl("mmethyl", method)) %>% 
    mutate(method = unname(dict[method])) %>%
    arrange(contrast, method, pvalue) %>%
    group_by(contrast, method, sub) %>%
    mutate(csum = cumsum(ID %in% topBROAD$ID)) %>% 
    mutate(rank = 1:n()) %>%
    mutate(cut = ifelse(sub == "c1", "Top 5000", 
                        ifelse(sub == "c2", "Top 10000",
                               ifelse(sub == "p1", "FDR < 0.01", "FDR < 0.05")))) %>%
    filter(rank <= 100) -> sub

p <- ggplot(sub, aes(x = rank, y = csum, colour = cut)) +
    geom_line() +
    facet_wrap(vars(contrast, method), ncol=2, nrow = 3, scales = "free") +
    geom_vline(xintercept = 10, linetype = "dotted") +
    labs(colour = "Sig. select", x = "Rank", y = "Cumulative no. immune sets") +
    theme(legend.position = "bottom")
p
```

```{r, fig.width=7, fig.height=8}
dat %>% filter(set == "BROAD") %>%
    filter(grepl("mmethyl", method)) %>% 
    mutate(method = unname(dict[method])) %>%
    arrange(contrast, method, pvalue) %>%
    group_by(contrast, method, sub) %>%
    mutate(csum = cumsum(ID %in% topBROAD$ID)) %>% 
    mutate(rank = 1:n()) %>%
    mutate(cut = ifelse(sub == "c1", "Top 5000", 
                        ifelse(sub == "c2", "Top 10000",
                               ifelse(sub == "p1", "FDR < 0.01", "FDR < 0.05")))) %>%
    filter(rank <= 100) -> sub

p <- ggplot(sub, aes(x = rank, y = csum, colour = cut)) +
    geom_line() +
    facet_wrap(vars(contrast, method), ncol=2, nrow = 3, scales = "free") +
    geom_vline(xintercept = 10, linetype = "dotted") +
    labs(colour = "Sig. select", x = "Rank", y = "Cumulative no. immune sets") +
    theme(legend.position = "bottom")
p
```
