---
title: "Gene set testing for Illumina HumanMethylation Arrays"
author: "Jovana Maksimovic"
date: "7/2/2019"
output: html_document
---

```{r libload, message=FALSE}
library(here)
library(minfi)
library(paletteer)
library(limma)
library(reshape2)
library(missMethyl)
library(ggplot2)
library(glue)
library(tibble)
library(dplyr)
library(curatedTCGAData)
library(MultiAssayExperiment)
library(TCGAutils)
library(DMRcate)
library(patchwork)
source(here("code/utility.R"))
```

# Load data

We are using publicly available kidney clear cell carcinoma (KIRC) 450k data from The Cancer Genome Atlas (TCGA). We are using only the normal samples to look at false discovery rate control by various methylation gene set testing methods.

Download the data using the `curatedTCGAData` Bioconductor package and then extract the normal samples. The data seems to have already been normalised(?) so we will only remove poor probes (>1 NA), SNP, multi-mapping and sex-chromosome probes. 

```{r}
kirc <- curatedTCGAData(diseaseCode = "KIRC", assays = "Methylation_methyl450", 
                        dry.run = FALSE)
kirc <- splitAssays(kirc, c("11")) # extract only the normal samples
exp <- experiments(kirc)[[1]]
meta <- colData(kirc)
betas <- as.matrix(assay(exp))
colnames(betas) <- substr(colnames(betas), 1, 12)
m <- match(colnames(betas), meta$patientID)
meta <- meta[m, ]
head(meta[, 1:5])
```

```{r}
betasNoNA <- betas[rowSums(is.na(betas)) == 0, ]
mds <- plotMDS(betasNoNA, gene.selection = "common", plot = FALSE)
dat <- tibble(x = mds$x, y = mds$y, gender = meta$gender)

ggplot(dat, aes(x = x, y = y, colour = gender)) +
  geom_point() +
  labs(x = "Principal Component 1", y = "Principal Component 2", 
       colour = "Gender")
```

```{r, message=FALSE}
betasFlt <- rmSNPandCH(betasNoNA, rmXY = TRUE, rmcrosshyb = TRUE)
mds <- plotMDS(betasFlt, gene.selection = "common", plot = FALSE)
dat <- tibble(x = mds$x, y = mds$y, gender = meta$gender)

ggplot(dat, aes(x = x, y = y, colour = gender)) +
  geom_point() +
  labs(x = "Principal Component 1", y = "Principal Component 2", 
       colour = "Gender")
```

```{r, fig.width=8, fig.height=7}
dat$age <- meta$years_to_birth
dat$race <- sub(" or", "\nor", meta$race)
dat$ethnicity <- sub(" or", "\nor", meta$ethnicity)

a <- ggplot(dat, aes(x = x, y = y, colour = age)) +
  geom_point() +
  labs(x = "Principal Component 1", y = "Principal Component 2", 
       colour = "Age") +
  viridis::scale_color_viridis(direction = -1)

b <- ggplot(dat, aes(x = x, y = y, colour = race)) +
  geom_point() +
  labs(x = "Principal Component 1", y = "Principal Component 2", 
       colour = "Race") +
  theme(legend.text = element_text(size = 7))

c <- ggplot(dat, aes(x = x, y = y, colour = ethnicity)) +
  geom_point() +
  labs(x = "Principal Component 1", y = "Principal Component 2", 
       colour = "Ethnicity")  +
  theme(legend.text = element_text(size = 7))

(b + c) / a
```

```{r}
dat <- as_tibble(melt(betasFlt))
colnames(dat) <- c("cpg", "ID", "beta")

p <- ggplot(dat, aes(x = beta)) + 
  geom_line(aes(color = ID), stat="density", size=0.5, alpha=0.5,
            show.legend = FALSE)

p + labs(x = "Beta value", y = "Density")
```

# FDR analysis

Run 100 null simulations comparing 5, 10, 20, 40, 80 randomly selected samples per "group". There should be no significant differential methylation between the groups as the data contains no signal. Consequently, we expect about 5% false gene set discoveries across the 100 simulations from methods that are "holding their size". We will compare gometh (with top 1000 and 5000 ranked CpGs selected), MethylGSA:GLM, MethylGSM:ORA, MethylGSA:GSEA and ChAMP:ebGSEA for the complete list of BROAD gene sets provided in the *ChAMP* package. 

```{r}
outFile <- here("data/TCGA.KIRC.rds")

if(!file.exists(outFile)){
    saveRDS(betasFlt, file = outFile)
}
```

Load the results of the FDR simulations.

```{r}
inFile <- here("output/FDR-analysis/FDR-analysis.rds")
if(file.exists(inFile)) dat <- as_tibble(readRDS(inFile))
```

The plots below shows that MethylGSA-ORA does not control the false discovery rate very well as the median proportion of p-value 0.05 for the 100 simulations is greater than 0.5. MethylGSA-GSEA does better, although its median false discovery rate is still relatively high at around 0.25. MethylGSA-GLM has good false discovery rate control with median FDR at around 0.05. ChAMP-ebGSEA is only slightly anti-conservative using both the wicox test (WT) and known population median test (KPMT) with a median FDR at around 0.06-0.08. MissMethyl-gometh is very consistent although somewhat conservative with a median FDR at around 0.02-0.03. 

```{r, fig.width=9}  
dat %>% group_by(simNo, sampleNo, method) %>%
  summarise(psig = sum(pvalue < 0.05)/length(pvalue)) -> psigData

p <- ggplot(psigData, aes(x = sampleNo, y = psig, fill = method)) + 
    geom_violin(scale = "width", width = 0.8)
p + stat_summary(geom = "point", size = 1, color = "white", 
                 position = position_dodge(0.8),
                 show.legend = FALSE, fun = median) +
  geom_hline(yintercept=0.05, linetype="dashed", color = "red") +
  labs(y="Prop. gene sets with p-value < 0.05", x="No. samples",
       fill="Method")
```
