---
title: "Gene set testing for Illumina HumanMethylation Arrays"
author: "Jovana Maksimovic"
date: "7/2/2019"
output: html_document
---

```{r libload, message=FALSE}
library(here)
library(minfi)
library(paletteer)
library(limma)
library(reshape2)
library(missMethyl)
library(ggplot2)
library(glue)
library(tibble)
library(dplyr)
library(curatedTCGAData)
library(MultiAssayExperiment)
library(TCGAutils)
library(DMRcate)
library(patchwork)
source(here("code/utility.R"))
```

# Load data

We are using publicly available kidney clear cell carcinoma (KIRC) 450k data from The Cancer Genome Atlas (TCGA). We are using only the normal samples to look at false discovery rate control by various methylation gene set testing methods.

Download the data using the `curatedTCGAData` Bioconductor package and then extract the normal samples. Remove poor probes, SNP, multi-mapping and sex-xhromosome probes. 

```{r}
kirc <- curatedTCGAData(diseaseCode = "KIRC", assays = "Methylation_methyl450", 
                        dry.run = FALSE)
kirc <- splitAssays(kirc, c("11")) # extract only the normal samples
exp <- experiments(kirc)[[1]]
meta <- colData(kirc)
betas <- as.matrix(assay(exp))
colnames(betas) <- substr(colnames(betas), 1, 12)
m <- match(colnames(betas), meta$patientID)
meta <- meta[m, ]
head(meta[, 1:5])
```

```{r}
betasNoNA <- betas[rowSums(is.na(betas)) == 0, ]
mds <- plotMDS(betasNoNA, gene.selection = "common", plot = FALSE)
dat <- tibble(x = mds$x, y = mds$y, gender = meta$gender)

ggplot(dat, aes(x = x, y = y, colour = gender)) +
  geom_point() +
  labs(x = "Principal Component 1", y = "Principal Component 2", 
       colour = "Gender")
```

```{r, message=FALSE}
betasFlt <- rmSNPandCH(betasNoNA, rmXY = TRUE, rmcrosshyb = TRUE)
mds <- plotMDS(betasFlt, gene.selection = "common", plot = FALSE)
dat <- tibble(x = mds$x, y = mds$y, gender = meta$gender)

ggplot(dat, aes(x = x, y = y, colour = gender)) +
  geom_point() +
  labs(x = "Principal Component 1", y = "Principal Component 2", 
       colour = "Gender")
```

```{r, fig.width=8, fig.height=7}
dat$age <- meta$years_to_birth
dat$race <- sub(" or", "\nor", meta$race)
dat$ethnicity <- sub(" or", "\nor", meta$ethnicity)

a <- ggplot(dat, aes(x = x, y = y, colour = age)) +
  geom_point() +
  labs(x = "Principal Component 1", y = "Principal Component 2", 
       colour = "Age") +
  viridis::scale_color_viridis(direction = -1)

b <- ggplot(dat, aes(x = x, y = y, colour = race)) +
  geom_point() +
  labs(x = "Principal Component 1", y = "Principal Component 2", 
       colour = "Race") +
  theme(legend.text = element_text(size = 7))

c <- ggplot(dat, aes(x = x, y = y, colour = ethnicity)) +
  geom_point() +
  labs(x = "Principal Component 1", y = "Principal Component 2", 
       colour = "Ethnicity")  +
  theme(legend.text = element_text(size = 7))

(b + c) / a
```

```{r}
dat <- as_tibble(melt(betasFlt))
colnames(dat) <- c("cpg", "ID", "beta")

p <- ggplot(dat, aes(x = beta)) + 
  geom_line(aes(color = ID), stat="density", size=0.5, alpha=0.5,
            show.legend = FALSE)

p + labs(x = "Beta value", y = "Density")
```

# FDR analysis

Run 100 null simulations comparing 5, 10, 20, 40, 80 randomly selected samples per "group". There should be no significant differential methylation between the groups as the data contains no signal. Consequently, we expect about 5% false gene set discoveries across the 100 simulations from methods that are "holding their size". We will compare gometh (with top 1000 and 5000 ranked CpGs selected), MethylGSA:GLM, MethylGSM:ORA, MethylGSA:GSEA and ChAMP:ebGSEA for the complete list of BROAD gene sets provided in the *ChAMP* package. 

```{r}
outFile <- here("data/TCGA.KIRC.rds")

if(!file.exists(outFile)){
    saveRDS(betasFlt, file = outFile)
}
```

Load the results of the FDR simulations.

```{r}
nullDataFile <- here("output/nullData.long.RData")

if(!file.exists(nullDataFile)){
  
  load(here("output/nullDat.gm.long.RData"))
  gmDat <- as_tibble(longDat)
  rm(longDat)

  load(here("data/nullLong.RData"))
  allDat <- as_tibble(longDat)
  rm(longDat)

  allDat %>% filter(method != "gm1" & method != "gm2") %>% 
    droplevels() -> allDat

  nullDat <- bind_rows(gmDat, allDat)

  nullDat %>% mutate_at(c("method","sim","samps"), as.factor) %>%
    mutate(method = recode_factor(method, gm1 = "a", gm2 = "b", glm = "c", ora = "d", 
                         gsea = "e", ebgs = "f", .default = levels(method))) %>%
    mutate(method = factor(method, labels = c("gometh-500", "gometh-1000", "mrra-glm", 
                                            "mrra-ora", "mrra-gsea", 
                                            "ebgsea"))) -> nullDat

  save(nullDat, file = here("output/nullData.long.RData"))
} else {
  load(here("output/nullData.long.RData"))
}
```

The plots below shows that mRRA-ORA does not control the false discovery rate very well as the median proportion of p-value 0.05 for the 100 simulations is greater than 0.75. mRRA-GSEA does better, although its median false discovery rate is still relatively high at around 0.2. ebGSEA and mRRA-GLM both have a median false discovery rate at around 0.05, although ebGSEA has a tail of simulations between 0.05 and 0.2. Gometh is very consistent althought somewhat conservative in terms of false discovery control. 

```{r, fig.width=9}  
nullDat %>% group_by(sim, samps, method) %>%
  summarise(psig = sum(pval < 0.05)/length(pval)) -> psigData

p <- ggplot(psigData, aes(x=samps, y=psig, fill=method)) + 
  geom_violin(scale = "width")
p + scale_fill_paletteer_d(package = "LaCroixColoR", palette = "PassionFruit") +
  stat_summary(geom="point", size=0.8, color="white", position = position_dodge(0.9),
               show.legend = FALSE, fun.y = median) +
  geom_hline(yintercept=0.05, linetype="dashed", color = "red") +
  labs(y="Prop. gene sets with p-value < 0.05", x="No. samples",
       fill="Method") +
  theme_minimal()
```
