---
title: "Gene set testing for Illumina HumanMethylation Arrays"
author: "Jovana Maksimovic"
date: "7/2/2019"
output: html_document
---

```{r libload, message=FALSE}
library(here)
library(minfi)
library(paletteer)
library(limma)
library(reshape2)
library(missMethyl)
library(ggplot2)
library(glue)
library(tibble)
library(dplyr)
library(curatedTCGAData)
library(MultiAssayExperiment)
library(TCGAutils)
library(DMRcate)
source(here("code/utility.R"))
```

```{r}
kirc <- curatedTCGAData(diseaseCode = "KIRC", assays = "Methylation_methyl450", 
                        dry.run = FALSE)
kirc <- splitAssays(kirc, c("11"))
exp <- experiments(kirc)[[1]]
meta <- colData(kirc)
betas <- as.matrix(assay(exp))
colnames(betas) <- substr(colnames(betas), 1, 12)
m <- match(colnames(betas), meta$patientID)
meta <- meta[m, ]
head(meta[, 1:5])
```

```{r}
betasNoNA <- betas[rowSums(is.na(betas)) == 0, ]
mds <- plotMDS(betasNoNA, gene.selection = "common", plot = FALSE)
dat <- tibble(x = mds$x, y = mds$y, gender = meta$gender)

ggplot(dat, aes(x = x, y = y, colour = gender)) +
  geom_point() +
  labs(x = "Principal Component 1", y = "Principal Component 2", 
       colour = "Gender")
```

```{r, message=FALSE}
betasFlt <- rmSNPandCH(betasNoNA, rmXY = TRUE, rmcrosshyb = TRUE)
mds <- plotMDS(betasFlt, gene.selection = "common", plot = FALSE)
dat <- tibble(x = mds$x, y = mds$y, gender = meta$gender)

ggplot(dat, aes(x = x, y = y, colour = gender)) +
  geom_point() +
  labs(x = "Principal Component 1", y = "Principal Component 2", 
       colour = "Gender")
```

```{r, fig.width=8, fig.height=7}
dat$age <- meta$years_to_birth
dat$race <- sub(" or", "\nor", meta$race)
dat$ethnicity <- sub(" or", "\nor", meta$ethnicity)

a <- ggplot(dat, aes(x = x, y = y, colour = age)) +
  geom_point() +
  labs(x = "Principal Component 1", y = "Principal Component 2", 
       colour = "Age") +
  viridis::scale_color_viridis(direction = -1)

b <- ggplot(dat, aes(x = x, y = y, colour = race)) +
  geom_point() +
  labs(x = "Principal Component 1", y = "Principal Component 2", 
       colour = "Race") +
  theme(legend.text = element_text(size = 7))

c <- ggplot(dat, aes(x = x, y = y, colour = ethnicity)) +
  geom_point() +
  labs(x = "Principal Component 1", y = "Principal Component 2", 
       colour = "Ethnicity")  +
  theme(legend.text = element_text(size = 7))

(b + c) / a
```

```{r}
dat <- as_tibble(melt(betasFlt))
colnames(dat) <- c("cpg", "ID", "beta")

p <- ggplot(dat, aes(x = beta)) + 
  geom_line(aes(color = ID), stat="density", size=0.5, alpha=0.5,
            show.legend = FALSE)

p + labs(x = "Beta value", y = "Density")
```

# Load data

We are using publicly available 450k data from a buc arthritis study [GSE42861](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=gse42861). Genome-wide DNA methylation level was studied to determine whether Rheumatoid arthritis patients (cases) have methylation differences compared to normal controls in peripheral blood leukocytes (PBLs). We are using only the normal samples to look at false discovery rate control by various methylation gene set testing methods.

Dwonload the data from GEO and then extract the normal samples. Remove poor probes, normalise and then remove SNPs, multi-mapping and sex-xhromosome probes. Save the processed data and the raw RG set for the normal samples.

```{r}
# load data
dataFile <- here("data/GSE42861-data.RData")
if(file.exists(dataFile)){
  # load processed data and sample information
  load(dataFile)
} else {
  # download data, normalise, filter and save objects
  readData(dataFile, dataSrc = "RA", saveRG = TRUE)
  # load processed data and sample information
  load(dataFile)
}
```

Get the `M` and $\beta$ values.

```{r}
mVals <- getM(fltGr)
bVals <- getBeta(fltGr)
```

## QC plots

Examine the mean detection p-values to identify any poor quality samples.  

```{r, fig.width=10}
# plot mean detection p-values across all samples
par(mar=c(10,4,4,2))
graphics::barplot(colMeans(detP), las=2, ylim=c(0,0.05),
         cex.names=0.5,ylab="Mean detection p-values", cex.axis = 0.8)
abline(h=0.05,col="red")
```

Examine the $\beta$ value distribution.

```{r}
# plot normalised beta value distribution
par(mfrow=c(1,1))
densityPlot(bVals, main="Normalized", legend=FALSE)
```

Use MDS plots to look at largest sources of variation. There is some structure present in principal component 2 that is not accounted for by sex, age or smoking status. There appears to be some separation of samples by "age" in principal component 1, with younger individuals tending towards the left and older individual towars the right. Age does not fully explain the pattern though; suspect it might be differences in cell type proportions.

```{r, fig.width=8, fig.height=7}
ksPal <- paletteer_d(package = "LaCroixColoR", palette = "KiwiSandia")
# MDS plots to look at largest sources of variation

colBy <- list(Sex = targets$`gender:ch1`,
              AgeInterval = findInterval(targets$`age:ch1`,c(20,30,40,50,60,70)),
              Smoking = targets$`smoking status:ch1`)
legendLabs <- list(Sex = levels(factor(targets$`gender:ch1`)),
              AgeInterval = c(20,30,40,50,60,70),
              Smoking = levels(factor(targets$`smoking status:ch1`)))
  
for(i in 1:length(colBy)){
par(mfrow=c(2,2), oma=c(0,0,2,0))
plotMDS(mVals, top=1000, gene.selection="common", pch = 19,
         col=ksPal[factor(colBy[[i]])])
legend("topright",legend = legendLabs[[i]], col = ksPal, pch = 19)
plotMDS(mVals, top=1000, gene.selection="common", pch = 19, dim.plot = c(1,3),
         col=ksPal[factor(colBy[[i]])])
plotMDS(mVals, top=1000, gene.selection="common", pch = 19, dim.plot = c(2,3),
         col=ksPal[factor(colBy[[i]])])
plotMDS(mVals, top=1000, gene.selection="common", pch = 19, dim.plot = c(3,4),
         col=ksPal[factor(colBy[[i]])])
mtext(names(colBy)[i],outer = TRUE)
}
```

We can see the separation in principal component 1 in the negative control probes but do not see the "age" effect. This suggests that the structure in principal component 2 is technical.

```{r}
load(here("data/GSE42861/rawRg.RData"))
m <- match(targets$id, colnames(rawRg))
rawRg <- rawRg[,m]

neg <- getINCs(rawRg) # get the Illumina negative control probes

plotMDS(neg, gene.selection = "common", pch = 19,
         col=ksPal[factor(colBy[[2]])])
legend("bottomleft",legend = legendLabs[[2]], col = ksPal, pch = 19)
```

## Estimate cell type proportions

Estimate the cell type proportions of the normal PBL samples. 

```{r}
forceRedo <- FALSE
cellCountFile <- here("data/cellCounts.RData")
  
if(!file.exists(cellCountFile) | forceRedo){
  estCounts <- estimateCellCounts(rawRg, compositeCellType = "Blood",
                                  processMethod = "auto", probeSelect = "auto",
                                  cellTypes = c("CD8T","CD4T", "NK","Bcell","Mono","Gran"),
                                  referencePlatform = c("IlluminaHumanMethylation450k"),
                                  returnAll = FALSE, meanPlot = FALSE, verbose = TRUE)
  save(estCounts, file = cellCountFile)

} else {
  load(cellCountFile)
}
```

There does appear to be a change in cell type proportions with age. Proportion of NK cells tends to increase with age and there seem to be reciprocal changes between granulocytes and T cells.

```{r}
cellDat <- data.frame(estCounts)
#cellDat$Tcell <- cellDat$CD4T + cellDat$CD8T
#cellDat <-subset(cellDat, select = -c(CD4T,CD8T))
cellDat$Sample <- rownames(cellDat)
cellDat$Age <- findInterval(targets$`age:ch1`,c(20,30,40,50,60,70))
cellDat <- reshape::melt(cellDat, id.vars = c("Sample","Age"))
colnames(cellDat) <- c("Sample","Age","Cell","Proportion")

cellDat <- as_tibble(cellDat)
cellDat %>% mutate(Age = replace(Age, Age == 1, 2)) %>%
  mutate(Age = replace(Age, Age == 6, 5)) %>%
  mutate_at(c("Age","Cell"), as.factor) -> cellDat

p <- ggplot(cellDat, aes(x=Cell, y=Proportion, fill=Age)) + 
  geom_violin()
p + scale_fill_brewer(palette="Set1", labels = c("<40","40","50",">60")) +
  stat_summary(geom="point", size=0.8, color="white", position = position_dodge(0.9),
               show.legend = FALSE, fun.y = median) +
  labs(y="Proportion", x="Cell Type",
       fill="Age") + 
  theme_minimal()
```

## Adjust data

Before performing the null analysis, we will adjust the data for the structure in the second principal component and the estimated cell type proportions.

```{r}
o <- order(rowVars(mVals), decreasing = TRUE)[1:1000]
var <- mVals[o,]
pca <- prcomp(var)

design <- model.matrix(~pca$rotation[,2] + estCounts[,1] + estCounts[,2] + estCounts[,3] + 
                         estCounts[,4] + estCounts[,5] + estCounts[,6])
fit <- lmFit(mVals, design)
fit <- eBayes(fit)

res <- mVals - fitted(fit)
```

The adjusted data no longer shows any obvious structure.   

```{r, fig.width=8, fig.height=7}
for(i in 1:length(colBy)){
par(mfrow=c(2,2), oma=c(0,0,2,0))
plotMDS(res, top=1000, gene.selection="common", pch = 19,
         col=ksPal[factor(colBy[[i]])])
legend("topright",legend = legendLabs[[i]], col = ksPal, pch = 19)
plotMDS(res, top=1000, gene.selection="common", pch = 19, dim.plot = c(1,3),
         col=ksPal[factor(colBy[[i]])])
plotMDS(res, top=1000, gene.selection="common", pch = 19, dim.plot = c(2,3),
         col=ksPal[factor(colBy[[i]])])
plotMDS(res, top=1000, gene.selection="common", pch = 19, dim.plot = c(3,4),
         col=ksPal[factor(colBy[[i]])])
mtext(names(colBy)[i],outer = TRUE)
}
```

# Null simulation analysis

Run 100 null simulations comparing 5, 10, 25, 50 and 100 randomly selected samples per group. There should be no significant differential methylation between the groups as the data contains no signal. Consequently, we expect about 5% false gene set discoveries across the 100 simulations from methods that are "holding their size". We will compare gometh (with top 500 and 1000 CpGs selected), mRRA-GLM, mRRA-ORA, mRRA-GSEA and ebGSEA for the complete list of BROAD gene sets provided in the *ChAMP* package. 

```{r}
forceRedo <- FALSE

# run sims on different numbers of samples
sampleNums <- c(5, 10, 25, 50, 100)
simNum <- 100 # number of simulations
scriptFile <- here("code/nullData.R")

inputFile <- here("data/inputNullDat.RData")
if(!file.exists(inputFile)) save(res, file = inputFile)

for(samp in sampleNums){
  nullDatFile <- here(glue("output/nullDat.{samp}.RData"))

  if(!file.exists(nullDatFile) | forceRedo){
    runBgAnalysis(scriptFile, args = c(inputFile, samp, simNum), 
                  outPrefix = glue("nullDat.{samp}"))
  } 
}
```

Load the results of the null data simulations.

```{r}
nullDataFile <- here("output/nullData.long.RData")

if(!file.exists(nullDataFile)){
  
  load(here("output/nullDat.gm.long.RData"))
  gmDat <- as_tibble(longDat)
  rm(longDat)

  load(here("data/nullLong.RData"))
  allDat <- as_tibble(longDat)
  rm(longDat)

  allDat %>% filter(method != "gm1" & method != "gm2") %>% 
    droplevels() -> allDat

  nullDat <- bind_rows(gmDat, allDat)

  nullDat %>% mutate_at(c("method","sim","samps"), as.factor) %>%
    mutate(method = recode_factor(method, gm1 = "a", gm2 = "b", glm = "c", ora = "d", 
                         gsea = "e", ebgs = "f", .default = levels(method))) %>%
    mutate(method = factor(method, labels = c("gometh-500", "gometh-1000", "mrra-glm", 
                                            "mrra-ora", "mrra-gsea", 
                                            "ebgsea"))) -> nullDat

  save(nullDat, file = here("output/nullData.long.RData"))
} else {
  load(here("output/nullData.long.RData"))
}
```

The plots below shows that mRRA-ORA does not control the false discovery rate very well as the median proportion of p-value 0.05 for the 100 simulations is greater than 0.75. mRRA-GSEA does better, although its median false discovery rate is still relatively high at around 0.2. ebGSEA and mRRA-GLM both have a median false discovery rate at around 0.05, although ebGSEA has a tail of simulations between 0.05 and 0.2. Gometh is very consistent althought somewhat conservative in terms of false discovery control. 

```{r, fig.width=9}  
nullDat %>% group_by(sim, samps, method) %>%
  summarise(psig = sum(pval < 0.05)/length(pval)) -> psigData

p <- ggplot(psigData, aes(x=samps, y=psig, fill=method)) + 
  geom_violin(scale = "width")
p + scale_fill_paletteer_d(package = "LaCroixColoR", palette = "PassionFruit") +
  stat_summary(geom="point", size=0.8, color="white", position = position_dodge(0.9),
               show.legend = FALSE, fun.y = median) +
  geom_hline(yintercept=0.05, linetype="dashed", color = "red") +
  labs(y="Prop. gene sets with p-value < 0.05", x="No. samples",
       fill="Method") +
  theme_minimal()
```
