---
title: "Getting started"
output:
  workflowr::wflow_html:
    toc: false
---

```{r knitr, include = FALSE}
knitr::opts_chunk$set(autodep        = TRUE,
                      cache          = FALSE,
                      cache.comments = FALSE,
                      echo           = TRUE,
                      error          = FALSE,
                      fig.align      = "center",
                      fig.width      = 10,
                      fig.height     = 8,
                      message        = FALSE,
                      warning        = FALSE)
```

This page describes how to download the data and code used in this analysis,
set up the project directory and rerun the analysis. We have use the
[`workflowr`][workflowr] package to organise the analysis and insert
reproducibilty information into the output documents. 

Getting the code {#code}
================

All the code and outputs of analysis are available from GitHub at 
https://github.com/Oshlack/methyl-geneset-testing. If you want to replicate the 
analysis you can either fork the repository and clone it or download the 
repository as a zipped directory.

Once you have a local copy of the repository you should see the following
directory structure:

* `analysis/` - Contains the RMarkdown documents with the various stages of
  analysis. These are numbered according to the order they should be run.
* `data/` - This directory contains the data files used in the analysis with
  sub-directories for different data types (see [Getting the data](#data) for
  details). Processed intermediate data files will also be placed here.
* `output/` - Directory for output files produced by the analysis, each analysis
  step has it's own sub-directory.
* `docs/` - This directory contains the analysis website hosted at 
  http://oshlacklab.com/methyl-geneset-testing, including image files.
* `code/` - R scripts with custom functions used in some analysis stages. There
  are sub-directories for scripts associated with different steps in the 
  anlaysis.
* `README.md` - README describing the project.
* `.Rprofile` - Custom R profile for the project including set up for 
  `workflowr`.
* `.gitignore` - Details of files and directories that are excluded from the
  repository.
* `_workflowr.yml` - Workflowr configuration file.
* `methyl-geneset-testing.Rproj` - RStudio project file.


Getting the data {#data}
================

In this project we have used data from several publicly avilable datasets. 
Flow-sorted, blood cell methylation data generated using Illumina 
HumanMethylationEPIC arrays. Normal kidney methylation data from The Cancer
Genome Atlas (TCGA) kidney clear-cell carcinoma (KIRC) cohort, which was 
generated using Illumina HumanMethylation450 arrays. These are both 
automatically downloaded as part of the analysis directly from the Bioconductor
[ExperimentHub](expt-hub). 

We use a flow-sorted, blood cell RNAseq dataset,
which can be downloaded from GEO at [GSE107011](geo-rnaseq) or SRA at 
[SRP125125](sra-rnaseq).

Once the RNAseq data has been downloaded it needs to be extracted, placed in the 
correct directories and quasi-mapped and quantified using [Salmon](https://combine-lab.github.io/salmon/). The approach we took is 
described [here](expressionGenesets.html). The downstream analysis code assumes 
the following directory structure inside the `data/` directory:

* `datasets`
    * `SRP125125`
        * `quants`
            * `SRR6298258_quant`
            * ...
            * ...
            * `SRR6298376_quant`

We use pre B-cell development Affymetrix gene expression array data, 
which can be downloaded from GEO at [GSE45460](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE45460).

For downstream analysis the CEL files for each sample are expected to be present 
in the following directory structure:

* `data`
    * `datasets`
        * `GSE45460`
       
We use publicly available 450K data generated from developing human B-cells,
which can be downloaded from: [GSE45459](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE45459) .
Specifically, the `GSE45459_Matrix_signal_intensities.txt.gz` file should be 
downloaded and placed in the `data\datasets` directory and unzipped using `gunzip GSE45459_Matrix_signal_intensities.txt.gz`. 

Some additional data files used during the analysis are provided as part of the
repository. 

* `genesets`
    * `GO-immune-system-process.txt`
    * `kegg-immune-related-pathways.csv`
* `datasets`
    * `SRP125125`
        * `SraRunTableFull.txt`
        
Intermediate data files created during the analysis will be placed in: 

* `annotations`
* `cache-intermediates`
* `cache-region`
* `cache-rnaseq`
* `cache-runtime`

These are used by later stages of the analysis so should not be moved, altered 
or deleted.

Running the analysis {#analysis}
====================

The analysis directory contains the following analysis files:

```{r, echo=FALSE}
list.files(path = here::here("analysis"), pattern = "0[1-8]_*")
```

As indicated by the numbering they should be run in this order. If you want to
rerun the entire analysis this can be easily done using `workflowr`.

```r
workflowr::wflow_build(republish = TRUE)
```

It is important to consider the computer and environment you are using before
doing this. Running this analysis from scratch requires a considerable amount
of time, disk space and memory. **Some stages of the analysis need to be executed
on a HPC to generate results required by downstream steps.** If you do no have 
access to a HPC to perform these analyses using the code provided, you can 
download pre-computed RDS files containing the results from [![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.4005288.svg)](https://doi.org/10.5281/zenodo.4005288).

To use the pre-computed RDS objects, after cloning or downloading the GitHub repository to your computer, please extract the `outputs.tar.gz` archive under the `output` directory, using `tar -xvf outputs.tar.gz`.

It is also possible to run individual stages of the analysis, either by
providing the names of the file you want to run to `workflowr::wflow_build()` or
by manually knitting the document (for example using the 'Kint' button in
RStudio).

Once **all** the analyses have been rerun, the manuscript figures can be generated
using the code provided [here](figures.html). 

[expt-hub]: https://bioconductor.org/packages/release/bioc/html/ExperimentHub.html "ExperimentHub"
[geo-rnaseq]: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE107011 "GSE107011"
[sra-rnaseq]: https://www.ncbi.nlm.nih.gov/sra?term=SRP125125 "SRP125125"
[workflowr]: https://jdblischak.github.io/workflowr/ "Workflowr"

```{r session-info, cache = FALSE, eval = FALSE}
devtools::session_info()
```
