---
title: "Getting started"
output:
  workflowr::wflow_html:
    toc: false
---

```{r knitr, include = FALSE}
knitr::opts_chunk$set(autodep        = TRUE,
                      cache          = FALSE,
                      cache.comments = FALSE,
                      echo           = TRUE,
                      error          = FALSE,
                      fig.align      = "center",
                      fig.width      = 10,
                      fig.height     = 8,
                      message        = FALSE,
                      warning        = FALSE)
```

This page describes how to download the data and code used in this analysis,
set up the project directory and rerun the analysis. We have use the
[`workflowr`][workflowr] package to organise the analysis and insert
reproducibilty information into the output documents. 

Getting the code
================

All the code and outputs of analysis are available from GitHub at 
https://github.com/Oshlack/methyl-geneset-testing. If you want to replicate the 
analysis you can either fork the repository and clone it or download the 
repository as a zipped directory.

Once you have a local copy of the repository you should see the following
directory structure:

* `analysis/` - Contains the RMarkdown documents with the various stages of
  analysis. These are numbered according to the order they should be run.
* `data/` - This directory contains the data files used in the analysis with
  each dataset in it's own sub-directory (see [Getting the data](#data) for
  details). Processed intermediate data files will also be placed here.
* `output/` - Directory for output files produced by the analysis, each analysis
  step has it's own sub-directory.
* `docs/` - This directory contains the analysis website hosted at 
  http://oshlacklab.com/methyl-geneset-testing, including image files.
* `R/` - R scripts with custom functions used in some analysis stages.
* `packrat/` - Directory created by `packrat` that contains details of the R
  packages and versions used in the analysis.
* `README.md` - README describing the project.
* `.Rprofile` - Custom R profile for the project including set up for
  `packrat` and `workflowr`.
* `.gitignore` - Details of files and directories that are excluded from the
  repository.
* `_workflowr.yml` - Workflowr configuration file.
* `methyl-geneset-testing.Rproj` - RStudio project file.


Getting the data {#data}
================

In this project we have used data from several publicly avilable datasets. 
Flow-sorted, blood cell methylation data generated using Illumina 
HumanMethylationEPIC arrays. Normal kidney methylation data from The Cancer
Genome Atlas (TCGA) kidney clear-cell carcinoma (KIRC) cohort, which was 
generated using Illumina HumanMethylation450 arrays. These are both 
automatically downloaded as part of the analysis directly from the Bioconductor
(ExperimentHub)[expt-hub]. 
We also use a flow-sorted, blood cell RNAseq dataset,
which can be downloaded from GEO at [GSE107011](geo-rnaseq) or SRA at 
[SRP125125](sra-rnaseq).

Once the datasets have been downloaded they need to be extracted, placed in the 
correct directorys and renamed. The analysis code assumes the following
directory structure inside the `data/` directory:

* `SRP125125/`
    * `quants/`
  
Additional data files used during the analysis are provided as part of the
repository. Intermediate data files created during the analysis will be placed
in `data/processed`. These are used by later stages of the analysis so should
not be moved, altered or deleted.

Running the analysis
====================

The analysis directory contains the following analysis files:

```{r, echo=FALSE}
list.files(path = here::here("analysis"), pattern = "Rmd")
```


As indicated by the numbering they should be run in this order. If you want to
rerun the entire analysis this can be easily done using `workflowr`.

```r
workflowr::wflow_build(republish = TRUE)
```

It is important to consider the computer and environment you are using before
doing this. Running this analysis from scratch requires a considerable amount
of time, disk space and memory. Some stages of the analysis also assume that
multiple (10) cores are available for processing. If you have fewer cores
available you will need to change the following line in the relevant files and
provide the number of cores that are available for use.

```r
bpparam <- MulticoreParam(workers = 10)
```

It is also possible to run individual stages of the analysis, either by
providing the names of the file you want to run to `workflowr::wflow_build()` or
by manually knitting the document (for example using the 'Kint' button in
RStudio).

[expt-hub]: https://bioconductor.org/packages/release/bioc/html/ExperimentHub.html "ExperimentHub"
[geo-rnaseq]: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE107011 "GSE107011"
[sra-rnaseq]: https://www.ncbi.nlm.nih.gov/sra?term=SRP125125 "SRP125125"
[workflowr]: https://jdblischak.github.io/workflowr/ "Workflowr"

```{r session-info, cache = FALSE, eval = FALSE}
devtools::session_info()
```
