---
title: "Home"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
---

Gene set enrichment analysis for genome-wide DNA methylation data
-----------------------------------------------------------------
This site contains the results of the analyses presented in 
[*“Gene set enrichment analysis for genome-wide DNA methylation data”*](https://www.biorxiv.org/content/10.1101/2020.08.24.265702v1). 
Follow the links below to view the different parts of the analysis. For details
on how to reproduce the complete analysis, please see the 
[Getting started](gettingStarted.html) page.


Abstract
--------
DNA methylation is one of the most commonly studied epigenetic marks, due to its role in disease and development. Illumina methylation arrays have been extensively used to measure methylation across the human genome. Methylation array analysis has primarily focused on preprocessing, normalisation and identification of differentially methylated CpGs and regions. GOmeth and GOregion are new methods for performing unbiased gene set testing following differential methylation analysis. Benchmarking analyses demonstrate GOmeth outperforms other approaches and GOregion is the first method for gene set testing of differentially methylated regions. Both methods are publicly available in the [missMethyl](https://www.bioconductor.org/packages/release/bioc/html/missMethyl.html) Bioconductor R package.


Authors
-------
Jovana Maksimovic^1,2,3^, Alicia Oshlack^1,4^, Belinda Phipson^1,2+^

^1^ Peter MacCallum Cancer Centre, Melbourne, Victoria, 3000, Australia
^2^ Department of Pediatrics, University of Melbourne, Parkville, Victoria, 
    3010, Australia
^3^ Murdoch Children’s Research Institute, Parkville, Victoria, 3052, Australia
^4^ School of Biosciences, University of Melbourne, Parkville, Victoria, 3010, 
    Australia
    
^+^ corresponding author
 
 
Analysis
--------
1. [Explore EPIC array bias](01_exploreArrayBiasEPIC.html)
   - Explore the various array biases on the EPIC array that affect gene set testing.
      
2. [Explore 450k array bias](02_exploreArrayBias450.html)
   - Explore the various array biases on the 450k array that affect gene set testing.

3.
    a) [Generate a blood cell RNAseq "truth" set](03_expressionGenesets.html)
       - Analyse an RNAseq sorted blood cell dataset and identify the top ranked 
         gene sets for each cell type comparison.


    b) [Generate a breast cancer cell-line RNAseq "truth" set](03_expressionGenesetsNew.html)
       - Analyse an RNAseq breast cancer cell-line dataset and identify the top ranked 
         gene sets for each cell type comparison.
      
4. [Compare FDR of different methods](04_fdrAnalysis.html)
   - Analyse the normal samples from a 450k array KIRC TCGA dataset using 
     various genset testing methods to estimate their false discovery rate control.
      
5.
    a) [Compare performance of different methods](05_exploreData.html)
       - Analyse an EPIC array sorted blood cell dataset using various gene set 
         testing methods. Compare how well the different methods perform using several 
         metrics.

    b) [Compare performance of different methods](05_exploreDataNew.html)
       - Analyse an EPIC array breast cancer cell-line dataset using various gene set 
         testing methods. Compare how well the different methods perform using several 
         metrics.
      
6. [Compare run-time of different methods](06_runTimeComparison.html)
   - Analyse an EPIC array sorted blood cell dataset using various gene set 
     testing methods. Compare the run-time of the different methods.
      
7. [Evaluate GOregion](07_regionAnalysis.html)
   - Evalulate GOregion, our extension of gometh for geneset testing of 
     differentially methylated regions (DMRs) identified by DMR finding software.

8. [Effect of gene set size parameters on methylGSA](methylGSAParamSweep.html)
   - Analyse an EPIC array sorted blood cell dataset using various gene set 
     testing methods. Compare the run-time of the different methods.
      

Licenses
--------
The code in this analysis is covered by the [MIT license][mit] and the written
content on this website is covered by a [Creative Commons CC-BY][cc] license.


Citations
---------
> Maksimovic J, Oshlack O, Phipson B. Gene set enrichment analysis for 
> genome-wide DNA methylation data. bioRxiv. 2020. 
> DOI: [10.1186/s13073-019-0615-0][pp].

[mit]: https://choosealicense.com/licenses/mit/ "MIT License"
[cc]: https://creativecommons.org/licenses/by/4.0/ "CC-BY License"
[pp]: https://www.biorxiv.org/content/10.1101/2020.08.24.265702v1 "Pre-print"
