---
title: "Gene set testing for Illumina HumanMethylation Arrays"
author: "Jovana Maksimovic"
date: "6/21/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r libload, message=FALSE}
library(here)
library(ChAMP)
library(minfi)
library(paletteer)
library(limma)
library(BiocParallel)
library(reshape2)
library(DMRcate)
library(missMethyl)
library(ggplot2)
library(glue)
library(UpSetR)
library(dplyr)
library(patchwork)
library(tibble)
library(TxDb.Hsapiens.UCSC.hg19.knownGene)
source(here("code/utility.R"))
```

# Load data

We are using publicly available EPIC data [GSE110554](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE110554) generated from flow sorted blood cells. The data is normalised and filtered (bad probes, multi-mapping probes, SNP probes, sex chromosomes).

```{r}
# load data
dataFile <- here("data/GSE110554-data.RData")
if(file.exists(dataFile)){
  # load processed data and sample information
  load(dataFile)
} else {
  # get data from experiment hub, normalise, filter and save objects
  readData(dataFile)
  # load processed data and sample information
  load(dataFile)
}
```

# Statistical analysis

Compare several sets of sorted immune cells. Consider results significant at FDR < 0.05 and delta beta ~ 10% (~ lfc = 0.5).

```{r}
mVals <- getM(fltGr)
bVals <- getBeta(fltGr)
```

```{r}
design <- model.matrix(~0+targets$CellType)
colnames(design) <- levels(factor(targets$CellType))
fit <- lmFit(mVals, design)
cont.matrix <- makeContrasts(CD4vCD8=CD4T-CD8T,
                             MonovNeu=Mono-Neu,
                             BcellvNK=Bcell-NK,
                             levels=design)
fit2 <- contrasts.fit(fit, cont.matrix)
tfit <- eBayes(fit2, robust=TRUE, trend=TRUE)
tfit <- treat(tfit, lfc = 0.5)
pval <- 0.05
fitSum <- summary(decideTests(tfit, p.value = pval))
fitSum
```

# Find differentially methylated regions

Identify differentially methylated regions using the DMRcate package. 

```{r, message=FALSE}
outFile <- here("data/dmrcate-results.rds")

if(!file.exists(outFile)){
  dmrList <- vector("list", ncol(fitSum))

  for(i in 1:ncol(fitSum)){
    cpgAnn <- cpg.annotate("array", mVals, what = "M", arraytype = "EPIC",
                           analysis.type = "differential", design = design, 
                           contrasts = TRUE, cont.matrix = cont.matrix, 
                           coef = colnames(fitSum)[i])
    dmrList[[i]] <- extractRanges(dmrcate(cpgAnn))

  }
  saveRDS(dmrList, file = outFile)
  
} else {
  dmrList <- readRDS(outFile)
  
}
```

## GO analysis of DMRs
Run GO analysis on the differentially methylated regions (DMRs) identified by DMRcate for each of the contrasts.

```{r}
outFile <- here("data/dmrcate-go.rds")
anno <- loadAnnotation(arrayType = "EPIC")
txdb <- TxDb.Hsapiens.UCSC.hg19.knownGene
hg19Genes <- GenomicFeatures::genes(txdb)
dmrGo <- NULL

if(!file.exists(outFile)){
    for(i in 1:length(dmrList)){
        
        keep <- (abs(dmrList[[i]]$meandiff) > 0.1 & dmrList[[i]]$no.cpgs >=3)
        
        overlaps <- findOverlaps(hg19Genes, dmrList[[i]][keep, ])
        sigGenes <- hg19Genes$gene_id[from(overlaps)]
        tmp <- topGO(goana(sigGenes), number = Inf)
        tmp <- rownames_to_column(tmp, var = "GO")[, c("GO", "P.DE")]
        tmp$method <- "goana"
        tmp$contrast <- colnames(cont.matrix)[i]
        dmrGo <- bind_rows(dmrGo, tmp)
  
        tmp <- topGSA(goregion(dmrList[[i]][keep, ], anno = anno, 
                               prior.prob = FALSE, array.type = "EPIC"), 
                      number = Inf)
        tmp <- rownames_to_column(tmp, var = "GO")[, c("GO", "P.DE")]
        tmp$method <- "goregion-hgt"
        tmp$contrast <- colnames(cont.matrix)[i]
        dmrGo <- bind_rows(dmrGo, tmp)
        
        tmp <- topGSA(goregion(dmrList[[i]][keep, ], anno = anno, 
                               array.type = "EPIC"), 
                      number = Inf)
        tmp <- rownames_to_column(tmp, var = "GO")[, c("GO", "P.DE")]
        tmp$method <- "goregion-gometh"
        tmp$contrast <- colnames(cont.matrix)[i]
        dmrGo <- bind_rows(dmrGo, tmp)
        
        tmp <- topGSA(gometh(rownames(topTreat(tfit, coef = i, num = 5000, 
                                               p.value = pval)), anno = anno, 
                             array.type = "EPIC"), number = Inf)
        tmp <- rownames_to_column(tmp, var = "GO")[, c("GO", "P.DE")]
        tmp$method <- "gometh-probe-top"
        tmp$contrast <- colnames(cont.matrix)[i]
        dmrGo <- bind_rows(dmrGo, tmp)
        
        tmp <- topGSA(gometh(rownames(topTreat(tfit, coef = i, num = Inf, 
                                               p.value = pval)), anno = anno, 
                             array.type = "EPIC"), number = Inf)
        tmp <- rownames_to_column(tmp, var = "GO")[, c("GO", "P.DE")]
        tmp$method <- "gometh-probe-fdr"
        tmp$contrast <- colnames(cont.matrix)[i]
        dmrGo <- bind_rows(dmrGo, tmp)
    }
    
    saveRDS(dmrGo, file = outFile)
    
} else {
    dmrGo <- readRDS(outFile)
    
}
```

### Compare GOregion with other approaches

```{r, fig.width=9}
immuneGO <- unique(read.csv(here("data/GO-immune-system-process.txt"), 
                            stringsAsFactors = FALSE, header = FALSE, 
                            col.names = "GOID"))

dmrGo %>% arrange(contrast, method, P.DE) %>%
    group_by(contrast, method) %>%
    mutate(csum = cumsum(GO %in% immuneGO$GOID)) %>%
    mutate(rank = 1:n()) %>%
    filter(rank <= 100) -> dat

p <- ggplot(dat, aes(x = rank, y = csum, colour = method)) +
    geom_line() +
    facet_wrap(vars(contrast), ncol=3) +
    geom_vline(xintercept = 10, linetype = "dotted") +
    labs(colour = "Method", x = "Rank", y = "Cumulative no. immune sets")
p
```

```{r}
immuneGO <- readRDS(here("data/RNAseq-GO.rds"))
immuneGO %>% group_by(contrast) %>%
    mutate(rank = 1:n()) %>%
    filter(rank <= 100) -> topSets
    
dat %>% arrange(contrast, method, P.DE) %>%
    group_by(contrast, method) %>%
    mutate(csum = cumsum(GO %in% topSets$ID[topSets$contrast %in% contrast])) %>%
    mutate(rank = 1:n()) %>%
    filter(rank <= 100) -> sub

p <- ggplot(sub, aes(x = rank, y = csum, colour = method)) +
    geom_line() +
    facet_wrap(vars(contrast), ncol=3) +
    geom_vline(xintercept = 10, linetype = "dotted") +
    labs(colour = "Method", x = "Rank", 
         y = glue("Cumulative no. RNAseq sets")) +
    theme(legend.position = "bottom")
p
```

Examine what the top 10 ranked gene sets are and how many genes they contain, for each method and comparison.

```{r, fig.height=11, fig.width=6}
terms <- missMethyl:::.getGO()$idTable
nGenes <- rownames_to_column(data.frame(n = sapply(missMethyl:::.getGO()$idList, 
                                                   length)), 
                             var = "ID")

dat %>% arrange(contrast, method, P.DE) %>% 
    group_by(contrast, method) %>%
    mutate(FDR = p.adjust(P.DE)) %>%
    filter(rank <= 10) %>% 
    inner_join(terms, by = c("GO" = "GOID")) %>%
    inner_join(nGenes, by = c("GO" = "ID")) -> sub

p <- vector("list", length(unique(sub$contrast)) * length(unique(sub$method)))
i = 1
for(cont in unique(sub$contrast)){
    c = 1
    for(meth in unique(sub$method)){
        tmp <- sub %>% filter(contrast == cont & method == meth) %>%
            mutate(rank = factor(rank), 
                   rank = factor(rank, levels = rev(levels(rank))))
        
        p[[i]] <- ggplot(tmp, aes(x = -log10(FDR), y = rank)) + 
            geom_point(aes(size = n), alpha = 0.5, 
                colour = scales::hue_pal()(length(unique(sub$method)))[c]) +
            scale_y_discrete(labels = rev(tmp$TERM)) +
            labs(y = "", size = "No. genes", title = meth) +
            theme(axis.text.y = element_text(size = 6),
                  plot.title = element_text(size = 8),
                  legend.position = "right", 
                  legend.key.size = unit(0.25, "cm"),
                  legend.text = element_text(size = 6),
                  legend.title = element_text(size = 8),
                  axis.text.x = element_text(size = 6),
                  axis.title.x = element_text(size = 8)) + 
            coord_cartesian(xlim = c(-log10(0.99), -log10(10^-100))) +
            geom_vline(xintercept = -log10(0.05), linetype = "dashed")
        i = i + 1
        c = c + 1
    }
}

(p[[1]] / p[[2]] / p[[3]] / p[[4]] / p[[5]]) + 
    plot_annotation(title = unique(sub$contrast)[1],
                    theme = theme(plot.title = element_text(size = 10))) 
(p[[6]] / p[[7]] / p[[8]] / p[[9]] / p[[10]]) + 
    plot_annotation(title = unique(sub$contrast)[2],
                    theme = theme(plot.title = element_text(size = 10))) 
(p[[11]] / p[[12]] / p[[13]] / p[[14]] / p[[15]]) + 
    plot_annotation(title = unique(sub$contrast)[3],
                    theme = theme(plot.title = element_text(size = 10))) 
```

## Compare characteristics of region-wise and probe-wise results

```{r, message=FALSE}
cpgs <- GRanges(seqnames = anno$chr, 
                ranges = IRanges(start = anno$pos, 
                                 end = anno$pos),
                strand = anno$strand,
                name = anno$Name)
dat <- NULL

for(i in 1:ncol(cont.matrix)){
  overlaps <- findOverlaps(cpgs, dmrList[[i]])
  tmp <- data.frame(cpgs = cpgs$name[from(overlaps)],
                    method = "dmrcate", 
                    contrast = colnames(cont.matrix)[i],
                    stringsAsFactors = FALSE)
  dat <- bind_rows(dat, tmp)
  
  tmp <- data.frame(cpgs = rownames(topTreat(tfit, coef = i, num = 5000)),
                    method = "probe-top",
                    contrast = colnames(cont.matrix)[i],
                    stringsAsFactors = FALSE)
  dat <- bind_rows(dat, tmp)
  
  tmp <- data.frame(cpgs = rownames(topTreat(tfit, coef = i, num = Inf, 
                                             p.value = pval)),
                    method = "probe-fdr",
                    contrast = colnames(cont.matrix)[i],
                    stringsAsFactors = FALSE)
  dat <- bind_rows(dat, tmp)
  
}

dat %>% group_by(contrast, method) %>% tally() -> sub

ggplot(sub, aes(x = method, y = n, fill = method)) +
    geom_bar(stat = "identity", show.legend = FALSE) +
    facet_wrap(vars(contrast)) + 
    labs(fill = "Method", y = "No. significant CpGs", x = "Method") + 
    theme(axis.text.x = element_text(angle = 45, hjust = 1))
```

```{r}
flatAnn <- loadFlatAnnotation(anno)

dat %>% group_by(contrast, method) %>%
    inner_join(flatAnn, by = c("cpgs" = "cpg")) %>% 
    group_by(contrast, method) %>%
    dplyr::select(group_cols(), entrezid) %>%
    distinct() %>%
    tally() -> sub

ggplot(sub, aes(x = method, y = n, fill = method)) +
    geom_bar(stat = "identity", show.legend = FALSE) +
    facet_wrap(vars(contrast)) + 
    labs(fill = "Method", y = "No. genes with sig. CpGs", x = "Method") + 
    theme(axis.text.x = element_text(angle = 45, hjust = 1))
```

```{r}
dat %>% group_by(contrast, method) %>%
    left_join(flatAnn, by = c("cpgs" = "cpg")) %>% 
    group_by(contrast, method) %>%
    dplyr::select(group_cols(), entrezid, cpgs) %>%
    summarise(prop = sum(!is.na(entrezid[!duplicated(cpgs)]))/
                  length(unique(cpgs))) -> sub

ggplot(sub, aes(x = method, y = prop, fill = method)) +
    geom_bar(stat = "identity", show.legend = FALSE) +
    facet_wrap(vars(contrast)) + 
    labs(fill = "Method", y = "Prop. sig. CpGs mapped to genes", x = "Method") + 
    theme(axis.text.x = element_text(angle = 45, hjust = 1))
```

```{r, fig.height=7, fig.width=6}
dat %>% group_by(contrast, method) %>%
    left_join(flatAnn, by = c("cpgs" = "cpg")) %>% 
    group_by(contrast, method) %>%
    dplyr::select(group_cols(), group, cpgs) %>%
    group_by(contrast, method, group) %>%
    tally() -> sub

ggplot(sub, aes(x = group, y = n, fill = method)) +
    geom_bar(stat = "identity", position = "dodge") +
    facet_wrap(vars(contrast), nrow = 3, ncol = 1, scales = "free_y") + 
    labs(fill = "Method", y = "No. sig. CpGs mapped to genomic features", 
         x = "Feature") + 
    theme(axis.text.x = element_text(angle = 45, hjust = 1))
```

## CpG overlap between region-wise and probe-wise approaches

Compare the CpGs covered by the different approaches, for the three contrasts.

```{r}
p <- vector("list", ncol(cont.matrix))

for(i in 1:ncol(cont.matrix)){
    dat %>% filter(contrast == colnames(cont.matrix)[i]) -> tmp
    tmp <- split(tmp$cpgs, f = tmp$method)
    p[[i]] <- upset(fromList(tmp), order.by = "freq", keep.order = TRUE, 
               sets = names(tmp))
    
}

p[[1]]
p[[2]]
p[[3]]
```

## Gene overlap between region-wise and probe-wise approaches

Compare the genes covered by the different approaches, for the three contrasts.

```{r}
p <- vector("list", ncol(cont.matrix))

for(i in 1:ncol(cont.matrix)){
    dat %>% filter(contrast == colnames(cont.matrix)[i]) %>%
        left_join(flatAnn, by = c("cpgs" = "cpg")) %>% 
        dplyr::select(method, entrezid) %>%
        distinct() -> tmp

    tmp <- split(tmp$entrezid, f = tmp$method)
    p[[i]] <- upset(fromList(tmp), order.by = "freq", keep.order = TRUE, 
               sets = names(tmp))
    
}

p[[1]]
p[[2]]
p[[3]]
```

## Effect of DMR cut offs i.e. num probes in region and absolute delta beta

```{r}
outFile <- here("data/dmrcate-params.rds")
dmrParams <- NULL

meanDiffs <- seq(0, 0.2, by = 0.1)
noCpgs <- 2:4

if(!file.exists(outFile)){
    for(i in 1:length(dmrList)){
        for(j in meanDiffs){
            for(k in noCpgs){
                keep <- (abs(dmrList[[i]]$meandiff) > j & 
                             dmrList[[i]]$no.cpgs >= k)
                
                tmp <- topGSA(goregion(dmrList[[i]][keep, ], anno = anno, 
                                       array.type = "EPIC"), 
                              number = Inf)
                tmp <- rownames_to_column(tmp, var = "GO")[, c("GO", "P.DE")]
                tmp$params <- glue("|Beta| = {j}; No. CpGs = {k}")
                tmp$contrast <- colnames(cont.matrix)[i]
                dmrParams <- bind_rows(dmrParams, tmp)
            }
        }
    }
    
    saveRDS(dmrParams, file = outFile)
    
} else {
    dmrParams <- readRDS(outFile)
    
}
```

Examine effect of changing DMr parameter cut offs on gene set rankings of GO categories in "immune system process".

```{r, fig.width=9}
immuneGO <- unique(read.csv(here("data/GO-immune-system-process.txt"), 
                            stringsAsFactors = FALSE, header = FALSE, 
                            col.names = "GOID"))

dmrParams %>% arrange(contrast, params, P.DE) %>%
    group_by(contrast, params) %>%
    mutate(csum = cumsum(GO %in% immuneGO$GOID)) %>%
    mutate(rank = 1:n()) %>%
    filter(rank <= 100) -> dat

p <- ggplot(dat, aes(x = rank, y = csum, colour = params)) +
    geom_line() +
    facet_wrap(vars(contrast), ncol=3) +
    geom_vline(xintercept = 10, linetype = "dotted") +
    labs(colour = "Parameters", x = "Rank", y = "Cumulative no. immune sets")
p
```

Examine effect of changing DMR parameter cut offs on gene set rankings on GO categories derived from RNAseq analysis.  

```{r, fig.width=9}
immuneGO <- readRDS(here("data/RNAseq-GO.rds"))
immuneGO %>% group_by(contrast) %>%
    mutate(rank = 1:n()) %>%
    filter(rank <= 100) -> topSets
    
dmrParams %>% arrange(contrast, params, P.DE) %>%
    group_by(contrast, params) %>%
    mutate(csum = cumsum(GO %in% topSets$ID[topSets$contrast %in% contrast])) %>%
    mutate(rank = 1:n()) %>%
    filter(rank <= 100) -> sub

p <- ggplot(sub, aes(x = rank, y = csum, colour = params)) +
    geom_line() +
    facet_wrap(vars(contrast), ncol=3) +
    geom_vline(xintercept = 10, linetype = "dotted") +
    labs(colour = "Parameters", x = "Rank", 
         y = glue("Cumulative no. RNAseq sets"))
p
```
